// @pulumi/azure/paloalto/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkAppliance = exports.NextGenerationFirewallVirtualNetworkPanorama = exports.NextGenerationFirewallVirtualNetworkLocalRulestack = exports.NextGenerationFirewallVirtualHubPanorama = exports.NextGenerationFirewallVirtualHubLocalRulestack = exports.LocalRulestackRule = exports.LocalRulestackPrefixList = exports.LocalRulestackOutboundUntrustCertificateAssociation = exports.LocalRulestackOutboundTrustCertificateAssociation = exports.LocalRulestackFqdnList = exports.LocalRulestackCertificate = exports.LocalRulestack = exports.getLocalRulestackOutput = exports.getLocalRulestack = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getLocalRulestack = null;
exports.getLocalRulestackOutput = null;
utilities.lazyLoad(exports, ["getLocalRulestack", "getLocalRulestackOutput"], () => require("./getLocalRulestack"));
exports.LocalRulestack = null;
utilities.lazyLoad(exports, ["LocalRulestack"], () => require("./localRulestack"));
exports.LocalRulestackCertificate = null;
utilities.lazyLoad(exports, ["LocalRulestackCertificate"], () => require("./localRulestackCertificate"));
exports.LocalRulestackFqdnList = null;
utilities.lazyLoad(exports, ["LocalRulestackFqdnList"], () => require("./localRulestackFqdnList"));
exports.LocalRulestackOutboundTrustCertificateAssociation = null;
utilities.lazyLoad(exports, ["LocalRulestackOutboundTrustCertificateAssociation"], () => require("./localRulestackOutboundTrustCertificateAssociation"));
exports.LocalRulestackOutboundUntrustCertificateAssociation = null;
utilities.lazyLoad(exports, ["LocalRulestackOutboundUntrustCertificateAssociation"], () => require("./localRulestackOutboundUntrustCertificateAssociation"));
exports.LocalRulestackPrefixList = null;
utilities.lazyLoad(exports, ["LocalRulestackPrefixList"], () => require("./localRulestackPrefixList"));
exports.LocalRulestackRule = null;
utilities.lazyLoad(exports, ["LocalRulestackRule"], () => require("./localRulestackRule"));
exports.NextGenerationFirewallVirtualHubLocalRulestack = null;
utilities.lazyLoad(exports, ["NextGenerationFirewallVirtualHubLocalRulestack"], () => require("./nextGenerationFirewallVirtualHubLocalRulestack"));
exports.NextGenerationFirewallVirtualHubPanorama = null;
utilities.lazyLoad(exports, ["NextGenerationFirewallVirtualHubPanorama"], () => require("./nextGenerationFirewallVirtualHubPanorama"));
exports.NextGenerationFirewallVirtualNetworkLocalRulestack = null;
utilities.lazyLoad(exports, ["NextGenerationFirewallVirtualNetworkLocalRulestack"], () => require("./nextGenerationFirewallVirtualNetworkLocalRulestack"));
exports.NextGenerationFirewallVirtualNetworkPanorama = null;
utilities.lazyLoad(exports, ["NextGenerationFirewallVirtualNetworkPanorama"], () => require("./nextGenerationFirewallVirtualNetworkPanorama"));
exports.VirtualNetworkAppliance = null;
utilities.lazyLoad(exports, ["VirtualNetworkAppliance"], () => require("./virtualNetworkAppliance"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:paloalto/localRulestack:LocalRulestack":
                return new exports.LocalRulestack(name, undefined, { urn });
            case "azure:paloalto/localRulestackCertificate:LocalRulestackCertificate":
                return new exports.LocalRulestackCertificate(name, undefined, { urn });
            case "azure:paloalto/localRulestackFqdnList:LocalRulestackFqdnList":
                return new exports.LocalRulestackFqdnList(name, undefined, { urn });
            case "azure:paloalto/localRulestackOutboundTrustCertificateAssociation:LocalRulestackOutboundTrustCertificateAssociation":
                return new exports.LocalRulestackOutboundTrustCertificateAssociation(name, undefined, { urn });
            case "azure:paloalto/localRulestackOutboundUntrustCertificateAssociation:LocalRulestackOutboundUntrustCertificateAssociation":
                return new exports.LocalRulestackOutboundUntrustCertificateAssociation(name, undefined, { urn });
            case "azure:paloalto/localRulestackPrefixList:LocalRulestackPrefixList":
                return new exports.LocalRulestackPrefixList(name, undefined, { urn });
            case "azure:paloalto/localRulestackRule:LocalRulestackRule":
                return new exports.LocalRulestackRule(name, undefined, { urn });
            case "azure:paloalto/nextGenerationFirewallVirtualHubLocalRulestack:NextGenerationFirewallVirtualHubLocalRulestack":
                return new exports.NextGenerationFirewallVirtualHubLocalRulestack(name, undefined, { urn });
            case "azure:paloalto/nextGenerationFirewallVirtualHubPanorama:NextGenerationFirewallVirtualHubPanorama":
                return new exports.NextGenerationFirewallVirtualHubPanorama(name, undefined, { urn });
            case "azure:paloalto/nextGenerationFirewallVirtualNetworkLocalRulestack:NextGenerationFirewallVirtualNetworkLocalRulestack":
                return new exports.NextGenerationFirewallVirtualNetworkLocalRulestack(name, undefined, { urn });
            case "azure:paloalto/nextGenerationFirewallVirtualNetworkPanorama:NextGenerationFirewallVirtualNetworkPanorama":
                return new exports.NextGenerationFirewallVirtualNetworkPanorama(name, undefined, { urn });
            case "azure:paloalto/virtualNetworkAppliance:VirtualNetworkAppliance":
                return new exports.VirtualNetworkAppliance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "paloalto/localRulestack", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/localRulestackCertificate", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/localRulestackFqdnList", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/localRulestackOutboundTrustCertificateAssociation", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/localRulestackOutboundUntrustCertificateAssociation", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/localRulestackPrefixList", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/localRulestackRule", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/nextGenerationFirewallVirtualHubLocalRulestack", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/nextGenerationFirewallVirtualHubPanorama", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/nextGenerationFirewallVirtualNetworkLocalRulestack", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/nextGenerationFirewallVirtualNetworkPanorama", _module);
pulumi.runtime.registerResourceModule("azure", "paloalto/virtualNetworkAppliance", _module);
