// @pulumi/azure/paloalto/localRulestackFqdnList.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalRulestackFqdnList = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Palo Alto Local Rulestack FQDN List.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "rg-example",
 *     location: "West Europe",
 * });
 * const exampleLocalRulestack = new azure.paloalto.LocalRulestack("example", {
 *     name: "example",
 *     resourceGroupName: exampleAzurermResrouceGroup.name,
 *     location: example.location,
 * });
 * const exampleLocalRulestackFqdnList = new azure.paloalto.LocalRulestackFqdnList("example", {
 *     name: "example",
 *     rulestackId: exampleLocalRulestack.id,
 *     fullyQualifiedDomainNames: ["contoso.com"],
 * });
 * ```
 *
 * ## Import
 *
 * Palo Alto Local Rulestack FQDN Lists can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:paloalto/localRulestackFqdnList:LocalRulestackFqdnList example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/PaloAltoNetworks.Cloudngfw/localRulestacks/myLocalRulestack/fqdnLists/myFQDNList1
 * ```
 */
class LocalRulestackFqdnList extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["auditComment"] = state ? state.auditComment : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["fullyQualifiedDomainNames"] = state ? state.fullyQualifiedDomainNames : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["rulestackId"] = state ? state.rulestackId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.fullyQualifiedDomainNames === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fullyQualifiedDomainNames'");
            }
            if ((!args || args.rulestackId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rulestackId'");
            }
            resourceInputs["auditComment"] = args ? args.auditComment : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["fullyQualifiedDomainNames"] = args ? args.fullyQualifiedDomainNames : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["rulestackId"] = args ? args.rulestackId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LocalRulestackFqdnList.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing LocalRulestackFqdnList resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LocalRulestackFqdnList(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocalRulestackFqdnList.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocalRulestackFqdnList.__pulumiType;
    }
}
exports.LocalRulestackFqdnList = LocalRulestackFqdnList;
/** @internal */
LocalRulestackFqdnList.__pulumiType = 'azure:paloalto/localRulestackFqdnList:LocalRulestackFqdnList';
