// @pulumi/azure-compliance-policies/azurenative/signalrservice/v20240101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignalRSharedPrivateLinkResource = exports.SignalRReplica = exports.SignalRPrivateEndpointConnection = exports.SignalRCustomDomain = exports.SignalRCustomCertificate = exports.SignalR = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240101preview_1 = require("@pulumi/azure-native/signalrservice/v20240101preview");
const v20240101preview_2 = require("@pulumi/azure-native/signalrservice/v20240101preview");
const v20240101preview_3 = require("@pulumi/azure-native/signalrservice/v20240101preview");
const v20240101preview_4 = require("@pulumi/azure-native/signalrservice/v20240101preview");
const v20240101preview_5 = require("@pulumi/azure-native/signalrservice/v20240101preview");
const v20240101preview_6 = require("@pulumi/azure-native/signalrservice/v20240101preview");
var SignalR;
(function (SignalR) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalR).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SignalR.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-signalrservice-v20240101preview-signalr-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalR).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_1.SignalR, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SignalR shouldn't use an unstable API (signalrservice.v20240101preview.SignalR). A compatible replacement can be found at 'signalrservice.SignalR'.");
            }),
        },
        vendors: ["azure"],
        services: ["signalrservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SignalR || (SignalR = {}));
exports.SignalR = SignalR;
var SignalRCustomCertificate;
(function (SignalRCustomCertificate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRCustomCertificate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SignalRCustomCertificate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-signalrservice-v20240101preview-signalrcustomcertificate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRCustomCertificate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_2.SignalRCustomCertificate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SignalRCustomCertificate shouldn't use an unstable API (signalrservice.v20240101preview.SignalRCustomCertificate). A compatible replacement can be found at 'signalrservice.SignalRCustomCertificate'.");
            }),
        },
        vendors: ["azure"],
        services: ["signalrservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SignalRCustomCertificate || (SignalRCustomCertificate = {}));
exports.SignalRCustomCertificate = SignalRCustomCertificate;
var SignalRCustomDomain;
(function (SignalRCustomDomain) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRCustomDomain).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SignalRCustomDomain.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-signalrservice-v20240101preview-signalrcustomdomain-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRCustomDomain).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_3.SignalRCustomDomain, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SignalRCustomDomain shouldn't use an unstable API (signalrservice.v20240101preview.SignalRCustomDomain). A compatible replacement can be found at 'signalrservice.SignalRCustomDomain'.");
            }),
        },
        vendors: ["azure"],
        services: ["signalrservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SignalRCustomDomain || (SignalRCustomDomain = {}));
exports.SignalRCustomDomain = SignalRCustomDomain;
var SignalRPrivateEndpointConnection;
(function (SignalRPrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRPrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SignalRPrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-signalrservice-v20240101preview-signalrprivateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRPrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_4.SignalRPrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SignalRPrivateEndpointConnection shouldn't use an unstable API (signalrservice.v20240101preview.SignalRPrivateEndpointConnection). A compatible replacement can be found at 'signalrservice.SignalRPrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["signalrservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SignalRPrivateEndpointConnection || (SignalRPrivateEndpointConnection = {}));
exports.SignalRPrivateEndpointConnection = SignalRPrivateEndpointConnection;
var SignalRReplica;
(function (SignalRReplica) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRReplica).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SignalRReplica.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-signalrservice-v20240101preview-signalrreplica-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRReplica).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_5.SignalRReplica, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SignalRReplica shouldn't use an unstable API (signalrservice.v20240101preview.SignalRReplica). A compatible replacement can be found at 'signalrservice.SignalRReplica'.");
            }),
        },
        vendors: ["azure"],
        services: ["signalrservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SignalRReplica || (SignalRReplica = {}));
exports.SignalRReplica = SignalRReplica;
var SignalRSharedPrivateLinkResource;
(function (SignalRSharedPrivateLinkResource) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRSharedPrivateLinkResource).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SignalRSharedPrivateLinkResource.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-signalrservice-v20240101preview-signalrsharedprivatelinkresource-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (signalrservice.v20240101preview.SignalRSharedPrivateLinkResource).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_6.SignalRSharedPrivateLinkResource, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SignalRSharedPrivateLinkResource shouldn't use an unstable API (signalrservice.v20240101preview.SignalRSharedPrivateLinkResource). A compatible replacement can be found at 'signalrservice.SignalRSharedPrivateLinkResource'.");
            }),
        },
        vendors: ["azure"],
        services: ["signalrservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SignalRSharedPrivateLinkResource || (SignalRSharedPrivateLinkResource = {}));
exports.SignalRSharedPrivateLinkResource = SignalRSharedPrivateLinkResource;
