// @pulumi/azure-compliance-policies/azurenative/storagepool/v20200315preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.IscsiTarget = exports.DiskPool = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20200315preview_1 = require("@pulumi/azure-native/storagepool/v20200315preview");
const v20200315preview_2 = require("@pulumi/azure-native/storagepool/v20200315preview");
var DiskPool;
(function (DiskPool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (storagepool.v20200315preview.DiskPool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DiskPool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-storagepool-v20200315preview-diskpool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (storagepool.v20200315preview.DiskPool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200315preview_1.DiskPool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DiskPool shouldn't use an unstable API (storagepool.v20200315preview.DiskPool). A compatible replacement can be found at 'storagepool.DiskPool'.");
            }),
        },
        vendors: ["azure"],
        services: ["storagepool"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DiskPool || (DiskPool = {}));
exports.DiskPool = DiskPool;
var IscsiTarget;
(function (IscsiTarget) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (storagepool.v20200315preview.IscsiTarget).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IscsiTarget.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-storagepool-v20200315preview-iscsitarget-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (storagepool.v20200315preview.IscsiTarget).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200315preview_2.IscsiTarget, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IscsiTarget shouldn't use an unstable API (storagepool.v20200315preview.IscsiTarget). A compatible replacement can be found at 'storagepool.IscsiTarget'.");
            }),
        },
        vendors: ["azure"],
        services: ["storagepool"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IscsiTarget || (IscsiTarget = {}));
exports.IscsiTarget = IscsiTarget;
