// @pulumi/azure-compliance-policies/azurenative/programmableconnectivity/v20240115preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperatorApiConnection = exports.Gateway = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240115preview_1 = require("@pulumi/azure-native/programmableconnectivity/v20240115preview");
const v20240115preview_2 = require("@pulumi/azure-native/programmableconnectivity/v20240115preview");
var Gateway;
(function (Gateway) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (programmableconnectivity.v20240115preview.Gateway).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Gateway.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-programmableconnectivity-v20240115preview-gateway-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (programmableconnectivity.v20240115preview.Gateway).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240115preview_1.Gateway, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Gateway shouldn't use an unstable API (programmableconnectivity.v20240115preview.Gateway). A compatible replacement can be found at 'programmableconnectivity.Gateway'.");
            }),
        },
        vendors: ["azure"],
        services: ["programmableconnectivity"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Gateway || (Gateway = {}));
exports.Gateway = Gateway;
var OperatorApiConnection;
(function (OperatorApiConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (programmableconnectivity.v20240115preview.OperatorApiConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OperatorApiConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-programmableconnectivity-v20240115preview-operatorapiconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (programmableconnectivity.v20240115preview.OperatorApiConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240115preview_2.OperatorApiConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OperatorApiConnection shouldn't use an unstable API (programmableconnectivity.v20240115preview.OperatorApiConnection). A compatible replacement can be found at 'programmableconnectivity.OperatorApiConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["programmableconnectivity"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OperatorApiConnection || (OperatorApiConnection = {}));
exports.OperatorApiConnection = OperatorApiConnection;
