// @pulumi/azure/streamanalytics/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamInputIotHub = exports.StreamInputEventHubV2 = exports.StreamInputEventHub = exports.StreamInputBlob = exports.ReferenceInputMssql = exports.ReferenceInputBlob = exports.OutputTable = exports.OutputSynapse = exports.OutputServicebusTopic = exports.OutputServiceBusQueue = exports.OutputPowerbi = exports.OutputMssql = exports.OutputFunction = exports.OutputEventHub = exports.OutputCosmosdb = exports.OutputBlob = exports.ManagedPrivateEndpoint = exports.JobSchedule = exports.Job = exports.getJobOutput = exports.getJob = exports.FunctionJavascriptUda = exports.FunctionJavaScriptUDF = exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.FunctionJavaScriptUDF = null;
utilities.lazyLoad(exports, ["FunctionJavaScriptUDF"], () => require("./functionJavaScriptUDF"));
exports.FunctionJavascriptUda = null;
utilities.lazyLoad(exports, ["FunctionJavascriptUda"], () => require("./functionJavascriptUda"));
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.JobSchedule = null;
utilities.lazyLoad(exports, ["JobSchedule"], () => require("./jobSchedule"));
exports.ManagedPrivateEndpoint = null;
utilities.lazyLoad(exports, ["ManagedPrivateEndpoint"], () => require("./managedPrivateEndpoint"));
exports.OutputBlob = null;
utilities.lazyLoad(exports, ["OutputBlob"], () => require("./outputBlob"));
exports.OutputCosmosdb = null;
utilities.lazyLoad(exports, ["OutputCosmosdb"], () => require("./outputCosmosdb"));
exports.OutputEventHub = null;
utilities.lazyLoad(exports, ["OutputEventHub"], () => require("./outputEventHub"));
exports.OutputFunction = null;
utilities.lazyLoad(exports, ["OutputFunction"], () => require("./outputFunction"));
exports.OutputMssql = null;
utilities.lazyLoad(exports, ["OutputMssql"], () => require("./outputMssql"));
exports.OutputPowerbi = null;
utilities.lazyLoad(exports, ["OutputPowerbi"], () => require("./outputPowerbi"));
exports.OutputServiceBusQueue = null;
utilities.lazyLoad(exports, ["OutputServiceBusQueue"], () => require("./outputServiceBusQueue"));
exports.OutputServicebusTopic = null;
utilities.lazyLoad(exports, ["OutputServicebusTopic"], () => require("./outputServicebusTopic"));
exports.OutputSynapse = null;
utilities.lazyLoad(exports, ["OutputSynapse"], () => require("./outputSynapse"));
exports.OutputTable = null;
utilities.lazyLoad(exports, ["OutputTable"], () => require("./outputTable"));
exports.ReferenceInputBlob = null;
utilities.lazyLoad(exports, ["ReferenceInputBlob"], () => require("./referenceInputBlob"));
exports.ReferenceInputMssql = null;
utilities.lazyLoad(exports, ["ReferenceInputMssql"], () => require("./referenceInputMssql"));
exports.StreamInputBlob = null;
utilities.lazyLoad(exports, ["StreamInputBlob"], () => require("./streamInputBlob"));
exports.StreamInputEventHub = null;
utilities.lazyLoad(exports, ["StreamInputEventHub"], () => require("./streamInputEventHub"));
exports.StreamInputEventHubV2 = null;
utilities.lazyLoad(exports, ["StreamInputEventHubV2"], () => require("./streamInputEventHubV2"));
exports.StreamInputIotHub = null;
utilities.lazyLoad(exports, ["StreamInputIotHub"], () => require("./streamInputIotHub"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:streamanalytics/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure:streamanalytics/functionJavaScriptUDF:FunctionJavaScriptUDF":
                return new exports.FunctionJavaScriptUDF(name, undefined, { urn });
            case "azure:streamanalytics/functionJavascriptUda:FunctionJavascriptUda":
                return new exports.FunctionJavascriptUda(name, undefined, { urn });
            case "azure:streamanalytics/job:Job":
                return new exports.Job(name, undefined, { urn });
            case "azure:streamanalytics/jobSchedule:JobSchedule":
                return new exports.JobSchedule(name, undefined, { urn });
            case "azure:streamanalytics/managedPrivateEndpoint:ManagedPrivateEndpoint":
                return new exports.ManagedPrivateEndpoint(name, undefined, { urn });
            case "azure:streamanalytics/outputBlob:OutputBlob":
                return new exports.OutputBlob(name, undefined, { urn });
            case "azure:streamanalytics/outputCosmosdb:OutputCosmosdb":
                return new exports.OutputCosmosdb(name, undefined, { urn });
            case "azure:streamanalytics/outputEventHub:OutputEventHub":
                return new exports.OutputEventHub(name, undefined, { urn });
            case "azure:streamanalytics/outputFunction:OutputFunction":
                return new exports.OutputFunction(name, undefined, { urn });
            case "azure:streamanalytics/outputMssql:OutputMssql":
                return new exports.OutputMssql(name, undefined, { urn });
            case "azure:streamanalytics/outputPowerbi:OutputPowerbi":
                return new exports.OutputPowerbi(name, undefined, { urn });
            case "azure:streamanalytics/outputServiceBusQueue:OutputServiceBusQueue":
                return new exports.OutputServiceBusQueue(name, undefined, { urn });
            case "azure:streamanalytics/outputServicebusTopic:OutputServicebusTopic":
                return new exports.OutputServicebusTopic(name, undefined, { urn });
            case "azure:streamanalytics/outputSynapse:OutputSynapse":
                return new exports.OutputSynapse(name, undefined, { urn });
            case "azure:streamanalytics/outputTable:OutputTable":
                return new exports.OutputTable(name, undefined, { urn });
            case "azure:streamanalytics/referenceInputBlob:ReferenceInputBlob":
                return new exports.ReferenceInputBlob(name, undefined, { urn });
            case "azure:streamanalytics/referenceInputMssql:ReferenceInputMssql":
                return new exports.ReferenceInputMssql(name, undefined, { urn });
            case "azure:streamanalytics/streamInputBlob:StreamInputBlob":
                return new exports.StreamInputBlob(name, undefined, { urn });
            case "azure:streamanalytics/streamInputEventHub:StreamInputEventHub":
                return new exports.StreamInputEventHub(name, undefined, { urn });
            case "azure:streamanalytics/streamInputEventHubV2:StreamInputEventHubV2":
                return new exports.StreamInputEventHubV2(name, undefined, { urn });
            case "azure:streamanalytics/streamInputIotHub:StreamInputIotHub":
                return new exports.StreamInputIotHub(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "streamanalytics/cluster", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/functionJavaScriptUDF", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/functionJavascriptUda", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/job", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/jobSchedule", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/managedPrivateEndpoint", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputBlob", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputCosmosdb", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputEventHub", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputFunction", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputMssql", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputPowerbi", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputServiceBusQueue", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputServicebusTopic", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputSynapse", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/outputTable", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/referenceInputBlob", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/referenceInputMssql", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/streamInputBlob", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/streamInputEventHub", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/streamInputEventHubV2", _module);
pulumi.runtime.registerResourceModule("azure", "streamanalytics/streamInputIotHub", _module);
