// @pulumi/azure-compliance-policies/azurenative/devcenter/v20241001preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schedule = exports.ProjectPolicy = exports.ProjectEnvironmentType = exports.ProjectCatalog = exports.Project = exports.Pool = exports.PlanMember = exports.Plan = exports.NetworkConnection = exports.Gallery = exports.EnvironmentType = exports.EncryptionSet = exports.DevCenter = exports.DevBoxDefinition = exports.CurationProfile = exports.Catalog = exports.AttachedNetworkByDevCenter = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20241001preview_1 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_2 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_3 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_4 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_5 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_6 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_7 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_8 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_9 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_10 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_11 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_12 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_13 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_14 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_15 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_16 = require("@pulumi/azure-native/devcenter/v20241001preview");
const v20241001preview_17 = require("@pulumi/azure-native/devcenter/v20241001preview");
var AttachedNetworkByDevCenter;
(function (AttachedNetworkByDevCenter) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.AttachedNetworkByDevCenter).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AttachedNetworkByDevCenter.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-attachednetworkbydevcenter-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.AttachedNetworkByDevCenter).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_1.AttachedNetworkByDevCenter, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AttachedNetworkByDevCenter shouldn't use an unstable API (devcenter.v20241001preview.AttachedNetworkByDevCenter). A compatible replacement can be found at 'devcenter.AttachedNetworkByDevCenter'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AttachedNetworkByDevCenter || (AttachedNetworkByDevCenter = {}));
exports.AttachedNetworkByDevCenter = AttachedNetworkByDevCenter;
var Catalog;
(function (Catalog) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Catalog).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Catalog.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-catalog-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Catalog).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_2.Catalog, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Catalog shouldn't use an unstable API (devcenter.v20241001preview.Catalog). A compatible replacement can be found at 'devcenter.Catalog'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Catalog || (Catalog = {}));
exports.Catalog = Catalog;
var CurationProfile;
(function (CurationProfile) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.CurationProfile).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CurationProfile.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-curationprofile-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.CurationProfile).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_3.CurationProfile, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CurationProfile shouldn't use an unstable API (devcenter.v20241001preview.CurationProfile). A compatible replacement can be found at 'devcenter.CurationProfile'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CurationProfile || (CurationProfile = {}));
exports.CurationProfile = CurationProfile;
var DevBoxDefinition;
(function (DevBoxDefinition) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.DevBoxDefinition).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DevBoxDefinition.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-devboxdefinition-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.DevBoxDefinition).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_4.DevBoxDefinition, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DevBoxDefinition shouldn't use an unstable API (devcenter.v20241001preview.DevBoxDefinition). A compatible replacement can be found at 'devcenter.DevBoxDefinition'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DevBoxDefinition || (DevBoxDefinition = {}));
exports.DevBoxDefinition = DevBoxDefinition;
var DevCenter;
(function (DevCenter) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.DevCenter).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DevCenter.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-devcenter-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.DevCenter).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_5.DevCenter, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DevCenter shouldn't use an unstable API (devcenter.v20241001preview.DevCenter). A compatible replacement can be found at 'devcenter.DevCenter'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DevCenter || (DevCenter = {}));
exports.DevCenter = DevCenter;
var EncryptionSet;
(function (EncryptionSet) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.EncryptionSet).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EncryptionSet.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-encryptionset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.EncryptionSet).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_6.EncryptionSet, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EncryptionSet shouldn't use an unstable API (devcenter.v20241001preview.EncryptionSet). A compatible replacement can be found at 'devcenter.EncryptionSet'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EncryptionSet || (EncryptionSet = {}));
exports.EncryptionSet = EncryptionSet;
var EnvironmentType;
(function (EnvironmentType) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.EnvironmentType).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EnvironmentType.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-environmenttype-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.EnvironmentType).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_7.EnvironmentType, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EnvironmentType shouldn't use an unstable API (devcenter.v20241001preview.EnvironmentType). A compatible replacement can be found at 'devcenter.EnvironmentType'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EnvironmentType || (EnvironmentType = {}));
exports.EnvironmentType = EnvironmentType;
var Gallery;
(function (Gallery) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Gallery).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Gallery.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-gallery-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Gallery).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_8.Gallery, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Gallery shouldn't use an unstable API (devcenter.v20241001preview.Gallery). A compatible replacement can be found at 'devcenter.Gallery'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Gallery || (Gallery = {}));
exports.Gallery = Gallery;
var NetworkConnection;
(function (NetworkConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.NetworkConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    NetworkConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-networkconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.NetworkConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_9.NetworkConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure NetworkConnection shouldn't use an unstable API (devcenter.v20241001preview.NetworkConnection). A compatible replacement can be found at 'devcenter.NetworkConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(NetworkConnection || (NetworkConnection = {}));
exports.NetworkConnection = NetworkConnection;
var Plan;
(function (Plan) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Plan).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Plan.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-plan-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Plan).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_10.Plan, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Plan shouldn't use an unstable API (devcenter.v20241001preview.Plan). A compatible replacement can be found at 'devcenter.Plan'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Plan || (Plan = {}));
exports.Plan = Plan;
var PlanMember;
(function (PlanMember) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.PlanMember).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PlanMember.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-planmember-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.PlanMember).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_11.PlanMember, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PlanMember shouldn't use an unstable API (devcenter.v20241001preview.PlanMember). A compatible replacement can be found at 'devcenter.PlanMember'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PlanMember || (PlanMember = {}));
exports.PlanMember = PlanMember;
var Pool;
(function (Pool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Pool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Pool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-pool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Pool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_12.Pool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Pool shouldn't use an unstable API (devcenter.v20241001preview.Pool). A compatible replacement can be found at 'devcenter.Pool'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Pool || (Pool = {}));
exports.Pool = Pool;
var Project;
(function (Project) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Project).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Project.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-project-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Project).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_13.Project, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Project shouldn't use an unstable API (devcenter.v20241001preview.Project). A compatible replacement can be found at 'devcenter.Project'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Project || (Project = {}));
exports.Project = Project;
var ProjectCatalog;
(function (ProjectCatalog) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.ProjectCatalog).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProjectCatalog.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-projectcatalog-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.ProjectCatalog).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_14.ProjectCatalog, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProjectCatalog shouldn't use an unstable API (devcenter.v20241001preview.ProjectCatalog). A compatible replacement can be found at 'devcenter.ProjectCatalog'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProjectCatalog || (ProjectCatalog = {}));
exports.ProjectCatalog = ProjectCatalog;
var ProjectEnvironmentType;
(function (ProjectEnvironmentType) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.ProjectEnvironmentType).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProjectEnvironmentType.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-projectenvironmenttype-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.ProjectEnvironmentType).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_15.ProjectEnvironmentType, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProjectEnvironmentType shouldn't use an unstable API (devcenter.v20241001preview.ProjectEnvironmentType). A compatible replacement can be found at 'devcenter.ProjectEnvironmentType'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProjectEnvironmentType || (ProjectEnvironmentType = {}));
exports.ProjectEnvironmentType = ProjectEnvironmentType;
var ProjectPolicy;
(function (ProjectPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.ProjectPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProjectPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-projectpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.ProjectPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_16.ProjectPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProjectPolicy shouldn't use an unstable API (devcenter.v20241001preview.ProjectPolicy). A compatible replacement can be found at 'devcenter.ProjectPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProjectPolicy || (ProjectPolicy = {}));
exports.ProjectPolicy = ProjectPolicy;
var Schedule;
(function (Schedule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Schedule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Schedule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devcenter-v20241001preview-schedule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devcenter.v20241001preview.Schedule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_17.Schedule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Schedule shouldn't use an unstable API (devcenter.v20241001preview.Schedule). A compatible replacement can be found at 'devcenter.Schedule'.");
            }),
        },
        vendors: ["azure"],
        services: ["devcenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Schedule || (Schedule = {}));
exports.Schedule = Schedule;
