// @pulumi/azure/containerapp/customDomain.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Container App Custom Domain.
 *
 * ### Managed Certificate
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as std from "@pulumi/std";
 *
 * const example = new azure.containerapp.CustomDomain("example", {
 *     name: std.trimprefix({
 *         input: api.fqdn,
 *         prefix: "asuid.",
 *     }).then(invoke => std.trimsuffix({
 *         input: invoke.result,
 *         suffix: ".",
 *     })).then(invoke => invoke.result),
 *     containerAppId: exampleAzurermContainerApp.id,
 * });
 * ```
 *
 * ## Import
 *
 * A Container App Custom Domain can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:containerapp/customDomain:CustomDomain example "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.App/containerApps/myContainerApp/customDomainName/mycustomdomain.example.com"
 * ```
 */
class CustomDomain extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["certificateBindingType"] = state ? state.certificateBindingType : undefined;
            resourceInputs["containerAppEnvironmentCertificateId"] = state ? state.containerAppEnvironmentCertificateId : undefined;
            resourceInputs["containerAppEnvironmentManagedCertificateId"] = state ? state.containerAppEnvironmentManagedCertificateId : undefined;
            resourceInputs["containerAppId"] = state ? state.containerAppId : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.containerAppId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerAppId'");
            }
            resourceInputs["certificateBindingType"] = args ? args.certificateBindingType : undefined;
            resourceInputs["containerAppEnvironmentCertificateId"] = args ? args.containerAppEnvironmentCertificateId : undefined;
            resourceInputs["containerAppId"] = args ? args.containerAppId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["containerAppEnvironmentManagedCertificateId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CustomDomain.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing CustomDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CustomDomain(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomDomain.__pulumiType;
    }
}
exports.CustomDomain = CustomDomain;
/** @internal */
CustomDomain.__pulumiType = 'azure:containerapp/customDomain:CustomDomain';
