// @pulumi/azure/containerapp/environmentCertificate.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentCertificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Container App Environment Certificate.
 *
 * ## Import
 *
 * A Container App Environment Certificate can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:containerapp/environmentCertificate:EnvironmentCertificate example "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.App/managedEnvironments/myenv/certificates/mycertificate"
 * ```
 */
class EnvironmentCertificate extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["certificateBlobBase64"] = state ? state.certificateBlobBase64 : undefined;
            resourceInputs["certificatePassword"] = state ? state.certificatePassword : undefined;
            resourceInputs["containerAppEnvironmentId"] = state ? state.containerAppEnvironmentId : undefined;
            resourceInputs["expirationDate"] = state ? state.expirationDate : undefined;
            resourceInputs["issueDate"] = state ? state.issueDate : undefined;
            resourceInputs["issuer"] = state ? state.issuer : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["subjectName"] = state ? state.subjectName : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["thumbprint"] = state ? state.thumbprint : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.certificateBlobBase64 === undefined) && !opts.urn) {
                throw new Error("Missing required property 'certificateBlobBase64'");
            }
            if ((!args || args.certificatePassword === undefined) && !opts.urn) {
                throw new Error("Missing required property 'certificatePassword'");
            }
            if ((!args || args.containerAppEnvironmentId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerAppEnvironmentId'");
            }
            resourceInputs["certificateBlobBase64"] = args ? args.certificateBlobBase64 : undefined;
            resourceInputs["certificatePassword"] = (args === null || args === void 0 ? void 0 : args.certificatePassword) ? pulumi.secret(args.certificatePassword) : undefined;
            resourceInputs["containerAppEnvironmentId"] = args ? args.containerAppEnvironmentId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["expirationDate"] = undefined /*out*/;
            resourceInputs["issueDate"] = undefined /*out*/;
            resourceInputs["issuer"] = undefined /*out*/;
            resourceInputs["subjectName"] = undefined /*out*/;
            resourceInputs["thumbprint"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["certificatePassword"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(EnvironmentCertificate.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing EnvironmentCertificate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EnvironmentCertificate(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EnvironmentCertificate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnvironmentCertificate.__pulumiType;
    }
}
exports.EnvironmentCertificate = EnvironmentCertificate;
/** @internal */
EnvironmentCertificate.__pulumiType = 'azure:containerapp/environmentCertificate:EnvironmentCertificate';
