// @pulumi/azure/containerapp/environment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Environment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Container App Environment.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleAnalyticsWorkspace = new azure.operationalinsights.AnalyticsWorkspace("example", {
 *     name: "acctest-01",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     sku: "PerGB2018",
 *     retentionInDays: 30,
 * });
 * const exampleEnvironment = new azure.containerapp.Environment("example", {
 *     name: "my-environment",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     logsDestination: "log-analytics",
 *     logAnalyticsWorkspaceId: exampleAnalyticsWorkspace.id,
 * });
 * ```
 *
 * ## Import
 *
 * A Container App Environment can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:containerapp/environment:Environment example "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.App/managedEnvironments/myEnvironment"
 * ```
 */
class Environment extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["customDomainVerificationId"] = state ? state.customDomainVerificationId : undefined;
            resourceInputs["daprApplicationInsightsConnectionString"] = state ? state.daprApplicationInsightsConnectionString : undefined;
            resourceInputs["defaultDomain"] = state ? state.defaultDomain : undefined;
            resourceInputs["dockerBridgeCidr"] = state ? state.dockerBridgeCidr : undefined;
            resourceInputs["infrastructureResourceGroupName"] = state ? state.infrastructureResourceGroupName : undefined;
            resourceInputs["infrastructureSubnetId"] = state ? state.infrastructureSubnetId : undefined;
            resourceInputs["internalLoadBalancerEnabled"] = state ? state.internalLoadBalancerEnabled : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["logAnalyticsWorkspaceId"] = state ? state.logAnalyticsWorkspaceId : undefined;
            resourceInputs["logsDestination"] = state ? state.logsDestination : undefined;
            resourceInputs["mutualTlsEnabled"] = state ? state.mutualTlsEnabled : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["platformReservedCidr"] = state ? state.platformReservedCidr : undefined;
            resourceInputs["platformReservedDnsIpAddress"] = state ? state.platformReservedDnsIpAddress : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["staticIpAddress"] = state ? state.staticIpAddress : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["workloadProfiles"] = state ? state.workloadProfiles : undefined;
            resourceInputs["zoneRedundancyEnabled"] = state ? state.zoneRedundancyEnabled : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["daprApplicationInsightsConnectionString"] = (args === null || args === void 0 ? void 0 : args.daprApplicationInsightsConnectionString) ? pulumi.secret(args.daprApplicationInsightsConnectionString) : undefined;
            resourceInputs["infrastructureResourceGroupName"] = args ? args.infrastructureResourceGroupName : undefined;
            resourceInputs["infrastructureSubnetId"] = args ? args.infrastructureSubnetId : undefined;
            resourceInputs["internalLoadBalancerEnabled"] = args ? args.internalLoadBalancerEnabled : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["logAnalyticsWorkspaceId"] = args ? args.logAnalyticsWorkspaceId : undefined;
            resourceInputs["logsDestination"] = args ? args.logsDestination : undefined;
            resourceInputs["mutualTlsEnabled"] = args ? args.mutualTlsEnabled : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["workloadProfiles"] = args ? args.workloadProfiles : undefined;
            resourceInputs["zoneRedundancyEnabled"] = args ? args.zoneRedundancyEnabled : undefined;
            resourceInputs["customDomainVerificationId"] = undefined /*out*/;
            resourceInputs["defaultDomain"] = undefined /*out*/;
            resourceInputs["dockerBridgeCidr"] = undefined /*out*/;
            resourceInputs["platformReservedCidr"] = undefined /*out*/;
            resourceInputs["platformReservedDnsIpAddress"] = undefined /*out*/;
            resourceInputs["staticIpAddress"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["daprApplicationInsightsConnectionString"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(Environment.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Environment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Environment(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Environment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Environment.__pulumiType;
    }
}
exports.Environment = Environment;
/** @internal */
Environment.__pulumiType = 'azure:containerapp/environment:Environment';
