// @pulumi/azure/containerapp/job.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Job = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Container App Job.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleAnalyticsWorkspace = new azure.operationalinsights.AnalyticsWorkspace("example", {
 *     name: "example-log-analytics-workspace",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     sku: "PerGB2018",
 *     retentionInDays: 30,
 * });
 * const exampleEnvironment = new azure.containerapp.Environment("example", {
 *     name: "example-container-app-environment",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     logAnalyticsWorkspaceId: exampleAnalyticsWorkspace.id,
 * });
 * const exampleJob = new azure.containerapp.Job("example", {
 *     name: "example-container-app-job",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     containerAppEnvironmentId: exampleEnvironment.id,
 *     replicaTimeoutInSeconds: 10,
 *     replicaRetryLimit: 10,
 *     manualTriggerConfig: {
 *         parallelism: 4,
 *         replicaCompletionCount: 1,
 *     },
 *     template: {
 *         containers: [{
 *             image: "repo/testcontainerAppsJob0:v1",
 *             name: "testcontainerappsjob0",
 *             readinessProbes: [{
 *                 transport: "HTTP",
 *                 port: 5000,
 *             }],
 *             livenessProbes: [{
 *                 transport: "HTTP",
 *                 port: 5000,
 *                 path: "/health",
 *                 headers: [{
 *                     name: "Cache-Control",
 *                     value: "no-cache",
 *                 }],
 *                 initialDelay: 5,
 *                 intervalSeconds: 20,
 *                 timeout: 2,
 *                 failureCountThreshold: 1,
 *             }],
 *             startupProbes: [{
 *                 transport: "TCP",
 *                 port: 5000,
 *             }],
 *             cpu: 0.5,
 *             memory: "1Gi",
 *         }],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * A Container App Job can be imported using the resource id, e.g.
 *
 * ```sh
 * $ pulumi import azure:containerapp/job:Job example "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/example-resources/providers/Microsoft.App/jobs/example-container-app-job"
 * ```
 */
class Job extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["containerAppEnvironmentId"] = state ? state.containerAppEnvironmentId : undefined;
            resourceInputs["eventStreamEndpoint"] = state ? state.eventStreamEndpoint : undefined;
            resourceInputs["eventTriggerConfig"] = state ? state.eventTriggerConfig : undefined;
            resourceInputs["identity"] = state ? state.identity : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["manualTriggerConfig"] = state ? state.manualTriggerConfig : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["outboundIpAddresses"] = state ? state.outboundIpAddresses : undefined;
            resourceInputs["registries"] = state ? state.registries : undefined;
            resourceInputs["replicaRetryLimit"] = state ? state.replicaRetryLimit : undefined;
            resourceInputs["replicaTimeoutInSeconds"] = state ? state.replicaTimeoutInSeconds : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["scheduleTriggerConfig"] = state ? state.scheduleTriggerConfig : undefined;
            resourceInputs["secrets"] = state ? state.secrets : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["template"] = state ? state.template : undefined;
            resourceInputs["workloadProfileName"] = state ? state.workloadProfileName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.containerAppEnvironmentId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerAppEnvironmentId'");
            }
            if ((!args || args.replicaTimeoutInSeconds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'replicaTimeoutInSeconds'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.template === undefined) && !opts.urn) {
                throw new Error("Missing required property 'template'");
            }
            resourceInputs["containerAppEnvironmentId"] = args ? args.containerAppEnvironmentId : undefined;
            resourceInputs["eventTriggerConfig"] = args ? args.eventTriggerConfig : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["manualTriggerConfig"] = args ? args.manualTriggerConfig : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["registries"] = args ? args.registries : undefined;
            resourceInputs["replicaRetryLimit"] = args ? args.replicaRetryLimit : undefined;
            resourceInputs["replicaTimeoutInSeconds"] = args ? args.replicaTimeoutInSeconds : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scheduleTriggerConfig"] = args ? args.scheduleTriggerConfig : undefined;
            resourceInputs["secrets"] = (args === null || args === void 0 ? void 0 : args.secrets) ? pulumi.secret(args.secrets) : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["template"] = args ? args.template : undefined;
            resourceInputs["workloadProfileName"] = args ? args.workloadProfileName : undefined;
            resourceInputs["eventStreamEndpoint"] = undefined /*out*/;
            resourceInputs["outboundIpAddresses"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["secrets"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(Job.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Job resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Job(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Job.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Job.__pulumiType;
    }
}
exports.Job = Job;
/** @internal */
Job.__pulumiType = 'azure:containerapp/job:Job';
