// @pulumi/azure-compliance-policies/azurenative/synapse/v20210401preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.KustoPool = exports.DatabasePrincipalAssignment = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20210401preview_1 = require("@pulumi/azure-native/synapse/v20210401preview");
const v20210401preview_2 = require("@pulumi/azure-native/synapse/v20210401preview");
var DatabasePrincipalAssignment;
(function (DatabasePrincipalAssignment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (synapse.v20210401preview.DatabasePrincipalAssignment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DatabasePrincipalAssignment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-synapse-v20210401preview-databaseprincipalassignment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (synapse.v20210401preview.DatabasePrincipalAssignment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210401preview_1.DatabasePrincipalAssignment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DatabasePrincipalAssignment shouldn't use an unstable API (synapse.v20210401preview.DatabasePrincipalAssignment). A compatible replacement can be found at 'synapse.DatabasePrincipalAssignment'.");
            }),
        },
        vendors: ["azure"],
        services: ["synapse"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DatabasePrincipalAssignment || (DatabasePrincipalAssignment = {}));
exports.DatabasePrincipalAssignment = DatabasePrincipalAssignment;
var KustoPool;
(function (KustoPool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (synapse.v20210401preview.KustoPool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    KustoPool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-synapse-v20210401preview-kustopool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (synapse.v20210401preview.KustoPool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210401preview_2.KustoPool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure KustoPool shouldn't use an unstable API (synapse.v20210401preview.KustoPool). A compatible replacement can be found at 'synapse.KustoPool'.");
            }),
        },
        vendors: ["azure"],
        services: ["synapse"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(KustoPool || (KustoPool = {}));
exports.KustoPool = KustoPool;
