// @pulumi/azure/lb/backendAddressPoolAddress.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackendAddressPoolAddress = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Backend Address within a Backend Address Pool.
 *
 * > **Note:** Backend Addresses can only be added to a `Standard` SKU Load Balancer.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = azure.network.getVirtualNetwork({
 *     name: "example-network",
 *     resourceGroupName: "example-resources",
 * });
 * const exampleGetLB = azure.lb.getLB({
 *     name: "example-lb",
 *     resourceGroupName: "example-resources",
 * });
 * const exampleGetBackendAddressPool = exampleGetLB.then(exampleGetLB => azure.lb.getBackendAddressPool({
 *     name: "first",
 *     loadbalancerId: exampleGetLB.id,
 * }));
 * const exampleBackendAddressPoolAddress = new azure.lb.BackendAddressPoolAddress("example", {
 *     name: "example",
 *     backendAddressPoolId: exampleGetBackendAddressPool.then(exampleGetBackendAddressPool => exampleGetBackendAddressPool.id),
 *     virtualNetworkId: example.then(example => example.id),
 *     ipAddress: "10.0.0.1",
 * });
 * const backend-pool-cr = exampleGetLB.then(exampleGetLB => azure.lb.getBackendAddressPool({
 *     name: "globalLBBackendPool",
 *     loadbalancerId: exampleGetLB.id,
 * }));
 * const example_1 = new azure.lb.BackendAddressPoolAddress("example-1", {
 *     name: "address1",
 *     backendAddressPoolId: backend_pool_cr.then(backend_pool_cr => backend_pool_cr.id),
 *     backendAddressIpConfigurationId: backend_lb_R1.frontendIpConfiguration[0].id,
 * });
 * const example_2 = new azure.lb.BackendAddressPoolAddress("example-2", {
 *     name: "address2",
 *     backendAddressPoolId: backend_pool_cr.then(backend_pool_cr => backend_pool_cr.id),
 *     backendAddressIpConfigurationId: backend_lb_R2.frontendIpConfiguration[0].id,
 * });
 * ```
 *
 * ## Import
 *
 * Backend Address Pool Addresses can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:lb/backendAddressPoolAddress:BackendAddressPoolAddress example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Network/loadBalancers/loadBalancer1/backendAddressPools/backendAddressPool1/addresses/address1
 * ```
 */
class BackendAddressPoolAddress extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["backendAddressIpConfigurationId"] = state ? state.backendAddressIpConfigurationId : undefined;
            resourceInputs["backendAddressPoolId"] = state ? state.backendAddressPoolId : undefined;
            resourceInputs["inboundNatRulePortMappings"] = state ? state.inboundNatRulePortMappings : undefined;
            resourceInputs["ipAddress"] = state ? state.ipAddress : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["virtualNetworkId"] = state ? state.virtualNetworkId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.backendAddressPoolId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backendAddressPoolId'");
            }
            resourceInputs["backendAddressIpConfigurationId"] = args ? args.backendAddressIpConfigurationId : undefined;
            resourceInputs["backendAddressPoolId"] = args ? args.backendAddressPoolId : undefined;
            resourceInputs["ipAddress"] = args ? args.ipAddress : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["virtualNetworkId"] = args ? args.virtualNetworkId : undefined;
            resourceInputs["inboundNatRulePortMappings"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BackendAddressPoolAddress.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing BackendAddressPoolAddress resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BackendAddressPoolAddress(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BackendAddressPoolAddress.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BackendAddressPoolAddress.__pulumiType;
    }
}
exports.BackendAddressPoolAddress = BackendAddressPoolAddress;
/** @internal */
BackendAddressPoolAddress.__pulumiType = 'azure:lb/backendAddressPoolAddress:BackendAddressPoolAddress';
