// @pulumi/azure/lb/natRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NatRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Load Balancer NAT Rule.
 *
 * > **NOTE:** This resource cannot be used with with virtual machine scale sets, instead use the `azure.lb.NatPool` resource.
 *
 * > **NOTE** When using this resource, the Load Balancer needs to have a FrontEnd IP Configuration Attached
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "LoadBalancerRG",
 *     location: "West Europe",
 * });
 * const examplePublicIp = new azure.network.PublicIp("example", {
 *     name: "PublicIPForLB",
 *     location: "West US",
 *     resourceGroupName: example.name,
 *     allocationMethod: "Static",
 * });
 * const exampleLoadBalancer = new azure.lb.LoadBalancer("example", {
 *     name: "TestLoadBalancer",
 *     location: "West US",
 *     resourceGroupName: example.name,
 *     frontendIpConfigurations: [{
 *         name: "PublicIPAddress",
 *         publicIpAddressId: examplePublicIp.id,
 *     }],
 * });
 * const exampleBackendAddressPool = new azure.lb.BackendAddressPool("example", {
 *     loadbalancerId: exampleLoadBalancer.id,
 *     name: "be",
 * });
 * const exampleNatRule = new azure.lb.NatRule("example", {
 *     resourceGroupName: example.name,
 *     loadbalancerId: exampleLoadBalancer.id,
 *     name: "RDPAccess",
 *     protocol: "Tcp",
 *     frontendPort: 3389,
 *     backendPort: 3389,
 *     frontendIpConfigurationName: "PublicIPAddress",
 * });
 * const example1 = new azure.lb.NatRule("example1", {
 *     resourceGroupName: example.name,
 *     loadbalancerId: exampleLoadBalancer.id,
 *     name: "RDPAccess",
 *     protocol: "Tcp",
 *     frontendPortStart: 3000,
 *     frontendPortEnd: 3389,
 *     backendPort: 3389,
 *     backendAddressPoolId: exampleBackendAddressPool.id,
 *     frontendIpConfigurationName: "PublicIPAddress",
 * });
 * ```
 *
 * ## Import
 *
 * Load Balancer NAT Rules can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:lb/natRule:NatRule example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Network/loadBalancers/lb1/inboundNatRules/rule1
 * ```
 */
class NatRule extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["backendAddressPoolId"] = state ? state.backendAddressPoolId : undefined;
            resourceInputs["backendIpConfigurationId"] = state ? state.backendIpConfigurationId : undefined;
            resourceInputs["backendPort"] = state ? state.backendPort : undefined;
            resourceInputs["enableFloatingIp"] = state ? state.enableFloatingIp : undefined;
            resourceInputs["enableTcpReset"] = state ? state.enableTcpReset : undefined;
            resourceInputs["frontendIpConfigurationId"] = state ? state.frontendIpConfigurationId : undefined;
            resourceInputs["frontendIpConfigurationName"] = state ? state.frontendIpConfigurationName : undefined;
            resourceInputs["frontendPort"] = state ? state.frontendPort : undefined;
            resourceInputs["frontendPortEnd"] = state ? state.frontendPortEnd : undefined;
            resourceInputs["frontendPortStart"] = state ? state.frontendPortStart : undefined;
            resourceInputs["idleTimeoutInMinutes"] = state ? state.idleTimeoutInMinutes : undefined;
            resourceInputs["loadbalancerId"] = state ? state.loadbalancerId : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["protocol"] = state ? state.protocol : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.backendPort === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backendPort'");
            }
            if ((!args || args.frontendIpConfigurationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'frontendIpConfigurationName'");
            }
            if ((!args || args.loadbalancerId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'loadbalancerId'");
            }
            if ((!args || args.protocol === undefined) && !opts.urn) {
                throw new Error("Missing required property 'protocol'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["backendAddressPoolId"] = args ? args.backendAddressPoolId : undefined;
            resourceInputs["backendPort"] = args ? args.backendPort : undefined;
            resourceInputs["enableFloatingIp"] = args ? args.enableFloatingIp : undefined;
            resourceInputs["enableTcpReset"] = args ? args.enableTcpReset : undefined;
            resourceInputs["frontendIpConfigurationName"] = args ? args.frontendIpConfigurationName : undefined;
            resourceInputs["frontendPort"] = args ? args.frontendPort : undefined;
            resourceInputs["frontendPortEnd"] = args ? args.frontendPortEnd : undefined;
            resourceInputs["frontendPortStart"] = args ? args.frontendPortStart : undefined;
            resourceInputs["idleTimeoutInMinutes"] = args ? args.idleTimeoutInMinutes : undefined;
            resourceInputs["loadbalancerId"] = args ? args.loadbalancerId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["protocol"] = args ? args.protocol : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["backendIpConfigurationId"] = undefined /*out*/;
            resourceInputs["frontendIpConfigurationId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NatRule.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NatRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NatRule(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NatRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NatRule.__pulumiType;
    }
}
exports.NatRule = NatRule;
/** @internal */
NatRule.__pulumiType = 'azure:lb/natRule:NatRule';
