// @pulumi/azure-compliance-policies/azurenative/maps/v20211201preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Creator = exports.Account = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20211201preview_1 = require("@pulumi/azure-native/maps/v20211201preview");
const v20211201preview_2 = require("@pulumi/azure-native/maps/v20211201preview");
var Account;
(function (Account) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (maps.v20211201preview.Account).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Account.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-maps-v20211201preview-account-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (maps.v20211201preview.Account).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20211201preview_1.Account, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Account shouldn't use an unstable API (maps.v20211201preview.Account). A compatible replacement can be found at 'maps.Account'.");
            }),
        },
        vendors: ["azure"],
        services: ["maps"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Account || (Account = {}));
exports.Account = Account;
var Creator;
(function (Creator) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (maps.v20211201preview.Creator).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Creator.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-maps-v20211201preview-creator-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (maps.v20211201preview.Creator).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20211201preview_2.Creator, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Creator shouldn't use an unstable API (maps.v20211201preview.Creator). A compatible replacement can be found at 'maps.Creator'.");
            }),
        },
        vendors: ["azure"],
        services: ["maps"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Creator || (Creator = {}));
exports.Creator = Creator;
