// @pulumi/azure-compliance-policies/azurenative/kubernetesruntime/v20231001preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageClass = exports.Service = exports.LoadBalancer = exports.BgpPeer = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20231001preview_1 = require("@pulumi/azure-native/kubernetesruntime/v20231001preview");
const v20231001preview_2 = require("@pulumi/azure-native/kubernetesruntime/v20231001preview");
const v20231001preview_3 = require("@pulumi/azure-native/kubernetesruntime/v20231001preview");
const v20231001preview_4 = require("@pulumi/azure-native/kubernetesruntime/v20231001preview");
var BgpPeer;
(function (BgpPeer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (kubernetesruntime.v20231001preview.BgpPeer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BgpPeer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-kubernetesruntime-v20231001preview-bgppeer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (kubernetesruntime.v20231001preview.BgpPeer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231001preview_1.BgpPeer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BgpPeer shouldn't use an unstable API (kubernetesruntime.v20231001preview.BgpPeer). A compatible replacement can be found at 'kubernetesruntime.BgpPeer'.");
            }),
        },
        vendors: ["azure"],
        services: ["kubernetesruntime"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BgpPeer || (BgpPeer = {}));
exports.BgpPeer = BgpPeer;
var LoadBalancer;
(function (LoadBalancer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (kubernetesruntime.v20231001preview.LoadBalancer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LoadBalancer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-kubernetesruntime-v20231001preview-loadbalancer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (kubernetesruntime.v20231001preview.LoadBalancer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231001preview_2.LoadBalancer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LoadBalancer shouldn't use an unstable API (kubernetesruntime.v20231001preview.LoadBalancer). A compatible replacement can be found at 'kubernetesruntime.LoadBalancer'.");
            }),
        },
        vendors: ["azure"],
        services: ["kubernetesruntime"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LoadBalancer || (LoadBalancer = {}));
exports.LoadBalancer = LoadBalancer;
var Service;
(function (Service) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (kubernetesruntime.v20231001preview.Service).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Service.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-kubernetesruntime-v20231001preview-service-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (kubernetesruntime.v20231001preview.Service).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231001preview_3.Service, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Service shouldn't use an unstable API (kubernetesruntime.v20231001preview.Service). A compatible replacement can be found at 'kubernetesruntime.Service'.");
            }),
        },
        vendors: ["azure"],
        services: ["kubernetesruntime"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Service || (Service = {}));
exports.Service = Service;
var StorageClass;
(function (StorageClass) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (kubernetesruntime.v20231001preview.StorageClass).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    StorageClass.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-kubernetesruntime-v20231001preview-storageclass-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (kubernetesruntime.v20231001preview.StorageClass).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231001preview_4.StorageClass, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure StorageClass shouldn't use an unstable API (kubernetesruntime.v20231001preview.StorageClass). A compatible replacement can be found at 'kubernetesruntime.StorageClass'.");
            }),
        },
        vendors: ["azure"],
        services: ["kubernetesruntime"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(StorageClass || (StorageClass = {}));
exports.StorageClass = StorageClass;
