// @pulumi/azure-compliance-policies/azurenative/dashboard/v20231001preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnection = exports.ManagedPrivateEndpoint = exports.IntegrationFabric = exports.Grafana = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20231001preview_1 = require("@pulumi/azure-native/dashboard/v20231001preview");
const v20231001preview_2 = require("@pulumi/azure-native/dashboard/v20231001preview");
const v20231001preview_3 = require("@pulumi/azure-native/dashboard/v20231001preview");
const v20231001preview_4 = require("@pulumi/azure-native/dashboard/v20231001preview");
var Grafana;
(function (Grafana) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dashboard.v20231001preview.Grafana).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Grafana.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dashboard-v20231001preview-grafana-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dashboard.v20231001preview.Grafana).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231001preview_1.Grafana, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Grafana shouldn't use an unstable API (dashboard.v20231001preview.Grafana). A compatible replacement can be found at 'dashboard.Grafana'.");
            }),
        },
        vendors: ["azure"],
        services: ["dashboard"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Grafana || (Grafana = {}));
exports.Grafana = Grafana;
var IntegrationFabric;
(function (IntegrationFabric) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dashboard.v20231001preview.IntegrationFabric).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IntegrationFabric.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dashboard-v20231001preview-integrationfabric-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dashboard.v20231001preview.IntegrationFabric).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231001preview_2.IntegrationFabric, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IntegrationFabric shouldn't use an unstable API (dashboard.v20231001preview.IntegrationFabric). A compatible replacement can be found at 'dashboard.IntegrationFabric'.");
            }),
        },
        vendors: ["azure"],
        services: ["dashboard"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IntegrationFabric || (IntegrationFabric = {}));
exports.IntegrationFabric = IntegrationFabric;
var ManagedPrivateEndpoint;
(function (ManagedPrivateEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dashboard.v20231001preview.ManagedPrivateEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagedPrivateEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dashboard-v20231001preview-managedprivateendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dashboard.v20231001preview.ManagedPrivateEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231001preview_3.ManagedPrivateEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagedPrivateEndpoint shouldn't use an unstable API (dashboard.v20231001preview.ManagedPrivateEndpoint). A compatible replacement can be found at 'dashboard.ManagedPrivateEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["dashboard"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagedPrivateEndpoint || (ManagedPrivateEndpoint = {}));
exports.ManagedPrivateEndpoint = ManagedPrivateEndpoint;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dashboard.v20231001preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dashboard-v20231001preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dashboard.v20231001preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231001preview_4.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (dashboard.v20231001preview.PrivateEndpointConnection). A compatible replacement can be found at 'dashboard.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["dashboard"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
