// @pulumi/azure-native/connectedvmwarevsphere/inventoryItem.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InventoryItem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Defines the inventory item.
 * Azure REST API version: 2022-07-15-preview. Prior API version in Azure Native 1.x: 2020-10-01-preview.
 *
 * Other available API versions: 2023-03-01-preview, 2023-10-01, 2023-12-01.
 */
class InventoryItem extends pulumi.CustomResource {
    /**
     * Get an existing InventoryItem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new InventoryItem(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of InventoryItem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InventoryItem.__pulumiType;
    }
    /**
     * Create a InventoryItem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.inventoryType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inventoryType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vcenterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vcenterName'");
            }
            resourceInputs["inventoryItemName"] = args ? args.inventoryItemName : undefined;
            resourceInputs["inventoryType"] = args ? args.inventoryType : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["managedResourceId"] = args ? args.managedResourceId : undefined;
            resourceInputs["moName"] = args ? args.moName : undefined;
            resourceInputs["moRefId"] = args ? args.moRefId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["vcenterName"] = args ? args.vcenterName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["inventoryType"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["managedResourceId"] = undefined /*out*/;
            resourceInputs["moName"] = undefined /*out*/;
            resourceInputs["moRefId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:connectedvmwarevsphere/v20201001preview:InventoryItem" }, { type: "azure-native:connectedvmwarevsphere/v20220110preview:InventoryItem" }, { type: "azure-native:connectedvmwarevsphere/v20220715preview:InventoryItem" }, { type: "azure-native:connectedvmwarevsphere/v20230301preview:InventoryItem" }, { type: "azure-native:connectedvmwarevsphere/v20231001:InventoryItem" }, { type: "azure-native:connectedvmwarevsphere/v20231201:InventoryItem" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(InventoryItem.__pulumiType, name, resourceInputs, opts);
    }
}
exports.InventoryItem = InventoryItem;
/** @internal */
InventoryItem.__pulumiType = 'azure-native:connectedvmwarevsphere:InventoryItem';
