// @pulumi/azure-compliance-policies/azurenative/containerservice/v20240202preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRun = exports.TrustedAccessRoleBinding = exports.Snapshot = exports.PrivateEndpointConnection = exports.ManagedClusterSnapshot = exports.ManagedCluster = exports.MaintenanceConfiguration = exports.FleetUpdateStrategy = exports.FleetMember = exports.Fleet = exports.AgentPool = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240202preview_1 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_2 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_3 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_4 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_5 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_6 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_7 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_8 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_9 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_10 = require("@pulumi/azure-native/containerservice/v20240202preview");
const v20240202preview_11 = require("@pulumi/azure-native/containerservice/v20240202preview");
var AgentPool;
(function (AgentPool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.AgentPool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AgentPool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-agentpool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.AgentPool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_1.AgentPool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AgentPool shouldn't use an unstable API (containerservice.v20240202preview.AgentPool). A compatible replacement can be found at 'containerservice.AgentPool'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AgentPool || (AgentPool = {}));
exports.AgentPool = AgentPool;
var Fleet;
(function (Fleet) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.Fleet).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Fleet.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-fleet-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.Fleet).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_2.Fleet, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Fleet shouldn't use an unstable API (containerservice.v20240202preview.Fleet). A compatible replacement can be found at 'containerservice.Fleet'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Fleet || (Fleet = {}));
exports.Fleet = Fleet;
var FleetMember;
(function (FleetMember) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.FleetMember).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FleetMember.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-fleetmember-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.FleetMember).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_3.FleetMember, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FleetMember shouldn't use an unstable API (containerservice.v20240202preview.FleetMember). A compatible replacement can be found at 'containerservice.FleetMember'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FleetMember || (FleetMember = {}));
exports.FleetMember = FleetMember;
var FleetUpdateStrategy;
(function (FleetUpdateStrategy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.FleetUpdateStrategy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FleetUpdateStrategy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-fleetupdatestrategy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.FleetUpdateStrategy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_4.FleetUpdateStrategy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FleetUpdateStrategy shouldn't use an unstable API (containerservice.v20240202preview.FleetUpdateStrategy). A compatible replacement can be found at 'containerservice.FleetUpdateStrategy'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FleetUpdateStrategy || (FleetUpdateStrategy = {}));
exports.FleetUpdateStrategy = FleetUpdateStrategy;
var MaintenanceConfiguration;
(function (MaintenanceConfiguration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.MaintenanceConfiguration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MaintenanceConfiguration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-maintenanceconfiguration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.MaintenanceConfiguration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_5.MaintenanceConfiguration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MaintenanceConfiguration shouldn't use an unstable API (containerservice.v20240202preview.MaintenanceConfiguration). A compatible replacement can be found at 'containerservice.MaintenanceConfiguration'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MaintenanceConfiguration || (MaintenanceConfiguration = {}));
exports.MaintenanceConfiguration = MaintenanceConfiguration;
var ManagedCluster;
(function (ManagedCluster) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.ManagedCluster).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagedCluster.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-managedcluster-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.ManagedCluster).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_6.ManagedCluster, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagedCluster shouldn't use an unstable API (containerservice.v20240202preview.ManagedCluster). A compatible replacement can be found at 'containerservice.ManagedCluster'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagedCluster || (ManagedCluster = {}));
exports.ManagedCluster = ManagedCluster;
var ManagedClusterSnapshot;
(function (ManagedClusterSnapshot) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.ManagedClusterSnapshot).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagedClusterSnapshot.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-managedclustersnapshot-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.ManagedClusterSnapshot).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_7.ManagedClusterSnapshot, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagedClusterSnapshot shouldn't use an unstable API (containerservice.v20240202preview.ManagedClusterSnapshot). A compatible replacement can be found at 'containerservice.ManagedClusterSnapshot'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagedClusterSnapshot || (ManagedClusterSnapshot = {}));
exports.ManagedClusterSnapshot = ManagedClusterSnapshot;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_8.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (containerservice.v20240202preview.PrivateEndpointConnection). A compatible replacement can be found at 'containerservice.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
var Snapshot;
(function (Snapshot) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.Snapshot).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Snapshot.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-snapshot-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.Snapshot).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_9.Snapshot, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Snapshot shouldn't use an unstable API (containerservice.v20240202preview.Snapshot). A compatible replacement can be found at 'containerservice.Snapshot'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Snapshot || (Snapshot = {}));
exports.Snapshot = Snapshot;
var TrustedAccessRoleBinding;
(function (TrustedAccessRoleBinding) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.TrustedAccessRoleBinding).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TrustedAccessRoleBinding.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-trustedaccessrolebinding-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.TrustedAccessRoleBinding).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_10.TrustedAccessRoleBinding, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TrustedAccessRoleBinding shouldn't use an unstable API (containerservice.v20240202preview.TrustedAccessRoleBinding). A compatible replacement can be found at 'containerservice.TrustedAccessRoleBinding'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TrustedAccessRoleBinding || (TrustedAccessRoleBinding = {}));
exports.TrustedAccessRoleBinding = TrustedAccessRoleBinding;
var UpdateRun;
(function (UpdateRun) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.UpdateRun).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    UpdateRun.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerservice-v20240202preview-updaterun-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerservice.v20240202preview.UpdateRun).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240202preview_11.UpdateRun, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure UpdateRun shouldn't use an unstable API (containerservice.v20240202preview.UpdateRun). A compatible replacement can be found at 'containerservice.UpdateRun'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(UpdateRun || (UpdateRun = {}));
exports.UpdateRun = UpdateRun;
