// @pulumi/azure-compliance-policies/azurenative/managedidentity/v20230731preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAssignedIdentity = exports.FederatedIdentityCredential = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230731preview_1 = require("@pulumi/azure-native/managedidentity/v20230731preview");
const v20230731preview_2 = require("@pulumi/azure-native/managedidentity/v20230731preview");
var FederatedIdentityCredential;
(function (FederatedIdentityCredential) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (managedidentity.v20230731preview.FederatedIdentityCredential).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FederatedIdentityCredential.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-managedidentity-v20230731preview-federatedidentitycredential-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (managedidentity.v20230731preview.FederatedIdentityCredential).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230731preview_1.FederatedIdentityCredential, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FederatedIdentityCredential shouldn't use an unstable API (managedidentity.v20230731preview.FederatedIdentityCredential). A compatible replacement can be found at 'managedidentity.FederatedIdentityCredential'.");
            }),
        },
        vendors: ["azure"],
        services: ["managedidentity"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FederatedIdentityCredential || (FederatedIdentityCredential = {}));
exports.FederatedIdentityCredential = FederatedIdentityCredential;
var UserAssignedIdentity;
(function (UserAssignedIdentity) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (managedidentity.v20230731preview.UserAssignedIdentity).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    UserAssignedIdentity.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-managedidentity-v20230731preview-userassignedidentity-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (managedidentity.v20230731preview.UserAssignedIdentity).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230731preview_2.UserAssignedIdentity, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure UserAssignedIdentity shouldn't use an unstable API (managedidentity.v20230731preview.UserAssignedIdentity). A compatible replacement can be found at 'managedidentity.UserAssignedIdentity'.");
            }),
        },
        vendors: ["azure"],
        services: ["managedidentity"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(UserAssignedIdentity || (UserAssignedIdentity = {}));
exports.UserAssignedIdentity = UserAssignedIdentity;
