// @pulumi/azure-compliance-policies/azurenative/cloudngfw/v20240207preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrefixListLocalRulestack = exports.PrefixListGlobalRulestack = exports.PreRule = exports.PostRule = exports.LocalRulestack = exports.LocalRule = exports.GlobalRulestack = exports.FqdnListLocalRulestack = exports.FqdnListGlobalRulestack = exports.Firewall = exports.CertificateObjectLocalRulestack = exports.CertificateObjectGlobalRulestack = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240207preview_1 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_2 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_3 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_4 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_5 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_6 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_7 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_8 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_9 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_10 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_11 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
const v20240207preview_12 = require("@pulumi/azure-native/cloudngfw/v20240207preview");
var CertificateObjectGlobalRulestack;
(function (CertificateObjectGlobalRulestack) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.CertificateObjectGlobalRulestack).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CertificateObjectGlobalRulestack.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-certificateobjectglobalrulestack-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.CertificateObjectGlobalRulestack).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_1.CertificateObjectGlobalRulestack, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CertificateObjectGlobalRulestack shouldn't use an unstable API (cloudngfw.v20240207preview.CertificateObjectGlobalRulestack). A compatible replacement can be found at 'cloudngfw.CertificateObjectGlobalRulestack'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CertificateObjectGlobalRulestack || (CertificateObjectGlobalRulestack = {}));
exports.CertificateObjectGlobalRulestack = CertificateObjectGlobalRulestack;
var CertificateObjectLocalRulestack;
(function (CertificateObjectLocalRulestack) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.CertificateObjectLocalRulestack).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CertificateObjectLocalRulestack.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-certificateobjectlocalrulestack-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.CertificateObjectLocalRulestack).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_2.CertificateObjectLocalRulestack, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CertificateObjectLocalRulestack shouldn't use an unstable API (cloudngfw.v20240207preview.CertificateObjectLocalRulestack). A compatible replacement can be found at 'cloudngfw.CertificateObjectLocalRulestack'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CertificateObjectLocalRulestack || (CertificateObjectLocalRulestack = {}));
exports.CertificateObjectLocalRulestack = CertificateObjectLocalRulestack;
var Firewall;
(function (Firewall) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.Firewall).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Firewall.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-firewall-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.Firewall).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_3.Firewall, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Firewall shouldn't use an unstable API (cloudngfw.v20240207preview.Firewall). A compatible replacement can be found at 'cloudngfw.Firewall'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Firewall || (Firewall = {}));
exports.Firewall = Firewall;
var FqdnListGlobalRulestack;
(function (FqdnListGlobalRulestack) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.FqdnListGlobalRulestack).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FqdnListGlobalRulestack.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-fqdnlistglobalrulestack-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.FqdnListGlobalRulestack).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_4.FqdnListGlobalRulestack, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FqdnListGlobalRulestack shouldn't use an unstable API (cloudngfw.v20240207preview.FqdnListGlobalRulestack). A compatible replacement can be found at 'cloudngfw.FqdnListGlobalRulestack'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FqdnListGlobalRulestack || (FqdnListGlobalRulestack = {}));
exports.FqdnListGlobalRulestack = FqdnListGlobalRulestack;
var FqdnListLocalRulestack;
(function (FqdnListLocalRulestack) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.FqdnListLocalRulestack).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FqdnListLocalRulestack.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-fqdnlistlocalrulestack-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.FqdnListLocalRulestack).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_5.FqdnListLocalRulestack, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FqdnListLocalRulestack shouldn't use an unstable API (cloudngfw.v20240207preview.FqdnListLocalRulestack). A compatible replacement can be found at 'cloudngfw.FqdnListLocalRulestack'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FqdnListLocalRulestack || (FqdnListLocalRulestack = {}));
exports.FqdnListLocalRulestack = FqdnListLocalRulestack;
var GlobalRulestack;
(function (GlobalRulestack) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.GlobalRulestack).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GlobalRulestack.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-globalrulestack-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.GlobalRulestack).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_6.GlobalRulestack, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GlobalRulestack shouldn't use an unstable API (cloudngfw.v20240207preview.GlobalRulestack). A compatible replacement can be found at 'cloudngfw.GlobalRulestack'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GlobalRulestack || (GlobalRulestack = {}));
exports.GlobalRulestack = GlobalRulestack;
var LocalRule;
(function (LocalRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.LocalRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LocalRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-localrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.LocalRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_7.LocalRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LocalRule shouldn't use an unstable API (cloudngfw.v20240207preview.LocalRule). A compatible replacement can be found at 'cloudngfw.LocalRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LocalRule || (LocalRule = {}));
exports.LocalRule = LocalRule;
var LocalRulestack;
(function (LocalRulestack) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.LocalRulestack).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LocalRulestack.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-localrulestack-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.LocalRulestack).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_8.LocalRulestack, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LocalRulestack shouldn't use an unstable API (cloudngfw.v20240207preview.LocalRulestack). A compatible replacement can be found at 'cloudngfw.LocalRulestack'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LocalRulestack || (LocalRulestack = {}));
exports.LocalRulestack = LocalRulestack;
var PostRule;
(function (PostRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.PostRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PostRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-postrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.PostRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_9.PostRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PostRule shouldn't use an unstable API (cloudngfw.v20240207preview.PostRule). A compatible replacement can be found at 'cloudngfw.PostRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PostRule || (PostRule = {}));
exports.PostRule = PostRule;
var PreRule;
(function (PreRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.PreRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PreRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-prerule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.PreRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_10.PreRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PreRule shouldn't use an unstable API (cloudngfw.v20240207preview.PreRule). A compatible replacement can be found at 'cloudngfw.PreRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PreRule || (PreRule = {}));
exports.PreRule = PreRule;
var PrefixListGlobalRulestack;
(function (PrefixListGlobalRulestack) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.PrefixListGlobalRulestack).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrefixListGlobalRulestack.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-prefixlistglobalrulestack-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.PrefixListGlobalRulestack).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_11.PrefixListGlobalRulestack, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrefixListGlobalRulestack shouldn't use an unstable API (cloudngfw.v20240207preview.PrefixListGlobalRulestack). A compatible replacement can be found at 'cloudngfw.PrefixListGlobalRulestack'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrefixListGlobalRulestack || (PrefixListGlobalRulestack = {}));
exports.PrefixListGlobalRulestack = PrefixListGlobalRulestack;
var PrefixListLocalRulestack;
(function (PrefixListLocalRulestack) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.PrefixListLocalRulestack).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrefixListLocalRulestack.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cloudngfw-v20240207preview-prefixlistlocalrulestack-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cloudngfw.v20240207preview.PrefixListLocalRulestack).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240207preview_12.PrefixListLocalRulestack, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrefixListLocalRulestack shouldn't use an unstable API (cloudngfw.v20240207preview.PrefixListLocalRulestack). A compatible replacement can be found at 'cloudngfw.PrefixListLocalRulestack'.");
            }),
        },
        vendors: ["azure"],
        services: ["cloudngfw"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrefixListLocalRulestack || (PrefixListLocalRulestack = {}));
exports.PrefixListLocalRulestack = PrefixListLocalRulestack;
