// @pulumi/azure-compliance-policies/azurenative/security/v20210701preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityConnector = exports.CustomEntityStoreAssignment = exports.CustomAssessmentAutomation = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20210701preview_1 = require("@pulumi/azure-native/security/v20210701preview");
const v20210701preview_2 = require("@pulumi/azure-native/security/v20210701preview");
const v20210701preview_3 = require("@pulumi/azure-native/security/v20210701preview");
var CustomAssessmentAutomation;
(function (CustomAssessmentAutomation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20210701preview.CustomAssessmentAutomation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomAssessmentAutomation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20210701preview-customassessmentautomation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20210701preview.CustomAssessmentAutomation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210701preview_1.CustomAssessmentAutomation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomAssessmentAutomation shouldn't use an unstable API (security.v20210701preview.CustomAssessmentAutomation). A compatible replacement can be found at 'security.CustomAssessmentAutomation'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomAssessmentAutomation || (CustomAssessmentAutomation = {}));
exports.CustomAssessmentAutomation = CustomAssessmentAutomation;
var CustomEntityStoreAssignment;
(function (CustomEntityStoreAssignment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20210701preview.CustomEntityStoreAssignment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomEntityStoreAssignment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20210701preview-customentitystoreassignment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20210701preview.CustomEntityStoreAssignment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210701preview_2.CustomEntityStoreAssignment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomEntityStoreAssignment shouldn't use an unstable API (security.v20210701preview.CustomEntityStoreAssignment). A compatible replacement can be found at 'security.CustomEntityStoreAssignment'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomEntityStoreAssignment || (CustomEntityStoreAssignment = {}));
exports.CustomEntityStoreAssignment = CustomEntityStoreAssignment;
var SecurityConnector;
(function (SecurityConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20210701preview.SecurityConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SecurityConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20210701preview-securityconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20210701preview.SecurityConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210701preview_3.SecurityConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SecurityConnector shouldn't use an unstable API (security.v20210701preview.SecurityConnector). A compatible replacement can be found at 'security.SecurityConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SecurityConnector || (SecurityConnector = {}));
exports.SecurityConnector = SecurityConnector;
