// @pulumi/azure-compliance-policies/azurenative/securityinsights/v20190101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Watchlist = exports.IPSyncer = exports.BookmarkRelation = exports.Bookmark = exports.AutomationRule = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20190101preview_1 = require("@pulumi/azure-native/securityinsights/v20190101preview");
const v20190101preview_2 = require("@pulumi/azure-native/securityinsights/v20190101preview");
const v20190101preview_3 = require("@pulumi/azure-native/securityinsights/v20190101preview");
const v20190101preview_4 = require("@pulumi/azure-native/securityinsights/v20190101preview");
const v20190101preview_5 = require("@pulumi/azure-native/securityinsights/v20190101preview");
var AutomationRule;
(function (AutomationRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.AutomationRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AutomationRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20190101preview-automationrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.AutomationRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20190101preview_1.AutomationRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AutomationRule shouldn't use an unstable API (securityinsights.v20190101preview.AutomationRule). A compatible replacement can be found at 'securityinsights.AutomationRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AutomationRule || (AutomationRule = {}));
exports.AutomationRule = AutomationRule;
var Bookmark;
(function (Bookmark) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.Bookmark).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Bookmark.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20190101preview-bookmark-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.Bookmark).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20190101preview_2.Bookmark, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Bookmark shouldn't use an unstable API (securityinsights.v20190101preview.Bookmark). A compatible replacement can be found at 'securityinsights.Bookmark'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Bookmark || (Bookmark = {}));
exports.Bookmark = Bookmark;
var BookmarkRelation;
(function (BookmarkRelation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.BookmarkRelation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BookmarkRelation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20190101preview-bookmarkrelation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.BookmarkRelation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20190101preview_3.BookmarkRelation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BookmarkRelation shouldn't use an unstable API (securityinsights.v20190101preview.BookmarkRelation). A compatible replacement can be found at 'securityinsights.BookmarkRelation'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BookmarkRelation || (BookmarkRelation = {}));
exports.BookmarkRelation = BookmarkRelation;
var IPSyncer;
(function (IPSyncer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.IPSyncer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IPSyncer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20190101preview-ipsyncer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.IPSyncer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20190101preview_4.IPSyncer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IPSyncer shouldn't use an unstable API (securityinsights.v20190101preview.IPSyncer). A compatible replacement can be found at 'securityinsights.IPSyncer'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IPSyncer || (IPSyncer = {}));
exports.IPSyncer = IPSyncer;
var Watchlist;
(function (Watchlist) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.Watchlist).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Watchlist.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20190101preview-watchlist-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20190101preview.Watchlist).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20190101preview_5.Watchlist, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Watchlist shouldn't use an unstable API (securityinsights.v20190101preview.Watchlist). A compatible replacement can be found at 'securityinsights.Watchlist'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Watchlist || (Watchlist = {}));
exports.Watchlist = Watchlist;
