// @pulumi/azure-compliance-policies/azurenative/eventgrid/v20240601preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicSpace = exports.TopicEventSubscription = exports.Topic = exports.SystemTopicEventSubscription = exports.SystemTopic = exports.PrivateEndpointConnection = exports.PermissionBinding = exports.PartnerTopicEventSubscription = exports.PartnerTopic = exports.PartnerRegistration = exports.PartnerNamespace = exports.PartnerDestination = exports.PartnerConfiguration = exports.NamespaceTopicEventSubscription = exports.NamespaceTopic = exports.Namespace = exports.EventSubscription = exports.DomainTopicEventSubscription = exports.DomainTopic = exports.DomainEventSubscription = exports.Domain = exports.ClientGroup = exports.Client = exports.Channel = exports.CaCertificate = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240601preview_1 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_2 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_3 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_4 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_5 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_6 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_7 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_8 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_9 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_10 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_11 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_12 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_13 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_14 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_15 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_16 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_17 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_18 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_19 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_20 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_21 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_22 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_23 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_24 = require("@pulumi/azure-native/eventgrid/v20240601preview");
const v20240601preview_25 = require("@pulumi/azure-native/eventgrid/v20240601preview");
var CaCertificate;
(function (CaCertificate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.CaCertificate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CaCertificate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-cacertificate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.CaCertificate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_1.CaCertificate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CaCertificate shouldn't use an unstable API (eventgrid.v20240601preview.CaCertificate). A compatible replacement can be found at 'eventgrid.CaCertificate'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CaCertificate || (CaCertificate = {}));
exports.CaCertificate = CaCertificate;
var Channel;
(function (Channel) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Channel).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Channel.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-channel-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Channel).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_2.Channel, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Channel shouldn't use an unstable API (eventgrid.v20240601preview.Channel). A compatible replacement can be found at 'eventgrid.Channel'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Channel || (Channel = {}));
exports.Channel = Channel;
var Client;
(function (Client) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Client).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Client.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-client-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Client).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_3.Client, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Client shouldn't use an unstable API (eventgrid.v20240601preview.Client). A compatible replacement can be found at 'eventgrid.Client'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Client || (Client = {}));
exports.Client = Client;
var ClientGroup;
(function (ClientGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.ClientGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ClientGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-clientgroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.ClientGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_4.ClientGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ClientGroup shouldn't use an unstable API (eventgrid.v20240601preview.ClientGroup). A compatible replacement can be found at 'eventgrid.ClientGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ClientGroup || (ClientGroup = {}));
exports.ClientGroup = ClientGroup;
var Domain;
(function (Domain) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Domain).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Domain.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-domain-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Domain).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_5.Domain, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Domain shouldn't use an unstable API (eventgrid.v20240601preview.Domain). A compatible replacement can be found at 'eventgrid.Domain'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Domain || (Domain = {}));
exports.Domain = Domain;
var DomainEventSubscription;
(function (DomainEventSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.DomainEventSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DomainEventSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-domaineventsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.DomainEventSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_6.DomainEventSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DomainEventSubscription shouldn't use an unstable API (eventgrid.v20240601preview.DomainEventSubscription). A compatible replacement can be found at 'eventgrid.DomainEventSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DomainEventSubscription || (DomainEventSubscription = {}));
exports.DomainEventSubscription = DomainEventSubscription;
var DomainTopic;
(function (DomainTopic) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.DomainTopic).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DomainTopic.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-domaintopic-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.DomainTopic).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_7.DomainTopic, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DomainTopic shouldn't use an unstable API (eventgrid.v20240601preview.DomainTopic). A compatible replacement can be found at 'eventgrid.DomainTopic'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DomainTopic || (DomainTopic = {}));
exports.DomainTopic = DomainTopic;
var DomainTopicEventSubscription;
(function (DomainTopicEventSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.DomainTopicEventSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DomainTopicEventSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-domaintopiceventsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.DomainTopicEventSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_8.DomainTopicEventSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DomainTopicEventSubscription shouldn't use an unstable API (eventgrid.v20240601preview.DomainTopicEventSubscription). A compatible replacement can be found at 'eventgrid.DomainTopicEventSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DomainTopicEventSubscription || (DomainTopicEventSubscription = {}));
exports.DomainTopicEventSubscription = DomainTopicEventSubscription;
var EventSubscription;
(function (EventSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.EventSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EventSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-eventsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.EventSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_9.EventSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EventSubscription shouldn't use an unstable API (eventgrid.v20240601preview.EventSubscription). A compatible replacement can be found at 'eventgrid.EventSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EventSubscription || (EventSubscription = {}));
exports.EventSubscription = EventSubscription;
var Namespace;
(function (Namespace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Namespace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Namespace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-namespace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Namespace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_10.Namespace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Namespace shouldn't use an unstable API (eventgrid.v20240601preview.Namespace). A compatible replacement can be found at 'eventgrid.Namespace'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Namespace || (Namespace = {}));
exports.Namespace = Namespace;
var NamespaceTopic;
(function (NamespaceTopic) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.NamespaceTopic).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    NamespaceTopic.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-namespacetopic-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.NamespaceTopic).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_11.NamespaceTopic, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure NamespaceTopic shouldn't use an unstable API (eventgrid.v20240601preview.NamespaceTopic). A compatible replacement can be found at 'eventgrid.NamespaceTopic'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(NamespaceTopic || (NamespaceTopic = {}));
exports.NamespaceTopic = NamespaceTopic;
var NamespaceTopicEventSubscription;
(function (NamespaceTopicEventSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.NamespaceTopicEventSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    NamespaceTopicEventSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-namespacetopiceventsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.NamespaceTopicEventSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_12.NamespaceTopicEventSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure NamespaceTopicEventSubscription shouldn't use an unstable API (eventgrid.v20240601preview.NamespaceTopicEventSubscription). A compatible replacement can be found at 'eventgrid.NamespaceTopicEventSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(NamespaceTopicEventSubscription || (NamespaceTopicEventSubscription = {}));
exports.NamespaceTopicEventSubscription = NamespaceTopicEventSubscription;
var PartnerConfiguration;
(function (PartnerConfiguration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerConfiguration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PartnerConfiguration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-partnerconfiguration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerConfiguration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_13.PartnerConfiguration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PartnerConfiguration shouldn't use an unstable API (eventgrid.v20240601preview.PartnerConfiguration). A compatible replacement can be found at 'eventgrid.PartnerConfiguration'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PartnerConfiguration || (PartnerConfiguration = {}));
exports.PartnerConfiguration = PartnerConfiguration;
var PartnerDestination;
(function (PartnerDestination) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerDestination).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PartnerDestination.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-partnerdestination-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerDestination).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_14.PartnerDestination, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PartnerDestination shouldn't use an unstable API (eventgrid.v20240601preview.PartnerDestination). A compatible replacement can be found at 'eventgrid.PartnerDestination'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PartnerDestination || (PartnerDestination = {}));
exports.PartnerDestination = PartnerDestination;
var PartnerNamespace;
(function (PartnerNamespace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerNamespace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PartnerNamespace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-partnernamespace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerNamespace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_15.PartnerNamespace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PartnerNamespace shouldn't use an unstable API (eventgrid.v20240601preview.PartnerNamespace). A compatible replacement can be found at 'eventgrid.PartnerNamespace'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PartnerNamespace || (PartnerNamespace = {}));
exports.PartnerNamespace = PartnerNamespace;
var PartnerRegistration;
(function (PartnerRegistration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerRegistration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PartnerRegistration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-partnerregistration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerRegistration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_16.PartnerRegistration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PartnerRegistration shouldn't use an unstable API (eventgrid.v20240601preview.PartnerRegistration). A compatible replacement can be found at 'eventgrid.PartnerRegistration'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PartnerRegistration || (PartnerRegistration = {}));
exports.PartnerRegistration = PartnerRegistration;
var PartnerTopic;
(function (PartnerTopic) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerTopic).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PartnerTopic.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-partnertopic-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerTopic).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_17.PartnerTopic, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PartnerTopic shouldn't use an unstable API (eventgrid.v20240601preview.PartnerTopic). A compatible replacement can be found at 'eventgrid.PartnerTopic'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PartnerTopic || (PartnerTopic = {}));
exports.PartnerTopic = PartnerTopic;
var PartnerTopicEventSubscription;
(function (PartnerTopicEventSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerTopicEventSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PartnerTopicEventSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-partnertopiceventsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PartnerTopicEventSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_18.PartnerTopicEventSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PartnerTopicEventSubscription shouldn't use an unstable API (eventgrid.v20240601preview.PartnerTopicEventSubscription). A compatible replacement can be found at 'eventgrid.PartnerTopicEventSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PartnerTopicEventSubscription || (PartnerTopicEventSubscription = {}));
exports.PartnerTopicEventSubscription = PartnerTopicEventSubscription;
var PermissionBinding;
(function (PermissionBinding) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PermissionBinding).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PermissionBinding.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-permissionbinding-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PermissionBinding).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_19.PermissionBinding, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PermissionBinding shouldn't use an unstable API (eventgrid.v20240601preview.PermissionBinding). A compatible replacement can be found at 'eventgrid.PermissionBinding'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PermissionBinding || (PermissionBinding = {}));
exports.PermissionBinding = PermissionBinding;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_20.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (eventgrid.v20240601preview.PrivateEndpointConnection). A compatible replacement can be found at 'eventgrid.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
var SystemTopic;
(function (SystemTopic) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.SystemTopic).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SystemTopic.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-systemtopic-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.SystemTopic).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_21.SystemTopic, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SystemTopic shouldn't use an unstable API (eventgrid.v20240601preview.SystemTopic). A compatible replacement can be found at 'eventgrid.SystemTopic'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SystemTopic || (SystemTopic = {}));
exports.SystemTopic = SystemTopic;
var SystemTopicEventSubscription;
(function (SystemTopicEventSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.SystemTopicEventSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SystemTopicEventSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-systemtopiceventsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.SystemTopicEventSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_22.SystemTopicEventSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SystemTopicEventSubscription shouldn't use an unstable API (eventgrid.v20240601preview.SystemTopicEventSubscription). A compatible replacement can be found at 'eventgrid.SystemTopicEventSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SystemTopicEventSubscription || (SystemTopicEventSubscription = {}));
exports.SystemTopicEventSubscription = SystemTopicEventSubscription;
var Topic;
(function (Topic) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Topic).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Topic.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-topic-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.Topic).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_23.Topic, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Topic shouldn't use an unstable API (eventgrid.v20240601preview.Topic). A compatible replacement can be found at 'eventgrid.Topic'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Topic || (Topic = {}));
exports.Topic = Topic;
var TopicEventSubscription;
(function (TopicEventSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.TopicEventSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TopicEventSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-topiceventsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.TopicEventSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_24.TopicEventSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TopicEventSubscription shouldn't use an unstable API (eventgrid.v20240601preview.TopicEventSubscription). A compatible replacement can be found at 'eventgrid.TopicEventSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TopicEventSubscription || (TopicEventSubscription = {}));
exports.TopicEventSubscription = TopicEventSubscription;
var TopicSpace;
(function (TopicSpace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.TopicSpace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TopicSpace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20240601preview-topicspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20240601preview.TopicSpace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_25.TopicSpace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TopicSpace shouldn't use an unstable API (eventgrid.v20240601preview.TopicSpace). A compatible replacement can be found at 'eventgrid.TopicSpace'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TopicSpace || (TopicSpace = {}));
exports.TopicSpace = TopicSpace;
