// @pulumi/azure-native/standbypool/standbyContainerGroupPool.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StandbyContainerGroupPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A StandbyContainerGroupPoolResource.
 * Azure REST API version: 2023-12-01-preview.
 *
 * Other available API versions: 2024-03-01, 2024-03-01-preview.
 */
class StandbyContainerGroupPool extends pulumi.CustomResource {
    /**
     * Get an existing StandbyContainerGroupPool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StandbyContainerGroupPool(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StandbyContainerGroupPool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StandbyContainerGroupPool.__pulumiType;
    }
    /**
     * Create a StandbyContainerGroupPool resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.containerGroupProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerGroupProperties'");
            }
            if ((!args || args.elasticityProfile === undefined) && !opts.urn) {
                throw new Error("Missing required property 'elasticityProfile'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["containerGroupProperties"] = args ? args.containerGroupProperties : undefined;
            resourceInputs["elasticityProfile"] = args ? args.elasticityProfile : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["standbyContainerGroupPoolName"] = args ? args.standbyContainerGroupPoolName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["containerGroupProperties"] = undefined /*out*/;
            resourceInputs["elasticityProfile"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:standbypool/v20231201preview:StandbyContainerGroupPool" }, { type: "azure-native:standbypool/v20240301:StandbyContainerGroupPool" }, { type: "azure-native:standbypool/v20240301preview:StandbyContainerGroupPool" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StandbyContainerGroupPool.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StandbyContainerGroupPool = StandbyContainerGroupPool;
/** @internal */
StandbyContainerGroupPool.__pulumiType = 'azure-native:standbypool:StandbyContainerGroupPool';
