// @pulumi/azure-compliance-policies/azurenative/voiceservices/v20221201preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestLine = exports.Contact = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20221201preview_1 = require("@pulumi/azure-native/voiceservices/v20221201preview");
const v20221201preview_2 = require("@pulumi/azure-native/voiceservices/v20221201preview");
var Contact;
(function (Contact) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (voiceservices.v20221201preview.Contact).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Contact.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-voiceservices-v20221201preview-contact-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (voiceservices.v20221201preview.Contact).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221201preview_1.Contact, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Contact shouldn't use an unstable API (voiceservices.v20221201preview.Contact). A compatible replacement can be found at 'voiceservices.Contact'.");
            }),
        },
        vendors: ["azure"],
        services: ["voiceservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Contact || (Contact = {}));
exports.Contact = Contact;
var TestLine;
(function (TestLine) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (voiceservices.v20221201preview.TestLine).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TestLine.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-voiceservices-v20221201preview-testline-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (voiceservices.v20221201preview.TestLine).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221201preview_2.TestLine, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TestLine shouldn't use an unstable API (voiceservices.v20221201preview.TestLine). A compatible replacement can be found at 'voiceservices.TestLine'.");
            }),
        },
        vendors: ["azure"],
        services: ["voiceservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TestLine || (TestLine = {}));
exports.TestLine = TestLine;
