// @pulumi/azure/search/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedPrivateLinkService = exports.Service = exports.getServiceOutput = exports.getService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.SharedPrivateLinkService = null;
utilities.lazyLoad(exports, ["SharedPrivateLinkService"], () => require("./sharedPrivateLinkService"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:search/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure:search/sharedPrivateLinkService:SharedPrivateLinkService":
                return new exports.SharedPrivateLinkService(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "search/service", _module);
pulumi.runtime.registerResourceModule("azure", "search/sharedPrivateLinkService", _module);
