// @pulumi/azure-compliance-policies/azurenative/chaos/v20231027preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Target = exports.PrivateAccess = exports.Experiment = exports.Capability = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20231027preview_1 = require("@pulumi/azure-native/chaos/v20231027preview");
const v20231027preview_2 = require("@pulumi/azure-native/chaos/v20231027preview");
const v20231027preview_3 = require("@pulumi/azure-native/chaos/v20231027preview");
const v20231027preview_4 = require("@pulumi/azure-native/chaos/v20231027preview");
var Capability;
(function (Capability) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (chaos.v20231027preview.Capability).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Capability.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-chaos-v20231027preview-capability-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (chaos.v20231027preview.Capability).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231027preview_1.Capability, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Capability shouldn't use an unstable API (chaos.v20231027preview.Capability). A compatible replacement can be found at 'chaos.Capability'.");
            }),
        },
        vendors: ["azure"],
        services: ["chaos"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Capability || (Capability = {}));
exports.Capability = Capability;
var Experiment;
(function (Experiment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (chaos.v20231027preview.Experiment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Experiment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-chaos-v20231027preview-experiment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (chaos.v20231027preview.Experiment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231027preview_2.Experiment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Experiment shouldn't use an unstable API (chaos.v20231027preview.Experiment). A compatible replacement can be found at 'chaos.Experiment'.");
            }),
        },
        vendors: ["azure"],
        services: ["chaos"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Experiment || (Experiment = {}));
exports.Experiment = Experiment;
var PrivateAccess;
(function (PrivateAccess) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (chaos.v20231027preview.PrivateAccess).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateAccess.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-chaos-v20231027preview-privateaccess-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (chaos.v20231027preview.PrivateAccess).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231027preview_3.PrivateAccess, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateAccess shouldn't use an unstable API (chaos.v20231027preview.PrivateAccess). A compatible replacement can be found at 'chaos.PrivateAccess'.");
            }),
        },
        vendors: ["azure"],
        services: ["chaos"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateAccess || (PrivateAccess = {}));
exports.PrivateAccess = PrivateAccess;
var Target;
(function (Target) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (chaos.v20231027preview.Target).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Target.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-chaos-v20231027preview-target-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (chaos.v20231027preview.Target).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231027preview_4.Target, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Target shouldn't use an unstable API (chaos.v20231027preview.Target). A compatible replacement can be found at 'chaos.Target'.");
            }),
        },
        vendors: ["azure"],
        services: ["chaos"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Target || (Target = {}));
exports.Target = Target;
