// @pulumi/azure-compliance-policies/azurenative/desktopvirtualization/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240808preview = exports.v20240408preview = exports.v20240306preview = exports.v20240116preview = exports.v20231101preview = exports.v20231004preview = exports.v20230707preview = exports.v20221014preview = exports.v20220401preview = exports.v20220210preview = exports.v20210201preview = void 0;
exports.v20210201preview = require("./v20210201preview");
exports.v20220210preview = require("./v20220210preview");
exports.v20220401preview = require("./v20220401preview");
exports.v20221014preview = require("./v20221014preview");
exports.v20230707preview = require("./v20230707preview");
exports.v20231004preview = require("./v20231004preview");
exports.v20231101preview = require("./v20231101preview");
exports.v20240116preview = require("./v20240116preview");
exports.v20240306preview = require("./v20240306preview");
exports.v20240408preview = require("./v20240408preview");
exports.v20240808preview = require("./v20240808preview");
