// @pulumi/azure-compliance-policies/azurenative/security/v20220701preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityConnectorApplication = exports.Application = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20220701preview_1 = require("@pulumi/azure-native/security/v20220701preview");
const v20220701preview_2 = require("@pulumi/azure-native/security/v20220701preview");
var Application;
(function (Application) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20220701preview.Application).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Application.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20220701preview-application-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20220701preview.Application).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220701preview_1.Application, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Application shouldn't use an unstable API (security.v20220701preview.Application). A compatible replacement can be found at 'security.Application'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Application || (Application = {}));
exports.Application = Application;
var SecurityConnectorApplication;
(function (SecurityConnectorApplication) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20220701preview.SecurityConnectorApplication).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SecurityConnectorApplication.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20220701preview-securityconnectorapplication-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20220701preview.SecurityConnectorApplication).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220701preview_2.SecurityConnectorApplication, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SecurityConnectorApplication shouldn't use an unstable API (security.v20220701preview.SecurityConnectorApplication). A compatible replacement can be found at 'security.SecurityConnectorApplication'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SecurityConnectorApplication || (SecurityConnectorApplication = {}));
exports.SecurityConnectorApplication = SecurityConnectorApplication;
