// @pulumi/azure-native/aadiam/privateLinkForAzureAd.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkForAzureAd = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * PrivateLink Policy configuration object.
 * Azure REST API version: 2020-03-01. Prior API version in Azure Native 1.x: 2020-03-01.
 */
class PrivateLinkForAzureAd extends pulumi.CustomResource {
    /**
     * Get an existing PrivateLinkForAzureAd resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateLinkForAzureAd(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateLinkForAzureAd.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateLinkForAzureAd.__pulumiType;
    }
    /**
     * Create a PrivateLinkForAzureAd resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["allTenants"] = args ? args.allTenants : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["ownerTenantId"] = args ? args.ownerTenantId : undefined;
            resourceInputs["policyName"] = args ? args.policyName : undefined;
            resourceInputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["subscriptionId"] = args ? args.subscriptionId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tenants"] = args ? args.tenants : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["allTenants"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["ownerTenantId"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["resourceName"] = undefined /*out*/;
            resourceInputs["subscriptionId"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["tenants"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:aadiam/v20200301:PrivateLinkForAzureAd" }, { type: "azure-native:aadiam/v20200301:privateLinkForAzureAd" }, { type: "azure-native:aadiam/v20200301preview:PrivateLinkForAzureAd" }, { type: "azure-native:aadiam/v20200301preview:privateLinkForAzureAd" }, { type: "azure-native:aadiam:privateLinkForAzureAd" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateLinkForAzureAd.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateLinkForAzureAd = PrivateLinkForAzureAd;
/** @internal */
PrivateLinkForAzureAd.__pulumiType = 'azure-native:aadiam:PrivateLinkForAzureAd';
