// @pulumi/azure-compliance-policies/azurenative/desktopvirtualization/v20240116preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.ScalingPlanPooledSchedule = exports.ScalingPlanPersonalSchedule = exports.ScalingPlan = exports.PrivateEndpointConnectionByWorkspace = exports.PrivateEndpointConnectionByHostPool = exports.MSIXPackage = exports.HostPool = exports.ApplicationGroup = exports.Application = exports.AppAttachPackage = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240116preview_1 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_2 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_3 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_4 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_5 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_6 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_7 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_8 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_9 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_10 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
const v20240116preview_11 = require("@pulumi/azure-native/desktopvirtualization/v20240116preview");
var AppAttachPackage;
(function (AppAttachPackage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.AppAttachPackage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AppAttachPackage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-appattachpackage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.AppAttachPackage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_1.AppAttachPackage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AppAttachPackage shouldn't use an unstable API (desktopvirtualization.v20240116preview.AppAttachPackage). A compatible replacement can be found at 'desktopvirtualization.AppAttachPackage'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AppAttachPackage || (AppAttachPackage = {}));
exports.AppAttachPackage = AppAttachPackage;
var Application;
(function (Application) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.Application).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Application.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-application-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.Application).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_2.Application, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Application shouldn't use an unstable API (desktopvirtualization.v20240116preview.Application). A compatible replacement can be found at 'desktopvirtualization.Application'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Application || (Application = {}));
exports.Application = Application;
var ApplicationGroup;
(function (ApplicationGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.ApplicationGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApplicationGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-applicationgroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.ApplicationGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_3.ApplicationGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApplicationGroup shouldn't use an unstable API (desktopvirtualization.v20240116preview.ApplicationGroup). A compatible replacement can be found at 'desktopvirtualization.ApplicationGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApplicationGroup || (ApplicationGroup = {}));
exports.ApplicationGroup = ApplicationGroup;
var HostPool;
(function (HostPool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.HostPool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    HostPool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-hostpool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.HostPool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_4.HostPool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure HostPool shouldn't use an unstable API (desktopvirtualization.v20240116preview.HostPool). A compatible replacement can be found at 'desktopvirtualization.HostPool'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(HostPool || (HostPool = {}));
exports.HostPool = HostPool;
var MSIXPackage;
(function (MSIXPackage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.MSIXPackage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MSIXPackage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-msixpackage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.MSIXPackage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_5.MSIXPackage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MSIXPackage shouldn't use an unstable API (desktopvirtualization.v20240116preview.MSIXPackage). A compatible replacement can be found at 'desktopvirtualization.MSIXPackage'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MSIXPackage || (MSIXPackage = {}));
exports.MSIXPackage = MSIXPackage;
var PrivateEndpointConnectionByHostPool;
(function (PrivateEndpointConnectionByHostPool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.PrivateEndpointConnectionByHostPool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnectionByHostPool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-privateendpointconnectionbyhostpool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.PrivateEndpointConnectionByHostPool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_6.PrivateEndpointConnectionByHostPool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnectionByHostPool shouldn't use an unstable API (desktopvirtualization.v20240116preview.PrivateEndpointConnectionByHostPool). A compatible replacement can be found at 'desktopvirtualization.PrivateEndpointConnectionByHostPool'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnectionByHostPool || (PrivateEndpointConnectionByHostPool = {}));
exports.PrivateEndpointConnectionByHostPool = PrivateEndpointConnectionByHostPool;
var PrivateEndpointConnectionByWorkspace;
(function (PrivateEndpointConnectionByWorkspace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.PrivateEndpointConnectionByWorkspace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnectionByWorkspace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-privateendpointconnectionbyworkspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.PrivateEndpointConnectionByWorkspace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_7.PrivateEndpointConnectionByWorkspace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnectionByWorkspace shouldn't use an unstable API (desktopvirtualization.v20240116preview.PrivateEndpointConnectionByWorkspace). A compatible replacement can be found at 'desktopvirtualization.PrivateEndpointConnectionByWorkspace'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnectionByWorkspace || (PrivateEndpointConnectionByWorkspace = {}));
exports.PrivateEndpointConnectionByWorkspace = PrivateEndpointConnectionByWorkspace;
var ScalingPlan;
(function (ScalingPlan) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.ScalingPlan).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ScalingPlan.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-scalingplan-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.ScalingPlan).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_8.ScalingPlan, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ScalingPlan shouldn't use an unstable API (desktopvirtualization.v20240116preview.ScalingPlan). A compatible replacement can be found at 'desktopvirtualization.ScalingPlan'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ScalingPlan || (ScalingPlan = {}));
exports.ScalingPlan = ScalingPlan;
var ScalingPlanPersonalSchedule;
(function (ScalingPlanPersonalSchedule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.ScalingPlanPersonalSchedule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ScalingPlanPersonalSchedule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-scalingplanpersonalschedule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.ScalingPlanPersonalSchedule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_9.ScalingPlanPersonalSchedule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ScalingPlanPersonalSchedule shouldn't use an unstable API (desktopvirtualization.v20240116preview.ScalingPlanPersonalSchedule). A compatible replacement can be found at 'desktopvirtualization.ScalingPlanPersonalSchedule'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ScalingPlanPersonalSchedule || (ScalingPlanPersonalSchedule = {}));
exports.ScalingPlanPersonalSchedule = ScalingPlanPersonalSchedule;
var ScalingPlanPooledSchedule;
(function (ScalingPlanPooledSchedule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.ScalingPlanPooledSchedule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ScalingPlanPooledSchedule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-scalingplanpooledschedule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.ScalingPlanPooledSchedule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_10.ScalingPlanPooledSchedule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ScalingPlanPooledSchedule shouldn't use an unstable API (desktopvirtualization.v20240116preview.ScalingPlanPooledSchedule). A compatible replacement can be found at 'desktopvirtualization.ScalingPlanPooledSchedule'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ScalingPlanPooledSchedule || (ScalingPlanPooledSchedule = {}));
exports.ScalingPlanPooledSchedule = ScalingPlanPooledSchedule;
var Workspace;
(function (Workspace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.Workspace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Workspace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20240116preview-workspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20240116preview.Workspace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240116preview_11.Workspace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Workspace shouldn't use an unstable API (desktopvirtualization.v20240116preview.Workspace). A compatible replacement can be found at 'desktopvirtualization.Workspace'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Workspace || (Workspace = {}));
exports.Workspace = Workspace;
