// @pulumi/azure-compliance-policies/azurenative/datashare/v20201001preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlobStorageAccountDataSetMapping = exports.BlobStorageAccountDataSet = exports.ADLSGen2StorageAccountDataSetMapping = exports.ADLSGen2StorageAccountDataSet = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20201001preview_1 = require("@pulumi/azure-native/datashare/v20201001preview");
const v20201001preview_2 = require("@pulumi/azure-native/datashare/v20201001preview");
const v20201001preview_3 = require("@pulumi/azure-native/datashare/v20201001preview");
const v20201001preview_4 = require("@pulumi/azure-native/datashare/v20201001preview");
var ADLSGen2StorageAccountDataSet;
(function (ADLSGen2StorageAccountDataSet) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datashare.v20201001preview.ADLSGen2StorageAccountDataSet).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ADLSGen2StorageAccountDataSet.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datashare-v20201001preview-adlsgen2storageaccountdataset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datashare.v20201001preview.ADLSGen2StorageAccountDataSet).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20201001preview_1.ADLSGen2StorageAccountDataSet, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ADLSGen2StorageAccountDataSet shouldn't use an unstable API (datashare.v20201001preview.ADLSGen2StorageAccountDataSet). A compatible replacement can be found at 'datashare.ADLSGen2StorageAccountDataSet'.");
            }),
        },
        vendors: ["azure"],
        services: ["datashare"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ADLSGen2StorageAccountDataSet || (ADLSGen2StorageAccountDataSet = {}));
exports.ADLSGen2StorageAccountDataSet = ADLSGen2StorageAccountDataSet;
var ADLSGen2StorageAccountDataSetMapping;
(function (ADLSGen2StorageAccountDataSetMapping) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datashare.v20201001preview.ADLSGen2StorageAccountDataSetMapping).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ADLSGen2StorageAccountDataSetMapping.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datashare-v20201001preview-adlsgen2storageaccountdatasetmapping-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datashare.v20201001preview.ADLSGen2StorageAccountDataSetMapping).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20201001preview_2.ADLSGen2StorageAccountDataSetMapping, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ADLSGen2StorageAccountDataSetMapping shouldn't use an unstable API (datashare.v20201001preview.ADLSGen2StorageAccountDataSetMapping). A compatible replacement can be found at 'datashare.ADLSGen2StorageAccountDataSetMapping'.");
            }),
        },
        vendors: ["azure"],
        services: ["datashare"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ADLSGen2StorageAccountDataSetMapping || (ADLSGen2StorageAccountDataSetMapping = {}));
exports.ADLSGen2StorageAccountDataSetMapping = ADLSGen2StorageAccountDataSetMapping;
var BlobStorageAccountDataSet;
(function (BlobStorageAccountDataSet) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datashare.v20201001preview.BlobStorageAccountDataSet).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BlobStorageAccountDataSet.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datashare-v20201001preview-blobstorageaccountdataset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datashare.v20201001preview.BlobStorageAccountDataSet).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20201001preview_3.BlobStorageAccountDataSet, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BlobStorageAccountDataSet shouldn't use an unstable API (datashare.v20201001preview.BlobStorageAccountDataSet). A compatible replacement can be found at 'datashare.BlobStorageAccountDataSet'.");
            }),
        },
        vendors: ["azure"],
        services: ["datashare"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BlobStorageAccountDataSet || (BlobStorageAccountDataSet = {}));
exports.BlobStorageAccountDataSet = BlobStorageAccountDataSet;
var BlobStorageAccountDataSetMapping;
(function (BlobStorageAccountDataSetMapping) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datashare.v20201001preview.BlobStorageAccountDataSetMapping).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BlobStorageAccountDataSetMapping.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datashare-v20201001preview-blobstorageaccountdatasetmapping-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datashare.v20201001preview.BlobStorageAccountDataSetMapping).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20201001preview_4.BlobStorageAccountDataSetMapping, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BlobStorageAccountDataSetMapping shouldn't use an unstable API (datashare.v20201001preview.BlobStorageAccountDataSetMapping). A compatible replacement can be found at 'datashare.BlobStorageAccountDataSetMapping'.");
            }),
        },
        vendors: ["azure"],
        services: ["datashare"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BlobStorageAccountDataSetMapping || (BlobStorageAccountDataSetMapping = {}));
exports.BlobStorageAccountDataSetMapping = BlobStorageAccountDataSetMapping;
