// @pulumi/azure-native/costmanagement/markupRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MarkupRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Markup rule
 * Azure REST API version: 2022-10-05-preview.
 */
class MarkupRule extends pulumi.CustomResource {
    /**
     * Get an existing MarkupRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MarkupRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MarkupRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MarkupRule.__pulumiType;
    }
    /**
     * Create a MarkupRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.billingAccountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'billingAccountId'");
            }
            if ((!args || args.billingProfileId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'billingProfileId'");
            }
            if ((!args || args.customerDetails === undefined) && !opts.urn) {
                throw new Error("Missing required property 'customerDetails'");
            }
            if ((!args || args.percentage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'percentage'");
            }
            if ((!args || args.startDate === undefined) && !opts.urn) {
                throw new Error("Missing required property 'startDate'");
            }
            resourceInputs["billingAccountId"] = args ? args.billingAccountId : undefined;
            resourceInputs["billingProfileId"] = args ? args.billingProfileId : undefined;
            resourceInputs["customerDetails"] = args ? args.customerDetails : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["endDate"] = args ? args.endDate : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["percentage"] = args ? args.percentage : undefined;
            resourceInputs["startDate"] = args ? args.startDate : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["customerDetails"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["endDate"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["percentage"] = undefined /*out*/;
            resourceInputs["startDate"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:costmanagement/v20221005preview:MarkupRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MarkupRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MarkupRule = MarkupRule;
/** @internal */
MarkupRule.__pulumiType = 'azure-native:costmanagement:MarkupRule';
