// @pulumi/azure-compliance-policies/azurenative/quota/v20241015preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupQuotaSubscription = exports.GroupQuota = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20241015preview_1 = require("@pulumi/azure-native/quota/v20241015preview");
const v20241015preview_2 = require("@pulumi/azure-native/quota/v20241015preview");
var GroupQuota;
(function (GroupQuota) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (quota.v20241015preview.GroupQuota).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GroupQuota.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-quota-v20241015preview-groupquota-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (quota.v20241015preview.GroupQuota).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241015preview_1.GroupQuota, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GroupQuota shouldn't use an unstable API (quota.v20241015preview.GroupQuota). A compatible replacement can be found at 'quota.GroupQuota'.");
            }),
        },
        vendors: ["azure"],
        services: ["quota"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GroupQuota || (GroupQuota = {}));
exports.GroupQuota = GroupQuota;
var GroupQuotaSubscription;
(function (GroupQuotaSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (quota.v20241015preview.GroupQuotaSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GroupQuotaSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-quota-v20241015preview-groupquotasubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (quota.v20241015preview.GroupQuotaSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241015preview_2.GroupQuotaSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GroupQuotaSubscription shouldn't use an unstable API (quota.v20241015preview.GroupQuotaSubscription). A compatible replacement can be found at 'quota.GroupQuotaSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["quota"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GroupQuotaSubscription || (GroupQuotaSubscription = {}));
exports.GroupQuotaSubscription = GroupQuotaSubscription;
