// @pulumi/azure-compliance-policies/azurenative/codesigning/v20240930preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodeSigningAccount = exports.CertificateProfile = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240930preview_1 = require("@pulumi/azure-native/codesigning/v20240930preview");
const v20240930preview_2 = require("@pulumi/azure-native/codesigning/v20240930preview");
var CertificateProfile;
(function (CertificateProfile) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (codesigning.v20240930preview.CertificateProfile).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CertificateProfile.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-codesigning-v20240930preview-certificateprofile-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (codesigning.v20240930preview.CertificateProfile).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240930preview_1.CertificateProfile, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CertificateProfile shouldn't use an unstable API (codesigning.v20240930preview.CertificateProfile). A compatible replacement can be found at 'codesigning.CertificateProfile'.");
            }),
        },
        vendors: ["azure"],
        services: ["codesigning"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CertificateProfile || (CertificateProfile = {}));
exports.CertificateProfile = CertificateProfile;
var CodeSigningAccount;
(function (CodeSigningAccount) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (codesigning.v20240930preview.CodeSigningAccount).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CodeSigningAccount.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-codesigning-v20240930preview-codesigningaccount-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (codesigning.v20240930preview.CodeSigningAccount).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240930preview_2.CodeSigningAccount, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CodeSigningAccount shouldn't use an unstable API (codesigning.v20240930preview.CodeSigningAccount). A compatible replacement can be found at 'codesigning.CodeSigningAccount'.");
            }),
        },
        vendors: ["azure"],
        services: ["codesigning"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CodeSigningAccount || (CodeSigningAccount = {}));
exports.CodeSigningAccount = CodeSigningAccount;
