// @pulumi/azure-compliance-policies/azurenative/healthbot/v20201208preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Bot = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20201208preview_1 = require("@pulumi/azure-native/healthbot/v20201208preview");
var Bot;
(function (Bot) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (healthbot.v20201208preview.Bot).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Bot.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-healthbot-v20201208preview-bot-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (healthbot.v20201208preview.Bot).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20201208preview_1.Bot, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Bot shouldn't use an unstable API (healthbot.v20201208preview.Bot). A compatible replacement can be found at 'healthbot.Bot'.");
            }),
        },
        vendors: ["azure"],
        services: ["healthbot"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Bot || (Bot = {}));
exports.Bot = Bot;
