// @pulumi/azure/cdn/frontdoorSecret.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrontdoorSecret = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Front Door (standard/premium) Secret.
 *
 * ```New-AzADServicePrincipal -ApplicationId "00000000-0000-0000-0000-000000000000"```
 *
 * | Object ID                                | Key Permissions | Secret Permissions   | Certificate Permissions                       |
 * |:-----------------------------------------|:---------------:|:--------------------:|:---------------------------------------------:|
 * | `Microsoft.Azure.Cdn` Object ID          | -               | **Get**              | -                                             |
 * | Your Personal AAD Object ID              | -               | **Get** and **List** | **Get**, **List**, **Purge** and **Recover**  |
 * | Terraform Service Principal              | -               | **Get**              | **Get**, **Import**, **Delete** and **Purge** |
 *
 * ->**NOTE:** You only need to add the `Access Policy` for your personal AAD Object ID if you are planning to view the `secrets` via the Azure Portal.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as azuread from "@pulumi/azuread";
 * import * as std from "@pulumi/std";
 *
 * const current = azure.core.getClientConfig({});
 * const frontdoor = azuread.getServicePrincipal({
 *     displayName: "Microsoft.Azure.Cdn",
 * });
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-cdn-frontdoor",
 *     location: "West Europe",
 * });
 * const exampleKeyVault = new azure.keyvault.KeyVault("example", {
 *     name: "example-keyvault",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     tenantId: current.then(current => current.tenantId),
 *     skuName: "premium",
 *     softDeleteRetentionDays: 7,
 *     networkAcls: {
 *         defaultAction: "Deny",
 *         bypass: "AzureServices",
 *         ipRules: ["10.0.0.0/24"],
 *     },
 *     accessPolicies: [
 *         {
 *             tenantId: current.then(current => current.tenantId),
 *             objectId: frontdoor.then(frontdoor => frontdoor.objectId),
 *             secretPermissions: ["Get"],
 *         },
 *         {
 *             tenantId: current.then(current => current.tenantId),
 *             objectId: current.then(current => current.objectId),
 *             certificatePermissions: [
 *                 "Get",
 *                 "Import",
 *                 "Delete",
 *                 "Purge",
 *             ],
 *             secretPermissions: ["Get"],
 *         },
 *     ],
 * });
 * const exampleCertificate = new azure.keyvault.Certificate("example", {
 *     name: "example-cert",
 *     keyVaultId: exampleKeyVault.id,
 *     certificate: {
 *         contents: std.filebase64({
 *             input: "my-certificate.pfx",
 *         }).then(invoke => invoke.result),
 *     },
 * });
 * const exampleFrontdoorProfile = new azure.cdn.FrontdoorProfile("example", {
 *     name: "example-cdn-profile",
 *     resourceGroupName: example.name,
 *     skuName: "Standard_AzureFrontDoor",
 * });
 * const exampleFrontdoorSecret = new azure.cdn.FrontdoorSecret("example", {
 *     name: "example-customer-managed-secret",
 *     cdnFrontdoorProfileId: exampleFrontdoorProfile.id,
 *     secret: {
 *         customerCertificates: [{
 *             keyVaultCertificateId: exampleCertificate.id,
 *         }],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Front Door Secrets can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:cdn/frontdoorSecret:FrontdoorSecret example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.Cdn/profiles/profile1/secrets/secrets1
 * ```
 */
class FrontdoorSecret extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["cdnFrontdoorProfileId"] = state ? state.cdnFrontdoorProfileId : undefined;
            resourceInputs["cdnFrontdoorProfileName"] = state ? state.cdnFrontdoorProfileName : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["secret"] = state ? state.secret : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.cdnFrontdoorProfileId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cdnFrontdoorProfileId'");
            }
            if ((!args || args.secret === undefined) && !opts.urn) {
                throw new Error("Missing required property 'secret'");
            }
            resourceInputs["cdnFrontdoorProfileId"] = args ? args.cdnFrontdoorProfileId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["secret"] = args ? args.secret : undefined;
            resourceInputs["cdnFrontdoorProfileName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(FrontdoorSecret.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing FrontdoorSecret resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new FrontdoorSecret(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FrontdoorSecret.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FrontdoorSecret.__pulumiType;
    }
}
exports.FrontdoorSecret = FrontdoorSecret;
/** @internal */
FrontdoorSecret.__pulumiType = 'azure:cdn/frontdoorSecret:FrontdoorSecret';
