// @pulumi/azure/voice/servicesCommunicationsGatewayTestLine.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServicesCommunicationsGatewayTestLine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Voice Services Communications Gateway Test Line.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Central US",
 * });
 * const exampleServicesCommunicationsGateway = new azure.voice.ServicesCommunicationsGateway("example", {
 *     name: "example-vcg",
 *     resourceGroupName: example.name,
 * });
 * const exampleServicesCommunicationsGatewayTestLine = new azure.voice.ServicesCommunicationsGatewayTestLine("example", {
 *     name: "example-vtl",
 *     location: "West Central US",
 *     voiceServicesCommunicationsGatewayId: exampleServicesCommunicationsGateway.id,
 *     phoneNumber: "123456789",
 *     purpose: "Automated",
 *     tags: {
 *         key: "value",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Voice Services Communications Gateway Test Line can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:voice/servicesCommunicationsGatewayTestLine:ServicesCommunicationsGatewayTestLine example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.VoiceServices/communicationsGateways/communicationsGateway1/testLines/testLine1
 * ```
 */
class ServicesCommunicationsGatewayTestLine extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["phoneNumber"] = state ? state.phoneNumber : undefined;
            resourceInputs["purpose"] = state ? state.purpose : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["voiceServicesCommunicationsGatewayId"] = state ? state.voiceServicesCommunicationsGatewayId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.phoneNumber === undefined) && !opts.urn) {
                throw new Error("Missing required property 'phoneNumber'");
            }
            if ((!args || args.purpose === undefined) && !opts.urn) {
                throw new Error("Missing required property 'purpose'");
            }
            if ((!args || args.voiceServicesCommunicationsGatewayId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'voiceServicesCommunicationsGatewayId'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["phoneNumber"] = args ? args.phoneNumber : undefined;
            resourceInputs["purpose"] = args ? args.purpose : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["voiceServicesCommunicationsGatewayId"] = args ? args.voiceServicesCommunicationsGatewayId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ServicesCommunicationsGatewayTestLine.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ServicesCommunicationsGatewayTestLine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServicesCommunicationsGatewayTestLine(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServicesCommunicationsGatewayTestLine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServicesCommunicationsGatewayTestLine.__pulumiType;
    }
}
exports.ServicesCommunicationsGatewayTestLine = ServicesCommunicationsGatewayTestLine;
/** @internal */
ServicesCommunicationsGatewayTestLine.__pulumiType = 'azure:voice/servicesCommunicationsGatewayTestLine:ServicesCommunicationsGatewayTestLine';
