// @pulumi/azure-compliance-policies/azurenative/containerregistry/v20230601preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook = exports.Token = exports.ScopeMap = exports.Replication = exports.Registry = exports.PrivateEndpointConnection = exports.PipelineRun = exports.ImportPipeline = exports.ExportPipeline = exports.CredentialSet = exports.ConnectedRegistry = exports.CacheRule = exports.ArchiveVersion = exports.Archife = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230601preview_1 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_2 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_3 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_4 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_5 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_6 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_7 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_8 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_9 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_10 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_11 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_12 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_13 = require("@pulumi/azure-native/containerregistry/v20230601preview");
const v20230601preview_14 = require("@pulumi/azure-native/containerregistry/v20230601preview");
var Archife;
(function (Archife) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Archife).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Archife.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-archife-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Archife).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_1.Archife, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Archife shouldn't use an unstable API (containerregistry.v20230601preview.Archife). A compatible replacement can be found at 'containerregistry.Archife'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Archife || (Archife = {}));
exports.Archife = Archife;
var ArchiveVersion;
(function (ArchiveVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ArchiveVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ArchiveVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-archiveversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ArchiveVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_2.ArchiveVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ArchiveVersion shouldn't use an unstable API (containerregistry.v20230601preview.ArchiveVersion). A compatible replacement can be found at 'containerregistry.ArchiveVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ArchiveVersion || (ArchiveVersion = {}));
exports.ArchiveVersion = ArchiveVersion;
var CacheRule;
(function (CacheRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.CacheRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CacheRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-cacherule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.CacheRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_3.CacheRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CacheRule shouldn't use an unstable API (containerregistry.v20230601preview.CacheRule). A compatible replacement can be found at 'containerregistry.CacheRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CacheRule || (CacheRule = {}));
exports.CacheRule = CacheRule;
var ConnectedRegistry;
(function (ConnectedRegistry) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ConnectedRegistry).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConnectedRegistry.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-connectedregistry-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ConnectedRegistry).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_4.ConnectedRegistry, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConnectedRegistry shouldn't use an unstable API (containerregistry.v20230601preview.ConnectedRegistry). A compatible replacement can be found at 'containerregistry.ConnectedRegistry'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConnectedRegistry || (ConnectedRegistry = {}));
exports.ConnectedRegistry = ConnectedRegistry;
var CredentialSet;
(function (CredentialSet) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.CredentialSet).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CredentialSet.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-credentialset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.CredentialSet).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_5.CredentialSet, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CredentialSet shouldn't use an unstable API (containerregistry.v20230601preview.CredentialSet). A compatible replacement can be found at 'containerregistry.CredentialSet'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CredentialSet || (CredentialSet = {}));
exports.CredentialSet = CredentialSet;
var ExportPipeline;
(function (ExportPipeline) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ExportPipeline).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ExportPipeline.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-exportpipeline-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ExportPipeline).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_6.ExportPipeline, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ExportPipeline shouldn't use an unstable API (containerregistry.v20230601preview.ExportPipeline). A compatible replacement can be found at 'containerregistry.ExportPipeline'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ExportPipeline || (ExportPipeline = {}));
exports.ExportPipeline = ExportPipeline;
var ImportPipeline;
(function (ImportPipeline) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ImportPipeline).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ImportPipeline.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-importpipeline-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ImportPipeline).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_7.ImportPipeline, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ImportPipeline shouldn't use an unstable API (containerregistry.v20230601preview.ImportPipeline). A compatible replacement can be found at 'containerregistry.ImportPipeline'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ImportPipeline || (ImportPipeline = {}));
exports.ImportPipeline = ImportPipeline;
var PipelineRun;
(function (PipelineRun) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.PipelineRun).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PipelineRun.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-pipelinerun-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.PipelineRun).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_8.PipelineRun, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PipelineRun shouldn't use an unstable API (containerregistry.v20230601preview.PipelineRun). A compatible replacement can be found at 'containerregistry.PipelineRun'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PipelineRun || (PipelineRun = {}));
exports.PipelineRun = PipelineRun;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_9.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (containerregistry.v20230601preview.PrivateEndpointConnection). A compatible replacement can be found at 'containerregistry.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
var Registry;
(function (Registry) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Registry).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Registry.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-registry-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Registry).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_10.Registry, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Registry shouldn't use an unstable API (containerregistry.v20230601preview.Registry). A compatible replacement can be found at 'containerregistry.Registry'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Registry || (Registry = {}));
exports.Registry = Registry;
var Replication;
(function (Replication) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Replication).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Replication.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-replication-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Replication).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_11.Replication, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Replication shouldn't use an unstable API (containerregistry.v20230601preview.Replication). A compatible replacement can be found at 'containerregistry.Replication'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Replication || (Replication = {}));
exports.Replication = Replication;
var ScopeMap;
(function (ScopeMap) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ScopeMap).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ScopeMap.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-scopemap-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.ScopeMap).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_12.ScopeMap, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ScopeMap shouldn't use an unstable API (containerregistry.v20230601preview.ScopeMap). A compatible replacement can be found at 'containerregistry.ScopeMap'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ScopeMap || (ScopeMap = {}));
exports.ScopeMap = ScopeMap;
var Token;
(function (Token) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Token).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Token.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-token-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Token).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_13.Token, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Token shouldn't use an unstable API (containerregistry.v20230601preview.Token). A compatible replacement can be found at 'containerregistry.Token'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Token || (Token = {}));
exports.Token = Token;
var Webhook;
(function (Webhook) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Webhook).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Webhook.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-containerregistry-v20230601preview-webhook-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (containerregistry.v20230601preview.Webhook).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_14.Webhook, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Webhook shouldn't use an unstable API (containerregistry.v20230601preview.Webhook). A compatible replacement can be found at 'containerregistry.Webhook'.");
            }),
        },
        vendors: ["azure"],
        services: ["containerregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Webhook || (Webhook = {}));
exports.Webhook = Webhook;
