// @pulumi/azure-compliance-policies/azurenative/extendedlocation/v20210831preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceSyncRule = exports.CustomLocation = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20210831preview_1 = require("@pulumi/azure-native/extendedlocation/v20210831preview");
const v20210831preview_2 = require("@pulumi/azure-native/extendedlocation/v20210831preview");
var CustomLocation;
(function (CustomLocation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (extendedlocation.v20210831preview.CustomLocation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomLocation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-extendedlocation-v20210831preview-customlocation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (extendedlocation.v20210831preview.CustomLocation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210831preview_1.CustomLocation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomLocation shouldn't use an unstable API (extendedlocation.v20210831preview.CustomLocation). A compatible replacement can be found at 'extendedlocation.CustomLocation'.");
            }),
        },
        vendors: ["azure"],
        services: ["extendedlocation"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomLocation || (CustomLocation = {}));
exports.CustomLocation = CustomLocation;
var ResourceSyncRule;
(function (ResourceSyncRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (extendedlocation.v20210831preview.ResourceSyncRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ResourceSyncRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-extendedlocation-v20210831preview-resourcesyncrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (extendedlocation.v20210831preview.ResourceSyncRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210831preview_2.ResourceSyncRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ResourceSyncRule shouldn't use an unstable API (extendedlocation.v20210831preview.ResourceSyncRule). A compatible replacement can be found at 'extendedlocation.ResourceSyncRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["extendedlocation"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ResourceSyncRule || (ResourceSyncRule = {}));
exports.ResourceSyncRule = ResourceSyncRule;
