// @pulumi/azure-compliance-policies/azurenative/blueprint/v20181101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplateArtifact = exports.RoleAssignmentArtifact = exports.PublishedBlueprint = exports.PolicyAssignmentArtifact = exports.Blueprint = exports.Assignment = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20181101preview_1 = require("@pulumi/azure-native/blueprint/v20181101preview");
const v20181101preview_2 = require("@pulumi/azure-native/blueprint/v20181101preview");
const v20181101preview_3 = require("@pulumi/azure-native/blueprint/v20181101preview");
const v20181101preview_4 = require("@pulumi/azure-native/blueprint/v20181101preview");
const v20181101preview_5 = require("@pulumi/azure-native/blueprint/v20181101preview");
const v20181101preview_6 = require("@pulumi/azure-native/blueprint/v20181101preview");
var Assignment;
(function (Assignment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.Assignment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Assignment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-blueprint-v20181101preview-assignment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.Assignment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20181101preview_1.Assignment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Assignment shouldn't use an unstable API (blueprint.v20181101preview.Assignment). A compatible replacement can be found at 'blueprint.Assignment'.");
            }),
        },
        vendors: ["azure"],
        services: ["blueprint"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Assignment || (Assignment = {}));
exports.Assignment = Assignment;
var Blueprint;
(function (Blueprint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.Blueprint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Blueprint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-blueprint-v20181101preview-blueprint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.Blueprint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20181101preview_2.Blueprint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Blueprint shouldn't use an unstable API (blueprint.v20181101preview.Blueprint). A compatible replacement can be found at 'blueprint.Blueprint'.");
            }),
        },
        vendors: ["azure"],
        services: ["blueprint"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Blueprint || (Blueprint = {}));
exports.Blueprint = Blueprint;
var PolicyAssignmentArtifact;
(function (PolicyAssignmentArtifact) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.PolicyAssignmentArtifact).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PolicyAssignmentArtifact.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-blueprint-v20181101preview-policyassignmentartifact-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.PolicyAssignmentArtifact).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20181101preview_3.PolicyAssignmentArtifact, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PolicyAssignmentArtifact shouldn't use an unstable API (blueprint.v20181101preview.PolicyAssignmentArtifact). A compatible replacement can be found at 'blueprint.PolicyAssignmentArtifact'.");
            }),
        },
        vendors: ["azure"],
        services: ["blueprint"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PolicyAssignmentArtifact || (PolicyAssignmentArtifact = {}));
exports.PolicyAssignmentArtifact = PolicyAssignmentArtifact;
var PublishedBlueprint;
(function (PublishedBlueprint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.PublishedBlueprint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PublishedBlueprint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-blueprint-v20181101preview-publishedblueprint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.PublishedBlueprint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20181101preview_4.PublishedBlueprint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PublishedBlueprint shouldn't use an unstable API (blueprint.v20181101preview.PublishedBlueprint). A compatible replacement can be found at 'blueprint.PublishedBlueprint'.");
            }),
        },
        vendors: ["azure"],
        services: ["blueprint"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PublishedBlueprint || (PublishedBlueprint = {}));
exports.PublishedBlueprint = PublishedBlueprint;
var RoleAssignmentArtifact;
(function (RoleAssignmentArtifact) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.RoleAssignmentArtifact).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RoleAssignmentArtifact.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-blueprint-v20181101preview-roleassignmentartifact-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.RoleAssignmentArtifact).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20181101preview_5.RoleAssignmentArtifact, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RoleAssignmentArtifact shouldn't use an unstable API (blueprint.v20181101preview.RoleAssignmentArtifact). A compatible replacement can be found at 'blueprint.RoleAssignmentArtifact'.");
            }),
        },
        vendors: ["azure"],
        services: ["blueprint"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RoleAssignmentArtifact || (RoleAssignmentArtifact = {}));
exports.RoleAssignmentArtifact = RoleAssignmentArtifact;
var TemplateArtifact;
(function (TemplateArtifact) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.TemplateArtifact).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TemplateArtifact.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-blueprint-v20181101preview-templateartifact-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (blueprint.v20181101preview.TemplateArtifact).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20181101preview_6.TemplateArtifact, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TemplateArtifact shouldn't use an unstable API (blueprint.v20181101preview.TemplateArtifact). A compatible replacement can be found at 'blueprint.TemplateArtifact'.");
            }),
        },
        vendors: ["azure"],
        services: ["blueprint"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TemplateArtifact || (TemplateArtifact = {}));
exports.TemplateArtifact = TemplateArtifact;
