// @pulumi/azure/cognitive/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAccountOutput = exports.getAccount = exports.Deployment = exports.AIServices = exports.AccountRaiPolicy = exports.AccountRaiBlocklist = exports.AccountCustomerManagedKey = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.AccountCustomerManagedKey = null;
utilities.lazyLoad(exports, ["AccountCustomerManagedKey"], () => require("./accountCustomerManagedKey"));
exports.AccountRaiBlocklist = null;
utilities.lazyLoad(exports, ["AccountRaiBlocklist"], () => require("./accountRaiBlocklist"));
exports.AccountRaiPolicy = null;
utilities.lazyLoad(exports, ["AccountRaiPolicy"], () => require("./accountRaiPolicy"));
exports.AIServices = null;
utilities.lazyLoad(exports, ["AIServices"], () => require("./aiservices"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:cognitive/aIServices:AIServices":
                return new exports.AIServices(name, undefined, { urn });
            case "azure:cognitive/account:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure:cognitive/accountCustomerManagedKey:AccountCustomerManagedKey":
                return new exports.AccountCustomerManagedKey(name, undefined, { urn });
            case "azure:cognitive/accountRaiBlocklist:AccountRaiBlocklist":
                return new exports.AccountRaiBlocklist(name, undefined, { urn });
            case "azure:cognitive/accountRaiPolicy:AccountRaiPolicy":
                return new exports.AccountRaiPolicy(name, undefined, { urn });
            case "azure:cognitive/deployment:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "cognitive/aIServices", _module);
pulumi.runtime.registerResourceModule("azure", "cognitive/account", _module);
pulumi.runtime.registerResourceModule("azure", "cognitive/accountCustomerManagedKey", _module);
pulumi.runtime.registerResourceModule("azure", "cognitive/accountRaiBlocklist", _module);
pulumi.runtime.registerResourceModule("azure", "cognitive/accountRaiPolicy", _module);
pulumi.runtime.registerResourceModule("azure", "cognitive/deployment", _module);
