// @pulumi/azure-compliance-policies/azurenative/apicenter/v20240601preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.Service = exports.MetadataSchema = exports.Environment = exports.Deployment = exports.ApiVersion = exports.ApiSource = exports.ApiDefinition = exports.Api = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240601preview_1 = require("@pulumi/azure-native/apicenter/v20240601preview");
const v20240601preview_2 = require("@pulumi/azure-native/apicenter/v20240601preview");
const v20240601preview_3 = require("@pulumi/azure-native/apicenter/v20240601preview");
const v20240601preview_4 = require("@pulumi/azure-native/apicenter/v20240601preview");
const v20240601preview_5 = require("@pulumi/azure-native/apicenter/v20240601preview");
const v20240601preview_6 = require("@pulumi/azure-native/apicenter/v20240601preview");
const v20240601preview_7 = require("@pulumi/azure-native/apicenter/v20240601preview");
const v20240601preview_8 = require("@pulumi/azure-native/apicenter/v20240601preview");
const v20240601preview_9 = require("@pulumi/azure-native/apicenter/v20240601preview");
var Api;
(function (Api) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Api).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Api.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apicenter-v20240601preview-api-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Api).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_1.Api, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Api shouldn't use an unstable API (apicenter.v20240601preview.Api). A compatible replacement can be found at 'apicenter.Api'.");
            }),
        },
        vendors: ["azure"],
        services: ["apicenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Api || (Api = {}));
exports.Api = Api;
var ApiDefinition;
(function (ApiDefinition) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.ApiDefinition).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiDefinition.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apicenter-v20240601preview-apidefinition-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.ApiDefinition).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_2.ApiDefinition, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiDefinition shouldn't use an unstable API (apicenter.v20240601preview.ApiDefinition). A compatible replacement can be found at 'apicenter.ApiDefinition'.");
            }),
        },
        vendors: ["azure"],
        services: ["apicenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiDefinition || (ApiDefinition = {}));
exports.ApiDefinition = ApiDefinition;
var ApiSource;
(function (ApiSource) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.ApiSource).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiSource.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apicenter-v20240601preview-apisource-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.ApiSource).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_3.ApiSource, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiSource shouldn't use an unstable API (apicenter.v20240601preview.ApiSource). A compatible replacement can be found at 'apicenter.ApiSource'.");
            }),
        },
        vendors: ["azure"],
        services: ["apicenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiSource || (ApiSource = {}));
exports.ApiSource = ApiSource;
var ApiVersion;
(function (ApiVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.ApiVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apicenter-v20240601preview-apiversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.ApiVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_4.ApiVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiVersion shouldn't use an unstable API (apicenter.v20240601preview.ApiVersion). A compatible replacement can be found at 'apicenter.ApiVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["apicenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiVersion || (ApiVersion = {}));
exports.ApiVersion = ApiVersion;
var Deployment;
(function (Deployment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Deployment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Deployment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apicenter-v20240601preview-deployment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Deployment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_5.Deployment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Deployment shouldn't use an unstable API (apicenter.v20240601preview.Deployment). A compatible replacement can be found at 'apicenter.Deployment'.");
            }),
        },
        vendors: ["azure"],
        services: ["apicenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Deployment || (Deployment = {}));
exports.Deployment = Deployment;
var Environment;
(function (Environment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Environment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Environment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apicenter-v20240601preview-environment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Environment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_6.Environment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Environment shouldn't use an unstable API (apicenter.v20240601preview.Environment). A compatible replacement can be found at 'apicenter.Environment'.");
            }),
        },
        vendors: ["azure"],
        services: ["apicenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Environment || (Environment = {}));
exports.Environment = Environment;
var MetadataSchema;
(function (MetadataSchema) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.MetadataSchema).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MetadataSchema.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apicenter-v20240601preview-metadataschema-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.MetadataSchema).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_7.MetadataSchema, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MetadataSchema shouldn't use an unstable API (apicenter.v20240601preview.MetadataSchema). A compatible replacement can be found at 'apicenter.MetadataSchema'.");
            }),
        },
        vendors: ["azure"],
        services: ["apicenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MetadataSchema || (MetadataSchema = {}));
exports.MetadataSchema = MetadataSchema;
var Service;
(function (Service) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Service).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Service.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apicenter-v20240601preview-service-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Service).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_8.Service, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Service shouldn't use an unstable API (apicenter.v20240601preview.Service). A compatible replacement can be found at 'apicenter.Service'.");
            }),
        },
        vendors: ["azure"],
        services: ["apicenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Service || (Service = {}));
exports.Service = Service;
var Workspace;
(function (Workspace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Workspace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Workspace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apicenter-v20240601preview-workspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apicenter.v20240601preview.Workspace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_9.Workspace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Workspace shouldn't use an unstable API (apicenter.v20240601preview.Workspace). A compatible replacement can be found at 'apicenter.Workspace'.");
            }),
        },
        vendors: ["azure"],
        services: ["apicenter"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Workspace || (Workspace = {}));
exports.Workspace = Workspace;
