// @pulumi/azure/pim/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleManagementPolicy = exports.getRoleManagementPolicyOutput = exports.getRoleManagementPolicy = exports.EligibleRoleAssignment = exports.ActiveRoleAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ActiveRoleAssignment = null;
utilities.lazyLoad(exports, ["ActiveRoleAssignment"], () => require("./activeRoleAssignment"));
exports.EligibleRoleAssignment = null;
utilities.lazyLoad(exports, ["EligibleRoleAssignment"], () => require("./eligibleRoleAssignment"));
exports.getRoleManagementPolicy = null;
exports.getRoleManagementPolicyOutput = null;
utilities.lazyLoad(exports, ["getRoleManagementPolicy", "getRoleManagementPolicyOutput"], () => require("./getRoleManagementPolicy"));
exports.RoleManagementPolicy = null;
utilities.lazyLoad(exports, ["RoleManagementPolicy"], () => require("./roleManagementPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:pim/activeRoleAssignment:ActiveRoleAssignment":
                return new exports.ActiveRoleAssignment(name, undefined, { urn });
            case "azure:pim/eligibleRoleAssignment:EligibleRoleAssignment":
                return new exports.EligibleRoleAssignment(name, undefined, { urn });
            case "azure:pim/roleManagementPolicy:RoleManagementPolicy":
                return new exports.RoleManagementPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "pim/activeRoleAssignment", _module);
pulumi.runtime.registerResourceModule("azure", "pim/eligibleRoleAssignment", _module);
pulumi.runtime.registerResourceModule("azure", "pim/roleManagementPolicy", _module);
