// @pulumi/azure-native/apimanagement/gatewayHostnameConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GatewayHostnameConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Gateway hostname configuration details.
 * Azure REST API version: 2022-08-01. Prior API version in Azure Native 1.x: 2020-12-01.
 *
 * Other available API versions: 2022-09-01-preview, 2023-03-01-preview, 2023-05-01-preview, 2023-09-01-preview, 2024-05-01, 2024-06-01-preview.
 */
class GatewayHostnameConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing GatewayHostnameConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GatewayHostnameConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GatewayHostnameConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GatewayHostnameConfiguration.__pulumiType;
    }
    /**
     * Create a GatewayHostnameConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.gatewayId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'gatewayId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["certificateId"] = args ? args.certificateId : undefined;
            resourceInputs["gatewayId"] = args ? args.gatewayId : undefined;
            resourceInputs["hcId"] = args ? args.hcId : undefined;
            resourceInputs["hostname"] = args ? args.hostname : undefined;
            resourceInputs["http2Enabled"] = args ? args.http2Enabled : undefined;
            resourceInputs["negotiateClientCertificate"] = args ? args.negotiateClientCertificate : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["tls10Enabled"] = args ? args.tls10Enabled : undefined;
            resourceInputs["tls11Enabled"] = args ? args.tls11Enabled : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["certificateId"] = undefined /*out*/;
            resourceInputs["hostname"] = undefined /*out*/;
            resourceInputs["http2Enabled"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["negotiateClientCertificate"] = undefined /*out*/;
            resourceInputs["tls10Enabled"] = undefined /*out*/;
            resourceInputs["tls11Enabled"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:apimanagement/v20191201:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20191201preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20200601preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20201201:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20210101preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20210401preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20210801:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20211201preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20220401preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20220801:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20220901preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20230301preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20230501preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20230901preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20240501:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20240601preview:GatewayHostnameConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GatewayHostnameConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GatewayHostnameConfiguration = GatewayHostnameConfiguration;
/** @internal */
GatewayHostnameConfiguration.__pulumiType = 'azure-native:apimanagement:GatewayHostnameConfiguration';
