// @pulumi/azure-native/apimanagement/workspaceApiOperation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceApiOperation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * API Operation details.
 * Azure REST API version: 2022-09-01-preview.
 *
 * Other available API versions: 2023-03-01-preview, 2023-05-01-preview, 2023-09-01-preview, 2024-05-01, 2024-06-01-preview.
 */
class WorkspaceApiOperation extends pulumi.CustomResource {
    /**
     * Get an existing WorkspaceApiOperation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkspaceApiOperation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkspaceApiOperation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkspaceApiOperation.__pulumiType;
    }
    /**
     * Create a WorkspaceApiOperation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.apiId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'apiId'");
            }
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.method === undefined) && !opts.urn) {
                throw new Error("Missing required property 'method'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            if ((!args || args.urlTemplate === undefined) && !opts.urn) {
                throw new Error("Missing required property 'urlTemplate'");
            }
            if ((!args || args.workspaceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceId'");
            }
            resourceInputs["apiId"] = args ? args.apiId : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["method"] = args ? args.method : undefined;
            resourceInputs["operationId"] = args ? args.operationId : undefined;
            resourceInputs["policies"] = args ? args.policies : undefined;
            resourceInputs["request"] = args ? args.request : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["responses"] = args ? args.responses : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["templateParameters"] = args ? args.templateParameters : undefined;
            resourceInputs["urlTemplate"] = args ? args.urlTemplate : undefined;
            resourceInputs["workspaceId"] = args ? args.workspaceId : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["method"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["policies"] = undefined /*out*/;
            resourceInputs["request"] = undefined /*out*/;
            resourceInputs["responses"] = undefined /*out*/;
            resourceInputs["templateParameters"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["urlTemplate"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:apimanagement/v20220901preview:WorkspaceApiOperation" }, { type: "azure-native:apimanagement/v20230301preview:WorkspaceApiOperation" }, { type: "azure-native:apimanagement/v20230501preview:WorkspaceApiOperation" }, { type: "azure-native:apimanagement/v20230901preview:WorkspaceApiOperation" }, { type: "azure-native:apimanagement/v20240501:WorkspaceApiOperation" }, { type: "azure-native:apimanagement/v20240601preview:WorkspaceApiOperation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkspaceApiOperation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkspaceApiOperation = WorkspaceApiOperation;
/** @internal */
WorkspaceApiOperation.__pulumiType = 'azure-native:apimanagement:WorkspaceApiOperation';
