// @pulumi/azure/storage/accountQueueProperties.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountQueueProperties = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages the Queue Properties of an Azure Storage Account.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleAccount = new azure.storage.Account("example", {
 *     name: "storageaccountname",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     accountTier: "Standard",
 *     accountReplicationType: "GRS",
 *     tags: {
 *         environment: "staging",
 *     },
 * });
 * const exampleAccountQueueProperties = new azure.storage.AccountQueueProperties("example", {
 *     storageAccountId: exampleAccount.id,
 *     corsRules: [{
 *         allowedOrigins: ["http://www.example.com"],
 *         exposedHeaders: ["x-tempo-*"],
 *         allowedHeaders: ["x-tempo-*"],
 *         allowedMethods: [
 *             "GET",
 *             "PUT",
 *         ],
 *         maxAgeInSeconds: 500,
 *     }],
 *     logging: {
 *         version: "1.0",
 *         "delete": true,
 *         read: true,
 *         write: true,
 *         retentionPolicyDays: 7,
 *     },
 *     hourMetrics: {
 *         version: "1.0",
 *         retentionPolicyDays: 7,
 *     },
 *     minuteMetrics: {
 *         version: "1.0",
 *         retentionPolicyDays: 7,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Storage Account Queue Properties can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:storage/accountQueueProperties:AccountQueueProperties queueprops /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myresourcegroup/providers/Microsoft.Storage/storageAccounts/myaccount
 * ```
 */
class AccountQueueProperties extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["corsRules"] = state ? state.corsRules : undefined;
            resourceInputs["hourMetrics"] = state ? state.hourMetrics : undefined;
            resourceInputs["logging"] = state ? state.logging : undefined;
            resourceInputs["minuteMetrics"] = state ? state.minuteMetrics : undefined;
            resourceInputs["storageAccountId"] = state ? state.storageAccountId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.storageAccountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageAccountId'");
            }
            resourceInputs["corsRules"] = args ? args.corsRules : undefined;
            resourceInputs["hourMetrics"] = args ? args.hourMetrics : undefined;
            resourceInputs["logging"] = args ? args.logging : undefined;
            resourceInputs["minuteMetrics"] = args ? args.minuteMetrics : undefined;
            resourceInputs["storageAccountId"] = args ? args.storageAccountId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AccountQueueProperties.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing AccountQueueProperties resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AccountQueueProperties(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AccountQueueProperties.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccountQueueProperties.__pulumiType;
    }
}
exports.AccountQueueProperties = AccountQueueProperties;
/** @internal */
AccountQueueProperties.__pulumiType = 'azure:storage/accountQueueProperties:AccountQueueProperties';
