// @pulumi/azure/storage/moverAgent.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MoverAgent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Storage Mover Agent.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "East US",
 * });
 * const exampleMover = new azure.storage.Mover("example", {
 *     name: "example-ssm",
 *     resourceGroupName: example.name,
 * });
 * const exampleMoverAgent = new azure.storage.MoverAgent("example", {
 *     name: "example-sa",
 *     storageMoverId: exampleMover.id,
 *     arcVirtualMachineId: pulumi.interpolate`${example.id}/providers/Microsoft.HybridCompute/machines/examples-hybridComputeName`,
 *     arcVirtualMachineUuid: "3bb2c024-eba9-4d18-9e7a-1d772fcc5fe9",
 *     description: "Example Agent Description",
 * });
 * ```
 *
 * ## Import
 *
 * Storage Mover Agent can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:storage/moverAgent:MoverAgent example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.StorageMover/storageMovers/storageMover1/agents/agent1
 * ```
 */
class MoverAgent extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arcVirtualMachineId"] = state ? state.arcVirtualMachineId : undefined;
            resourceInputs["arcVirtualMachineUuid"] = state ? state.arcVirtualMachineUuid : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["storageMoverId"] = state ? state.storageMoverId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.arcVirtualMachineId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'arcVirtualMachineId'");
            }
            if ((!args || args.arcVirtualMachineUuid === undefined) && !opts.urn) {
                throw new Error("Missing required property 'arcVirtualMachineUuid'");
            }
            if ((!args || args.storageMoverId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageMoverId'");
            }
            resourceInputs["arcVirtualMachineId"] = args ? args.arcVirtualMachineId : undefined;
            resourceInputs["arcVirtualMachineUuid"] = args ? args.arcVirtualMachineUuid : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["storageMoverId"] = args ? args.storageMoverId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MoverAgent.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MoverAgent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MoverAgent(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MoverAgent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MoverAgent.__pulumiType;
    }
}
exports.MoverAgent = MoverAgent;
/** @internal */
MoverAgent.__pulumiType = 'azure:storage/moverAgent:MoverAgent';
