// @pulumi/azure-compliance-policies/azurenative/costmanagement/v20221005preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewByScope = exports.View = exports.TagInheritanceSetting = exports.MarkupRule = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20221005preview_1 = require("@pulumi/azure-native/costmanagement/v20221005preview");
const v20221005preview_2 = require("@pulumi/azure-native/costmanagement/v20221005preview");
const v20221005preview_3 = require("@pulumi/azure-native/costmanagement/v20221005preview");
const v20221005preview_4 = require("@pulumi/azure-native/costmanagement/v20221005preview");
var MarkupRule;
(function (MarkupRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20221005preview.MarkupRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MarkupRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-costmanagement-v20221005preview-markuprule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20221005preview.MarkupRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221005preview_1.MarkupRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MarkupRule shouldn't use an unstable API (costmanagement.v20221005preview.MarkupRule).");
            }),
        },
        vendors: ["azure"],
        services: ["costmanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MarkupRule || (MarkupRule = {}));
exports.MarkupRule = MarkupRule;
var TagInheritanceSetting;
(function (TagInheritanceSetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20221005preview.TagInheritanceSetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TagInheritanceSetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-costmanagement-v20221005preview-taginheritancesetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20221005preview.TagInheritanceSetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221005preview_2.TagInheritanceSetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TagInheritanceSetting shouldn't use an unstable API (costmanagement.v20221005preview.TagInheritanceSetting).");
            }),
        },
        vendors: ["azure"],
        services: ["costmanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TagInheritanceSetting || (TagInheritanceSetting = {}));
exports.TagInheritanceSetting = TagInheritanceSetting;
var View;
(function (View) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20221005preview.View).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    View.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-costmanagement-v20221005preview-view-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20221005preview.View).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221005preview_3.View, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure View shouldn't use an unstable API (costmanagement.v20221005preview.View). A compatible replacement can be found at 'costmanagement.View'.");
            }),
        },
        vendors: ["azure"],
        services: ["costmanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(View || (View = {}));
exports.View = View;
var ViewByScope;
(function (ViewByScope) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20221005preview.ViewByScope).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ViewByScope.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-costmanagement-v20221005preview-viewbyscope-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20221005preview.ViewByScope).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221005preview_4.ViewByScope, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ViewByScope shouldn't use an unstable API (costmanagement.v20221005preview.ViewByScope). A compatible replacement can be found at 'costmanagement.ViewByScope'.");
            }),
        },
        vendors: ["azure"],
        services: ["costmanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ViewByScope || (ViewByScope = {}));
exports.ViewByScope = ViewByScope;
