// @pulumi/azure-compliance-policies/azurenative/securityinsights/v20240101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreatIntelligenceAlertRule = exports.TIDataConnector = exports.SentinelOnboardingState = exports.ScheduledAlertRule = exports.RestApiPollerDataConnector = exports.OfficePowerBIDataConnector = exports.OfficeIRMDataConnector = exports.OfficeDataConnector = exports.OfficeATPDataConnector = exports.Office365ProjectDataConnector = exports.NrtAlertRule = exports.MicrosoftSecurityIncidentCreationAlertRule = exports.MicrosoftPurviewInformationProtectionDataConnector = exports.Metadata = exports.MTPDataConnector = exports.MSTIDataConnector = exports.MLBehaviorAnalyticsAlertRule = exports.MDATPDataConnector = exports.MCASDataConnector = exports.IoTDataConnector = exports.IncidentTask = exports.IncidentRelation = exports.IncidentComment = exports.Incident = exports.HuntRelation = exports.HuntComment = exports.Hunt = exports.GCPDataConnector = exports.FusionAlertRule = exports.FileImport = exports.EyesOn = exports.EntityAnalytics = exports.Dynamics365DataConnector = exports.CustomizableConnectorDefinition = exports.ContentTemplate = exports.ContentPackage = exports.CodelessUiDataConnector = exports.CodelessApiPollingDataConnector = exports.BookmarkRelation = exports.Bookmark = exports.AwsS3DataConnector = exports.AwsCloudTrailDataConnector = exports.AutomationRule = exports.AnomalySecurityMLAnalyticsSettings = exports.Anomalies = exports.ActivityCustomEntityQuery = exports.Action = exports.ASCDataConnector = exports.AATPDataConnector = exports.AADDataConnector = void 0;
exports.WorkspaceManagerMember = exports.WorkspaceManagerGroup = exports.WorkspaceManagerConfiguration = exports.WorkspaceManagerAssignment = exports.WatchlistItem = exports.Watchlist = exports.Ueba = exports.TiTaxiiDataConnector = exports.ThreatIntelligenceIndicator = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240101preview_1 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_2 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_3 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_4 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_5 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_6 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_7 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_8 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_9 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_10 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_11 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_12 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_13 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_14 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_15 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_16 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_17 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_18 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_19 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_20 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_21 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_22 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_23 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_24 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_25 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_26 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_27 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_28 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_29 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_30 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_31 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_32 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_33 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_34 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_35 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_36 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_37 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_38 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_39 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_40 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_41 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_42 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_43 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_44 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_45 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_46 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_47 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_48 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_49 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_50 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_51 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_52 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_53 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_54 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_55 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_56 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_57 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_58 = require("@pulumi/azure-native/securityinsights/v20240101preview");
const v20240101preview_59 = require("@pulumi/azure-native/securityinsights/v20240101preview");
var AADDataConnector;
(function (AADDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AADDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AADDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-aaddataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AADDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_1.AADDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AADDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.AADDataConnector). A compatible replacement can be found at 'securityinsights.AADDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AADDataConnector || (AADDataConnector = {}));
exports.AADDataConnector = AADDataConnector;
var AATPDataConnector;
(function (AATPDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AATPDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AATPDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-aatpdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AATPDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_2.AATPDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AATPDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.AATPDataConnector). A compatible replacement can be found at 'securityinsights.AATPDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AATPDataConnector || (AATPDataConnector = {}));
exports.AATPDataConnector = AATPDataConnector;
var ASCDataConnector;
(function (ASCDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ASCDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ASCDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-ascdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ASCDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_3.ASCDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ASCDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.ASCDataConnector). A compatible replacement can be found at 'securityinsights.ASCDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ASCDataConnector || (ASCDataConnector = {}));
exports.ASCDataConnector = ASCDataConnector;
var Action;
(function (Action) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Action).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Action.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-action-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Action).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_4.Action, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Action shouldn't use an unstable API (securityinsights.v20240101preview.Action). A compatible replacement can be found at 'securityinsights.Action'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Action || (Action = {}));
exports.Action = Action;
var ActivityCustomEntityQuery;
(function (ActivityCustomEntityQuery) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ActivityCustomEntityQuery).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ActivityCustomEntityQuery.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-activitycustomentityquery-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ActivityCustomEntityQuery).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_5.ActivityCustomEntityQuery, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ActivityCustomEntityQuery shouldn't use an unstable API (securityinsights.v20240101preview.ActivityCustomEntityQuery). A compatible replacement can be found at 'securityinsights.ActivityCustomEntityQuery'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ActivityCustomEntityQuery || (ActivityCustomEntityQuery = {}));
exports.ActivityCustomEntityQuery = ActivityCustomEntityQuery;
var Anomalies;
(function (Anomalies) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Anomalies).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Anomalies.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-anomalies-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Anomalies).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_6.Anomalies, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Anomalies shouldn't use an unstable API (securityinsights.v20240101preview.Anomalies). A compatible replacement can be found at 'securityinsights.Anomalies'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Anomalies || (Anomalies = {}));
exports.Anomalies = Anomalies;
var AnomalySecurityMLAnalyticsSettings;
(function (AnomalySecurityMLAnalyticsSettings) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AnomalySecurityMLAnalyticsSettings).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AnomalySecurityMLAnalyticsSettings.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-anomalysecuritymlanalyticssettings-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AnomalySecurityMLAnalyticsSettings).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_7.AnomalySecurityMLAnalyticsSettings, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AnomalySecurityMLAnalyticsSettings shouldn't use an unstable API (securityinsights.v20240101preview.AnomalySecurityMLAnalyticsSettings). A compatible replacement can be found at 'securityinsights.AnomalySecurityMLAnalyticsSettings'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AnomalySecurityMLAnalyticsSettings || (AnomalySecurityMLAnalyticsSettings = {}));
exports.AnomalySecurityMLAnalyticsSettings = AnomalySecurityMLAnalyticsSettings;
var AutomationRule;
(function (AutomationRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AutomationRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AutomationRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-automationrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AutomationRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_8.AutomationRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AutomationRule shouldn't use an unstable API (securityinsights.v20240101preview.AutomationRule). A compatible replacement can be found at 'securityinsights.AutomationRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AutomationRule || (AutomationRule = {}));
exports.AutomationRule = AutomationRule;
var AwsCloudTrailDataConnector;
(function (AwsCloudTrailDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AwsCloudTrailDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AwsCloudTrailDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-awscloudtraildataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AwsCloudTrailDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_9.AwsCloudTrailDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AwsCloudTrailDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.AwsCloudTrailDataConnector). A compatible replacement can be found at 'securityinsights.AwsCloudTrailDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AwsCloudTrailDataConnector || (AwsCloudTrailDataConnector = {}));
exports.AwsCloudTrailDataConnector = AwsCloudTrailDataConnector;
var AwsS3DataConnector;
(function (AwsS3DataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AwsS3DataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AwsS3DataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-awss3dataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.AwsS3DataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_10.AwsS3DataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AwsS3DataConnector shouldn't use an unstable API (securityinsights.v20240101preview.AwsS3DataConnector). A compatible replacement can be found at 'securityinsights.AwsS3DataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AwsS3DataConnector || (AwsS3DataConnector = {}));
exports.AwsS3DataConnector = AwsS3DataConnector;
var Bookmark;
(function (Bookmark) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Bookmark).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Bookmark.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-bookmark-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Bookmark).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_11.Bookmark, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Bookmark shouldn't use an unstable API (securityinsights.v20240101preview.Bookmark). A compatible replacement can be found at 'securityinsights.Bookmark'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Bookmark || (Bookmark = {}));
exports.Bookmark = Bookmark;
var BookmarkRelation;
(function (BookmarkRelation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.BookmarkRelation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BookmarkRelation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-bookmarkrelation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.BookmarkRelation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_12.BookmarkRelation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BookmarkRelation shouldn't use an unstable API (securityinsights.v20240101preview.BookmarkRelation). A compatible replacement can be found at 'securityinsights.BookmarkRelation'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BookmarkRelation || (BookmarkRelation = {}));
exports.BookmarkRelation = BookmarkRelation;
var CodelessApiPollingDataConnector;
(function (CodelessApiPollingDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.CodelessApiPollingDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CodelessApiPollingDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-codelessapipollingdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.CodelessApiPollingDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_13.CodelessApiPollingDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CodelessApiPollingDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.CodelessApiPollingDataConnector). A compatible replacement can be found at 'securityinsights.CodelessApiPollingDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CodelessApiPollingDataConnector || (CodelessApiPollingDataConnector = {}));
exports.CodelessApiPollingDataConnector = CodelessApiPollingDataConnector;
var CodelessUiDataConnector;
(function (CodelessUiDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.CodelessUiDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CodelessUiDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-codelessuidataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.CodelessUiDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_14.CodelessUiDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CodelessUiDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.CodelessUiDataConnector). A compatible replacement can be found at 'securityinsights.CodelessUiDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CodelessUiDataConnector || (CodelessUiDataConnector = {}));
exports.CodelessUiDataConnector = CodelessUiDataConnector;
var ContentPackage;
(function (ContentPackage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ContentPackage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ContentPackage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-contentpackage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ContentPackage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_15.ContentPackage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ContentPackage shouldn't use an unstable API (securityinsights.v20240101preview.ContentPackage). A compatible replacement can be found at 'securityinsights.ContentPackage'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ContentPackage || (ContentPackage = {}));
exports.ContentPackage = ContentPackage;
var ContentTemplate;
(function (ContentTemplate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ContentTemplate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ContentTemplate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-contenttemplate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ContentTemplate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_16.ContentTemplate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ContentTemplate shouldn't use an unstable API (securityinsights.v20240101preview.ContentTemplate). A compatible replacement can be found at 'securityinsights.ContentTemplate'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ContentTemplate || (ContentTemplate = {}));
exports.ContentTemplate = ContentTemplate;
var CustomizableConnectorDefinition;
(function (CustomizableConnectorDefinition) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.CustomizableConnectorDefinition).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomizableConnectorDefinition.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-customizableconnectordefinition-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.CustomizableConnectorDefinition).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_17.CustomizableConnectorDefinition, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomizableConnectorDefinition shouldn't use an unstable API (securityinsights.v20240101preview.CustomizableConnectorDefinition). A compatible replacement can be found at 'securityinsights.CustomizableConnectorDefinition'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomizableConnectorDefinition || (CustomizableConnectorDefinition = {}));
exports.CustomizableConnectorDefinition = CustomizableConnectorDefinition;
var Dynamics365DataConnector;
(function (Dynamics365DataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Dynamics365DataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Dynamics365DataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-dynamics365dataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Dynamics365DataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_18.Dynamics365DataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Dynamics365DataConnector shouldn't use an unstable API (securityinsights.v20240101preview.Dynamics365DataConnector). A compatible replacement can be found at 'securityinsights.Dynamics365DataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Dynamics365DataConnector || (Dynamics365DataConnector = {}));
exports.Dynamics365DataConnector = Dynamics365DataConnector;
var EntityAnalytics;
(function (EntityAnalytics) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.EntityAnalytics).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EntityAnalytics.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-entityanalytics-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.EntityAnalytics).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_19.EntityAnalytics, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EntityAnalytics shouldn't use an unstable API (securityinsights.v20240101preview.EntityAnalytics). A compatible replacement can be found at 'securityinsights.EntityAnalytics'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EntityAnalytics || (EntityAnalytics = {}));
exports.EntityAnalytics = EntityAnalytics;
var EyesOn;
(function (EyesOn) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.EyesOn).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EyesOn.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-eyeson-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.EyesOn).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_20.EyesOn, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EyesOn shouldn't use an unstable API (securityinsights.v20240101preview.EyesOn). A compatible replacement can be found at 'securityinsights.EyesOn'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EyesOn || (EyesOn = {}));
exports.EyesOn = EyesOn;
var FileImport;
(function (FileImport) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.FileImport).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FileImport.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-fileimport-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.FileImport).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_21.FileImport, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FileImport shouldn't use an unstable API (securityinsights.v20240101preview.FileImport). A compatible replacement can be found at 'securityinsights.FileImport'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FileImport || (FileImport = {}));
exports.FileImport = FileImport;
var FusionAlertRule;
(function (FusionAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.FusionAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FusionAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-fusionalertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.FusionAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_22.FusionAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FusionAlertRule shouldn't use an unstable API (securityinsights.v20240101preview.FusionAlertRule). A compatible replacement can be found at 'securityinsights.FusionAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FusionAlertRule || (FusionAlertRule = {}));
exports.FusionAlertRule = FusionAlertRule;
var GCPDataConnector;
(function (GCPDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.GCPDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GCPDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-gcpdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.GCPDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_23.GCPDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GCPDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.GCPDataConnector). A compatible replacement can be found at 'securityinsights.GCPDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GCPDataConnector || (GCPDataConnector = {}));
exports.GCPDataConnector = GCPDataConnector;
var Hunt;
(function (Hunt) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Hunt).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Hunt.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-hunt-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Hunt).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_24.Hunt, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Hunt shouldn't use an unstable API (securityinsights.v20240101preview.Hunt). A compatible replacement can be found at 'securityinsights.Hunt'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Hunt || (Hunt = {}));
exports.Hunt = Hunt;
var HuntComment;
(function (HuntComment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.HuntComment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    HuntComment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-huntcomment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.HuntComment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_25.HuntComment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure HuntComment shouldn't use an unstable API (securityinsights.v20240101preview.HuntComment). A compatible replacement can be found at 'securityinsights.HuntComment'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(HuntComment || (HuntComment = {}));
exports.HuntComment = HuntComment;
var HuntRelation;
(function (HuntRelation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.HuntRelation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    HuntRelation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-huntrelation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.HuntRelation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_26.HuntRelation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure HuntRelation shouldn't use an unstable API (securityinsights.v20240101preview.HuntRelation). A compatible replacement can be found at 'securityinsights.HuntRelation'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(HuntRelation || (HuntRelation = {}));
exports.HuntRelation = HuntRelation;
var Incident;
(function (Incident) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Incident).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Incident.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-incident-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Incident).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_27.Incident, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Incident shouldn't use an unstable API (securityinsights.v20240101preview.Incident). A compatible replacement can be found at 'securityinsights.Incident'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Incident || (Incident = {}));
exports.Incident = Incident;
var IncidentComment;
(function (IncidentComment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.IncidentComment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IncidentComment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-incidentcomment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.IncidentComment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_28.IncidentComment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IncidentComment shouldn't use an unstable API (securityinsights.v20240101preview.IncidentComment). A compatible replacement can be found at 'securityinsights.IncidentComment'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IncidentComment || (IncidentComment = {}));
exports.IncidentComment = IncidentComment;
var IncidentRelation;
(function (IncidentRelation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.IncidentRelation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IncidentRelation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-incidentrelation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.IncidentRelation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_29.IncidentRelation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IncidentRelation shouldn't use an unstable API (securityinsights.v20240101preview.IncidentRelation). A compatible replacement can be found at 'securityinsights.IncidentRelation'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IncidentRelation || (IncidentRelation = {}));
exports.IncidentRelation = IncidentRelation;
var IncidentTask;
(function (IncidentTask) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.IncidentTask).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IncidentTask.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-incidenttask-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.IncidentTask).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_30.IncidentTask, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IncidentTask shouldn't use an unstable API (securityinsights.v20240101preview.IncidentTask). A compatible replacement can be found at 'securityinsights.IncidentTask'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IncidentTask || (IncidentTask = {}));
exports.IncidentTask = IncidentTask;
var IoTDataConnector;
(function (IoTDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.IoTDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IoTDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-iotdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.IoTDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_31.IoTDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IoTDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.IoTDataConnector). A compatible replacement can be found at 'securityinsights.IoTDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IoTDataConnector || (IoTDataConnector = {}));
exports.IoTDataConnector = IoTDataConnector;
var MCASDataConnector;
(function (MCASDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MCASDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MCASDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-mcasdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MCASDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_32.MCASDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MCASDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.MCASDataConnector). A compatible replacement can be found at 'securityinsights.MCASDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MCASDataConnector || (MCASDataConnector = {}));
exports.MCASDataConnector = MCASDataConnector;
var MDATPDataConnector;
(function (MDATPDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MDATPDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MDATPDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-mdatpdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MDATPDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_33.MDATPDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MDATPDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.MDATPDataConnector). A compatible replacement can be found at 'securityinsights.MDATPDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MDATPDataConnector || (MDATPDataConnector = {}));
exports.MDATPDataConnector = MDATPDataConnector;
var MLBehaviorAnalyticsAlertRule;
(function (MLBehaviorAnalyticsAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MLBehaviorAnalyticsAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MLBehaviorAnalyticsAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-mlbehavioranalyticsalertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MLBehaviorAnalyticsAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_34.MLBehaviorAnalyticsAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MLBehaviorAnalyticsAlertRule shouldn't use an unstable API (securityinsights.v20240101preview.MLBehaviorAnalyticsAlertRule). A compatible replacement can be found at 'securityinsights.MLBehaviorAnalyticsAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MLBehaviorAnalyticsAlertRule || (MLBehaviorAnalyticsAlertRule = {}));
exports.MLBehaviorAnalyticsAlertRule = MLBehaviorAnalyticsAlertRule;
var MSTIDataConnector;
(function (MSTIDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MSTIDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MSTIDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-mstidataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MSTIDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_35.MSTIDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MSTIDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.MSTIDataConnector). A compatible replacement can be found at 'securityinsights.MSTIDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MSTIDataConnector || (MSTIDataConnector = {}));
exports.MSTIDataConnector = MSTIDataConnector;
var MTPDataConnector;
(function (MTPDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MTPDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MTPDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-mtpdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MTPDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_36.MTPDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MTPDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.MTPDataConnector). A compatible replacement can be found at 'securityinsights.MTPDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MTPDataConnector || (MTPDataConnector = {}));
exports.MTPDataConnector = MTPDataConnector;
var Metadata;
(function (Metadata) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Metadata).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Metadata.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-metadata-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Metadata).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_37.Metadata, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Metadata shouldn't use an unstable API (securityinsights.v20240101preview.Metadata). A compatible replacement can be found at 'securityinsights.Metadata'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Metadata || (Metadata = {}));
exports.Metadata = Metadata;
var MicrosoftPurviewInformationProtectionDataConnector;
(function (MicrosoftPurviewInformationProtectionDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MicrosoftPurviewInformationProtectionDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MicrosoftPurviewInformationProtectionDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-microsoftpurviewinformationprotectiondataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MicrosoftPurviewInformationProtectionDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_38.MicrosoftPurviewInformationProtectionDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MicrosoftPurviewInformationProtectionDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.MicrosoftPurviewInformationProtectionDataConnector). A compatible replacement can be found at 'securityinsights.MicrosoftPurviewInformationProtectionDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MicrosoftPurviewInformationProtectionDataConnector || (MicrosoftPurviewInformationProtectionDataConnector = {}));
exports.MicrosoftPurviewInformationProtectionDataConnector = MicrosoftPurviewInformationProtectionDataConnector;
var MicrosoftSecurityIncidentCreationAlertRule;
(function (MicrosoftSecurityIncidentCreationAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MicrosoftSecurityIncidentCreationAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MicrosoftSecurityIncidentCreationAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-microsoftsecurityincidentcreationalertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.MicrosoftSecurityIncidentCreationAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_39.MicrosoftSecurityIncidentCreationAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MicrosoftSecurityIncidentCreationAlertRule shouldn't use an unstable API (securityinsights.v20240101preview.MicrosoftSecurityIncidentCreationAlertRule). A compatible replacement can be found at 'securityinsights.MicrosoftSecurityIncidentCreationAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MicrosoftSecurityIncidentCreationAlertRule || (MicrosoftSecurityIncidentCreationAlertRule = {}));
exports.MicrosoftSecurityIncidentCreationAlertRule = MicrosoftSecurityIncidentCreationAlertRule;
var NrtAlertRule;
(function (NrtAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.NrtAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    NrtAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-nrtalertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.NrtAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_40.NrtAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure NrtAlertRule shouldn't use an unstable API (securityinsights.v20240101preview.NrtAlertRule). A compatible replacement can be found at 'securityinsights.NrtAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(NrtAlertRule || (NrtAlertRule = {}));
exports.NrtAlertRule = NrtAlertRule;
var Office365ProjectDataConnector;
(function (Office365ProjectDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Office365ProjectDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Office365ProjectDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-office365projectdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Office365ProjectDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_41.Office365ProjectDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Office365ProjectDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.Office365ProjectDataConnector). A compatible replacement can be found at 'securityinsights.Office365ProjectDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Office365ProjectDataConnector || (Office365ProjectDataConnector = {}));
exports.Office365ProjectDataConnector = Office365ProjectDataConnector;
var OfficeATPDataConnector;
(function (OfficeATPDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.OfficeATPDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OfficeATPDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-officeatpdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.OfficeATPDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_42.OfficeATPDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OfficeATPDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.OfficeATPDataConnector). A compatible replacement can be found at 'securityinsights.OfficeATPDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OfficeATPDataConnector || (OfficeATPDataConnector = {}));
exports.OfficeATPDataConnector = OfficeATPDataConnector;
var OfficeDataConnector;
(function (OfficeDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.OfficeDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OfficeDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-officedataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.OfficeDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_43.OfficeDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OfficeDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.OfficeDataConnector). A compatible replacement can be found at 'securityinsights.OfficeDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OfficeDataConnector || (OfficeDataConnector = {}));
exports.OfficeDataConnector = OfficeDataConnector;
var OfficeIRMDataConnector;
(function (OfficeIRMDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.OfficeIRMDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OfficeIRMDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-officeirmdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.OfficeIRMDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_44.OfficeIRMDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OfficeIRMDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.OfficeIRMDataConnector). A compatible replacement can be found at 'securityinsights.OfficeIRMDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OfficeIRMDataConnector || (OfficeIRMDataConnector = {}));
exports.OfficeIRMDataConnector = OfficeIRMDataConnector;
var OfficePowerBIDataConnector;
(function (OfficePowerBIDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.OfficePowerBIDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OfficePowerBIDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-officepowerbidataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.OfficePowerBIDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_45.OfficePowerBIDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OfficePowerBIDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.OfficePowerBIDataConnector). A compatible replacement can be found at 'securityinsights.OfficePowerBIDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OfficePowerBIDataConnector || (OfficePowerBIDataConnector = {}));
exports.OfficePowerBIDataConnector = OfficePowerBIDataConnector;
var RestApiPollerDataConnector;
(function (RestApiPollerDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.RestApiPollerDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RestApiPollerDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-restapipollerdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.RestApiPollerDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_46.RestApiPollerDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RestApiPollerDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.RestApiPollerDataConnector). A compatible replacement can be found at 'securityinsights.RestApiPollerDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RestApiPollerDataConnector || (RestApiPollerDataConnector = {}));
exports.RestApiPollerDataConnector = RestApiPollerDataConnector;
var ScheduledAlertRule;
(function (ScheduledAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ScheduledAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ScheduledAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-scheduledalertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ScheduledAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_47.ScheduledAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ScheduledAlertRule shouldn't use an unstable API (securityinsights.v20240101preview.ScheduledAlertRule). A compatible replacement can be found at 'securityinsights.ScheduledAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ScheduledAlertRule || (ScheduledAlertRule = {}));
exports.ScheduledAlertRule = ScheduledAlertRule;
var SentinelOnboardingState;
(function (SentinelOnboardingState) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.SentinelOnboardingState).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SentinelOnboardingState.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-sentinelonboardingstate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.SentinelOnboardingState).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_48.SentinelOnboardingState, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SentinelOnboardingState shouldn't use an unstable API (securityinsights.v20240101preview.SentinelOnboardingState). A compatible replacement can be found at 'securityinsights.SentinelOnboardingState'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SentinelOnboardingState || (SentinelOnboardingState = {}));
exports.SentinelOnboardingState = SentinelOnboardingState;
var TIDataConnector;
(function (TIDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.TIDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TIDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-tidataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.TIDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_49.TIDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TIDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.TIDataConnector). A compatible replacement can be found at 'securityinsights.TIDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TIDataConnector || (TIDataConnector = {}));
exports.TIDataConnector = TIDataConnector;
var ThreatIntelligenceAlertRule;
(function (ThreatIntelligenceAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ThreatIntelligenceAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ThreatIntelligenceAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-threatintelligencealertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ThreatIntelligenceAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_50.ThreatIntelligenceAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ThreatIntelligenceAlertRule shouldn't use an unstable API (securityinsights.v20240101preview.ThreatIntelligenceAlertRule). A compatible replacement can be found at 'securityinsights.ThreatIntelligenceAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ThreatIntelligenceAlertRule || (ThreatIntelligenceAlertRule = {}));
exports.ThreatIntelligenceAlertRule = ThreatIntelligenceAlertRule;
var ThreatIntelligenceIndicator;
(function (ThreatIntelligenceIndicator) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ThreatIntelligenceIndicator).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ThreatIntelligenceIndicator.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-threatintelligenceindicator-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.ThreatIntelligenceIndicator).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_51.ThreatIntelligenceIndicator, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ThreatIntelligenceIndicator shouldn't use an unstable API (securityinsights.v20240101preview.ThreatIntelligenceIndicator). A compatible replacement can be found at 'securityinsights.ThreatIntelligenceIndicator'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ThreatIntelligenceIndicator || (ThreatIntelligenceIndicator = {}));
exports.ThreatIntelligenceIndicator = ThreatIntelligenceIndicator;
var TiTaxiiDataConnector;
(function (TiTaxiiDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.TiTaxiiDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TiTaxiiDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-titaxiidataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.TiTaxiiDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_52.TiTaxiiDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TiTaxiiDataConnector shouldn't use an unstable API (securityinsights.v20240101preview.TiTaxiiDataConnector). A compatible replacement can be found at 'securityinsights.TiTaxiiDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TiTaxiiDataConnector || (TiTaxiiDataConnector = {}));
exports.TiTaxiiDataConnector = TiTaxiiDataConnector;
var Ueba;
(function (Ueba) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Ueba).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Ueba.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-ueba-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Ueba).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_53.Ueba, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Ueba shouldn't use an unstable API (securityinsights.v20240101preview.Ueba). A compatible replacement can be found at 'securityinsights.Ueba'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Ueba || (Ueba = {}));
exports.Ueba = Ueba;
var Watchlist;
(function (Watchlist) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Watchlist).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Watchlist.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-watchlist-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.Watchlist).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_54.Watchlist, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Watchlist shouldn't use an unstable API (securityinsights.v20240101preview.Watchlist). A compatible replacement can be found at 'securityinsights.Watchlist'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Watchlist || (Watchlist = {}));
exports.Watchlist = Watchlist;
var WatchlistItem;
(function (WatchlistItem) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WatchlistItem).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WatchlistItem.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-watchlistitem-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WatchlistItem).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_55.WatchlistItem, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WatchlistItem shouldn't use an unstable API (securityinsights.v20240101preview.WatchlistItem). A compatible replacement can be found at 'securityinsights.WatchlistItem'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WatchlistItem || (WatchlistItem = {}));
exports.WatchlistItem = WatchlistItem;
var WorkspaceManagerAssignment;
(function (WorkspaceManagerAssignment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WorkspaceManagerAssignment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceManagerAssignment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-workspacemanagerassignment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WorkspaceManagerAssignment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_56.WorkspaceManagerAssignment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceManagerAssignment shouldn't use an unstable API (securityinsights.v20240101preview.WorkspaceManagerAssignment). A compatible replacement can be found at 'securityinsights.WorkspaceManagerAssignment'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceManagerAssignment || (WorkspaceManagerAssignment = {}));
exports.WorkspaceManagerAssignment = WorkspaceManagerAssignment;
var WorkspaceManagerConfiguration;
(function (WorkspaceManagerConfiguration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WorkspaceManagerConfiguration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceManagerConfiguration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-workspacemanagerconfiguration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WorkspaceManagerConfiguration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_57.WorkspaceManagerConfiguration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceManagerConfiguration shouldn't use an unstable API (securityinsights.v20240101preview.WorkspaceManagerConfiguration). A compatible replacement can be found at 'securityinsights.WorkspaceManagerConfiguration'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceManagerConfiguration || (WorkspaceManagerConfiguration = {}));
exports.WorkspaceManagerConfiguration = WorkspaceManagerConfiguration;
var WorkspaceManagerGroup;
(function (WorkspaceManagerGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WorkspaceManagerGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceManagerGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-workspacemanagergroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WorkspaceManagerGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_58.WorkspaceManagerGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceManagerGroup shouldn't use an unstable API (securityinsights.v20240101preview.WorkspaceManagerGroup). A compatible replacement can be found at 'securityinsights.WorkspaceManagerGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceManagerGroup || (WorkspaceManagerGroup = {}));
exports.WorkspaceManagerGroup = WorkspaceManagerGroup;
var WorkspaceManagerMember;
(function (WorkspaceManagerMember) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WorkspaceManagerMember).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceManagerMember.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20240101preview-workspacemanagermember-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20240101preview.WorkspaceManagerMember).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_59.WorkspaceManagerMember, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceManagerMember shouldn't use an unstable API (securityinsights.v20240101preview.WorkspaceManagerMember). A compatible replacement can be found at 'securityinsights.WorkspaceManagerMember'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceManagerMember || (WorkspaceManagerMember = {}));
exports.WorkspaceManagerMember = WorkspaceManagerMember;
