// @pulumi/azure-compliance-policies/azurenative/security/v20210801preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Standard = exports.Assignment = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20210801preview_1 = require("@pulumi/azure-native/security/v20210801preview");
const v20210801preview_2 = require("@pulumi/azure-native/security/v20210801preview");
var Assignment;
(function (Assignment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20210801preview.Assignment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Assignment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20210801preview-assignment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20210801preview.Assignment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210801preview_1.Assignment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Assignment shouldn't use an unstable API (security.v20210801preview.Assignment). A compatible replacement can be found at 'security.Assignment'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Assignment || (Assignment = {}));
exports.Assignment = Assignment;
var Standard;
(function (Standard) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20210801preview.Standard).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Standard.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20210801preview-standard-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20210801preview.Standard).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210801preview_2.Standard, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Standard shouldn't use an unstable API (security.v20210801preview.Standard). A compatible replacement can be found at 'security.Standard'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Standard || (Standard = {}));
exports.Standard = Standard;
