// @pulumi/azure/dataprotection/backupPolicyBlobStorage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupPolicyBlobStorage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Backup Policy Blob Storage.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleBackupVault = new azure.dataprotection.BackupVault("example", {
 *     name: "example-backup-vault",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     datastoreType: "VaultStore",
 *     redundancy: "LocallyRedundant",
 * });
 * const exampleBackupPolicyBlobStorage = new azure.dataprotection.BackupPolicyBlobStorage("example", {
 *     name: "example-backup-policy",
 *     vaultId: exampleBackupVault.id,
 *     operationalDefaultRetentionDuration: "P30D",
 * });
 * ```
 *
 * ## Import
 *
 * Backup Policy Blob Storages can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:dataprotection/backupPolicyBlobStorage:BackupPolicyBlobStorage example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.DataProtection/backupVaults/vault1/backupPolicies/backupPolicy1
 * ```
 */
class BackupPolicyBlobStorage extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["backupRepeatingTimeIntervals"] = state ? state.backupRepeatingTimeIntervals : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["operationalDefaultRetentionDuration"] = state ? state.operationalDefaultRetentionDuration : undefined;
            resourceInputs["retentionRules"] = state ? state.retentionRules : undefined;
            resourceInputs["timeZone"] = state ? state.timeZone : undefined;
            resourceInputs["vaultDefaultRetentionDuration"] = state ? state.vaultDefaultRetentionDuration : undefined;
            resourceInputs["vaultId"] = state ? state.vaultId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.vaultId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vaultId'");
            }
            resourceInputs["backupRepeatingTimeIntervals"] = args ? args.backupRepeatingTimeIntervals : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["operationalDefaultRetentionDuration"] = args ? args.operationalDefaultRetentionDuration : undefined;
            resourceInputs["retentionRules"] = args ? args.retentionRules : undefined;
            resourceInputs["timeZone"] = args ? args.timeZone : undefined;
            resourceInputs["vaultDefaultRetentionDuration"] = args ? args.vaultDefaultRetentionDuration : undefined;
            resourceInputs["vaultId"] = args ? args.vaultId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BackupPolicyBlobStorage.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing BackupPolicyBlobStorage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BackupPolicyBlobStorage(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BackupPolicyBlobStorage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BackupPolicyBlobStorage.__pulumiType;
    }
}
exports.BackupPolicyBlobStorage = BackupPolicyBlobStorage;
/** @internal */
BackupPolicyBlobStorage.__pulumiType = 'azure:dataprotection/backupPolicyBlobStorage:BackupPolicyBlobStorage';
