// @pulumi/azure-compliance-policies/azurenative/testbase/v20220401preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestBaseAccount = exports.Package = exports.FavoriteProcess = exports.CustomerEvent = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20220401preview_1 = require("@pulumi/azure-native/testbase/v20220401preview");
const v20220401preview_2 = require("@pulumi/azure-native/testbase/v20220401preview");
const v20220401preview_3 = require("@pulumi/azure-native/testbase/v20220401preview");
const v20220401preview_4 = require("@pulumi/azure-native/testbase/v20220401preview");
var CustomerEvent;
(function (CustomerEvent) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20220401preview.CustomerEvent).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomerEvent.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20220401preview-customerevent-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20220401preview.CustomerEvent).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220401preview_1.CustomerEvent, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomerEvent shouldn't use an unstable API (testbase.v20220401preview.CustomerEvent). A compatible replacement can be found at 'testbase.CustomerEvent'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomerEvent || (CustomerEvent = {}));
exports.CustomerEvent = CustomerEvent;
var FavoriteProcess;
(function (FavoriteProcess) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20220401preview.FavoriteProcess).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FavoriteProcess.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20220401preview-favoriteprocess-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20220401preview.FavoriteProcess).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220401preview_2.FavoriteProcess, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FavoriteProcess shouldn't use an unstable API (testbase.v20220401preview.FavoriteProcess). A compatible replacement can be found at 'testbase.FavoriteProcess'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FavoriteProcess || (FavoriteProcess = {}));
exports.FavoriteProcess = FavoriteProcess;
var Package;
(function (Package) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20220401preview.Package).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Package.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20220401preview-package-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20220401preview.Package).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220401preview_3.Package, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Package shouldn't use an unstable API (testbase.v20220401preview.Package). A compatible replacement can be found at 'testbase.Package'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Package || (Package = {}));
exports.Package = Package;
var TestBaseAccount;
(function (TestBaseAccount) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20220401preview.TestBaseAccount).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TestBaseAccount.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20220401preview-testbaseaccount-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20220401preview.TestBaseAccount).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220401preview_4.TestBaseAccount, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TestBaseAccount shouldn't use an unstable API (testbase.v20220401preview.TestBaseAccount). A compatible replacement can be found at 'testbase.TestBaseAccount'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TestBaseAccount || (TestBaseAccount = {}));
exports.TestBaseAccount = TestBaseAccount;
