// @pulumi/azure-compliance-policies/azurenative/apimanagement/v20230301preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPolicy = exports.ProductGroupLink = exports.ProductGroup = exports.ProductApiLink = exports.ProductApi = exports.Product = exports.PrivateEndpointConnectionByName = exports.PolicyFragment = exports.Policy = exports.OpenIdConnectProvider = exports.NotificationRecipientUser = exports.NotificationRecipientEmail = exports.NamedValue = exports.Logger = exports.IdentityProvider = exports.GroupUser = exports.Group = exports.GraphQLApiResolverPolicy = exports.GraphQLApiResolver = exports.GlobalSchema = exports.GatewayHostnameConfiguration = exports.GatewayCertificateAuthority = exports.GatewayApiEntityTag = exports.Gateway = exports.EmailTemplate = exports.Documentation = exports.Diagnostic = exports.ContentType = exports.ContentItem = exports.Certificate = exports.Cache = exports.Backend = exports.AuthorizationServer = exports.AuthorizationProvider = exports.AuthorizationAccessPolicy = exports.Authorization = exports.ApiWiki = exports.ApiVersionSet = exports.ApiTagDescription = exports.ApiSchema = exports.ApiRelease = exports.ApiPolicy = exports.ApiOperationPolicy = exports.ApiOperation = exports.ApiManagementService = exports.ApiIssueComment = exports.ApiIssueAttachment = exports.ApiIssue = exports.ApiDiagnostic = exports.Api = void 0;
exports.WorkspaceTagProductLink = exports.WorkspaceTagOperationLink = exports.WorkspaceTagApiLink = exports.WorkspaceTag = exports.WorkspaceSubscription = exports.WorkspaceProductPolicy = exports.WorkspaceProductGroupLink = exports.WorkspaceProductApiLink = exports.WorkspaceProduct = exports.WorkspacePolicyFragment = exports.WorkspacePolicy = exports.WorkspaceNotificationRecipientUser = exports.WorkspaceNotificationRecipientEmail = exports.WorkspaceNamedValue = exports.WorkspaceGroupUser = exports.WorkspaceGroup = exports.WorkspaceGlobalSchema = exports.WorkspaceApiVersionSet = exports.WorkspaceApiSchema = exports.WorkspaceApiRelease = exports.WorkspaceApiPolicy = exports.WorkspaceApiOperationPolicy = exports.WorkspaceApiOperation = exports.WorkspaceApi = exports.Workspace = exports.User = exports.TagProductLink = exports.TagOperationLink = exports.TagByProduct = exports.TagByOperation = exports.TagByApi = exports.TagApiLink = exports.Tag = exports.Subscription = exports.ProductWiki = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230301preview_1 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_2 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_3 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_4 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_5 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_6 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_7 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_8 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_9 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_10 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_11 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_12 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_13 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_14 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_15 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_16 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_17 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_18 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_19 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_20 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_21 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_22 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_23 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_24 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_25 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_26 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_27 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_28 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_29 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_30 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_31 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_32 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_33 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_34 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_35 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_36 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_37 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_38 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_39 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_40 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_41 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_42 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_43 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_44 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_45 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_46 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_47 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_48 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_49 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_50 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_51 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_52 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_53 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_54 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_55 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_56 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_57 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_58 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_59 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_60 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_61 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_62 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_63 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_64 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_65 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_66 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_67 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_68 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_69 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_70 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_71 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_72 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_73 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_74 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_75 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_76 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_77 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_78 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_79 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_80 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_81 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_82 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_83 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_84 = require("@pulumi/azure-native/apimanagement/v20230301preview");
const v20230301preview_85 = require("@pulumi/azure-native/apimanagement/v20230301preview");
var Api;
(function (Api) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Api).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Api.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-api-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Api).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_1.Api, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Api shouldn't use an unstable API (apimanagement.v20230301preview.Api). A compatible replacement can be found at 'apimanagement.Api'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Api || (Api = {}));
exports.Api = Api;
var ApiDiagnostic;
(function (ApiDiagnostic) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiDiagnostic).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiDiagnostic.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apidiagnostic-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiDiagnostic).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_2.ApiDiagnostic, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiDiagnostic shouldn't use an unstable API (apimanagement.v20230301preview.ApiDiagnostic). A compatible replacement can be found at 'apimanagement.ApiDiagnostic'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiDiagnostic || (ApiDiagnostic = {}));
exports.ApiDiagnostic = ApiDiagnostic;
var ApiIssue;
(function (ApiIssue) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiIssue).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiIssue.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apiissue-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiIssue).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_3.ApiIssue, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiIssue shouldn't use an unstable API (apimanagement.v20230301preview.ApiIssue). A compatible replacement can be found at 'apimanagement.ApiIssue'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiIssue || (ApiIssue = {}));
exports.ApiIssue = ApiIssue;
var ApiIssueAttachment;
(function (ApiIssueAttachment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiIssueAttachment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiIssueAttachment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apiissueattachment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiIssueAttachment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_4.ApiIssueAttachment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiIssueAttachment shouldn't use an unstable API (apimanagement.v20230301preview.ApiIssueAttachment). A compatible replacement can be found at 'apimanagement.ApiIssueAttachment'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiIssueAttachment || (ApiIssueAttachment = {}));
exports.ApiIssueAttachment = ApiIssueAttachment;
var ApiIssueComment;
(function (ApiIssueComment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiIssueComment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiIssueComment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apiissuecomment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiIssueComment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_5.ApiIssueComment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiIssueComment shouldn't use an unstable API (apimanagement.v20230301preview.ApiIssueComment). A compatible replacement can be found at 'apimanagement.ApiIssueComment'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiIssueComment || (ApiIssueComment = {}));
exports.ApiIssueComment = ApiIssueComment;
var ApiManagementService;
(function (ApiManagementService) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiManagementService).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiManagementService.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apimanagementservice-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiManagementService).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_6.ApiManagementService, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiManagementService shouldn't use an unstable API (apimanagement.v20230301preview.ApiManagementService). A compatible replacement can be found at 'apimanagement.ApiManagementService'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiManagementService || (ApiManagementService = {}));
exports.ApiManagementService = ApiManagementService;
var ApiOperation;
(function (ApiOperation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiOperation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiOperation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apioperation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiOperation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_7.ApiOperation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiOperation shouldn't use an unstable API (apimanagement.v20230301preview.ApiOperation). A compatible replacement can be found at 'apimanagement.ApiOperation'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiOperation || (ApiOperation = {}));
exports.ApiOperation = ApiOperation;
var ApiOperationPolicy;
(function (ApiOperationPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiOperationPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiOperationPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apioperationpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiOperationPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_8.ApiOperationPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiOperationPolicy shouldn't use an unstable API (apimanagement.v20230301preview.ApiOperationPolicy). A compatible replacement can be found at 'apimanagement.ApiOperationPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiOperationPolicy || (ApiOperationPolicy = {}));
exports.ApiOperationPolicy = ApiOperationPolicy;
var ApiPolicy;
(function (ApiPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apipolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_9.ApiPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiPolicy shouldn't use an unstable API (apimanagement.v20230301preview.ApiPolicy). A compatible replacement can be found at 'apimanagement.ApiPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiPolicy || (ApiPolicy = {}));
exports.ApiPolicy = ApiPolicy;
var ApiRelease;
(function (ApiRelease) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiRelease).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiRelease.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apirelease-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiRelease).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_10.ApiRelease, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiRelease shouldn't use an unstable API (apimanagement.v20230301preview.ApiRelease). A compatible replacement can be found at 'apimanagement.ApiRelease'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiRelease || (ApiRelease = {}));
exports.ApiRelease = ApiRelease;
var ApiSchema;
(function (ApiSchema) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiSchema).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiSchema.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apischema-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiSchema).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_11.ApiSchema, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiSchema shouldn't use an unstable API (apimanagement.v20230301preview.ApiSchema). A compatible replacement can be found at 'apimanagement.ApiSchema'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiSchema || (ApiSchema = {}));
exports.ApiSchema = ApiSchema;
var ApiTagDescription;
(function (ApiTagDescription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiTagDescription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiTagDescription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apitagdescription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiTagDescription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_12.ApiTagDescription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiTagDescription shouldn't use an unstable API (apimanagement.v20230301preview.ApiTagDescription). A compatible replacement can be found at 'apimanagement.ApiTagDescription'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiTagDescription || (ApiTagDescription = {}));
exports.ApiTagDescription = ApiTagDescription;
var ApiVersionSet;
(function (ApiVersionSet) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiVersionSet).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiVersionSet.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apiversionset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiVersionSet).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_13.ApiVersionSet, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiVersionSet shouldn't use an unstable API (apimanagement.v20230301preview.ApiVersionSet). A compatible replacement can be found at 'apimanagement.ApiVersionSet'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiVersionSet || (ApiVersionSet = {}));
exports.ApiVersionSet = ApiVersionSet;
var ApiWiki;
(function (ApiWiki) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiWiki).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiWiki.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-apiwiki-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ApiWiki).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_14.ApiWiki, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiWiki shouldn't use an unstable API (apimanagement.v20230301preview.ApiWiki). A compatible replacement can be found at 'apimanagement.ApiWiki'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiWiki || (ApiWiki = {}));
exports.ApiWiki = ApiWiki;
var Authorization;
(function (Authorization) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Authorization).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Authorization.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-authorization-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Authorization).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_15.Authorization, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Authorization shouldn't use an unstable API (apimanagement.v20230301preview.Authorization). A compatible replacement can be found at 'apimanagement.Authorization'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Authorization || (Authorization = {}));
exports.Authorization = Authorization;
var AuthorizationAccessPolicy;
(function (AuthorizationAccessPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.AuthorizationAccessPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AuthorizationAccessPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-authorizationaccesspolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.AuthorizationAccessPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_16.AuthorizationAccessPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AuthorizationAccessPolicy shouldn't use an unstable API (apimanagement.v20230301preview.AuthorizationAccessPolicy). A compatible replacement can be found at 'apimanagement.AuthorizationAccessPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AuthorizationAccessPolicy || (AuthorizationAccessPolicy = {}));
exports.AuthorizationAccessPolicy = AuthorizationAccessPolicy;
var AuthorizationProvider;
(function (AuthorizationProvider) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.AuthorizationProvider).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AuthorizationProvider.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-authorizationprovider-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.AuthorizationProvider).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_17.AuthorizationProvider, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AuthorizationProvider shouldn't use an unstable API (apimanagement.v20230301preview.AuthorizationProvider). A compatible replacement can be found at 'apimanagement.AuthorizationProvider'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AuthorizationProvider || (AuthorizationProvider = {}));
exports.AuthorizationProvider = AuthorizationProvider;
var AuthorizationServer;
(function (AuthorizationServer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.AuthorizationServer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AuthorizationServer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-authorizationserver-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.AuthorizationServer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_18.AuthorizationServer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AuthorizationServer shouldn't use an unstable API (apimanagement.v20230301preview.AuthorizationServer). A compatible replacement can be found at 'apimanagement.AuthorizationServer'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AuthorizationServer || (AuthorizationServer = {}));
exports.AuthorizationServer = AuthorizationServer;
var Backend;
(function (Backend) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Backend).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Backend.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-backend-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Backend).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_19.Backend, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Backend shouldn't use an unstable API (apimanagement.v20230301preview.Backend). A compatible replacement can be found at 'apimanagement.Backend'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Backend || (Backend = {}));
exports.Backend = Backend;
var Cache;
(function (Cache) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Cache).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Cache.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-cache-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Cache).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_20.Cache, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Cache shouldn't use an unstable API (apimanagement.v20230301preview.Cache). A compatible replacement can be found at 'apimanagement.Cache'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Cache || (Cache = {}));
exports.Cache = Cache;
var Certificate;
(function (Certificate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Certificate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Certificate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-certificate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Certificate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_21.Certificate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Certificate shouldn't use an unstable API (apimanagement.v20230301preview.Certificate). A compatible replacement can be found at 'apimanagement.Certificate'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Certificate || (Certificate = {}));
exports.Certificate = Certificate;
var ContentItem;
(function (ContentItem) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ContentItem).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ContentItem.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-contentitem-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ContentItem).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_22.ContentItem, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ContentItem shouldn't use an unstable API (apimanagement.v20230301preview.ContentItem). A compatible replacement can be found at 'apimanagement.ContentItem'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ContentItem || (ContentItem = {}));
exports.ContentItem = ContentItem;
var ContentType;
(function (ContentType) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ContentType).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ContentType.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-contenttype-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ContentType).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_23.ContentType, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ContentType shouldn't use an unstable API (apimanagement.v20230301preview.ContentType). A compatible replacement can be found at 'apimanagement.ContentType'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ContentType || (ContentType = {}));
exports.ContentType = ContentType;
var Diagnostic;
(function (Diagnostic) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Diagnostic).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Diagnostic.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-diagnostic-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Diagnostic).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_24.Diagnostic, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Diagnostic shouldn't use an unstable API (apimanagement.v20230301preview.Diagnostic). A compatible replacement can be found at 'apimanagement.Diagnostic'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Diagnostic || (Diagnostic = {}));
exports.Diagnostic = Diagnostic;
var Documentation;
(function (Documentation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Documentation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Documentation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-documentation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Documentation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_25.Documentation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Documentation shouldn't use an unstable API (apimanagement.v20230301preview.Documentation). A compatible replacement can be found at 'apimanagement.Documentation'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Documentation || (Documentation = {}));
exports.Documentation = Documentation;
var EmailTemplate;
(function (EmailTemplate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.EmailTemplate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EmailTemplate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-emailtemplate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.EmailTemplate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_26.EmailTemplate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EmailTemplate shouldn't use an unstable API (apimanagement.v20230301preview.EmailTemplate). A compatible replacement can be found at 'apimanagement.EmailTemplate'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EmailTemplate || (EmailTemplate = {}));
exports.EmailTemplate = EmailTemplate;
var Gateway;
(function (Gateway) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Gateway).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Gateway.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-gateway-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Gateway).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_27.Gateway, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Gateway shouldn't use an unstable API (apimanagement.v20230301preview.Gateway). A compatible replacement can be found at 'apimanagement.Gateway'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Gateway || (Gateway = {}));
exports.Gateway = Gateway;
var GatewayApiEntityTag;
(function (GatewayApiEntityTag) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GatewayApiEntityTag).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GatewayApiEntityTag.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-gatewayapientitytag-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GatewayApiEntityTag).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_28.GatewayApiEntityTag, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GatewayApiEntityTag shouldn't use an unstable API (apimanagement.v20230301preview.GatewayApiEntityTag). A compatible replacement can be found at 'apimanagement.GatewayApiEntityTag'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GatewayApiEntityTag || (GatewayApiEntityTag = {}));
exports.GatewayApiEntityTag = GatewayApiEntityTag;
var GatewayCertificateAuthority;
(function (GatewayCertificateAuthority) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GatewayCertificateAuthority).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GatewayCertificateAuthority.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-gatewaycertificateauthority-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GatewayCertificateAuthority).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_29.GatewayCertificateAuthority, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GatewayCertificateAuthority shouldn't use an unstable API (apimanagement.v20230301preview.GatewayCertificateAuthority). A compatible replacement can be found at 'apimanagement.GatewayCertificateAuthority'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GatewayCertificateAuthority || (GatewayCertificateAuthority = {}));
exports.GatewayCertificateAuthority = GatewayCertificateAuthority;
var GatewayHostnameConfiguration;
(function (GatewayHostnameConfiguration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GatewayHostnameConfiguration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GatewayHostnameConfiguration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-gatewayhostnameconfiguration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GatewayHostnameConfiguration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_30.GatewayHostnameConfiguration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GatewayHostnameConfiguration shouldn't use an unstable API (apimanagement.v20230301preview.GatewayHostnameConfiguration). A compatible replacement can be found at 'apimanagement.GatewayHostnameConfiguration'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GatewayHostnameConfiguration || (GatewayHostnameConfiguration = {}));
exports.GatewayHostnameConfiguration = GatewayHostnameConfiguration;
var GlobalSchema;
(function (GlobalSchema) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GlobalSchema).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GlobalSchema.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-globalschema-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GlobalSchema).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_31.GlobalSchema, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GlobalSchema shouldn't use an unstable API (apimanagement.v20230301preview.GlobalSchema). A compatible replacement can be found at 'apimanagement.GlobalSchema'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GlobalSchema || (GlobalSchema = {}));
exports.GlobalSchema = GlobalSchema;
var GraphQLApiResolver;
(function (GraphQLApiResolver) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GraphQLApiResolver).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GraphQLApiResolver.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-graphqlapiresolver-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GraphQLApiResolver).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_32.GraphQLApiResolver, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GraphQLApiResolver shouldn't use an unstable API (apimanagement.v20230301preview.GraphQLApiResolver). A compatible replacement can be found at 'apimanagement.GraphQLApiResolver'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GraphQLApiResolver || (GraphQLApiResolver = {}));
exports.GraphQLApiResolver = GraphQLApiResolver;
var GraphQLApiResolverPolicy;
(function (GraphQLApiResolverPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GraphQLApiResolverPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GraphQLApiResolverPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-graphqlapiresolverpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GraphQLApiResolverPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_33.GraphQLApiResolverPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GraphQLApiResolverPolicy shouldn't use an unstable API (apimanagement.v20230301preview.GraphQLApiResolverPolicy). A compatible replacement can be found at 'apimanagement.GraphQLApiResolverPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GraphQLApiResolverPolicy || (GraphQLApiResolverPolicy = {}));
exports.GraphQLApiResolverPolicy = GraphQLApiResolverPolicy;
var Group;
(function (Group) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Group).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Group.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-group-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Group).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_34.Group, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Group shouldn't use an unstable API (apimanagement.v20230301preview.Group). A compatible replacement can be found at 'apimanagement.Group'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Group || (Group = {}));
exports.Group = Group;
var GroupUser;
(function (GroupUser) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GroupUser).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GroupUser.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-groupuser-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.GroupUser).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_35.GroupUser, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GroupUser shouldn't use an unstable API (apimanagement.v20230301preview.GroupUser). A compatible replacement can be found at 'apimanagement.GroupUser'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GroupUser || (GroupUser = {}));
exports.GroupUser = GroupUser;
var IdentityProvider;
(function (IdentityProvider) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.IdentityProvider).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IdentityProvider.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-identityprovider-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.IdentityProvider).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_36.IdentityProvider, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IdentityProvider shouldn't use an unstable API (apimanagement.v20230301preview.IdentityProvider). A compatible replacement can be found at 'apimanagement.IdentityProvider'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IdentityProvider || (IdentityProvider = {}));
exports.IdentityProvider = IdentityProvider;
var Logger;
(function (Logger) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Logger).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Logger.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-logger-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Logger).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_37.Logger, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Logger shouldn't use an unstable API (apimanagement.v20230301preview.Logger). A compatible replacement can be found at 'apimanagement.Logger'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Logger || (Logger = {}));
exports.Logger = Logger;
var NamedValue;
(function (NamedValue) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.NamedValue).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    NamedValue.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-namedvalue-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.NamedValue).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_38.NamedValue, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure NamedValue shouldn't use an unstable API (apimanagement.v20230301preview.NamedValue). A compatible replacement can be found at 'apimanagement.NamedValue'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(NamedValue || (NamedValue = {}));
exports.NamedValue = NamedValue;
var NotificationRecipientEmail;
(function (NotificationRecipientEmail) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.NotificationRecipientEmail).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    NotificationRecipientEmail.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-notificationrecipientemail-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.NotificationRecipientEmail).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_39.NotificationRecipientEmail, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure NotificationRecipientEmail shouldn't use an unstable API (apimanagement.v20230301preview.NotificationRecipientEmail). A compatible replacement can be found at 'apimanagement.NotificationRecipientEmail'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(NotificationRecipientEmail || (NotificationRecipientEmail = {}));
exports.NotificationRecipientEmail = NotificationRecipientEmail;
var NotificationRecipientUser;
(function (NotificationRecipientUser) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.NotificationRecipientUser).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    NotificationRecipientUser.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-notificationrecipientuser-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.NotificationRecipientUser).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_40.NotificationRecipientUser, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure NotificationRecipientUser shouldn't use an unstable API (apimanagement.v20230301preview.NotificationRecipientUser). A compatible replacement can be found at 'apimanagement.NotificationRecipientUser'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(NotificationRecipientUser || (NotificationRecipientUser = {}));
exports.NotificationRecipientUser = NotificationRecipientUser;
var OpenIdConnectProvider;
(function (OpenIdConnectProvider) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.OpenIdConnectProvider).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OpenIdConnectProvider.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-openidconnectprovider-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.OpenIdConnectProvider).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_41.OpenIdConnectProvider, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OpenIdConnectProvider shouldn't use an unstable API (apimanagement.v20230301preview.OpenIdConnectProvider). A compatible replacement can be found at 'apimanagement.OpenIdConnectProvider'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OpenIdConnectProvider || (OpenIdConnectProvider = {}));
exports.OpenIdConnectProvider = OpenIdConnectProvider;
var Policy;
(function (Policy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Policy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Policy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-policy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Policy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_42.Policy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Policy shouldn't use an unstable API (apimanagement.v20230301preview.Policy). A compatible replacement can be found at 'apimanagement.Policy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Policy || (Policy = {}));
exports.Policy = Policy;
var PolicyFragment;
(function (PolicyFragment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.PolicyFragment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PolicyFragment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-policyfragment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.PolicyFragment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_43.PolicyFragment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PolicyFragment shouldn't use an unstable API (apimanagement.v20230301preview.PolicyFragment). A compatible replacement can be found at 'apimanagement.PolicyFragment'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PolicyFragment || (PolicyFragment = {}));
exports.PolicyFragment = PolicyFragment;
var PrivateEndpointConnectionByName;
(function (PrivateEndpointConnectionByName) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.PrivateEndpointConnectionByName).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnectionByName.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-privateendpointconnectionbyname-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.PrivateEndpointConnectionByName).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_44.PrivateEndpointConnectionByName, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnectionByName shouldn't use an unstable API (apimanagement.v20230301preview.PrivateEndpointConnectionByName). A compatible replacement can be found at 'apimanagement.PrivateEndpointConnectionByName'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnectionByName || (PrivateEndpointConnectionByName = {}));
exports.PrivateEndpointConnectionByName = PrivateEndpointConnectionByName;
var Product;
(function (Product) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Product).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Product.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-product-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Product).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_45.Product, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Product shouldn't use an unstable API (apimanagement.v20230301preview.Product). A compatible replacement can be found at 'apimanagement.Product'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Product || (Product = {}));
exports.Product = Product;
var ProductApi;
(function (ProductApi) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductApi).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProductApi.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-productapi-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductApi).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_46.ProductApi, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProductApi shouldn't use an unstable API (apimanagement.v20230301preview.ProductApi). A compatible replacement can be found at 'apimanagement.ProductApi'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProductApi || (ProductApi = {}));
exports.ProductApi = ProductApi;
var ProductApiLink;
(function (ProductApiLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductApiLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProductApiLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-productapilink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductApiLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_47.ProductApiLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProductApiLink shouldn't use an unstable API (apimanagement.v20230301preview.ProductApiLink). A compatible replacement can be found at 'apimanagement.ProductApiLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProductApiLink || (ProductApiLink = {}));
exports.ProductApiLink = ProductApiLink;
var ProductGroup;
(function (ProductGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProductGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-productgroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_48.ProductGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProductGroup shouldn't use an unstable API (apimanagement.v20230301preview.ProductGroup). A compatible replacement can be found at 'apimanagement.ProductGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProductGroup || (ProductGroup = {}));
exports.ProductGroup = ProductGroup;
var ProductGroupLink;
(function (ProductGroupLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductGroupLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProductGroupLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-productgrouplink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductGroupLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_49.ProductGroupLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProductGroupLink shouldn't use an unstable API (apimanagement.v20230301preview.ProductGroupLink). A compatible replacement can be found at 'apimanagement.ProductGroupLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProductGroupLink || (ProductGroupLink = {}));
exports.ProductGroupLink = ProductGroupLink;
var ProductPolicy;
(function (ProductPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProductPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-productpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_50.ProductPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProductPolicy shouldn't use an unstable API (apimanagement.v20230301preview.ProductPolicy). A compatible replacement can be found at 'apimanagement.ProductPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProductPolicy || (ProductPolicy = {}));
exports.ProductPolicy = ProductPolicy;
var ProductWiki;
(function (ProductWiki) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductWiki).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProductWiki.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-productwiki-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.ProductWiki).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_51.ProductWiki, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProductWiki shouldn't use an unstable API (apimanagement.v20230301preview.ProductWiki). A compatible replacement can be found at 'apimanagement.ProductWiki'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProductWiki || (ProductWiki = {}));
exports.ProductWiki = ProductWiki;
var Subscription;
(function (Subscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Subscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Subscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-subscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Subscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_52.Subscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Subscription shouldn't use an unstable API (apimanagement.v20230301preview.Subscription). A compatible replacement can be found at 'apimanagement.Subscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Subscription || (Subscription = {}));
exports.Subscription = Subscription;
var Tag;
(function (Tag) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Tag).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Tag.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-tag-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Tag).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_53.Tag, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Tag shouldn't use an unstable API (apimanagement.v20230301preview.Tag). A compatible replacement can be found at 'apimanagement.Tag'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Tag || (Tag = {}));
exports.Tag = Tag;
var TagApiLink;
(function (TagApiLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagApiLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TagApiLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-tagapilink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagApiLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_54.TagApiLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TagApiLink shouldn't use an unstable API (apimanagement.v20230301preview.TagApiLink). A compatible replacement can be found at 'apimanagement.TagApiLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TagApiLink || (TagApiLink = {}));
exports.TagApiLink = TagApiLink;
var TagByApi;
(function (TagByApi) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagByApi).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TagByApi.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-tagbyapi-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagByApi).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_55.TagByApi, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TagByApi shouldn't use an unstable API (apimanagement.v20230301preview.TagByApi). A compatible replacement can be found at 'apimanagement.TagByApi'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TagByApi || (TagByApi = {}));
exports.TagByApi = TagByApi;
var TagByOperation;
(function (TagByOperation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagByOperation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TagByOperation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-tagbyoperation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagByOperation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_56.TagByOperation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TagByOperation shouldn't use an unstable API (apimanagement.v20230301preview.TagByOperation). A compatible replacement can be found at 'apimanagement.TagByOperation'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TagByOperation || (TagByOperation = {}));
exports.TagByOperation = TagByOperation;
var TagByProduct;
(function (TagByProduct) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagByProduct).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TagByProduct.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-tagbyproduct-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagByProduct).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_57.TagByProduct, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TagByProduct shouldn't use an unstable API (apimanagement.v20230301preview.TagByProduct). A compatible replacement can be found at 'apimanagement.TagByProduct'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TagByProduct || (TagByProduct = {}));
exports.TagByProduct = TagByProduct;
var TagOperationLink;
(function (TagOperationLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagOperationLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TagOperationLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-tagoperationlink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagOperationLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_58.TagOperationLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TagOperationLink shouldn't use an unstable API (apimanagement.v20230301preview.TagOperationLink). A compatible replacement can be found at 'apimanagement.TagOperationLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TagOperationLink || (TagOperationLink = {}));
exports.TagOperationLink = TagOperationLink;
var TagProductLink;
(function (TagProductLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagProductLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TagProductLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-tagproductlink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.TagProductLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_59.TagProductLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TagProductLink shouldn't use an unstable API (apimanagement.v20230301preview.TagProductLink). A compatible replacement can be found at 'apimanagement.TagProductLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TagProductLink || (TagProductLink = {}));
exports.TagProductLink = TagProductLink;
var User;
(function (User) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.User).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    User.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-user-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.User).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_60.User, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure User shouldn't use an unstable API (apimanagement.v20230301preview.User). A compatible replacement can be found at 'apimanagement.User'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(User || (User = {}));
exports.User = User;
var Workspace;
(function (Workspace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Workspace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Workspace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.Workspace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_61.Workspace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Workspace shouldn't use an unstable API (apimanagement.v20230301preview.Workspace). A compatible replacement can be found at 'apimanagement.Workspace'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Workspace || (Workspace = {}));
exports.Workspace = Workspace;
var WorkspaceApi;
(function (WorkspaceApi) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApi).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceApi.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceapi-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApi).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_62.WorkspaceApi, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceApi shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceApi). A compatible replacement can be found at 'apimanagement.WorkspaceApi'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceApi || (WorkspaceApi = {}));
exports.WorkspaceApi = WorkspaceApi;
var WorkspaceApiOperation;
(function (WorkspaceApiOperation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiOperation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceApiOperation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceapioperation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiOperation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_63.WorkspaceApiOperation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceApiOperation shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceApiOperation). A compatible replacement can be found at 'apimanagement.WorkspaceApiOperation'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceApiOperation || (WorkspaceApiOperation = {}));
exports.WorkspaceApiOperation = WorkspaceApiOperation;
var WorkspaceApiOperationPolicy;
(function (WorkspaceApiOperationPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiOperationPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceApiOperationPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceapioperationpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiOperationPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_64.WorkspaceApiOperationPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceApiOperationPolicy shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceApiOperationPolicy). A compatible replacement can be found at 'apimanagement.WorkspaceApiOperationPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceApiOperationPolicy || (WorkspaceApiOperationPolicy = {}));
exports.WorkspaceApiOperationPolicy = WorkspaceApiOperationPolicy;
var WorkspaceApiPolicy;
(function (WorkspaceApiPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceApiPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceapipolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_65.WorkspaceApiPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceApiPolicy shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceApiPolicy). A compatible replacement can be found at 'apimanagement.WorkspaceApiPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceApiPolicy || (WorkspaceApiPolicy = {}));
exports.WorkspaceApiPolicy = WorkspaceApiPolicy;
var WorkspaceApiRelease;
(function (WorkspaceApiRelease) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiRelease).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceApiRelease.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceapirelease-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiRelease).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_66.WorkspaceApiRelease, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceApiRelease shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceApiRelease). A compatible replacement can be found at 'apimanagement.WorkspaceApiRelease'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceApiRelease || (WorkspaceApiRelease = {}));
exports.WorkspaceApiRelease = WorkspaceApiRelease;
var WorkspaceApiSchema;
(function (WorkspaceApiSchema) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiSchema).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceApiSchema.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceapischema-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiSchema).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_67.WorkspaceApiSchema, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceApiSchema shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceApiSchema). A compatible replacement can be found at 'apimanagement.WorkspaceApiSchema'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceApiSchema || (WorkspaceApiSchema = {}));
exports.WorkspaceApiSchema = WorkspaceApiSchema;
var WorkspaceApiVersionSet;
(function (WorkspaceApiVersionSet) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiVersionSet).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceApiVersionSet.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceapiversionset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceApiVersionSet).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_68.WorkspaceApiVersionSet, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceApiVersionSet shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceApiVersionSet). A compatible replacement can be found at 'apimanagement.WorkspaceApiVersionSet'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceApiVersionSet || (WorkspaceApiVersionSet = {}));
exports.WorkspaceApiVersionSet = WorkspaceApiVersionSet;
var WorkspaceGlobalSchema;
(function (WorkspaceGlobalSchema) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceGlobalSchema).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceGlobalSchema.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceglobalschema-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceGlobalSchema).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_69.WorkspaceGlobalSchema, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceGlobalSchema shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceGlobalSchema). A compatible replacement can be found at 'apimanagement.WorkspaceGlobalSchema'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceGlobalSchema || (WorkspaceGlobalSchema = {}));
exports.WorkspaceGlobalSchema = WorkspaceGlobalSchema;
var WorkspaceGroup;
(function (WorkspaceGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacegroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_70.WorkspaceGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceGroup shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceGroup). A compatible replacement can be found at 'apimanagement.WorkspaceGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceGroup || (WorkspaceGroup = {}));
exports.WorkspaceGroup = WorkspaceGroup;
var WorkspaceGroupUser;
(function (WorkspaceGroupUser) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceGroupUser).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceGroupUser.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacegroupuser-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceGroupUser).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_71.WorkspaceGroupUser, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceGroupUser shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceGroupUser). A compatible replacement can be found at 'apimanagement.WorkspaceGroupUser'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceGroupUser || (WorkspaceGroupUser = {}));
exports.WorkspaceGroupUser = WorkspaceGroupUser;
var WorkspaceNamedValue;
(function (WorkspaceNamedValue) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceNamedValue).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceNamedValue.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacenamedvalue-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceNamedValue).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_72.WorkspaceNamedValue, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceNamedValue shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceNamedValue). A compatible replacement can be found at 'apimanagement.WorkspaceNamedValue'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceNamedValue || (WorkspaceNamedValue = {}));
exports.WorkspaceNamedValue = WorkspaceNamedValue;
var WorkspaceNotificationRecipientEmail;
(function (WorkspaceNotificationRecipientEmail) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceNotificationRecipientEmail).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceNotificationRecipientEmail.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacenotificationrecipientemail-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceNotificationRecipientEmail).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_73.WorkspaceNotificationRecipientEmail, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceNotificationRecipientEmail shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceNotificationRecipientEmail). A compatible replacement can be found at 'apimanagement.WorkspaceNotificationRecipientEmail'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceNotificationRecipientEmail || (WorkspaceNotificationRecipientEmail = {}));
exports.WorkspaceNotificationRecipientEmail = WorkspaceNotificationRecipientEmail;
var WorkspaceNotificationRecipientUser;
(function (WorkspaceNotificationRecipientUser) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceNotificationRecipientUser).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceNotificationRecipientUser.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacenotificationrecipientuser-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceNotificationRecipientUser).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_74.WorkspaceNotificationRecipientUser, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceNotificationRecipientUser shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceNotificationRecipientUser). A compatible replacement can be found at 'apimanagement.WorkspaceNotificationRecipientUser'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceNotificationRecipientUser || (WorkspaceNotificationRecipientUser = {}));
exports.WorkspaceNotificationRecipientUser = WorkspaceNotificationRecipientUser;
var WorkspacePolicy;
(function (WorkspacePolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspacePolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspacePolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacepolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspacePolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_75.WorkspacePolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspacePolicy shouldn't use an unstable API (apimanagement.v20230301preview.WorkspacePolicy). A compatible replacement can be found at 'apimanagement.WorkspacePolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspacePolicy || (WorkspacePolicy = {}));
exports.WorkspacePolicy = WorkspacePolicy;
var WorkspacePolicyFragment;
(function (WorkspacePolicyFragment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspacePolicyFragment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspacePolicyFragment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacepolicyfragment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspacePolicyFragment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_76.WorkspacePolicyFragment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspacePolicyFragment shouldn't use an unstable API (apimanagement.v20230301preview.WorkspacePolicyFragment). A compatible replacement can be found at 'apimanagement.WorkspacePolicyFragment'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspacePolicyFragment || (WorkspacePolicyFragment = {}));
exports.WorkspacePolicyFragment = WorkspacePolicyFragment;
var WorkspaceProduct;
(function (WorkspaceProduct) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceProduct).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceProduct.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceproduct-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceProduct).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_77.WorkspaceProduct, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceProduct shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceProduct). A compatible replacement can be found at 'apimanagement.WorkspaceProduct'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceProduct || (WorkspaceProduct = {}));
exports.WorkspaceProduct = WorkspaceProduct;
var WorkspaceProductApiLink;
(function (WorkspaceProductApiLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceProductApiLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceProductApiLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceproductapilink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceProductApiLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_78.WorkspaceProductApiLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceProductApiLink shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceProductApiLink). A compatible replacement can be found at 'apimanagement.WorkspaceProductApiLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceProductApiLink || (WorkspaceProductApiLink = {}));
exports.WorkspaceProductApiLink = WorkspaceProductApiLink;
var WorkspaceProductGroupLink;
(function (WorkspaceProductGroupLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceProductGroupLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceProductGroupLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceproductgrouplink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceProductGroupLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_79.WorkspaceProductGroupLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceProductGroupLink shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceProductGroupLink). A compatible replacement can be found at 'apimanagement.WorkspaceProductGroupLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceProductGroupLink || (WorkspaceProductGroupLink = {}));
exports.WorkspaceProductGroupLink = WorkspaceProductGroupLink;
var WorkspaceProductPolicy;
(function (WorkspaceProductPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceProductPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceProductPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspaceproductpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceProductPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_80.WorkspaceProductPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceProductPolicy shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceProductPolicy). A compatible replacement can be found at 'apimanagement.WorkspaceProductPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceProductPolicy || (WorkspaceProductPolicy = {}));
exports.WorkspaceProductPolicy = WorkspaceProductPolicy;
var WorkspaceSubscription;
(function (WorkspaceSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacesubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_81.WorkspaceSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceSubscription shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceSubscription). A compatible replacement can be found at 'apimanagement.WorkspaceSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceSubscription || (WorkspaceSubscription = {}));
exports.WorkspaceSubscription = WorkspaceSubscription;
var WorkspaceTag;
(function (WorkspaceTag) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceTag).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceTag.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacetag-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceTag).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_82.WorkspaceTag, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceTag shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceTag). A compatible replacement can be found at 'apimanagement.WorkspaceTag'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceTag || (WorkspaceTag = {}));
exports.WorkspaceTag = WorkspaceTag;
var WorkspaceTagApiLink;
(function (WorkspaceTagApiLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceTagApiLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceTagApiLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacetagapilink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceTagApiLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_83.WorkspaceTagApiLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceTagApiLink shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceTagApiLink). A compatible replacement can be found at 'apimanagement.WorkspaceTagApiLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceTagApiLink || (WorkspaceTagApiLink = {}));
exports.WorkspaceTagApiLink = WorkspaceTagApiLink;
var WorkspaceTagOperationLink;
(function (WorkspaceTagOperationLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceTagOperationLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceTagOperationLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacetagoperationlink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceTagOperationLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_84.WorkspaceTagOperationLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceTagOperationLink shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceTagOperationLink). A compatible replacement can be found at 'apimanagement.WorkspaceTagOperationLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceTagOperationLink || (WorkspaceTagOperationLink = {}));
exports.WorkspaceTagOperationLink = WorkspaceTagOperationLink;
var WorkspaceTagProductLink;
(function (WorkspaceTagProductLink) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceTagProductLink).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceTagProductLink.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20230301preview-workspacetagproductlink-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20230301preview.WorkspaceTagProductLink).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_85.WorkspaceTagProductLink, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceTagProductLink shouldn't use an unstable API (apimanagement.v20230301preview.WorkspaceTagProductLink). A compatible replacement can be found at 'apimanagement.WorkspaceTagProductLink'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceTagProductLink || (WorkspaceTagProductLink = {}));
exports.WorkspaceTagProductLink = WorkspaceTagProductLink;
