// @pulumi/azure-native/machinelearningservices/modelContainer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelContainer = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Azure Resource Manager resource envelope.
 * Azure REST API version: 2023-04-01. Prior API version in Azure Native 1.x: 2021-03-01-preview.
 *
 * Other available API versions: 2021-03-01-preview, 2022-02-01-preview, 2023-04-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2023-10-01, 2024-01-01-preview, 2024-04-01, 2024-04-01-preview, 2024-07-01-preview, 2024-10-01, 2024-10-01-preview.
 */
class ModelContainer extends pulumi.CustomResource {
    /**
     * Get an existing ModelContainer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ModelContainer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ModelContainer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ModelContainer.__pulumiType;
    }
    /**
     * Create a ModelContainer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.modelContainerProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelContainerProperties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["modelContainerProperties"] = args ? (args.modelContainerProperties ? pulumi.output(args.modelContainerProperties).apply(inputs.machinelearningservices.modelContainerArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["modelContainerProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20210301preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20220201preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20220501:ModelContainer" }, { type: "azure-native:machinelearningservices/v20220601preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20221001:ModelContainer" }, { type: "azure-native:machinelearningservices/v20221001preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20221201preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20230201preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20230401:ModelContainer" }, { type: "azure-native:machinelearningservices/v20230401preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20230601preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20230801preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20231001:ModelContainer" }, { type: "azure-native:machinelearningservices/v20240101preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20240401:ModelContainer" }, { type: "azure-native:machinelearningservices/v20240401preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20240701preview:ModelContainer" }, { type: "azure-native:machinelearningservices/v20241001:ModelContainer" }, { type: "azure-native:machinelearningservices/v20241001preview:ModelContainer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ModelContainer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ModelContainer = ModelContainer;
/** @internal */
ModelContainer.__pulumiType = 'azure-native:machinelearningservices:ModelContainer';
