// @pulumi/azure/frontdoor/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RulesEngine = exports.Frontdoor = exports.FirewallPolicy = exports.CustomHttpsConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CustomHttpsConfiguration = null;
utilities.lazyLoad(exports, ["CustomHttpsConfiguration"], () => require("./customHttpsConfiguration"));
exports.FirewallPolicy = null;
utilities.lazyLoad(exports, ["FirewallPolicy"], () => require("./firewallPolicy"));
exports.Frontdoor = null;
utilities.lazyLoad(exports, ["Frontdoor"], () => require("./frontdoor"));
exports.RulesEngine = null;
utilities.lazyLoad(exports, ["RulesEngine"], () => require("./rulesEngine"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:frontdoor/customHttpsConfiguration:CustomHttpsConfiguration":
                return new exports.CustomHttpsConfiguration(name, undefined, { urn });
            case "azure:frontdoor/firewallPolicy:FirewallPolicy":
                return new exports.FirewallPolicy(name, undefined, { urn });
            case "azure:frontdoor/frontdoor:Frontdoor":
                return new exports.Frontdoor(name, undefined, { urn });
            case "azure:frontdoor/rulesEngine:RulesEngine":
                return new exports.RulesEngine(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "frontdoor/customHttpsConfiguration", _module);
pulumi.runtime.registerResourceModule("azure", "frontdoor/firewallPolicy", _module);
pulumi.runtime.registerResourceModule("azure", "frontdoor/frontdoor", _module);
pulumi.runtime.registerResourceModule("azure", "frontdoor/rulesEngine", _module);
