// @pulumi/azure/authorization/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAssignedIdentity = exports.RoleDefinition = exports.getUserAssignedIdentityOutput = exports.getUserAssignedIdentity = exports.getRoleDefinitionOutput = exports.getRoleDefinition = exports.Assignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Assignment = null;
utilities.lazyLoad(exports, ["Assignment"], () => require("./assignment"));
exports.getRoleDefinition = null;
exports.getRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getRoleDefinition", "getRoleDefinitionOutput"], () => require("./getRoleDefinition"));
exports.getUserAssignedIdentity = null;
exports.getUserAssignedIdentityOutput = null;
utilities.lazyLoad(exports, ["getUserAssignedIdentity", "getUserAssignedIdentityOutput"], () => require("./getUserAssignedIdentity"));
exports.RoleDefinition = null;
utilities.lazyLoad(exports, ["RoleDefinition"], () => require("./roleDefinition"));
exports.UserAssignedIdentity = null;
utilities.lazyLoad(exports, ["UserAssignedIdentity"], () => require("./userAssignedIdentity"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:authorization/assignment:Assignment":
                return new exports.Assignment(name, undefined, { urn });
            case "azure:authorization/roleDefinition:RoleDefinition":
                return new exports.RoleDefinition(name, undefined, { urn });
            case "azure:authorization/userAssignedIdentity:UserAssignedIdentity":
                return new exports.UserAssignedIdentity(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "authorization/assignment", _module);
pulumi.runtime.registerResourceModule("azure", "authorization/roleDefinition", _module);
pulumi.runtime.registerResourceModule("azure", "authorization/userAssignedIdentity", _module);
