// @pulumi/azure-compliance-policies/azurenative/powerplatform/v20201030preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnection = exports.EnterprisePolicy = exports.Account = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20201030preview_1 = require("@pulumi/azure-native/powerplatform/v20201030preview");
const v20201030preview_2 = require("@pulumi/azure-native/powerplatform/v20201030preview");
const v20201030preview_3 = require("@pulumi/azure-native/powerplatform/v20201030preview");
var Account;
(function (Account) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (powerplatform.v20201030preview.Account).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Account.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-powerplatform-v20201030preview-account-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (powerplatform.v20201030preview.Account).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20201030preview_1.Account, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Account shouldn't use an unstable API (powerplatform.v20201030preview.Account). A compatible replacement can be found at 'powerplatform.Account'.");
            }),
        },
        vendors: ["azure"],
        services: ["powerplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Account || (Account = {}));
exports.Account = Account;
var EnterprisePolicy;
(function (EnterprisePolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (powerplatform.v20201030preview.EnterprisePolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EnterprisePolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-powerplatform-v20201030preview-enterprisepolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (powerplatform.v20201030preview.EnterprisePolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20201030preview_2.EnterprisePolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EnterprisePolicy shouldn't use an unstable API (powerplatform.v20201030preview.EnterprisePolicy). A compatible replacement can be found at 'powerplatform.EnterprisePolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["powerplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EnterprisePolicy || (EnterprisePolicy = {}));
exports.EnterprisePolicy = EnterprisePolicy;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (powerplatform.v20201030preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-powerplatform-v20201030preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (powerplatform.v20201030preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20201030preview_3.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (powerplatform.v20201030preview.PrivateEndpointConnection). A compatible replacement can be found at 'powerplatform.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["powerplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
