// @pulumi/azure-compliance-policies/azurenative/resources/v20191001preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzurePowerShellScript = exports.AzureCliScript = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20191001preview_1 = require("@pulumi/azure-native/resources/v20191001preview");
const v20191001preview_2 = require("@pulumi/azure-native/resources/v20191001preview");
var AzureCliScript;
(function (AzureCliScript) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (resources.v20191001preview.AzureCliScript).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AzureCliScript.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-resources-v20191001preview-azurecliscript-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (resources.v20191001preview.AzureCliScript).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20191001preview_1.AzureCliScript, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AzureCliScript shouldn't use an unstable API (resources.v20191001preview.AzureCliScript). A compatible replacement can be found at 'resources.AzureCliScript'.");
            }),
        },
        vendors: ["azure"],
        services: ["resources"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AzureCliScript || (AzureCliScript = {}));
exports.AzureCliScript = AzureCliScript;
var AzurePowerShellScript;
(function (AzurePowerShellScript) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (resources.v20191001preview.AzurePowerShellScript).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AzurePowerShellScript.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-resources-v20191001preview-azurepowershellscript-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (resources.v20191001preview.AzurePowerShellScript).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20191001preview_2.AzurePowerShellScript, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AzurePowerShellScript shouldn't use an unstable API (resources.v20191001preview.AzurePowerShellScript). A compatible replacement can be found at 'resources.AzurePowerShellScript'.");
            }),
        },
        vendors: ["azure"],
        services: ["resources"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AzurePowerShellScript || (AzurePowerShellScript = {}));
exports.AzurePowerShellScript = AzurePowerShellScript;
