// @pulumi/azure-compliance-policies/azurenative/security/v20170801preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceSetting = exports.SecurityContact = exports.IotSecuritySolution = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20170801preview_1 = require("@pulumi/azure-native/security/v20170801preview");
const v20170801preview_2 = require("@pulumi/azure-native/security/v20170801preview");
const v20170801preview_3 = require("@pulumi/azure-native/security/v20170801preview");
var IotSecuritySolution;
(function (IotSecuritySolution) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20170801preview.IotSecuritySolution).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IotSecuritySolution.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20170801preview-iotsecuritysolution-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20170801preview.IotSecuritySolution).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20170801preview_1.IotSecuritySolution, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IotSecuritySolution shouldn't use an unstable API (security.v20170801preview.IotSecuritySolution). A compatible replacement can be found at 'security.IotSecuritySolution'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IotSecuritySolution || (IotSecuritySolution = {}));
exports.IotSecuritySolution = IotSecuritySolution;
var SecurityContact;
(function (SecurityContact) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20170801preview.SecurityContact).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SecurityContact.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20170801preview-securitycontact-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20170801preview.SecurityContact).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20170801preview_2.SecurityContact, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SecurityContact shouldn't use an unstable API (security.v20170801preview.SecurityContact). A compatible replacement can be found at 'security.SecurityContact'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SecurityContact || (SecurityContact = {}));
exports.SecurityContact = SecurityContact;
var WorkspaceSetting;
(function (WorkspaceSetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (security.v20170801preview.WorkspaceSetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceSetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-security-v20170801preview-workspacesetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (security.v20170801preview.WorkspaceSetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20170801preview_3.WorkspaceSetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceSetting shouldn't use an unstable API (security.v20170801preview.WorkspaceSetting). A compatible replacement can be found at 'security.WorkspaceSetting'.");
            }),
        },
        vendors: ["azure"],
        services: ["security"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceSetting || (WorkspaceSetting = {}));
exports.WorkspaceSetting = WorkspaceSetting;
