// @pulumi/azure/datadog/monitorTagRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorTagRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages TagRules on the datadog Monitor.
 *
 * ## Example Usage
 *
 * ### Adding TagRules on monitor
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-datadog",
 *     location: "West US 2",
 * });
 * const exampleMonitor = new azure.datadog.Monitor("example", {
 *     name: "example-monitor",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     datadogOrganization: {
 *         apiKey: "XXXX",
 *         applicationKey: "XXXX",
 *     },
 *     user: {
 *         name: "Example",
 *         email: "abc@xyz.com",
 *     },
 *     skuName: "Linked",
 *     identity: {
 *         type: "SystemAssigned",
 *     },
 * });
 * const exampleMonitorTagRule = new azure.datadog.MonitorTagRule("example", {
 *     datadogMonitorId: exampleMonitor.id,
 *     logs: [{
 *         subscriptionLogEnabled: true,
 *     }],
 *     metrics: [{
 *         filters: [{
 *             name: "Test",
 *             value: "Logs",
 *             action: "Include",
 *         }],
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * Tag Rules on the Datadog Monitor can be imported using the `tag rule resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:datadog/monitorTagRule:MonitorTagRule example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Datadog/monitors/monitor1/tagRules/default
 * ```
 */
class MonitorTagRule extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["datadogMonitorId"] = state ? state.datadogMonitorId : undefined;
            resourceInputs["logs"] = state ? state.logs : undefined;
            resourceInputs["metrics"] = state ? state.metrics : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.datadogMonitorId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'datadogMonitorId'");
            }
            resourceInputs["datadogMonitorId"] = args ? args.datadogMonitorId : undefined;
            resourceInputs["logs"] = args ? args.logs : undefined;
            resourceInputs["metrics"] = args ? args.metrics : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MonitorTagRule.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MonitorTagRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MonitorTagRule(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MonitorTagRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MonitorTagRule.__pulumiType;
    }
}
exports.MonitorTagRule = MonitorTagRule;
/** @internal */
MonitorTagRule.__pulumiType = 'azure:datadog/monitorTagRule:MonitorTagRule';
