// @pulumi/azure-compliance-policies/azurenative/scom/v20230707preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitoredResource = exports.ManagedGateway = exports.Instance = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230707preview_1 = require("@pulumi/azure-native/scom/v20230707preview");
const v20230707preview_2 = require("@pulumi/azure-native/scom/v20230707preview");
const v20230707preview_3 = require("@pulumi/azure-native/scom/v20230707preview");
var Instance;
(function (Instance) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (scom.v20230707preview.Instance).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Instance.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-scom-v20230707preview-instance-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (scom.v20230707preview.Instance).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230707preview_1.Instance, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Instance shouldn't use an unstable API (scom.v20230707preview.Instance). A compatible replacement can be found at 'scom.Instance'.");
            }),
        },
        vendors: ["azure"],
        services: ["scom"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Instance || (Instance = {}));
exports.Instance = Instance;
var ManagedGateway;
(function (ManagedGateway) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (scom.v20230707preview.ManagedGateway).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagedGateway.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-scom-v20230707preview-managedgateway-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (scom.v20230707preview.ManagedGateway).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230707preview_2.ManagedGateway, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagedGateway shouldn't use an unstable API (scom.v20230707preview.ManagedGateway). A compatible replacement can be found at 'scom.ManagedGateway'.");
            }),
        },
        vendors: ["azure"],
        services: ["scom"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagedGateway || (ManagedGateway = {}));
exports.ManagedGateway = ManagedGateway;
var MonitoredResource;
(function (MonitoredResource) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (scom.v20230707preview.MonitoredResource).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MonitoredResource.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-scom-v20230707preview-monitoredresource-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (scom.v20230707preview.MonitoredResource).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230707preview_3.MonitoredResource, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MonitoredResource shouldn't use an unstable API (scom.v20230707preview.MonitoredResource). A compatible replacement can be found at 'scom.MonitoredResource'.");
            }),
        },
        vendors: ["azure"],
        services: ["scom"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MonitoredResource || (MonitoredResource = {}));
exports.MonitoredResource = MonitoredResource;
