// @pulumi/azure/maintenance/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPublicConfigurationsOutput = exports.getPublicConfigurations = exports.getConfigurationOutput = exports.getConfiguration = exports.Configuration = exports.AssignmentVirtualMachineScaleSet = exports.AssignmentVirtualMachine = exports.AssignmentDynamicScope = exports.AssignmentDedicatedHost = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AssignmentDedicatedHost = null;
utilities.lazyLoad(exports, ["AssignmentDedicatedHost"], () => require("./assignmentDedicatedHost"));
exports.AssignmentDynamicScope = null;
utilities.lazyLoad(exports, ["AssignmentDynamicScope"], () => require("./assignmentDynamicScope"));
exports.AssignmentVirtualMachine = null;
utilities.lazyLoad(exports, ["AssignmentVirtualMachine"], () => require("./assignmentVirtualMachine"));
exports.AssignmentVirtualMachineScaleSet = null;
utilities.lazyLoad(exports, ["AssignmentVirtualMachineScaleSet"], () => require("./assignmentVirtualMachineScaleSet"));
exports.Configuration = null;
utilities.lazyLoad(exports, ["Configuration"], () => require("./configuration"));
exports.getConfiguration = null;
exports.getConfigurationOutput = null;
utilities.lazyLoad(exports, ["getConfiguration", "getConfigurationOutput"], () => require("./getConfiguration"));
exports.getPublicConfigurations = null;
exports.getPublicConfigurationsOutput = null;
utilities.lazyLoad(exports, ["getPublicConfigurations", "getPublicConfigurationsOutput"], () => require("./getPublicConfigurations"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:maintenance/assignmentDedicatedHost:AssignmentDedicatedHost":
                return new exports.AssignmentDedicatedHost(name, undefined, { urn });
            case "azure:maintenance/assignmentDynamicScope:AssignmentDynamicScope":
                return new exports.AssignmentDynamicScope(name, undefined, { urn });
            case "azure:maintenance/assignmentVirtualMachine:AssignmentVirtualMachine":
                return new exports.AssignmentVirtualMachine(name, undefined, { urn });
            case "azure:maintenance/assignmentVirtualMachineScaleSet:AssignmentVirtualMachineScaleSet":
                return new exports.AssignmentVirtualMachineScaleSet(name, undefined, { urn });
            case "azure:maintenance/configuration:Configuration":
                return new exports.Configuration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "maintenance/assignmentDedicatedHost", _module);
pulumi.runtime.registerResourceModule("azure", "maintenance/assignmentDynamicScope", _module);
pulumi.runtime.registerResourceModule("azure", "maintenance/assignmentVirtualMachine", _module);
pulumi.runtime.registerResourceModule("azure", "maintenance/assignmentVirtualMachineScaleSet", _module);
pulumi.runtime.registerResourceModule("azure", "maintenance/configuration", _module);
