// @pulumi/azure-native/appplatform/storage.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Storage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Storage resource payload.
 * Azure REST API version: 2023-05-01-preview. Prior API version in Azure Native 1.x: 2021-09-01-preview.
 *
 * Other available API versions: 2023-07-01-preview, 2023-09-01-preview, 2023-11-01-preview, 2023-12-01, 2024-01-01-preview, 2024-05-01-preview.
 */
class Storage extends pulumi.CustomResource {
    /**
     * Get an existing Storage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Storage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Storage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Storage.__pulumiType;
    }
    /**
     * Create a Storage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["storageName"] = args ? args.storageName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:appplatform/v20210901preview:Storage" }, { type: "azure-native:appplatform/v20220101preview:Storage" }, { type: "azure-native:appplatform/v20220301preview:Storage" }, { type: "azure-native:appplatform/v20220501preview:Storage" }, { type: "azure-native:appplatform/v20220901preview:Storage" }, { type: "azure-native:appplatform/v20221101preview:Storage" }, { type: "azure-native:appplatform/v20221201:Storage" }, { type: "azure-native:appplatform/v20230101preview:Storage" }, { type: "azure-native:appplatform/v20230301preview:Storage" }, { type: "azure-native:appplatform/v20230501preview:Storage" }, { type: "azure-native:appplatform/v20230701preview:Storage" }, { type: "azure-native:appplatform/v20230901preview:Storage" }, { type: "azure-native:appplatform/v20231101preview:Storage" }, { type: "azure-native:appplatform/v20231201:Storage" }, { type: "azure-native:appplatform/v20240101preview:Storage" }, { type: "azure-native:appplatform/v20240501preview:Storage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Storage.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Storage = Storage;
/** @internal */
Storage.__pulumiType = 'azure-native:appplatform:Storage';
