// @pulumi/azure-compliance-policies/azurenative/databricks/v20240901preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.VNetPeering = exports.PrivateEndpointConnection = exports.AccessConnector = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240901preview_1 = require("@pulumi/azure-native/databricks/v20240901preview");
const v20240901preview_2 = require("@pulumi/azure-native/databricks/v20240901preview");
const v20240901preview_3 = require("@pulumi/azure-native/databricks/v20240901preview");
const v20240901preview_4 = require("@pulumi/azure-native/databricks/v20240901preview");
var AccessConnector;
(function (AccessConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (databricks.v20240901preview.AccessConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AccessConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-databricks-v20240901preview-accessconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (databricks.v20240901preview.AccessConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_1.AccessConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AccessConnector shouldn't use an unstable API (databricks.v20240901preview.AccessConnector). A compatible replacement can be found at 'databricks.AccessConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["databricks"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AccessConnector || (AccessConnector = {}));
exports.AccessConnector = AccessConnector;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (databricks.v20240901preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-databricks-v20240901preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (databricks.v20240901preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_2.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (databricks.v20240901preview.PrivateEndpointConnection). A compatible replacement can be found at 'databricks.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["databricks"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
var VNetPeering;
(function (VNetPeering) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (databricks.v20240901preview.VNetPeering).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VNetPeering.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-databricks-v20240901preview-vnetpeering-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (databricks.v20240901preview.VNetPeering).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_3.VNetPeering, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VNetPeering shouldn't use an unstable API (databricks.v20240901preview.VNetPeering). A compatible replacement can be found at 'databricks.VNetPeering'.");
            }),
        },
        vendors: ["azure"],
        services: ["databricks"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VNetPeering || (VNetPeering = {}));
exports.VNetPeering = VNetPeering;
var Workspace;
(function (Workspace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (databricks.v20240901preview.Workspace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Workspace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-databricks-v20240901preview-workspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (databricks.v20240901preview.Workspace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_4.Workspace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Workspace shouldn't use an unstable API (databricks.v20240901preview.Workspace). A compatible replacement can be found at 'databricks.Workspace'.");
            }),
        },
        vendors: ["azure"],
        services: ["databricks"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Workspace || (Workspace = {}));
exports.Workspace = Workspace;
