// @pulumi/azure-compliance-policies/azurenative/eventgrid/v20211015preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerRegistration = exports.PartnerDestination = exports.EventChannel = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20211015preview_1 = require("@pulumi/azure-native/eventgrid/v20211015preview");
const v20211015preview_2 = require("@pulumi/azure-native/eventgrid/v20211015preview");
const v20211015preview_3 = require("@pulumi/azure-native/eventgrid/v20211015preview");
var EventChannel;
(function (EventChannel) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20211015preview.EventChannel).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EventChannel.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20211015preview-eventchannel-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20211015preview.EventChannel).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20211015preview_1.EventChannel, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EventChannel shouldn't use an unstable API (eventgrid.v20211015preview.EventChannel). A compatible replacement can be found at 'eventgrid.EventChannel'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EventChannel || (EventChannel = {}));
exports.EventChannel = EventChannel;
var PartnerDestination;
(function (PartnerDestination) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20211015preview.PartnerDestination).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PartnerDestination.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20211015preview-partnerdestination-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20211015preview.PartnerDestination).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20211015preview_2.PartnerDestination, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PartnerDestination shouldn't use an unstable API (eventgrid.v20211015preview.PartnerDestination). A compatible replacement can be found at 'eventgrid.PartnerDestination'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PartnerDestination || (PartnerDestination = {}));
exports.PartnerDestination = PartnerDestination;
var PartnerRegistration;
(function (PartnerRegistration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20211015preview.PartnerRegistration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PartnerRegistration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-eventgrid-v20211015preview-partnerregistration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (eventgrid.v20211015preview.PartnerRegistration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20211015preview_3.PartnerRegistration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PartnerRegistration shouldn't use an unstable API (eventgrid.v20211015preview.PartnerRegistration). A compatible replacement can be found at 'eventgrid.PartnerRegistration'.");
            }),
        },
        vendors: ["azure"],
        services: ["eventgrid"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PartnerRegistration || (PartnerRegistration = {}));
exports.PartnerRegistration = PartnerRegistration;
