// @pulumi/azure/dashboard/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrafanaManagedPrivateEndpoint = exports.Grafana = exports.getGrafanaOutput = exports.getGrafana = exports.azurerm_portal_dashboardOutput = exports.azurerm_portal_dashboard = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.azurerm_portal_dashboard = null;
exports.azurerm_portal_dashboardOutput = null;
utilities.lazyLoad(exports, ["azurerm_portal_dashboard", "azurerm_portal_dashboardOutput"], () => require("./azurerm_portal_dashboard"));
exports.getGrafana = null;
exports.getGrafanaOutput = null;
utilities.lazyLoad(exports, ["getGrafana", "getGrafanaOutput"], () => require("./getGrafana"));
exports.Grafana = null;
utilities.lazyLoad(exports, ["Grafana"], () => require("./grafana"));
exports.GrafanaManagedPrivateEndpoint = null;
utilities.lazyLoad(exports, ["GrafanaManagedPrivateEndpoint"], () => require("./grafanaManagedPrivateEndpoint"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:dashboard/grafana:Grafana":
                return new exports.Grafana(name, undefined, { urn });
            case "azure:dashboard/grafanaManagedPrivateEndpoint:GrafanaManagedPrivateEndpoint":
                return new exports.GrafanaManagedPrivateEndpoint(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "dashboard/grafana", _module);
pulumi.runtime.registerResourceModule("azure", "dashboard/grafanaManagedPrivateEndpoint", _module);
