// @pulumi/azure/appplatform/springCloudGatewayRouteConfig.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpringCloudGatewayRouteConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Spring Cloud Gateway Route Config.
 *
 * > **NOTE:** This resource is applicable only for Spring Cloud Service with enterprise tier.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example",
 *     location: "West Europe",
 * });
 * const exampleSpringCloudService = new azure.appplatform.SpringCloudService("example", {
 *     name: "example",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     skuName: "E0",
 * });
 * const exampleSpringCloudApp = new azure.appplatform.SpringCloudApp("example", {
 *     name: "example",
 *     resourceGroupName: example.name,
 *     serviceName: exampleSpringCloudService.name,
 * });
 * const exampleSpringCloudGateway = new azure.appplatform.SpringCloudGateway("example", {
 *     name: "default",
 *     springCloudServiceId: exampleSpringCloudService.id,
 * });
 * const exampleSpringCloudGatewayRouteConfig = new azure.appplatform.SpringCloudGatewayRouteConfig("example", {
 *     name: "example",
 *     springCloudGatewayId: exampleSpringCloudGateway.id,
 *     springCloudAppId: exampleSpringCloudApp.id,
 *     protocol: "HTTPS",
 *     routes: [{
 *         description: "example description",
 *         filters: [
 *             "StripPrefix=2",
 *             "RateLimit=1,1s",
 *         ],
 *         order: 1,
 *         predicates: ["Path=/api5/customer/**"],
 *         ssoValidationEnabled: true,
 *         title: "myApp route config",
 *         tokenRelay: true,
 *         uri: "https://www.example.com",
 *         classificationTags: [
 *             "tag1",
 *             "tag2",
 *         ],
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * Spring Cloud Gateway Route Configs can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:appplatform/springCloudGatewayRouteConfig:SpringCloudGatewayRouteConfig example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.AppPlatform/spring/service1/gateways/gateway1/routeConfigs/routeConfig1
 * ```
 */
class SpringCloudGatewayRouteConfig extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["filters"] = state ? state.filters : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["openApi"] = state ? state.openApi : undefined;
            resourceInputs["predicates"] = state ? state.predicates : undefined;
            resourceInputs["protocol"] = state ? state.protocol : undefined;
            resourceInputs["routes"] = state ? state.routes : undefined;
            resourceInputs["springCloudAppId"] = state ? state.springCloudAppId : undefined;
            resourceInputs["springCloudGatewayId"] = state ? state.springCloudGatewayId : undefined;
            resourceInputs["ssoValidationEnabled"] = state ? state.ssoValidationEnabled : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.protocol === undefined) && !opts.urn) {
                throw new Error("Missing required property 'protocol'");
            }
            if ((!args || args.springCloudGatewayId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'springCloudGatewayId'");
            }
            resourceInputs["filters"] = args ? args.filters : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["openApi"] = args ? args.openApi : undefined;
            resourceInputs["predicates"] = args ? args.predicates : undefined;
            resourceInputs["protocol"] = args ? args.protocol : undefined;
            resourceInputs["routes"] = args ? args.routes : undefined;
            resourceInputs["springCloudAppId"] = args ? args.springCloudAppId : undefined;
            resourceInputs["springCloudGatewayId"] = args ? args.springCloudGatewayId : undefined;
            resourceInputs["ssoValidationEnabled"] = args ? args.ssoValidationEnabled : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SpringCloudGatewayRouteConfig.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SpringCloudGatewayRouteConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SpringCloudGatewayRouteConfig(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SpringCloudGatewayRouteConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SpringCloudGatewayRouteConfig.__pulumiType;
    }
}
exports.SpringCloudGatewayRouteConfig = SpringCloudGatewayRouteConfig;
/** @internal */
SpringCloudGatewayRouteConfig.__pulumiType = 'azure:appplatform/springCloudGatewayRouteConfig:SpringCloudGatewayRouteConfig';
