// @pulumi/azure/appplatform/springCloudBuildPackBinding.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpringCloudBuildPackBinding = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Spring Cloud Build Pack Binding.
 *
 * > **NOTE:** This resource is applicable only for Spring Cloud Service with enterprise tier.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleSpringCloudService = new azure.appplatform.SpringCloudService("example", {
 *     name: "example-springcloud",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     skuName: "E0",
 * });
 * const exampleSpringCloudBuilder = new azure.appplatform.SpringCloudBuilder("example", {
 *     name: "example",
 *     springCloudServiceId: exampleSpringCloudService.id,
 *     buildPackGroups: [{
 *         name: "mix",
 *         buildPackIds: ["tanzu-Build Packs/java-azure"],
 *     }],
 *     stack: {
 *         id: "io.Build Packs.stacks.bionic",
 *         version: "base",
 *     },
 * });
 * const exampleSpringCloudBuildPackBinding = new azure.appplatform.SpringCloudBuildPackBinding("example", {
 *     name: "example",
 *     springCloudBuilderId: exampleSpringCloudBuilder.id,
 *     bindingType: "ApplicationInsights",
 *     launch: {
 *         properties: {
 *             abc: "def",
 *             "any-string": "any-string",
 *             "sampling-rate": "12.0",
 *         },
 *         secrets: {
 *             "connection-string": "XXXXXXXXXXXXXXXXX=XXXXXXXXXXXXX-XXXXXXXXXXXXXXXXXXX;XXXXXXXXXXXXXXXXX=XXXXXXXXXXXXXXXXXXX",
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Spring Cloud Build Pack Bindings can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:appplatform/springCloudBuildPackBinding:SpringCloudBuildPackBinding example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.AppPlatform/spring/service1/buildServices/buildService1/builders/builder1/buildPackBindings/binding1
 * ```
 */
class SpringCloudBuildPackBinding extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["bindingType"] = state ? state.bindingType : undefined;
            resourceInputs["launch"] = state ? state.launch : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["springCloudBuilderId"] = state ? state.springCloudBuilderId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.springCloudBuilderId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'springCloudBuilderId'");
            }
            resourceInputs["bindingType"] = args ? args.bindingType : undefined;
            resourceInputs["launch"] = args ? args.launch : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["springCloudBuilderId"] = args ? args.springCloudBuilderId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SpringCloudBuildPackBinding.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SpringCloudBuildPackBinding resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SpringCloudBuildPackBinding(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SpringCloudBuildPackBinding.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SpringCloudBuildPackBinding.__pulumiType;
    }
}
exports.SpringCloudBuildPackBinding = SpringCloudBuildPackBinding;
/** @internal */
SpringCloudBuildPackBinding.__pulumiType = 'azure:appplatform/springCloudBuildPackBinding:SpringCloudBuildPackBinding';
