// @pulumi/azure/appplatform/springCloudNewRelicApplicationPerformanceMonitoring.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpringCloudNewRelicApplicationPerformanceMonitoring = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * > **NOTE:** This resource is only applicable for Spring Cloud Service enterprise tier
 *
 * Manages a Spring Cloud Application Performance Monitoring resource for New Relic.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example",
 *     location: "West Europe",
 * });
 * const exampleSpringCloudService = new azure.appplatform.SpringCloudService("example", {
 *     name: "example",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     skuName: "E0",
 * });
 * const exampleSpringCloudNewRelicApplicationPerformanceMonitoring = new azure.appplatform.SpringCloudNewRelicApplicationPerformanceMonitoring("example", {
 *     name: "example",
 *     springCloudServiceId: exampleSpringCloudService.id,
 *     appName: "example-app-name",
 *     licenseKey: "example-license-key",
 *     appServerPort: 8080,
 *     labels: {
 *         tagName1: "tagValue1",
 *         tagName2: "tagValue2",
 *     },
 *     globallyEnabled: true,
 * });
 * ```
 *
 * ## Import
 *
 * Spring Cloud Application Performance Monitoring resource for New Relic can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:appplatform/springCloudNewRelicApplicationPerformanceMonitoring:SpringCloudNewRelicApplicationPerformanceMonitoring example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.AppPlatform/spring/service1/apms/apm1
 * ```
 */
class SpringCloudNewRelicApplicationPerformanceMonitoring extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["agentEnabled"] = state ? state.agentEnabled : undefined;
            resourceInputs["appName"] = state ? state.appName : undefined;
            resourceInputs["appServerPort"] = state ? state.appServerPort : undefined;
            resourceInputs["auditModeEnabled"] = state ? state.auditModeEnabled : undefined;
            resourceInputs["autoAppNamingEnabled"] = state ? state.autoAppNamingEnabled : undefined;
            resourceInputs["autoTransactionNamingEnabled"] = state ? state.autoTransactionNamingEnabled : undefined;
            resourceInputs["customTracingEnabled"] = state ? state.customTracingEnabled : undefined;
            resourceInputs["globallyEnabled"] = state ? state.globallyEnabled : undefined;
            resourceInputs["labels"] = state ? state.labels : undefined;
            resourceInputs["licenseKey"] = state ? state.licenseKey : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["springCloudServiceId"] = state ? state.springCloudServiceId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.appName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'appName'");
            }
            if ((!args || args.licenseKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'licenseKey'");
            }
            if ((!args || args.springCloudServiceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'springCloudServiceId'");
            }
            resourceInputs["agentEnabled"] = args ? args.agentEnabled : undefined;
            resourceInputs["appName"] = args ? args.appName : undefined;
            resourceInputs["appServerPort"] = args ? args.appServerPort : undefined;
            resourceInputs["auditModeEnabled"] = args ? args.auditModeEnabled : undefined;
            resourceInputs["autoAppNamingEnabled"] = args ? args.autoAppNamingEnabled : undefined;
            resourceInputs["autoTransactionNamingEnabled"] = args ? args.autoTransactionNamingEnabled : undefined;
            resourceInputs["customTracingEnabled"] = args ? args.customTracingEnabled : undefined;
            resourceInputs["globallyEnabled"] = args ? args.globallyEnabled : undefined;
            resourceInputs["labels"] = args ? args.labels : undefined;
            resourceInputs["licenseKey"] = (args === null || args === void 0 ? void 0 : args.licenseKey) ? pulumi.secret(args.licenseKey) : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["springCloudServiceId"] = args ? args.springCloudServiceId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["licenseKey"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(SpringCloudNewRelicApplicationPerformanceMonitoring.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SpringCloudNewRelicApplicationPerformanceMonitoring resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SpringCloudNewRelicApplicationPerformanceMonitoring(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SpringCloudNewRelicApplicationPerformanceMonitoring.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SpringCloudNewRelicApplicationPerformanceMonitoring.__pulumiType;
    }
}
exports.SpringCloudNewRelicApplicationPerformanceMonitoring = SpringCloudNewRelicApplicationPerformanceMonitoring;
/** @internal */
SpringCloudNewRelicApplicationPerformanceMonitoring.__pulumiType = 'azure:appplatform/springCloudNewRelicApplicationPerformanceMonitoring:SpringCloudNewRelicApplicationPerformanceMonitoring';
