// @pulumi/azure/appplatform/springCloudBuilder.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpringCloudBuilder = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Spring Cloud Builder.
 *
 * > **NOTE:** This resource is applicable only for Spring Cloud Service with enterprise tier.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleSpringCloudService = new azure.appplatform.SpringCloudService("example", {
 *     name: "example-springcloud",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     skuName: "E0",
 * });
 * const exampleSpringCloudBuilder = new azure.appplatform.SpringCloudBuilder("example", {
 *     name: "example",
 *     springCloudServiceId: exampleSpringCloudService.id,
 *     buildPackGroups: [{
 *         name: "mix",
 *         buildPackIds: ["tanzu-buildpacks/java-azure"],
 *     }],
 *     stack: {
 *         id: "io.buildpacks.stacks.bionic",
 *         version: "base",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Spring Cloud Builders can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:appplatform/springCloudBuilder:SpringCloudBuilder example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resourceGroup1/providers/Microsoft.AppPlatform/spring/service1/buildServices/buildService1/builders/builder1
 * ```
 */
class SpringCloudBuilder extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["buildPackGroups"] = state ? state.buildPackGroups : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["springCloudServiceId"] = state ? state.springCloudServiceId : undefined;
            resourceInputs["stack"] = state ? state.stack : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.buildPackGroups === undefined) && !opts.urn) {
                throw new Error("Missing required property 'buildPackGroups'");
            }
            if ((!args || args.springCloudServiceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'springCloudServiceId'");
            }
            if ((!args || args.stack === undefined) && !opts.urn) {
                throw new Error("Missing required property 'stack'");
            }
            resourceInputs["buildPackGroups"] = args ? args.buildPackGroups : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["springCloudServiceId"] = args ? args.springCloudServiceId : undefined;
            resourceInputs["stack"] = args ? args.stack : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SpringCloudBuilder.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SpringCloudBuilder resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SpringCloudBuilder(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SpringCloudBuilder.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SpringCloudBuilder.__pulumiType;
    }
}
exports.SpringCloudBuilder = SpringCloudBuilder;
/** @internal */
SpringCloudBuilder.__pulumiType = 'azure:appplatform/springCloudBuilder:SpringCloudBuilder';
