// @pulumi/azure-compliance-policies/azurenative/devhub/v20240801preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workflow = exports.IacProfile = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240801preview_1 = require("@pulumi/azure-native/devhub/v20240801preview");
const v20240801preview_2 = require("@pulumi/azure-native/devhub/v20240801preview");
var IacProfile;
(function (IacProfile) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devhub.v20240801preview.IacProfile).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IacProfile.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devhub-v20240801preview-iacprofile-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devhub.v20240801preview.IacProfile).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240801preview_1.IacProfile, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IacProfile shouldn't use an unstable API (devhub.v20240801preview.IacProfile). A compatible replacement can be found at 'devhub.IacProfile'.");
            }),
        },
        vendors: ["azure"],
        services: ["devhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IacProfile || (IacProfile = {}));
exports.IacProfile = IacProfile;
var Workflow;
(function (Workflow) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devhub.v20240801preview.Workflow).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Workflow.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devhub-v20240801preview-workflow-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devhub.v20240801preview.Workflow).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240801preview_2.Workflow, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Workflow shouldn't use an unstable API (devhub.v20240801preview.Workflow). A compatible replacement can be found at 'devhub.Workflow'.");
            }),
        },
        vendors: ["azure"],
        services: ["devhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Workflow || (Workflow = {}));
exports.Workflow = Workflow;
