// @pulumi/azure/dns/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Zone = exports.TxtRecord = exports.SrvRecord = exports.PtrRecord = exports.NsRecord = exports.MxRecord = exports.getZoneOutput = exports.getZone = exports.getTxtRecordOutput = exports.getTxtRecord = exports.getSrvRecordOutput = exports.getSrvRecord = exports.getSoaRecordOutput = exports.getSoaRecord = exports.getPtrRecordOutput = exports.getPtrRecord = exports.getNsRecordOutput = exports.getNsRecord = exports.getMxRecordOutput = exports.getMxRecord = exports.getCnameRecordOutput = exports.getCnameRecord = exports.getCAARecordOutput = exports.getCAARecord = exports.getARecordOutput = exports.getARecord = exports.getAAAARecordOutput = exports.getAAAARecord = exports.CNameRecord = exports.CaaRecord = exports.ARecord = exports.AaaaRecord = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AaaaRecord = null;
utilities.lazyLoad(exports, ["AaaaRecord"], () => require("./aaaaRecord"));
exports.ARecord = null;
utilities.lazyLoad(exports, ["ARecord"], () => require("./arecord"));
exports.CaaRecord = null;
utilities.lazyLoad(exports, ["CaaRecord"], () => require("./caaRecord"));
exports.CNameRecord = null;
utilities.lazyLoad(exports, ["CNameRecord"], () => require("./cnameRecord"));
exports.getAAAARecord = null;
exports.getAAAARecordOutput = null;
utilities.lazyLoad(exports, ["getAAAARecord", "getAAAARecordOutput"], () => require("./getAAAARecord"));
exports.getARecord = null;
exports.getARecordOutput = null;
utilities.lazyLoad(exports, ["getARecord", "getARecordOutput"], () => require("./getARecord"));
exports.getCAARecord = null;
exports.getCAARecordOutput = null;
utilities.lazyLoad(exports, ["getCAARecord", "getCAARecordOutput"], () => require("./getCAARecord"));
exports.getCnameRecord = null;
exports.getCnameRecordOutput = null;
utilities.lazyLoad(exports, ["getCnameRecord", "getCnameRecordOutput"], () => require("./getCnameRecord"));
exports.getMxRecord = null;
exports.getMxRecordOutput = null;
utilities.lazyLoad(exports, ["getMxRecord", "getMxRecordOutput"], () => require("./getMxRecord"));
exports.getNsRecord = null;
exports.getNsRecordOutput = null;
utilities.lazyLoad(exports, ["getNsRecord", "getNsRecordOutput"], () => require("./getNsRecord"));
exports.getPtrRecord = null;
exports.getPtrRecordOutput = null;
utilities.lazyLoad(exports, ["getPtrRecord", "getPtrRecordOutput"], () => require("./getPtrRecord"));
exports.getSoaRecord = null;
exports.getSoaRecordOutput = null;
utilities.lazyLoad(exports, ["getSoaRecord", "getSoaRecordOutput"], () => require("./getSoaRecord"));
exports.getSrvRecord = null;
exports.getSrvRecordOutput = null;
utilities.lazyLoad(exports, ["getSrvRecord", "getSrvRecordOutput"], () => require("./getSrvRecord"));
exports.getTxtRecord = null;
exports.getTxtRecordOutput = null;
utilities.lazyLoad(exports, ["getTxtRecord", "getTxtRecordOutput"], () => require("./getTxtRecord"));
exports.getZone = null;
exports.getZoneOutput = null;
utilities.lazyLoad(exports, ["getZone", "getZoneOutput"], () => require("./getZone"));
exports.MxRecord = null;
utilities.lazyLoad(exports, ["MxRecord"], () => require("./mxRecord"));
exports.NsRecord = null;
utilities.lazyLoad(exports, ["NsRecord"], () => require("./nsRecord"));
exports.PtrRecord = null;
utilities.lazyLoad(exports, ["PtrRecord"], () => require("./ptrRecord"));
exports.SrvRecord = null;
utilities.lazyLoad(exports, ["SrvRecord"], () => require("./srvRecord"));
exports.TxtRecord = null;
utilities.lazyLoad(exports, ["TxtRecord"], () => require("./txtRecord"));
exports.Zone = null;
utilities.lazyLoad(exports, ["Zone"], () => require("./zone"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:dns/aRecord:ARecord":
                return new exports.ARecord(name, undefined, { urn });
            case "azure:dns/aaaaRecord:AaaaRecord":
                return new exports.AaaaRecord(name, undefined, { urn });
            case "azure:dns/cNameRecord:CNameRecord":
                return new exports.CNameRecord(name, undefined, { urn });
            case "azure:dns/caaRecord:CaaRecord":
                return new exports.CaaRecord(name, undefined, { urn });
            case "azure:dns/mxRecord:MxRecord":
                return new exports.MxRecord(name, undefined, { urn });
            case "azure:dns/nsRecord:NsRecord":
                return new exports.NsRecord(name, undefined, { urn });
            case "azure:dns/ptrRecord:PtrRecord":
                return new exports.PtrRecord(name, undefined, { urn });
            case "azure:dns/srvRecord:SrvRecord":
                return new exports.SrvRecord(name, undefined, { urn });
            case "azure:dns/txtRecord:TxtRecord":
                return new exports.TxtRecord(name, undefined, { urn });
            case "azure:dns/zone:Zone":
                return new exports.Zone(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "dns/aRecord", _module);
pulumi.runtime.registerResourceModule("azure", "dns/aaaaRecord", _module);
pulumi.runtime.registerResourceModule("azure", "dns/cNameRecord", _module);
pulumi.runtime.registerResourceModule("azure", "dns/caaRecord", _module);
pulumi.runtime.registerResourceModule("azure", "dns/mxRecord", _module);
pulumi.runtime.registerResourceModule("azure", "dns/nsRecord", _module);
pulumi.runtime.registerResourceModule("azure", "dns/ptrRecord", _module);
pulumi.runtime.registerResourceModule("azure", "dns/srvRecord", _module);
pulumi.runtime.registerResourceModule("azure", "dns/txtRecord", _module);
pulumi.runtime.registerResourceModule("azure", "dns/zone", _module);
