// @pulumi/azure/marketplace/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleAssignment = exports.getAgreementOutput = exports.getAgreement = exports.Agreement = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Agreement = null;
utilities.lazyLoad(exports, ["Agreement"], () => require("./agreement"));
exports.getAgreement = null;
exports.getAgreementOutput = null;
utilities.lazyLoad(exports, ["getAgreement", "getAgreementOutput"], () => require("./getAgreement"));
exports.RoleAssignment = null;
utilities.lazyLoad(exports, ["RoleAssignment"], () => require("./roleAssignment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:marketplace/agreement:Agreement":
                return new exports.Agreement(name, undefined, { urn });
            case "azure:marketplace/roleAssignment:RoleAssignment":
                return new exports.RoleAssignment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "marketplace/agreement", _module);
pulumi.runtime.registerResourceModule("azure", "marketplace/roleAssignment", _module);
