// @pulumi/azure-compliance-policies/azurenative/connectedvmwarevsphere/v20230301preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetwork = exports.VirtualMachineTemplate = exports.VirtualMachineInstance = exports.VirtualMachine = exports.VMInstanceGuestAgent = exports.VCenter = exports.ResourcePool = exports.MachineExtension = exports.InventoryItem = exports.HybridIdentityMetadatum = exports.Host = exports.GuestAgent = exports.Datastore = exports.Cluster = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230301preview_1 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_2 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_3 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_4 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_5 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_6 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_7 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_8 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_9 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_10 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_11 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_12 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_13 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
const v20230301preview_14 = require("@pulumi/azure-native/connectedvmwarevsphere/v20230301preview");
var Cluster;
(function (Cluster) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.Cluster).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Cluster.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-cluster-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.Cluster).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_1.Cluster, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Cluster shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.Cluster). A compatible replacement can be found at 'connectedvmwarevsphere.Cluster'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Cluster || (Cluster = {}));
exports.Cluster = Cluster;
var Datastore;
(function (Datastore) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.Datastore).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Datastore.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-datastore-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.Datastore).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_2.Datastore, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Datastore shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.Datastore). A compatible replacement can be found at 'connectedvmwarevsphere.Datastore'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Datastore || (Datastore = {}));
exports.Datastore = Datastore;
var GuestAgent;
(function (GuestAgent) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.GuestAgent).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GuestAgent.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-guestagent-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.GuestAgent).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_3.GuestAgent, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GuestAgent shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.GuestAgent). A compatible replacement can be found at 'connectedvmwarevsphere.GuestAgent'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GuestAgent || (GuestAgent = {}));
exports.GuestAgent = GuestAgent;
var Host;
(function (Host) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.Host).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Host.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-host-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.Host).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_4.Host, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Host shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.Host). A compatible replacement can be found at 'connectedvmwarevsphere.Host'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Host || (Host = {}));
exports.Host = Host;
var HybridIdentityMetadatum;
(function (HybridIdentityMetadatum) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.HybridIdentityMetadatum).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    HybridIdentityMetadatum.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-hybrididentitymetadatum-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.HybridIdentityMetadatum).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_5.HybridIdentityMetadatum, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure HybridIdentityMetadatum shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.HybridIdentityMetadatum). A compatible replacement can be found at 'connectedvmwarevsphere.HybridIdentityMetadatum'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(HybridIdentityMetadatum || (HybridIdentityMetadatum = {}));
exports.HybridIdentityMetadatum = HybridIdentityMetadatum;
var InventoryItem;
(function (InventoryItem) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.InventoryItem).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    InventoryItem.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-inventoryitem-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.InventoryItem).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_6.InventoryItem, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure InventoryItem shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.InventoryItem). A compatible replacement can be found at 'connectedvmwarevsphere.InventoryItem'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(InventoryItem || (InventoryItem = {}));
exports.InventoryItem = InventoryItem;
var MachineExtension;
(function (MachineExtension) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.MachineExtension).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MachineExtension.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-machineextension-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.MachineExtension).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_7.MachineExtension, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MachineExtension shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.MachineExtension). A compatible replacement can be found at 'connectedvmwarevsphere.MachineExtension'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MachineExtension || (MachineExtension = {}));
exports.MachineExtension = MachineExtension;
var ResourcePool;
(function (ResourcePool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.ResourcePool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ResourcePool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-resourcepool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.ResourcePool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_8.ResourcePool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ResourcePool shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.ResourcePool). A compatible replacement can be found at 'connectedvmwarevsphere.ResourcePool'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ResourcePool || (ResourcePool = {}));
exports.ResourcePool = ResourcePool;
var VCenter;
(function (VCenter) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VCenter).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VCenter.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-vcenter-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VCenter).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_9.VCenter, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VCenter shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.VCenter). A compatible replacement can be found at 'connectedvmwarevsphere.VCenter'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VCenter || (VCenter = {}));
exports.VCenter = VCenter;
var VMInstanceGuestAgent;
(function (VMInstanceGuestAgent) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VMInstanceGuestAgent).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VMInstanceGuestAgent.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-vminstanceguestagent-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VMInstanceGuestAgent).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_10.VMInstanceGuestAgent, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VMInstanceGuestAgent shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.VMInstanceGuestAgent). A compatible replacement can be found at 'connectedvmwarevsphere.VMInstanceGuestAgent'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VMInstanceGuestAgent || (VMInstanceGuestAgent = {}));
exports.VMInstanceGuestAgent = VMInstanceGuestAgent;
var VirtualMachine;
(function (VirtualMachine) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VirtualMachine).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VirtualMachine.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-virtualmachine-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VirtualMachine).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_11.VirtualMachine, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VirtualMachine shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.VirtualMachine). A compatible replacement can be found at 'connectedvmwarevsphere.VirtualMachine'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VirtualMachine || (VirtualMachine = {}));
exports.VirtualMachine = VirtualMachine;
var VirtualMachineInstance;
(function (VirtualMachineInstance) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VirtualMachineInstance).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VirtualMachineInstance.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-virtualmachineinstance-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VirtualMachineInstance).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_12.VirtualMachineInstance, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VirtualMachineInstance shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.VirtualMachineInstance). A compatible replacement can be found at 'connectedvmwarevsphere.VirtualMachineInstance'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VirtualMachineInstance || (VirtualMachineInstance = {}));
exports.VirtualMachineInstance = VirtualMachineInstance;
var VirtualMachineTemplate;
(function (VirtualMachineTemplate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VirtualMachineTemplate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VirtualMachineTemplate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-virtualmachinetemplate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VirtualMachineTemplate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_13.VirtualMachineTemplate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VirtualMachineTemplate shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.VirtualMachineTemplate). A compatible replacement can be found at 'connectedvmwarevsphere.VirtualMachineTemplate'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VirtualMachineTemplate || (VirtualMachineTemplate = {}));
exports.VirtualMachineTemplate = VirtualMachineTemplate;
var VirtualNetwork;
(function (VirtualNetwork) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VirtualNetwork).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VirtualNetwork.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-connectedvmwarevsphere-v20230301preview-virtualnetwork-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (connectedvmwarevsphere.v20230301preview.VirtualNetwork).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230301preview_14.VirtualNetwork, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VirtualNetwork shouldn't use an unstable API (connectedvmwarevsphere.v20230301preview.VirtualNetwork). A compatible replacement can be found at 'connectedvmwarevsphere.VirtualNetwork'.");
            }),
        },
        vendors: ["azure"],
        services: ["connectedvmwarevsphere"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VirtualNetwork || (VirtualNetwork = {}));
exports.VirtualNetwork = VirtualNetwork;
