// @pulumi/azure-compliance-policies/azurenative/documentdb/v20241001preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnection = exports.MongoCluster = exports.FirewallRule = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20241001preview_1 = require("@pulumi/azure-native/documentdb/v20241001preview");
const v20241001preview_2 = require("@pulumi/azure-native/documentdb/v20241001preview");
const v20241001preview_3 = require("@pulumi/azure-native/documentdb/v20241001preview");
var FirewallRule;
(function (FirewallRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20241001preview.FirewallRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FirewallRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20241001preview-firewallrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20241001preview.FirewallRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_1.FirewallRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FirewallRule shouldn't use an unstable API (documentdb.v20241001preview.FirewallRule). A compatible replacement can be found at 'documentdb.FirewallRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FirewallRule || (FirewallRule = {}));
exports.FirewallRule = FirewallRule;
var MongoCluster;
(function (MongoCluster) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20241001preview.MongoCluster).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MongoCluster.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20241001preview-mongocluster-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20241001preview.MongoCluster).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_2.MongoCluster, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MongoCluster shouldn't use an unstable API (documentdb.v20241001preview.MongoCluster). A compatible replacement can be found at 'documentdb.MongoCluster'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MongoCluster || (MongoCluster = {}));
exports.MongoCluster = MongoCluster;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20241001preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20241001preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20241001preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_3.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (documentdb.v20241001preview.PrivateEndpointConnection). A compatible replacement can be found at 'documentdb.v20240701.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
