// @pulumi/azure-compliance-policies/azurenative/iotfirmwaredefense/v20230208preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.Firmware = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230208preview_1 = require("@pulumi/azure-native/iotfirmwaredefense/v20230208preview");
const v20230208preview_2 = require("@pulumi/azure-native/iotfirmwaredefense/v20230208preview");
var Firmware;
(function (Firmware) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotfirmwaredefense.v20230208preview.Firmware).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Firmware.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotfirmwaredefense-v20230208preview-firmware-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotfirmwaredefense.v20230208preview.Firmware).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230208preview_1.Firmware, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Firmware shouldn't use an unstable API (iotfirmwaredefense.v20230208preview.Firmware). A compatible replacement can be found at 'iotfirmwaredefense.Firmware'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotfirmwaredefense"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Firmware || (Firmware = {}));
exports.Firmware = Firmware;
var Workspace;
(function (Workspace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotfirmwaredefense.v20230208preview.Workspace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Workspace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotfirmwaredefense-v20230208preview-workspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotfirmwaredefense.v20230208preview.Workspace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230208preview_2.Workspace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Workspace shouldn't use an unstable API (iotfirmwaredefense.v20230208preview.Workspace). A compatible replacement can be found at 'iotfirmwaredefense.Workspace'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotfirmwaredefense"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Workspace || (Workspace = {}));
exports.Workspace = Workspace;
