// @pulumi/azure-compliance-policies/azurenative/operationalinsights/v20190901preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryPack = exports.Query = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20190901preview_1 = require("@pulumi/azure-native/operationalinsights/v20190901preview");
const v20190901preview_2 = require("@pulumi/azure-native/operationalinsights/v20190901preview");
var Query;
(function (Query) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (operationalinsights.v20190901preview.Query).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Query.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-operationalinsights-v20190901preview-query-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (operationalinsights.v20190901preview.Query).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20190901preview_1.Query, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Query shouldn't use an unstable API (operationalinsights.v20190901preview.Query). A compatible replacement can be found at 'operationalinsights.Query'.");
            }),
        },
        vendors: ["azure"],
        services: ["operationalinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Query || (Query = {}));
exports.Query = Query;
var QueryPack;
(function (QueryPack) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (operationalinsights.v20190901preview.QueryPack).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    QueryPack.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-operationalinsights-v20190901preview-querypack-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (operationalinsights.v20190901preview.QueryPack).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20190901preview_2.QueryPack, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure QueryPack shouldn't use an unstable API (operationalinsights.v20190901preview.QueryPack). A compatible replacement can be found at 'operationalinsights.QueryPack'.");
            }),
        },
        vendors: ["azure"],
        services: ["operationalinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(QueryPack || (QueryPack = {}));
exports.QueryPack = QueryPack;
