// @pulumi/azure-compliance-policies/azurenative/azuredata/v20190724preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlServerRegistration = exports.SqlServer = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20190724preview_1 = require("@pulumi/azure-native/azuredata/v20190724preview");
const v20190724preview_2 = require("@pulumi/azure-native/azuredata/v20190724preview");
var SqlServer;
(function (SqlServer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azuredata.v20190724preview.SqlServer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SqlServer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azuredata-v20190724preview-sqlserver-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azuredata.v20190724preview.SqlServer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20190724preview_1.SqlServer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SqlServer shouldn't use an unstable API (azuredata.v20190724preview.SqlServer). A compatible replacement can be found at 'azuredata.SqlServer'.");
            }),
        },
        vendors: ["azure"],
        services: ["azuredata"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SqlServer || (SqlServer = {}));
exports.SqlServer = SqlServer;
var SqlServerRegistration;
(function (SqlServerRegistration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azuredata.v20190724preview.SqlServerRegistration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SqlServerRegistration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azuredata-v20190724preview-sqlserverregistration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azuredata.v20190724preview.SqlServerRegistration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20190724preview_2.SqlServerRegistration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SqlServerRegistration shouldn't use an unstable API (azuredata.v20190724preview.SqlServerRegistration). A compatible replacement can be found at 'azuredata.SqlServerRegistration'.");
            }),
        },
        vendors: ["azure"],
        services: ["azuredata"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SqlServerRegistration || (SqlServerRegistration = {}));
exports.SqlServerRegistration = SqlServerRegistration;
