// @pulumi/azure/orbital/contact.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Contact = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an orbital contact.
 *
 * > **Note:** The `azure.orbital.Contact` resource has been deprecated and will be removed in v5.0 of the AzureRM Provider.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "rg-example",
 *     location: "West Europe",
 * });
 * const exampleSpacecraft = new azure.orbital.Spacecraft("example", {
 *     name: "example-spacecraft",
 *     resourceGroupName: example.name,
 *     location: "westeurope",
 *     noradId: "12345",
 *     links: [{
 *         bandwidthMhz: 100,
 *         centerFrequencyMhz: 101,
 *         direction: "Uplink",
 *         polarization: "LHCP",
 *         name: "examplename",
 *     }],
 *     twoLineElements: [
 *         "1 23455U 94089A   97320.90946019  .00000140  00000-0  10191-3 0  2621",
 *         "2 23455  99.0090 272.6745 0008546 223.1686 136.8816 14.11711747148495",
 *     ],
 *     titleLine: "AQUA",
 *     tags: {
 *         "aks-managed-cluster-name": "9a57225d-a405-4d40-aa46-f13d2342abef",
 *     },
 * });
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("example", {
 *     name: "example-vnet",
 *     addressSpaces: ["10.0.0.0/16"],
 *     location: example.location,
 *     resourceGroupName: example.name,
 * });
 * const exampleSubnet = new azure.network.Subnet("example", {
 *     name: "example-subnet",
 *     resourceGroupName: example.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.0.1.0/24"],
 *     delegations: [{
 *         name: "orbitalgateway",
 *         serviceDelegation: {
 *             name: "Microsoft.Orbital/orbitalGateways",
 *             actions: [
 *                 "Microsoft.Network/publicIPAddresses/join/action",
 *                 "Microsoft.Network/virtualNetworks/subnets/join/action",
 *                 "Microsoft.Network/virtualNetworks/read",
 *                 "Microsoft.Network/publicIPAddresses/read",
 *             ],
 *         },
 *     }],
 * });
 * const exampleContactProfile = new azure.orbital.ContactProfile("example", {
 *     name: "example-contactprofile",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     minimumVariableContactDuration: "PT1M",
 *     autoTracking: "disabled",
 *     links: [{
 *         channels: [{
 *             name: "channelname",
 *             bandwidthMhz: 100,
 *             centerFrequencyMhz: 101,
 *             endPoints: [{
 *                 endPointName: "AQUA_command",
 *                 ipAddress: "10.0.1.0",
 *                 port: "49153",
 *                 protocol: "TCP",
 *             }],
 *         }],
 *         direction: "Uplink",
 *         name: "RHCP_UL",
 *         polarization: "RHCP",
 *     }],
 *     networkConfigurationSubnetId: exampleSubnet.id,
 * });
 * const exampleContact = new azure.orbital.Contact("example", {
 *     name: "example-contact",
 *     spacecraftId: exampleSpacecraft.id,
 *     reservationStartTime: "2020-07-16T20:35:00.00Z",
 *     reservationEndTime: "2020-07-16T20:55:00.00Z",
 *     groundStationName: "WESTUS2_0",
 *     contactProfileId: exampleContactProfile.id,
 * });
 * ```
 *
 * ## Import
 *
 * Spacecraft can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:orbital/contact:Contact example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Orbital/spacecrafts/spacecraft1/contacts/contact1
 * ```
 */
class Contact extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["contactProfileId"] = state ? state.contactProfileId : undefined;
            resourceInputs["groundStationName"] = state ? state.groundStationName : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["reservationEndTime"] = state ? state.reservationEndTime : undefined;
            resourceInputs["reservationStartTime"] = state ? state.reservationStartTime : undefined;
            resourceInputs["spacecraftId"] = state ? state.spacecraftId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.contactProfileId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'contactProfileId'");
            }
            if ((!args || args.groundStationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groundStationName'");
            }
            if ((!args || args.reservationEndTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'reservationEndTime'");
            }
            if ((!args || args.reservationStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'reservationStartTime'");
            }
            if ((!args || args.spacecraftId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'spacecraftId'");
            }
            resourceInputs["contactProfileId"] = args ? args.contactProfileId : undefined;
            resourceInputs["groundStationName"] = args ? args.groundStationName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["reservationEndTime"] = args ? args.reservationEndTime : undefined;
            resourceInputs["reservationStartTime"] = args ? args.reservationStartTime : undefined;
            resourceInputs["spacecraftId"] = args ? args.spacecraftId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Contact.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Contact resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Contact(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Contact.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Contact.__pulumiType;
    }
}
exports.Contact = Contact;
/** @internal */
Contact.__pulumiType = 'azure:orbital/contact:Contact';
