// @pulumi/azure/elasticsan/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeGroup = exports.Volume = exports.getVolumeSnapshotOutput = exports.getVolumeSnapshot = exports.getVolumeGroupOutput = exports.getVolumeGroup = exports.getOutput = exports.get = exports.ElasticSan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ElasticSan = null;
utilities.lazyLoad(exports, ["ElasticSan"], () => require("./elasticSan"));
exports.get = null;
exports.getOutput = null;
utilities.lazyLoad(exports, ["get", "getOutput"], () => require("./get"));
exports.getVolumeGroup = null;
exports.getVolumeGroupOutput = null;
utilities.lazyLoad(exports, ["getVolumeGroup", "getVolumeGroupOutput"], () => require("./getVolumeGroup"));
exports.getVolumeSnapshot = null;
exports.getVolumeSnapshotOutput = null;
utilities.lazyLoad(exports, ["getVolumeSnapshot", "getVolumeSnapshotOutput"], () => require("./getVolumeSnapshot"));
exports.Volume = null;
utilities.lazyLoad(exports, ["Volume"], () => require("./volume"));
exports.VolumeGroup = null;
utilities.lazyLoad(exports, ["VolumeGroup"], () => require("./volumeGroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:elasticsan/elasticSan:ElasticSan":
                return new exports.ElasticSan(name, undefined, { urn });
            case "azure:elasticsan/volume:Volume":
                return new exports.Volume(name, undefined, { urn });
            case "azure:elasticsan/volumeGroup:VolumeGroup":
                return new exports.VolumeGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "elasticsan/elasticSan", _module);
pulumi.runtime.registerResourceModule("azure", "elasticsan/volume", _module);
pulumi.runtime.registerResourceModule("azure", "elasticsan/volumeGroup", _module);
