// @pulumi/azure-compliance-policies/azurenative/intune/v20150114privatepreview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.IoMAMPolicyByName = exports.AndroidMAMPolicyByName = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20150114privatepreview_1 = require("@pulumi/azure-native/intune/v20150114privatepreview");
const v20150114privatepreview_2 = require("@pulumi/azure-native/intune/v20150114privatepreview");
var AndroidMAMPolicyByName;
(function (AndroidMAMPolicyByName) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (intune.v20150114privatepreview.AndroidMAMPolicyByName).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AndroidMAMPolicyByName.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-intune-v20150114privatepreview-androidmampolicybyname-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (intune.v20150114privatepreview.AndroidMAMPolicyByName).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20150114privatepreview_1.AndroidMAMPolicyByName, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AndroidMAMPolicyByName shouldn't use an unstable API (intune.v20150114privatepreview.AndroidMAMPolicyByName). A compatible replacement can be found at 'intune.AndroidMAMPolicyByName'.");
            }),
        },
        vendors: ["azure"],
        services: ["intune"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AndroidMAMPolicyByName || (AndroidMAMPolicyByName = {}));
exports.AndroidMAMPolicyByName = AndroidMAMPolicyByName;
var IoMAMPolicyByName;
(function (IoMAMPolicyByName) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (intune.v20150114privatepreview.IoMAMPolicyByName).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IoMAMPolicyByName.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-intune-v20150114privatepreview-iomampolicybyname-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (intune.v20150114privatepreview.IoMAMPolicyByName).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20150114privatepreview_2.IoMAMPolicyByName, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IoMAMPolicyByName shouldn't use an unstable API (intune.v20150114privatepreview.IoMAMPolicyByName). A compatible replacement can be found at 'intune.IoMAMPolicyByName'.");
            }),
        },
        vendors: ["azure"],
        services: ["intune"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IoMAMPolicyByName || (IoMAMPolicyByName = {}));
exports.IoMAMPolicyByName = IoMAMPolicyByName;
