// @pulumi/azure/mssql/jobSchedule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Elastic Job Schedule.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resource-group",
 *     location: "East US",
 * });
 * const exampleServer = new azure.mssql.Server("example", {
 *     name: "example-server",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     version: "12.0",
 *     administratorLogin: "4dm1n157r470r",
 *     administratorLoginPassword: "4-v3ry-53cr37-p455w0rd",
 * });
 * const exampleDatabase = new azure.mssql.Database("example", {
 *     name: "example-db",
 *     serverId: exampleServer.id,
 *     collation: "SQL_Latin1_General_CP1_CI_AS",
 *     skuName: "S1",
 * });
 * const exampleJobAgent = new azure.mssql.JobAgent("example", {
 *     name: "example-job-agent",
 *     location: example.location,
 *     databaseId: exampleDatabase.id,
 * });
 * const exampleJobCredential = new azure.mssql.JobCredential("example", {
 *     name: "example-job-credential",
 *     jobAgentId: exampleJobAgent.id,
 *     username: "my-username",
 *     password: "MyP4ssw0rd!!!",
 * });
 * const exampleJob = new azure.mssql.Job("example", {
 *     name: "example-job",
 *     jobAgentId: exampleJobAgent.id,
 * });
 * const exampleJobSchedule = new azure.mssql.JobSchedule("example", {
 *     jobId: exampleJob.id,
 *     type: "Recurring",
 *     enabled: true,
 *     endTime: "2025-12-01T00:00:00Z",
 *     interval: "PT5M",
 *     startTime: "2025-01-01T00:00:00Z",
 * });
 * ```
 *
 * ## Import
 *
 * Elastic Job Schedules can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:mssql/jobSchedule:JobSchedule example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Sql/servers/myserver1/jobAgents/myjobagent1/jobs/myjob1
 * ```
 */
class JobSchedule extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["endTime"] = state ? state.endTime : undefined;
            resourceInputs["interval"] = state ? state.interval : undefined;
            resourceInputs["jobId"] = state ? state.jobId : undefined;
            resourceInputs["startTime"] = state ? state.startTime : undefined;
            resourceInputs["type"] = state ? state.type : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.jobId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'jobId'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["endTime"] = args ? args.endTime : undefined;
            resourceInputs["interval"] = args ? args.interval : undefined;
            resourceInputs["jobId"] = args ? args.jobId : undefined;
            resourceInputs["startTime"] = args ? args.startTime : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(JobSchedule.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing JobSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new JobSchedule(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of JobSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === JobSchedule.__pulumiType;
    }
}
exports.JobSchedule = JobSchedule;
/** @internal */
JobSchedule.__pulumiType = 'azure:mssql/jobSchedule:JobSchedule';
