// @pulumi/azure-native/iotoperationsmq/kafkaConnectorTopicMap.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KafkaConnectorTopicMap = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * MQ kafkaConnector/topicMap resource
 * Azure REST API version: 2023-10-04-preview.
 */
class KafkaConnectorTopicMap extends pulumi.CustomResource {
    /**
     * Get an existing KafkaConnectorTopicMap resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new KafkaConnectorTopicMap(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KafkaConnectorTopicMap.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KafkaConnectorTopicMap.__pulumiType;
    }
    /**
     * Create a KafkaConnectorTopicMap resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.extendedLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'extendedLocation'");
            }
            if ((!args || args.kafkaConnectorName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kafkaConnectorName'");
            }
            if ((!args || args.kafkaConnectorRef === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kafkaConnectorRef'");
            }
            if ((!args || args.mqName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mqName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.routes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'routes'");
            }
            resourceInputs["batching"] = args ? (args.batching ? pulumi.output(args.batching).apply(inputs.iotoperationsmq.kafkaTopicMapBatchingArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["compression"] = (_a = (args ? args.compression : undefined)) !== null && _a !== void 0 ? _a : "none";
            resourceInputs["copyMqttProperties"] = args ? args.copyMqttProperties : undefined;
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["kafkaConnectorName"] = args ? args.kafkaConnectorName : undefined;
            resourceInputs["kafkaConnectorRef"] = args ? args.kafkaConnectorRef : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["mqName"] = args ? args.mqName : undefined;
            resourceInputs["partitionKeyProperty"] = args ? args.partitionKeyProperty : undefined;
            resourceInputs["partitionStrategy"] = (_b = (args ? args.partitionStrategy : undefined)) !== null && _b !== void 0 ? _b : "default";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["routes"] = args ? args.routes : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["topicMapName"] = args ? args.topicMapName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["batching"] = undefined /*out*/;
            resourceInputs["compression"] = undefined /*out*/;
            resourceInputs["copyMqttProperties"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["kafkaConnectorRef"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["partitionKeyProperty"] = undefined /*out*/;
            resourceInputs["partitionStrategy"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["routes"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:iotoperationsmq/v20231004preview:KafkaConnectorTopicMap" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(KafkaConnectorTopicMap.__pulumiType, name, resourceInputs, opts);
    }
}
exports.KafkaConnectorTopicMap = KafkaConnectorTopicMap;
/** @internal */
KafkaConnectorTopicMap.__pulumiType = 'azure-native:iotoperationsmq:KafkaConnectorTopicMap';
