// @pulumi/azure-native/containerservice/fleetMember.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FleetMember = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A member of the Fleet. It contains a reference to an existing Kubernetes cluster on Azure.
 * Azure REST API version: 2023-03-15-preview.
 *
 * Other available API versions: 2022-07-02-preview, 2023-06-15-preview, 2023-08-15-preview, 2023-10-15, 2024-02-02-preview, 2024-04-01, 2024-05-02-preview.
 */
class FleetMember extends pulumi.CustomResource {
    /**
     * Get an existing FleetMember resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FleetMember(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FleetMember.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FleetMember.__pulumiType;
    }
    /**
     * Create a FleetMember resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterResourceId'");
            }
            if ((!args || args.fleetName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fleetName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["clusterResourceId"] = args ? args.clusterResourceId : undefined;
            resourceInputs["fleetMemberName"] = args ? args.fleetMemberName : undefined;
            resourceInputs["fleetName"] = args ? args.fleetName : undefined;
            resourceInputs["group"] = args ? args.group : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["clusterResourceId"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["group"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:containerservice/v20220602preview:FleetMember" }, { type: "azure-native:containerservice/v20220702preview:FleetMember" }, { type: "azure-native:containerservice/v20220902preview:FleetMember" }, { type: "azure-native:containerservice/v20230315preview:FleetMember" }, { type: "azure-native:containerservice/v20230615preview:FleetMember" }, { type: "azure-native:containerservice/v20230815preview:FleetMember" }, { type: "azure-native:containerservice/v20231015:FleetMember" }, { type: "azure-native:containerservice/v20240202preview:FleetMember" }, { type: "azure-native:containerservice/v20240401:FleetMember" }, { type: "azure-native:containerservice/v20240502preview:FleetMember" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FleetMember.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FleetMember = FleetMember;
/** @internal */
FleetMember.__pulumiType = 'azure-native:containerservice:FleetMember';
