// @pulumi/azure/iot/iotHubDeviceUpdateInstance.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotHubDeviceUpdateInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an IoT Hub Device Update Instance.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "East US",
 * });
 * const exampleIotHubDeviceUpdateAccount = new azure.iot.IotHubDeviceUpdateAccount("example", {
 *     name: "example",
 *     resourceGroupName: example.name,
 *     location: example.location,
 * });
 * const exampleIoTHub = new azure.iot.IoTHub("example", {
 *     name: "example",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     sku: {
 *         name: "S1",
 *         capacity: 1,
 *     },
 * });
 * const exampleAccount = new azure.storage.Account("example", {
 *     name: "example",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 * });
 * const exampleIotHubDeviceUpdateInstance = new azure.iot.IotHubDeviceUpdateInstance("example", {
 *     name: "example",
 *     deviceUpdateAccountId: exampleIotHubDeviceUpdateAccount.id,
 *     iothubId: exampleIoTHub.id,
 *     diagnosticEnabled: true,
 *     diagnosticStorageAccount: {
 *         connectionString: exampleAccount.primaryConnectionString,
 *         id: exampleAccount.id,
 *     },
 *     tags: {
 *         key: "value",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * IoT Hub Device Update Instance can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:iot/iotHubDeviceUpdateInstance:IotHubDeviceUpdateInstance example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.DeviceUpdate/accounts/account1/instances/instance1
 * ```
 */
class IotHubDeviceUpdateInstance extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deviceUpdateAccountId"] = state ? state.deviceUpdateAccountId : undefined;
            resourceInputs["diagnosticEnabled"] = state ? state.diagnosticEnabled : undefined;
            resourceInputs["diagnosticStorageAccount"] = state ? state.diagnosticStorageAccount : undefined;
            resourceInputs["iothubId"] = state ? state.iothubId : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.deviceUpdateAccountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceUpdateAccountId'");
            }
            if ((!args || args.iothubId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iothubId'");
            }
            resourceInputs["deviceUpdateAccountId"] = args ? args.deviceUpdateAccountId : undefined;
            resourceInputs["diagnosticEnabled"] = args ? args.diagnosticEnabled : undefined;
            resourceInputs["diagnosticStorageAccount"] = args ? args.diagnosticStorageAccount : undefined;
            resourceInputs["iothubId"] = args ? args.iothubId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(IotHubDeviceUpdateInstance.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing IotHubDeviceUpdateInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new IotHubDeviceUpdateInstance(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IotHubDeviceUpdateInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IotHubDeviceUpdateInstance.__pulumiType;
    }
}
exports.IotHubDeviceUpdateInstance = IotHubDeviceUpdateInstance;
/** @internal */
IotHubDeviceUpdateInstance.__pulumiType = 'azure:iot/iotHubDeviceUpdateInstance:IotHubDeviceUpdateInstance';
