// @pulumi/azure-native/voiceservices/communicationsGateway.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommunicationsGateway = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A CommunicationsGateway resource
 * Azure REST API version: 2023-04-03. Prior API version in Azure Native 1.x: 2022-12-01-preview.
 *
 * Other available API versions: 2023-09-01.
 */
class CommunicationsGateway extends pulumi.CustomResource {
    /**
     * Get an existing CommunicationsGateway resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CommunicationsGateway(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CommunicationsGateway.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CommunicationsGateway.__pulumiType;
    }
    /**
     * Create a CommunicationsGateway resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.codecs === undefined) && !opts.urn) {
                throw new Error("Missing required property 'codecs'");
            }
            if ((!args || args.connectivity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectivity'");
            }
            if ((!args || args.e911Type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'e911Type'");
            }
            if ((!args || args.platforms === undefined) && !opts.urn) {
                throw new Error("Missing required property 'platforms'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceLocations === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceLocations'");
            }
            resourceInputs["apiBridge"] = args ? args.apiBridge : undefined;
            resourceInputs["autoGeneratedDomainNameLabelScope"] = (_a = (args ? args.autoGeneratedDomainNameLabelScope : undefined)) !== null && _a !== void 0 ? _a : "TenantReuse";
            resourceInputs["codecs"] = args ? args.codecs : undefined;
            resourceInputs["communicationsGatewayName"] = args ? args.communicationsGatewayName : undefined;
            resourceInputs["connectivity"] = args ? args.connectivity : undefined;
            resourceInputs["e911Type"] = args ? args.e911Type : undefined;
            resourceInputs["emergencyDialStrings"] = args ? args.emergencyDialStrings : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["integratedMcpEnabled"] = (_b = (args ? args.integratedMcpEnabled : undefined)) !== null && _b !== void 0 ? _b : false;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["onPremMcpEnabled"] = (_c = (args ? args.onPremMcpEnabled : undefined)) !== null && _c !== void 0 ? _c : false;
            resourceInputs["platforms"] = args ? args.platforms : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceLocations"] = args ? args.serviceLocations : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["teamsVoicemailPilotNumber"] = args ? args.teamsVoicemailPilotNumber : undefined;
            resourceInputs["autoGeneratedDomainNameLabel"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["apiBridge"] = undefined /*out*/;
            resourceInputs["autoGeneratedDomainNameLabel"] = undefined /*out*/;
            resourceInputs["autoGeneratedDomainNameLabelScope"] = undefined /*out*/;
            resourceInputs["codecs"] = undefined /*out*/;
            resourceInputs["connectivity"] = undefined /*out*/;
            resourceInputs["e911Type"] = undefined /*out*/;
            resourceInputs["emergencyDialStrings"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["integratedMcpEnabled"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["onPremMcpEnabled"] = undefined /*out*/;
            resourceInputs["platforms"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["serviceLocations"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["teamsVoicemailPilotNumber"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:voiceservices/v20221201preview:CommunicationsGateway" }, { type: "azure-native:voiceservices/v20230131:CommunicationsGateway" }, { type: "azure-native:voiceservices/v20230403:CommunicationsGateway" }, { type: "azure-native:voiceservices/v20230901:CommunicationsGateway" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CommunicationsGateway.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CommunicationsGateway = CommunicationsGateway;
/** @internal */
CommunicationsGateway.__pulumiType = 'azure-native:voiceservices:CommunicationsGateway';
