// @pulumi/azure-compliance-policies/azurenative/dbformysql/v20200701privatepreview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerKey = exports.Server = exports.Configuration = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20200701privatepreview_1 = require("@pulumi/azure-native/dbformysql/v20200701privatepreview");
const v20200701privatepreview_2 = require("@pulumi/azure-native/dbformysql/v20200701privatepreview");
const v20200701privatepreview_3 = require("@pulumi/azure-native/dbformysql/v20200701privatepreview");
var Configuration;
(function (Configuration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dbformysql.v20200701privatepreview.Configuration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Configuration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dbformysql-v20200701privatepreview-configuration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dbformysql.v20200701privatepreview.Configuration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200701privatepreview_1.Configuration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Configuration shouldn't use an unstable API (dbformysql.v20200701privatepreview.Configuration). A compatible replacement can be found at 'dbformysql.v20220101.Configuration'.");
            }),
        },
        vendors: ["azure"],
        services: ["dbformysql"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Configuration || (Configuration = {}));
exports.Configuration = Configuration;
var Server;
(function (Server) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dbformysql.v20200701privatepreview.Server).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Server.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dbformysql-v20200701privatepreview-server-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dbformysql.v20200701privatepreview.Server).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200701privatepreview_2.Server, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Server shouldn't use an unstable API (dbformysql.v20200701privatepreview.Server). A compatible replacement can be found at 'dbformysql.v20220101.Server'.");
            }),
        },
        vendors: ["azure"],
        services: ["dbformysql"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Server || (Server = {}));
exports.Server = Server;
var ServerKey;
(function (ServerKey) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dbformysql.v20200701privatepreview.ServerKey).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ServerKey.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dbformysql-v20200701privatepreview-serverkey-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dbformysql.v20200701privatepreview.ServerKey).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200701privatepreview_3.ServerKey, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ServerKey shouldn't use an unstable API (dbformysql.v20200701privatepreview.ServerKey).");
            }),
        },
        vendors: ["azure"],
        services: ["dbformysql"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ServerKey || (ServerKey = {}));
exports.ServerKey = ServerKey;
