// @pulumi/azure-compliance-policies/azurenative/migrate/v20180901preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Solution = exports.MigrateProject = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20180901preview_1 = require("@pulumi/azure-native/migrate/v20180901preview");
const v20180901preview_2 = require("@pulumi/azure-native/migrate/v20180901preview");
var MigrateProject;
(function (MigrateProject) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (migrate.v20180901preview.MigrateProject).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MigrateProject.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-migrate-v20180901preview-migrateproject-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (migrate.v20180901preview.MigrateProject).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_1.MigrateProject, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MigrateProject shouldn't use an unstable API (migrate.v20180901preview.MigrateProject). A compatible replacement can be found at 'migrate.MigrateProject'.");
            }),
        },
        vendors: ["azure"],
        services: ["migrate"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MigrateProject || (MigrateProject = {}));
exports.MigrateProject = MigrateProject;
var Solution;
(function (Solution) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (migrate.v20180901preview.Solution).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Solution.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-migrate-v20180901preview-solution-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (migrate.v20180901preview.Solution).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_2.Solution, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Solution shouldn't use an unstable API (migrate.v20180901preview.Solution). A compatible replacement can be found at 'migrate.Solution'.");
            }),
        },
        vendors: ["azure"],
        services: ["migrate"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Solution || (Solution = {}));
exports.Solution = Solution;
