// @pulumi/azure-compliance-policies/azurenative/iotoperations/v20240815preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Instance = exports.DataflowProfile = exports.DataflowEndpoint = exports.Dataflow = exports.BrokerListener = exports.BrokerAuthorization = exports.BrokerAuthentication = exports.Broker = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240815preview_1 = require("@pulumi/azure-native/iotoperations/v20240815preview");
const v20240815preview_2 = require("@pulumi/azure-native/iotoperations/v20240815preview");
const v20240815preview_3 = require("@pulumi/azure-native/iotoperations/v20240815preview");
const v20240815preview_4 = require("@pulumi/azure-native/iotoperations/v20240815preview");
const v20240815preview_5 = require("@pulumi/azure-native/iotoperations/v20240815preview");
const v20240815preview_6 = require("@pulumi/azure-native/iotoperations/v20240815preview");
const v20240815preview_7 = require("@pulumi/azure-native/iotoperations/v20240815preview");
const v20240815preview_8 = require("@pulumi/azure-native/iotoperations/v20240815preview");
var Broker;
(function (Broker) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.Broker).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Broker.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotoperations-v20240815preview-broker-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.Broker).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240815preview_1.Broker, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Broker shouldn't use an unstable API (iotoperations.v20240815preview.Broker). A compatible replacement can be found at 'iotoperations.Broker'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotoperations"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Broker || (Broker = {}));
exports.Broker = Broker;
var BrokerAuthentication;
(function (BrokerAuthentication) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.BrokerAuthentication).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BrokerAuthentication.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotoperations-v20240815preview-brokerauthentication-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.BrokerAuthentication).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240815preview_2.BrokerAuthentication, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BrokerAuthentication shouldn't use an unstable API (iotoperations.v20240815preview.BrokerAuthentication). A compatible replacement can be found at 'iotoperations.BrokerAuthentication'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotoperations"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BrokerAuthentication || (BrokerAuthentication = {}));
exports.BrokerAuthentication = BrokerAuthentication;
var BrokerAuthorization;
(function (BrokerAuthorization) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.BrokerAuthorization).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BrokerAuthorization.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotoperations-v20240815preview-brokerauthorization-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.BrokerAuthorization).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240815preview_3.BrokerAuthorization, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BrokerAuthorization shouldn't use an unstable API (iotoperations.v20240815preview.BrokerAuthorization). A compatible replacement can be found at 'iotoperations.BrokerAuthorization'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotoperations"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BrokerAuthorization || (BrokerAuthorization = {}));
exports.BrokerAuthorization = BrokerAuthorization;
var BrokerListener;
(function (BrokerListener) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.BrokerListener).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BrokerListener.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotoperations-v20240815preview-brokerlistener-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.BrokerListener).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240815preview_4.BrokerListener, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BrokerListener shouldn't use an unstable API (iotoperations.v20240815preview.BrokerListener). A compatible replacement can be found at 'iotoperations.BrokerListener'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotoperations"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BrokerListener || (BrokerListener = {}));
exports.BrokerListener = BrokerListener;
var Dataflow;
(function (Dataflow) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.Dataflow).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Dataflow.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotoperations-v20240815preview-dataflow-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.Dataflow).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240815preview_5.Dataflow, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Dataflow shouldn't use an unstable API (iotoperations.v20240815preview.Dataflow). A compatible replacement can be found at 'iotoperations.Dataflow'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotoperations"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Dataflow || (Dataflow = {}));
exports.Dataflow = Dataflow;
var DataflowEndpoint;
(function (DataflowEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.DataflowEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DataflowEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotoperations-v20240815preview-dataflowendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.DataflowEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240815preview_6.DataflowEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DataflowEndpoint shouldn't use an unstable API (iotoperations.v20240815preview.DataflowEndpoint). A compatible replacement can be found at 'iotoperations.DataflowEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotoperations"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DataflowEndpoint || (DataflowEndpoint = {}));
exports.DataflowEndpoint = DataflowEndpoint;
var DataflowProfile;
(function (DataflowProfile) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.DataflowProfile).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DataflowProfile.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotoperations-v20240815preview-dataflowprofile-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.DataflowProfile).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240815preview_7.DataflowProfile, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DataflowProfile shouldn't use an unstable API (iotoperations.v20240815preview.DataflowProfile). A compatible replacement can be found at 'iotoperations.DataflowProfile'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotoperations"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DataflowProfile || (DataflowProfile = {}));
exports.DataflowProfile = DataflowProfile;
var Instance;
(function (Instance) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.Instance).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Instance.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-iotoperations-v20240815preview-instance-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (iotoperations.v20240815preview.Instance).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240815preview_8.Instance, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Instance shouldn't use an unstable API (iotoperations.v20240815preview.Instance). A compatible replacement can be found at 'iotoperations.Instance'.");
            }),
        },
        vendors: ["azure"],
        services: ["iotoperations"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Instance || (Instance = {}));
exports.Instance = Instance;
