// @pulumi/azure-compliance-policies/azurenative/appconfiguration/v20230801preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Replica = exports.PrivateEndpointConnection = exports.KeyValue = exports.ConfigurationStore = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230801preview_1 = require("@pulumi/azure-native/appconfiguration/v20230801preview");
const v20230801preview_2 = require("@pulumi/azure-native/appconfiguration/v20230801preview");
const v20230801preview_3 = require("@pulumi/azure-native/appconfiguration/v20230801preview");
const v20230801preview_4 = require("@pulumi/azure-native/appconfiguration/v20230801preview");
var ConfigurationStore;
(function (ConfigurationStore) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appconfiguration.v20230801preview.ConfigurationStore).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConfigurationStore.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appconfiguration-v20230801preview-configurationstore-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appconfiguration.v20230801preview.ConfigurationStore).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_1.ConfigurationStore, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConfigurationStore shouldn't use an unstable API (appconfiguration.v20230801preview.ConfigurationStore). A compatible replacement can be found at 'appconfiguration.ConfigurationStore'.");
            }),
        },
        vendors: ["azure"],
        services: ["appconfiguration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConfigurationStore || (ConfigurationStore = {}));
exports.ConfigurationStore = ConfigurationStore;
var KeyValue;
(function (KeyValue) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appconfiguration.v20230801preview.KeyValue).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    KeyValue.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appconfiguration-v20230801preview-keyvalue-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appconfiguration.v20230801preview.KeyValue).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_2.KeyValue, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure KeyValue shouldn't use an unstable API (appconfiguration.v20230801preview.KeyValue). A compatible replacement can be found at 'appconfiguration.KeyValue'.");
            }),
        },
        vendors: ["azure"],
        services: ["appconfiguration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(KeyValue || (KeyValue = {}));
exports.KeyValue = KeyValue;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appconfiguration.v20230801preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appconfiguration-v20230801preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appconfiguration.v20230801preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_3.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (appconfiguration.v20230801preview.PrivateEndpointConnection). A compatible replacement can be found at 'appconfiguration.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["appconfiguration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
var Replica;
(function (Replica) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appconfiguration.v20230801preview.Replica).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Replica.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appconfiguration-v20230801preview-replica-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appconfiguration.v20230801preview.Replica).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_4.Replica, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Replica shouldn't use an unstable API (appconfiguration.v20230801preview.Replica). A compatible replacement can be found at 'appconfiguration.Replica'.");
            }),
        },
        vendors: ["azure"],
        services: ["appconfiguration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Replica || (Replica = {}));
exports.Replica = Replica;
