// @pulumi/azure-compliance-policies/azurenative/appplatform/v20240101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Storage = exports.ServiceRegistry = exports.Service = exports.MonitoringSetting = exports.GatewayRouteConfig = exports.GatewayCustomDomain = exports.Gateway = exports.DevToolPortal = exports.Deployment = exports.CustomizedAccelerator = exports.CustomDomain = exports.ContainerRegistry = exports.ConfigurationService = exports.ConfigServer = exports.Certificate = exports.BuildpackBinding = exports.BuildServiceBuilder = exports.BuildServiceBuild = exports.BuildServiceAgentPool = exports.Binding = exports.ApplicationLiveView = exports.ApplicationAccelerator = exports.App = exports.Apm = exports.ApiPortalCustomDomain = exports.ApiPortal = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240101preview_1 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_2 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_3 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_4 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_5 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_6 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_7 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_8 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_9 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_10 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_11 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_12 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_13 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_14 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_15 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_16 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_17 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_18 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_19 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_20 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_21 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_22 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_23 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_24 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_25 = require("@pulumi/azure-native/appplatform/v20240101preview");
const v20240101preview_26 = require("@pulumi/azure-native/appplatform/v20240101preview");
var ApiPortal;
(function (ApiPortal) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ApiPortal).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiPortal.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-apiportal-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ApiPortal).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_1.ApiPortal, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiPortal shouldn't use an unstable API (appplatform.v20240101preview.ApiPortal). A compatible replacement can be found at 'appplatform.ApiPortal'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiPortal || (ApiPortal = {}));
exports.ApiPortal = ApiPortal;
var ApiPortalCustomDomain;
(function (ApiPortalCustomDomain) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ApiPortalCustomDomain).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiPortalCustomDomain.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-apiportalcustomdomain-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ApiPortalCustomDomain).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_2.ApiPortalCustomDomain, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiPortalCustomDomain shouldn't use an unstable API (appplatform.v20240101preview.ApiPortalCustomDomain). A compatible replacement can be found at 'appplatform.ApiPortalCustomDomain'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiPortalCustomDomain || (ApiPortalCustomDomain = {}));
exports.ApiPortalCustomDomain = ApiPortalCustomDomain;
var Apm;
(function (Apm) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Apm).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Apm.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-apm-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Apm).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_3.Apm, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Apm shouldn't use an unstable API (appplatform.v20240101preview.Apm). A compatible replacement can be found at 'appplatform.Apm'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Apm || (Apm = {}));
exports.Apm = Apm;
var App;
(function (App) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.App).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    App.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-app-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.App).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_4.App, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure App shouldn't use an unstable API (appplatform.v20240101preview.App). A compatible replacement can be found at 'appplatform.App'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(App || (App = {}));
exports.App = App;
var ApplicationAccelerator;
(function (ApplicationAccelerator) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ApplicationAccelerator).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApplicationAccelerator.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-applicationaccelerator-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ApplicationAccelerator).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_5.ApplicationAccelerator, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApplicationAccelerator shouldn't use an unstable API (appplatform.v20240101preview.ApplicationAccelerator). A compatible replacement can be found at 'appplatform.ApplicationAccelerator'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApplicationAccelerator || (ApplicationAccelerator = {}));
exports.ApplicationAccelerator = ApplicationAccelerator;
var ApplicationLiveView;
(function (ApplicationLiveView) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ApplicationLiveView).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApplicationLiveView.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-applicationliveview-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ApplicationLiveView).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_6.ApplicationLiveView, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApplicationLiveView shouldn't use an unstable API (appplatform.v20240101preview.ApplicationLiveView). A compatible replacement can be found at 'appplatform.ApplicationLiveView'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApplicationLiveView || (ApplicationLiveView = {}));
exports.ApplicationLiveView = ApplicationLiveView;
var Binding;
(function (Binding) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Binding).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Binding.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-binding-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Binding).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_7.Binding, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Binding shouldn't use an unstable API (appplatform.v20240101preview.Binding). A compatible replacement can be found at 'appplatform.Binding'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Binding || (Binding = {}));
exports.Binding = Binding;
var BuildServiceAgentPool;
(function (BuildServiceAgentPool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.BuildServiceAgentPool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BuildServiceAgentPool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-buildserviceagentpool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.BuildServiceAgentPool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_8.BuildServiceAgentPool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BuildServiceAgentPool shouldn't use an unstable API (appplatform.v20240101preview.BuildServiceAgentPool). A compatible replacement can be found at 'appplatform.BuildServiceAgentPool'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BuildServiceAgentPool || (BuildServiceAgentPool = {}));
exports.BuildServiceAgentPool = BuildServiceAgentPool;
var BuildServiceBuild;
(function (BuildServiceBuild) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.BuildServiceBuild).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BuildServiceBuild.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-buildservicebuild-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.BuildServiceBuild).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_9.BuildServiceBuild, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BuildServiceBuild shouldn't use an unstable API (appplatform.v20240101preview.BuildServiceBuild). A compatible replacement can be found at 'appplatform.BuildServiceBuild'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BuildServiceBuild || (BuildServiceBuild = {}));
exports.BuildServiceBuild = BuildServiceBuild;
var BuildServiceBuilder;
(function (BuildServiceBuilder) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.BuildServiceBuilder).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BuildServiceBuilder.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-buildservicebuilder-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.BuildServiceBuilder).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_10.BuildServiceBuilder, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BuildServiceBuilder shouldn't use an unstable API (appplatform.v20240101preview.BuildServiceBuilder). A compatible replacement can be found at 'appplatform.BuildServiceBuilder'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BuildServiceBuilder || (BuildServiceBuilder = {}));
exports.BuildServiceBuilder = BuildServiceBuilder;
var BuildpackBinding;
(function (BuildpackBinding) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.BuildpackBinding).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BuildpackBinding.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-buildpackbinding-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.BuildpackBinding).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_11.BuildpackBinding, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BuildpackBinding shouldn't use an unstable API (appplatform.v20240101preview.BuildpackBinding). A compatible replacement can be found at 'appplatform.BuildpackBinding'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BuildpackBinding || (BuildpackBinding = {}));
exports.BuildpackBinding = BuildpackBinding;
var Certificate;
(function (Certificate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Certificate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Certificate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-certificate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Certificate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_12.Certificate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Certificate shouldn't use an unstable API (appplatform.v20240101preview.Certificate). A compatible replacement can be found at 'appplatform.Certificate'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Certificate || (Certificate = {}));
exports.Certificate = Certificate;
var ConfigServer;
(function (ConfigServer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ConfigServer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConfigServer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-configserver-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ConfigServer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_13.ConfigServer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConfigServer shouldn't use an unstable API (appplatform.v20240101preview.ConfigServer). A compatible replacement can be found at 'appplatform.ConfigServer'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConfigServer || (ConfigServer = {}));
exports.ConfigServer = ConfigServer;
var ConfigurationService;
(function (ConfigurationService) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ConfigurationService).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConfigurationService.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-configurationservice-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ConfigurationService).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_14.ConfigurationService, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConfigurationService shouldn't use an unstable API (appplatform.v20240101preview.ConfigurationService). A compatible replacement can be found at 'appplatform.ConfigurationService'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConfigurationService || (ConfigurationService = {}));
exports.ConfigurationService = ConfigurationService;
var ContainerRegistry;
(function (ContainerRegistry) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ContainerRegistry).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ContainerRegistry.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-containerregistry-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ContainerRegistry).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_15.ContainerRegistry, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ContainerRegistry shouldn't use an unstable API (appplatform.v20240101preview.ContainerRegistry). A compatible replacement can be found at 'appplatform.ContainerRegistry'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ContainerRegistry || (ContainerRegistry = {}));
exports.ContainerRegistry = ContainerRegistry;
var CustomDomain;
(function (CustomDomain) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.CustomDomain).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomDomain.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-customdomain-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.CustomDomain).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_16.CustomDomain, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomDomain shouldn't use an unstable API (appplatform.v20240101preview.CustomDomain). A compatible replacement can be found at 'appplatform.CustomDomain'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomDomain || (CustomDomain = {}));
exports.CustomDomain = CustomDomain;
var CustomizedAccelerator;
(function (CustomizedAccelerator) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.CustomizedAccelerator).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomizedAccelerator.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-customizedaccelerator-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.CustomizedAccelerator).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_17.CustomizedAccelerator, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomizedAccelerator shouldn't use an unstable API (appplatform.v20240101preview.CustomizedAccelerator). A compatible replacement can be found at 'appplatform.CustomizedAccelerator'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomizedAccelerator || (CustomizedAccelerator = {}));
exports.CustomizedAccelerator = CustomizedAccelerator;
var Deployment;
(function (Deployment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Deployment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Deployment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-deployment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Deployment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_18.Deployment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Deployment shouldn't use an unstable API (appplatform.v20240101preview.Deployment). A compatible replacement can be found at 'appplatform.Deployment'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Deployment || (Deployment = {}));
exports.Deployment = Deployment;
var DevToolPortal;
(function (DevToolPortal) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.DevToolPortal).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DevToolPortal.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-devtoolportal-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.DevToolPortal).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_19.DevToolPortal, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DevToolPortal shouldn't use an unstable API (appplatform.v20240101preview.DevToolPortal). A compatible replacement can be found at 'appplatform.DevToolPortal'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DevToolPortal || (DevToolPortal = {}));
exports.DevToolPortal = DevToolPortal;
var Gateway;
(function (Gateway) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Gateway).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Gateway.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-gateway-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Gateway).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_20.Gateway, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Gateway shouldn't use an unstable API (appplatform.v20240101preview.Gateway). A compatible replacement can be found at 'appplatform.Gateway'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Gateway || (Gateway = {}));
exports.Gateway = Gateway;
var GatewayCustomDomain;
(function (GatewayCustomDomain) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.GatewayCustomDomain).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GatewayCustomDomain.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-gatewaycustomdomain-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.GatewayCustomDomain).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_21.GatewayCustomDomain, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GatewayCustomDomain shouldn't use an unstable API (appplatform.v20240101preview.GatewayCustomDomain). A compatible replacement can be found at 'appplatform.GatewayCustomDomain'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GatewayCustomDomain || (GatewayCustomDomain = {}));
exports.GatewayCustomDomain = GatewayCustomDomain;
var GatewayRouteConfig;
(function (GatewayRouteConfig) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.GatewayRouteConfig).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GatewayRouteConfig.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-gatewayrouteconfig-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.GatewayRouteConfig).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_22.GatewayRouteConfig, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GatewayRouteConfig shouldn't use an unstable API (appplatform.v20240101preview.GatewayRouteConfig). A compatible replacement can be found at 'appplatform.GatewayRouteConfig'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GatewayRouteConfig || (GatewayRouteConfig = {}));
exports.GatewayRouteConfig = GatewayRouteConfig;
var MonitoringSetting;
(function (MonitoringSetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.MonitoringSetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MonitoringSetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-monitoringsetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.MonitoringSetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_23.MonitoringSetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MonitoringSetting shouldn't use an unstable API (appplatform.v20240101preview.MonitoringSetting). A compatible replacement can be found at 'appplatform.MonitoringSetting'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MonitoringSetting || (MonitoringSetting = {}));
exports.MonitoringSetting = MonitoringSetting;
var Service;
(function (Service) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Service).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Service.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-service-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Service).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_24.Service, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Service shouldn't use an unstable API (appplatform.v20240101preview.Service). A compatible replacement can be found at 'appplatform.Service'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Service || (Service = {}));
exports.Service = Service;
var ServiceRegistry;
(function (ServiceRegistry) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ServiceRegistry).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ServiceRegistry.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-serviceregistry-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.ServiceRegistry).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_25.ServiceRegistry, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ServiceRegistry shouldn't use an unstable API (appplatform.v20240101preview.ServiceRegistry). A compatible replacement can be found at 'appplatform.ServiceRegistry'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ServiceRegistry || (ServiceRegistry = {}));
exports.ServiceRegistry = ServiceRegistry;
var Storage;
(function (Storage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Storage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Storage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-appplatform-v20240101preview-storage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (appplatform.v20240101preview.Storage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_26.Storage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Storage shouldn't use an unstable API (appplatform.v20240101preview.Storage). A compatible replacement can be found at 'appplatform.Storage'.");
            }),
        },
        vendors: ["azure"],
        services: ["appplatform"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Storage || (Storage = {}));
exports.Storage = Storage;
