// @pulumi/azure-compliance-policies/azurenative/media/v20180601preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingEndpoint = exports.LiveEvent = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20180601preview_1 = require("@pulumi/azure-native/media/v20180601preview");
const v20180601preview_2 = require("@pulumi/azure-native/media/v20180601preview");
var LiveEvent;
(function (LiveEvent) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (media.v20180601preview.LiveEvent).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LiveEvent.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-media-v20180601preview-liveevent-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (media.v20180601preview.LiveEvent).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180601preview_1.LiveEvent, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LiveEvent shouldn't use an unstable API (media.v20180601preview.LiveEvent). A compatible replacement can be found at 'media.LiveEvent'.");
            }),
        },
        vendors: ["azure"],
        services: ["media"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LiveEvent || (LiveEvent = {}));
exports.LiveEvent = LiveEvent;
var StreamingEndpoint;
(function (StreamingEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (media.v20180601preview.StreamingEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    StreamingEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-media-v20180601preview-streamingendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (media.v20180601preview.StreamingEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180601preview_2.StreamingEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure StreamingEndpoint shouldn't use an unstable API (media.v20180601preview.StreamingEndpoint). A compatible replacement can be found at 'media.StreamingEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["media"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(StreamingEndpoint || (StreamingEndpoint = {}));
exports.StreamingEndpoint = StreamingEndpoint;
