// @pulumi/azure-native/sql/backupLongTermRetentionPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupLongTermRetentionPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A long term retention policy.
 * Azure REST API version: 2017-03-01-preview.
 */
class BackupLongTermRetentionPolicy extends pulumi.CustomResource {
    /**
     * Get an existing BackupLongTermRetentionPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BackupLongTermRetentionPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BackupLongTermRetentionPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BackupLongTermRetentionPolicy.__pulumiType;
    }
    /**
     * Create a BackupLongTermRetentionPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["monthlyRetention"] = args ? args.monthlyRetention : undefined;
            resourceInputs["policyName"] = args ? args.policyName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverName"] = args ? args.serverName : undefined;
            resourceInputs["weekOfYear"] = args ? args.weekOfYear : undefined;
            resourceInputs["weeklyRetention"] = args ? args.weeklyRetention : undefined;
            resourceInputs["yearlyRetention"] = args ? args.yearlyRetention : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["monthlyRetention"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["weekOfYear"] = undefined /*out*/;
            resourceInputs["weeklyRetention"] = undefined /*out*/;
            resourceInputs["yearlyRetention"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20170301preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20200202preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20200801preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20201101preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20210201preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20210501preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20210801preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20211101:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20211101preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20220201preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20220501preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20220801preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20221101preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20230201preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20230501preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20230801preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20240501preview:BackupLongTermRetentionPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BackupLongTermRetentionPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BackupLongTermRetentionPolicy = BackupLongTermRetentionPolicy;
/** @internal */
BackupLongTermRetentionPolicy.__pulumiType = 'azure-native:sql:BackupLongTermRetentionPolicy';
