// @pulumi/azure-native/sql/jobPrivateEndpoint.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobPrivateEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A job agent private endpoint.
 * Azure REST API version: 2023-05-01-preview.
 *
 * Other available API versions: 2023-08-01-preview, 2024-05-01-preview.
 */
class JobPrivateEndpoint extends pulumi.CustomResource {
    /**
     * Get an existing JobPrivateEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new JobPrivateEndpoint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of JobPrivateEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === JobPrivateEndpoint.__pulumiType;
    }
    /**
     * Create a JobPrivateEndpoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.jobAgentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'jobAgentName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            if ((!args || args.targetServerAzureResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetServerAzureResourceId'");
            }
            resourceInputs["jobAgentName"] = args ? args.jobAgentName : undefined;
            resourceInputs["privateEndpointName"] = args ? args.privateEndpointName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverName"] = args ? args.serverName : undefined;
            resourceInputs["targetServerAzureResourceId"] = args ? args.targetServerAzureResourceId : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpointId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpointId"] = undefined /*out*/;
            resourceInputs["targetServerAzureResourceId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20230501preview:JobPrivateEndpoint" }, { type: "azure-native:sql/v20230801preview:JobPrivateEndpoint" }, { type: "azure-native:sql/v20240501preview:JobPrivateEndpoint" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(JobPrivateEndpoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.JobPrivateEndpoint = JobPrivateEndpoint;
/** @internal */
JobPrivateEndpoint.__pulumiType = 'azure-native:sql:JobPrivateEndpoint';
