// @pulumi/azure-compliance-policies/azurenative/securityinsights/v20220101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Watchlist = exports.EntityAnalytics = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20220101preview_1 = require("@pulumi/azure-native/securityinsights/v20220101preview");
const v20220101preview_2 = require("@pulumi/azure-native/securityinsights/v20220101preview");
var EntityAnalytics;
(function (EntityAnalytics) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20220101preview.EntityAnalytics).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EntityAnalytics.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20220101preview-entityanalytics-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20220101preview.EntityAnalytics).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220101preview_1.EntityAnalytics, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EntityAnalytics shouldn't use an unstable API (securityinsights.v20220101preview.EntityAnalytics). A compatible replacement can be found at 'securityinsights.EntityAnalytics'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EntityAnalytics || (EntityAnalytics = {}));
exports.EntityAnalytics = EntityAnalytics;
var Watchlist;
(function (Watchlist) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20220101preview.Watchlist).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Watchlist.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20220101preview-watchlist-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20220101preview.Watchlist).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220101preview_2.Watchlist, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Watchlist shouldn't use an unstable API (securityinsights.v20220101preview.Watchlist). A compatible replacement can be found at 'securityinsights.Watchlist'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Watchlist || (Watchlist = {}));
exports.Watchlist = Watchlist;
