// @pulumi/azure-native/servicefabric/v20241101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeType = exports.ManagedClusterService = exports.ManagedClusterApplicationTypeVersion = exports.ManagedClusterApplicationType = exports.ManagedClusterApplication = exports.ManagedCluster = exports.listNodeTypeFaultSimulationOutput = exports.listNodeTypeFaultSimulation = exports.listManagedClusterFaultSimulationOutput = exports.listManagedClusterFaultSimulation = exports.getmanagedMaintenanceWindowStatusOutput = exports.getmanagedMaintenanceWindowStatus = exports.getmanagedAzResiliencyStatusOutput = exports.getmanagedAzResiliencyStatus = exports.getNodeTypeFaultSimulationOutput = exports.getNodeTypeFaultSimulation = exports.getNodeTypeOutput = exports.getNodeType = exports.getManagedClusterServiceOutput = exports.getManagedClusterService = exports.getManagedClusterFaultSimulationOutput = exports.getManagedClusterFaultSimulation = exports.getManagedClusterApplicationTypeVersionOutput = exports.getManagedClusterApplicationTypeVersion = exports.getManagedClusterApplicationTypeOutput = exports.getManagedClusterApplicationType = exports.getManagedClusterApplicationOutput = exports.getManagedClusterApplication = exports.getManagedClusterOutput = exports.getManagedCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getManagedCluster = null;
exports.getManagedClusterOutput = null;
utilities.lazyLoad(exports, ["getManagedCluster", "getManagedClusterOutput"], () => require("./getManagedCluster"));
exports.getManagedClusterApplication = null;
exports.getManagedClusterApplicationOutput = null;
utilities.lazyLoad(exports, ["getManagedClusterApplication", "getManagedClusterApplicationOutput"], () => require("./getManagedClusterApplication"));
exports.getManagedClusterApplicationType = null;
exports.getManagedClusterApplicationTypeOutput = null;
utilities.lazyLoad(exports, ["getManagedClusterApplicationType", "getManagedClusterApplicationTypeOutput"], () => require("./getManagedClusterApplicationType"));
exports.getManagedClusterApplicationTypeVersion = null;
exports.getManagedClusterApplicationTypeVersionOutput = null;
utilities.lazyLoad(exports, ["getManagedClusterApplicationTypeVersion", "getManagedClusterApplicationTypeVersionOutput"], () => require("./getManagedClusterApplicationTypeVersion"));
exports.getManagedClusterFaultSimulation = null;
exports.getManagedClusterFaultSimulationOutput = null;
utilities.lazyLoad(exports, ["getManagedClusterFaultSimulation", "getManagedClusterFaultSimulationOutput"], () => require("./getManagedClusterFaultSimulation"));
exports.getManagedClusterService = null;
exports.getManagedClusterServiceOutput = null;
utilities.lazyLoad(exports, ["getManagedClusterService", "getManagedClusterServiceOutput"], () => require("./getManagedClusterService"));
exports.getNodeType = null;
exports.getNodeTypeOutput = null;
utilities.lazyLoad(exports, ["getNodeType", "getNodeTypeOutput"], () => require("./getNodeType"));
exports.getNodeTypeFaultSimulation = null;
exports.getNodeTypeFaultSimulationOutput = null;
utilities.lazyLoad(exports, ["getNodeTypeFaultSimulation", "getNodeTypeFaultSimulationOutput"], () => require("./getNodeTypeFaultSimulation"));
exports.getmanagedAzResiliencyStatus = null;
exports.getmanagedAzResiliencyStatusOutput = null;
utilities.lazyLoad(exports, ["getmanagedAzResiliencyStatus", "getmanagedAzResiliencyStatusOutput"], () => require("./getmanagedAzResiliencyStatus"));
exports.getmanagedMaintenanceWindowStatus = null;
exports.getmanagedMaintenanceWindowStatusOutput = null;
utilities.lazyLoad(exports, ["getmanagedMaintenanceWindowStatus", "getmanagedMaintenanceWindowStatusOutput"], () => require("./getmanagedMaintenanceWindowStatus"));
exports.listManagedClusterFaultSimulation = null;
exports.listManagedClusterFaultSimulationOutput = null;
utilities.lazyLoad(exports, ["listManagedClusterFaultSimulation", "listManagedClusterFaultSimulationOutput"], () => require("./listManagedClusterFaultSimulation"));
exports.listNodeTypeFaultSimulation = null;
exports.listNodeTypeFaultSimulationOutput = null;
utilities.lazyLoad(exports, ["listNodeTypeFaultSimulation", "listNodeTypeFaultSimulationOutput"], () => require("./listNodeTypeFaultSimulation"));
exports.ManagedCluster = null;
utilities.lazyLoad(exports, ["ManagedCluster"], () => require("./managedCluster"));
exports.ManagedClusterApplication = null;
utilities.lazyLoad(exports, ["ManagedClusterApplication"], () => require("./managedClusterApplication"));
exports.ManagedClusterApplicationType = null;
utilities.lazyLoad(exports, ["ManagedClusterApplicationType"], () => require("./managedClusterApplicationType"));
exports.ManagedClusterApplicationTypeVersion = null;
utilities.lazyLoad(exports, ["ManagedClusterApplicationTypeVersion"], () => require("./managedClusterApplicationTypeVersion"));
exports.ManagedClusterService = null;
utilities.lazyLoad(exports, ["ManagedClusterService"], () => require("./managedClusterService"));
exports.NodeType = null;
utilities.lazyLoad(exports, ["NodeType"], () => require("./nodeType"));
// Export enums:
__exportStar(require("../../types/enums/servicefabric/v20241101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:servicefabric/v20241101preview:ManagedCluster":
                return new exports.ManagedCluster(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:ManagedClusterApplication":
                return new exports.ManagedClusterApplication(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:ManagedClusterApplicationType":
                return new exports.ManagedClusterApplicationType(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:ManagedClusterApplicationTypeVersion":
                return new exports.ManagedClusterApplicationTypeVersion(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:ManagedClusterService":
                return new exports.ManagedClusterService(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:NodeType":
                return new exports.NodeType(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "servicefabric/v20241101preview", _module);
