// @pulumi/azure-compliance-policies/azurenative/testbase/v20231101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestBaseAccount = exports.Package = exports.ImageDefinition = exports.FavoriteProcess = exports.DraftPackage = exports.CustomerEvent = exports.CustomImage = exports.Credential = exports.ActionRequest = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20231101preview_1 = require("@pulumi/azure-native/testbase/v20231101preview");
const v20231101preview_2 = require("@pulumi/azure-native/testbase/v20231101preview");
const v20231101preview_3 = require("@pulumi/azure-native/testbase/v20231101preview");
const v20231101preview_4 = require("@pulumi/azure-native/testbase/v20231101preview");
const v20231101preview_5 = require("@pulumi/azure-native/testbase/v20231101preview");
const v20231101preview_6 = require("@pulumi/azure-native/testbase/v20231101preview");
const v20231101preview_7 = require("@pulumi/azure-native/testbase/v20231101preview");
const v20231101preview_8 = require("@pulumi/azure-native/testbase/v20231101preview");
const v20231101preview_9 = require("@pulumi/azure-native/testbase/v20231101preview");
var ActionRequest;
(function (ActionRequest) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.ActionRequest).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ActionRequest.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20231101preview-actionrequest-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.ActionRequest).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_1.ActionRequest, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ActionRequest shouldn't use an unstable API (testbase.v20231101preview.ActionRequest). A compatible replacement can be found at 'testbase.ActionRequest'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ActionRequest || (ActionRequest = {}));
exports.ActionRequest = ActionRequest;
var Credential;
(function (Credential) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.Credential).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Credential.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20231101preview-credential-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.Credential).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_2.Credential, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Credential shouldn't use an unstable API (testbase.v20231101preview.Credential). A compatible replacement can be found at 'testbase.Credential'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Credential || (Credential = {}));
exports.Credential = Credential;
var CustomImage;
(function (CustomImage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.CustomImage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomImage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20231101preview-customimage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.CustomImage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_3.CustomImage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomImage shouldn't use an unstable API (testbase.v20231101preview.CustomImage). A compatible replacement can be found at 'testbase.CustomImage'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomImage || (CustomImage = {}));
exports.CustomImage = CustomImage;
var CustomerEvent;
(function (CustomerEvent) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.CustomerEvent).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomerEvent.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20231101preview-customerevent-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.CustomerEvent).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_4.CustomerEvent, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomerEvent shouldn't use an unstable API (testbase.v20231101preview.CustomerEvent). A compatible replacement can be found at 'testbase.CustomerEvent'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomerEvent || (CustomerEvent = {}));
exports.CustomerEvent = CustomerEvent;
var DraftPackage;
(function (DraftPackage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.DraftPackage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DraftPackage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20231101preview-draftpackage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.DraftPackage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_5.DraftPackage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DraftPackage shouldn't use an unstable API (testbase.v20231101preview.DraftPackage). A compatible replacement can be found at 'testbase.DraftPackage'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DraftPackage || (DraftPackage = {}));
exports.DraftPackage = DraftPackage;
var FavoriteProcess;
(function (FavoriteProcess) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.FavoriteProcess).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FavoriteProcess.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20231101preview-favoriteprocess-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.FavoriteProcess).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_6.FavoriteProcess, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FavoriteProcess shouldn't use an unstable API (testbase.v20231101preview.FavoriteProcess). A compatible replacement can be found at 'testbase.FavoriteProcess'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FavoriteProcess || (FavoriteProcess = {}));
exports.FavoriteProcess = FavoriteProcess;
var ImageDefinition;
(function (ImageDefinition) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.ImageDefinition).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ImageDefinition.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20231101preview-imagedefinition-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.ImageDefinition).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_7.ImageDefinition, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ImageDefinition shouldn't use an unstable API (testbase.v20231101preview.ImageDefinition). A compatible replacement can be found at 'testbase.ImageDefinition'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ImageDefinition || (ImageDefinition = {}));
exports.ImageDefinition = ImageDefinition;
var Package;
(function (Package) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.Package).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Package.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20231101preview-package-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.Package).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_8.Package, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Package shouldn't use an unstable API (testbase.v20231101preview.Package). A compatible replacement can be found at 'testbase.Package'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Package || (Package = {}));
exports.Package = Package;
var TestBaseAccount;
(function (TestBaseAccount) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.TestBaseAccount).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TestBaseAccount.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-testbase-v20231101preview-testbaseaccount-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (testbase.v20231101preview.TestBaseAccount).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_9.TestBaseAccount, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TestBaseAccount shouldn't use an unstable API (testbase.v20231101preview.TestBaseAccount). A compatible replacement can be found at 'testbase.TestBaseAccount'.");
            }),
        },
        vendors: ["azure"],
        services: ["testbase"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TestBaseAccount || (TestBaseAccount = {}));
exports.TestBaseAccount = TestBaseAccount;
