// @pulumi/azure-compliance-policies/azurenative/resources/v20220801preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentStackAtSubscription = exports.DeploymentStackAtResourceGroup = exports.DeploymentStackAtManagementGroup = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20220801preview_1 = require("@pulumi/azure-native/resources/v20220801preview");
const v20220801preview_2 = require("@pulumi/azure-native/resources/v20220801preview");
const v20220801preview_3 = require("@pulumi/azure-native/resources/v20220801preview");
var DeploymentStackAtManagementGroup;
(function (DeploymentStackAtManagementGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (resources.v20220801preview.DeploymentStackAtManagementGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DeploymentStackAtManagementGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-resources-v20220801preview-deploymentstackatmanagementgroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (resources.v20220801preview.DeploymentStackAtManagementGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220801preview_1.DeploymentStackAtManagementGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DeploymentStackAtManagementGroup shouldn't use an unstable API (resources.v20220801preview.DeploymentStackAtManagementGroup).");
            }),
        },
        vendors: ["azure"],
        services: ["resources"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DeploymentStackAtManagementGroup || (DeploymentStackAtManagementGroup = {}));
exports.DeploymentStackAtManagementGroup = DeploymentStackAtManagementGroup;
var DeploymentStackAtResourceGroup;
(function (DeploymentStackAtResourceGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (resources.v20220801preview.DeploymentStackAtResourceGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DeploymentStackAtResourceGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-resources-v20220801preview-deploymentstackatresourcegroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (resources.v20220801preview.DeploymentStackAtResourceGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220801preview_2.DeploymentStackAtResourceGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DeploymentStackAtResourceGroup shouldn't use an unstable API (resources.v20220801preview.DeploymentStackAtResourceGroup).");
            }),
        },
        vendors: ["azure"],
        services: ["resources"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DeploymentStackAtResourceGroup || (DeploymentStackAtResourceGroup = {}));
exports.DeploymentStackAtResourceGroup = DeploymentStackAtResourceGroup;
var DeploymentStackAtSubscription;
(function (DeploymentStackAtSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (resources.v20220801preview.DeploymentStackAtSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DeploymentStackAtSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-resources-v20220801preview-deploymentstackatsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (resources.v20220801preview.DeploymentStackAtSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220801preview_3.DeploymentStackAtSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DeploymentStackAtSubscription shouldn't use an unstable API (resources.v20220801preview.DeploymentStackAtSubscription).");
            }),
        },
        vendors: ["azure"],
        services: ["resources"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DeploymentStackAtSubscription || (DeploymentStackAtSubscription = {}));
exports.DeploymentStackAtSubscription = DeploymentStackAtSubscription;
