// @pulumi/azure/arckubernetes/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisionedCluster = exports.FluxConfiguration = exports.ClusterExtension = exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ClusterExtension = null;
utilities.lazyLoad(exports, ["ClusterExtension"], () => require("./clusterExtension"));
exports.FluxConfiguration = null;
utilities.lazyLoad(exports, ["FluxConfiguration"], () => require("./fluxConfiguration"));
exports.ProvisionedCluster = null;
utilities.lazyLoad(exports, ["ProvisionedCluster"], () => require("./provisionedCluster"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:arckubernetes/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure:arckubernetes/clusterExtension:ClusterExtension":
                return new exports.ClusterExtension(name, undefined, { urn });
            case "azure:arckubernetes/fluxConfiguration:FluxConfiguration":
                return new exports.FluxConfiguration(name, undefined, { urn });
            case "azure:arckubernetes/provisionedCluster:ProvisionedCluster":
                return new exports.ProvisionedCluster(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "arckubernetes/cluster", _module);
pulumi.runtime.registerResourceModule("azure", "arckubernetes/clusterExtension", _module);
pulumi.runtime.registerResourceModule("azure", "arckubernetes/fluxConfiguration", _module);
pulumi.runtime.registerResourceModule("azure", "arckubernetes/provisionedCluster", _module);
