// @pulumi/azure/arckubernetes/provisionedCluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisionedCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Arc Kubernetes Provisioned Cluster.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as azuread from "@pulumi/azuread";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const current = azure.core.getClientConfig({});
 * const exampleGroup = new azuread.Group("example", {
 *     displayName: "example-adg",
 *     owners: [current.then(current => current.objectId)],
 *     securityEnabled: true,
 * });
 * const exampleProvisionedCluster = new azure.arckubernetes.ProvisionedCluster("example", {
 *     name: "example-akpc",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     azureActiveDirectory: {
 *         azureRbacEnabled: true,
 *         adminGroupObjectIds: [exampleGroup.id],
 *         tenantId: current.then(current => current.tenantId),
 *     },
 *     identity: {
 *         type: "SystemAssigned",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Arc Kubernetes Provisioned Clusters can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:arckubernetes/provisionedCluster:ProvisionedCluster example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.Kubernetes/connectedClusters/cluster1
 * ```
 */
class ProvisionedCluster extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["agentVersion"] = state ? state.agentVersion : undefined;
            resourceInputs["arcAgentAutoUpgradeEnabled"] = state ? state.arcAgentAutoUpgradeEnabled : undefined;
            resourceInputs["arcAgentDesiredVersion"] = state ? state.arcAgentDesiredVersion : undefined;
            resourceInputs["azureActiveDirectory"] = state ? state.azureActiveDirectory : undefined;
            resourceInputs["distribution"] = state ? state.distribution : undefined;
            resourceInputs["identity"] = state ? state.identity : undefined;
            resourceInputs["infrastructure"] = state ? state.infrastructure : undefined;
            resourceInputs["kubernetesVersion"] = state ? state.kubernetesVersion : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["offering"] = state ? state.offering : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["totalCoreCount"] = state ? state.totalCoreCount : undefined;
            resourceInputs["totalNodeCount"] = state ? state.totalNodeCount : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.identity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'identity'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["arcAgentAutoUpgradeEnabled"] = args ? args.arcAgentAutoUpgradeEnabled : undefined;
            resourceInputs["arcAgentDesiredVersion"] = args ? args.arcAgentDesiredVersion : undefined;
            resourceInputs["azureActiveDirectory"] = args ? args.azureActiveDirectory : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["agentVersion"] = undefined /*out*/;
            resourceInputs["distribution"] = undefined /*out*/;
            resourceInputs["infrastructure"] = undefined /*out*/;
            resourceInputs["kubernetesVersion"] = undefined /*out*/;
            resourceInputs["offering"] = undefined /*out*/;
            resourceInputs["totalCoreCount"] = undefined /*out*/;
            resourceInputs["totalNodeCount"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ProvisionedCluster.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ProvisionedCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ProvisionedCluster(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ProvisionedCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ProvisionedCluster.__pulumiType;
    }
}
exports.ProvisionedCluster = ProvisionedCluster;
/** @internal */
ProvisionedCluster.__pulumiType = 'azure:arckubernetes/provisionedCluster:ProvisionedCluster';
