// @pulumi/azure/stack/hciMarketplaceGalleryImage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HciMarketplaceGalleryImage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure Stack HCI Marketplace Gallery Image.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as azuread from "@pulumi/azuread";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("example", {
 *     name: "examples",
 *     location: "West Europe",
 * });
 * const example = azure.core.getClientConfig({});
 * // service principal of 'Microsoft.AzureStackHCI Resource Provider'
 * const hciRp = azuread.getServicePrincipal({
 *     clientId: "1412d89f-b8a8-4111-b4fd-e82905cbd85d",
 * });
 * const exampleAssignment = new azure.authorization.Assignment("example", {
 *     scope: exampleResourceGroup.id,
 *     roleDefinitionName: "Azure Connected Machine Resource Manager",
 *     principalId: hciRp.then(hciRp => hciRp.objectId),
 * });
 * const exampleHciMarketplaceGalleryImage = new azure.stack.HciMarketplaceGalleryImage("example", {
 *     name: "example-mgi",
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     customLocationId: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.ExtendedLocation/customLocations/cl1",
 *     hypervGeneration: "V2",
 *     osType: "Windows",
 *     version: "20348.2655.240905",
 *     identifier: {
 *         publisher: "MicrosoftWindowsServer",
 *         offer: "WindowsServer",
 *         sku: "2022-datacenter-azure-edition-core",
 *     },
 *     tags: {
 *         foo: "bar",
 *         env: "example",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Azure Stack HCI Marketplace Gallery Images can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:stack/hciMarketplaceGalleryImage:HciMarketplaceGalleryImage example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.AzureStackHCI/marketplaceGalleryImages/image1
 * ```
 */
class HciMarketplaceGalleryImage extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["customLocationId"] = state ? state.customLocationId : undefined;
            resourceInputs["hypervGeneration"] = state ? state.hypervGeneration : undefined;
            resourceInputs["identifier"] = state ? state.identifier : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["osType"] = state ? state.osType : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["storagePathId"] = state ? state.storagePathId : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["version"] = state ? state.version : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.customLocationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'customLocationId'");
            }
            if ((!args || args.hypervGeneration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hypervGeneration'");
            }
            if ((!args || args.identifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'identifier'");
            }
            if ((!args || args.osType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'osType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.version === undefined) && !opts.urn) {
                throw new Error("Missing required property 'version'");
            }
            resourceInputs["customLocationId"] = args ? args.customLocationId : undefined;
            resourceInputs["hypervGeneration"] = args ? args.hypervGeneration : undefined;
            resourceInputs["identifier"] = args ? args.identifier : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["osType"] = args ? args.osType : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["storagePathId"] = args ? args.storagePathId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(HciMarketplaceGalleryImage.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing HciMarketplaceGalleryImage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HciMarketplaceGalleryImage(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HciMarketplaceGalleryImage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HciMarketplaceGalleryImage.__pulumiType;
    }
}
exports.HciMarketplaceGalleryImage = HciMarketplaceGalleryImage;
/** @internal */
HciMarketplaceGalleryImage.__pulumiType = 'azure:stack/hciMarketplaceGalleryImage:HciMarketplaceGalleryImage';
