// @pulumi/azure-compliance-policies/azurenative/datamigration/v20220330preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Task = exports.SqlMigrationService = exports.ServiceTask = exports.Service = exports.Project = exports.File = exports.DatabaseMigrationsSqlDb = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20220330preview_1 = require("@pulumi/azure-native/datamigration/v20220330preview");
const v20220330preview_2 = require("@pulumi/azure-native/datamigration/v20220330preview");
const v20220330preview_3 = require("@pulumi/azure-native/datamigration/v20220330preview");
const v20220330preview_4 = require("@pulumi/azure-native/datamigration/v20220330preview");
const v20220330preview_5 = require("@pulumi/azure-native/datamigration/v20220330preview");
const v20220330preview_6 = require("@pulumi/azure-native/datamigration/v20220330preview");
const v20220330preview_7 = require("@pulumi/azure-native/datamigration/v20220330preview");
var DatabaseMigrationsSqlDb;
(function (DatabaseMigrationsSqlDb) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.DatabaseMigrationsSqlDb).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DatabaseMigrationsSqlDb.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datamigration-v20220330preview-databasemigrationssqldb-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.DatabaseMigrationsSqlDb).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220330preview_1.DatabaseMigrationsSqlDb, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DatabaseMigrationsSqlDb shouldn't use an unstable API (datamigration.v20220330preview.DatabaseMigrationsSqlDb). A compatible replacement can be found at 'datamigration.DatabaseMigrationsSqlDb'.");
            }),
        },
        vendors: ["azure"],
        services: ["datamigration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DatabaseMigrationsSqlDb || (DatabaseMigrationsSqlDb = {}));
exports.DatabaseMigrationsSqlDb = DatabaseMigrationsSqlDb;
var File;
(function (File) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.File).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    File.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datamigration-v20220330preview-file-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.File).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220330preview_2.File, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure File shouldn't use an unstable API (datamigration.v20220330preview.File). A compatible replacement can be found at 'datamigration.File'.");
            }),
        },
        vendors: ["azure"],
        services: ["datamigration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(File || (File = {}));
exports.File = File;
var Project;
(function (Project) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.Project).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Project.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datamigration-v20220330preview-project-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.Project).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220330preview_3.Project, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Project shouldn't use an unstable API (datamigration.v20220330preview.Project). A compatible replacement can be found at 'datamigration.Project'.");
            }),
        },
        vendors: ["azure"],
        services: ["datamigration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Project || (Project = {}));
exports.Project = Project;
var Service;
(function (Service) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.Service).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Service.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datamigration-v20220330preview-service-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.Service).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220330preview_4.Service, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Service shouldn't use an unstable API (datamigration.v20220330preview.Service). A compatible replacement can be found at 'datamigration.Service'.");
            }),
        },
        vendors: ["azure"],
        services: ["datamigration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Service || (Service = {}));
exports.Service = Service;
var ServiceTask;
(function (ServiceTask) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.ServiceTask).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ServiceTask.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datamigration-v20220330preview-servicetask-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.ServiceTask).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220330preview_5.ServiceTask, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ServiceTask shouldn't use an unstable API (datamigration.v20220330preview.ServiceTask). A compatible replacement can be found at 'datamigration.v20210630.ServiceTask'.");
            }),
        },
        vendors: ["azure"],
        services: ["datamigration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ServiceTask || (ServiceTask = {}));
exports.ServiceTask = ServiceTask;
var SqlMigrationService;
(function (SqlMigrationService) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.SqlMigrationService).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SqlMigrationService.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datamigration-v20220330preview-sqlmigrationservice-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.SqlMigrationService).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220330preview_6.SqlMigrationService, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SqlMigrationService shouldn't use an unstable API (datamigration.v20220330preview.SqlMigrationService). A compatible replacement can be found at 'datamigration.SqlMigrationService'.");
            }),
        },
        vendors: ["azure"],
        services: ["datamigration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SqlMigrationService || (SqlMigrationService = {}));
exports.SqlMigrationService = SqlMigrationService;
var Task;
(function (Task) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.Task).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Task.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-datamigration-v20220330preview-task-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (datamigration.v20220330preview.Task).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220330preview_7.Task, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Task shouldn't use an unstable API (datamigration.v20220330preview.Task). A compatible replacement can be found at 'datamigration.Task'.");
            }),
        },
        vendors: ["azure"],
        services: ["datamigration"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Task || (Task = {}));
exports.Task = Task;
