// @pulumi/azure-native/netapp/subvolume.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subvolume = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Subvolume Information properties
 * Azure REST API version: 2022-11-01. Prior API version in Azure Native 1.x: 2021-10-01.
 *
 * Other available API versions: 2022-11-01-preview, 2023-05-01, 2023-05-01-preview, 2023-07-01, 2023-07-01-preview, 2023-11-01, 2023-11-01-preview, 2024-01-01, 2024-03-01, 2024-03-01-preview, 2024-05-01, 2024-05-01-preview, 2024-07-01, 2024-07-01-preview, 2024-09-01, 2024-09-01-preview.
 */
class Subvolume extends pulumi.CustomResource {
    /**
     * Get an existing Subvolume resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Subvolume(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Subvolume.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Subvolume.__pulumiType;
    }
    /**
     * Create a Subvolume resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.poolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'poolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.volumeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'volumeName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["parentPath"] = args ? args.parentPath : undefined;
            resourceInputs["path"] = args ? args.path : undefined;
            resourceInputs["poolName"] = args ? args.poolName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["size"] = args ? args.size : undefined;
            resourceInputs["subvolumeName"] = args ? args.subvolumeName : undefined;
            resourceInputs["volumeName"] = args ? args.volumeName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["parentPath"] = undefined /*out*/;
            resourceInputs["path"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:netapp/v20211001:Subvolume" }, { type: "azure-native:netapp/v20220101:Subvolume" }, { type: "azure-native:netapp/v20220301:Subvolume" }, { type: "azure-native:netapp/v20220501:Subvolume" }, { type: "azure-native:netapp/v20220901:Subvolume" }, { type: "azure-native:netapp/v20221101:Subvolume" }, { type: "azure-native:netapp/v20221101preview:Subvolume" }, { type: "azure-native:netapp/v20230501:Subvolume" }, { type: "azure-native:netapp/v20230501preview:Subvolume" }, { type: "azure-native:netapp/v20230701:Subvolume" }, { type: "azure-native:netapp/v20230701preview:Subvolume" }, { type: "azure-native:netapp/v20231101:Subvolume" }, { type: "azure-native:netapp/v20231101preview:Subvolume" }, { type: "azure-native:netapp/v20240101:Subvolume" }, { type: "azure-native:netapp/v20240301:Subvolume" }, { type: "azure-native:netapp/v20240301preview:Subvolume" }, { type: "azure-native:netapp/v20240501:Subvolume" }, { type: "azure-native:netapp/v20240501preview:Subvolume" }, { type: "azure-native:netapp/v20240701:Subvolume" }, { type: "azure-native:netapp/v20240701preview:Subvolume" }, { type: "azure-native:netapp/v20240901:Subvolume" }, { type: "azure-native:netapp/v20240901preview:Subvolume" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Subvolume.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Subvolume = Subvolume;
/** @internal */
Subvolume.__pulumiType = 'azure-native:netapp:Subvolume';
