// @pulumi/azure-compliance-policies/azurenative/azurestack/v20200601preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Registration = exports.LinkedSubscription = exports.CustomerSubscription = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20200601preview_1 = require("@pulumi/azure-native/azurestack/v20200601preview");
const v20200601preview_2 = require("@pulumi/azure-native/azurestack/v20200601preview");
const v20200601preview_3 = require("@pulumi/azure-native/azurestack/v20200601preview");
var CustomerSubscription;
(function (CustomerSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestack.v20200601preview.CustomerSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomerSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestack-v20200601preview-customersubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestack.v20200601preview.CustomerSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200601preview_1.CustomerSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomerSubscription shouldn't use an unstable API (azurestack.v20200601preview.CustomerSubscription). A compatible replacement can be found at 'azurestack.CustomerSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestack"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomerSubscription || (CustomerSubscription = {}));
exports.CustomerSubscription = CustomerSubscription;
var LinkedSubscription;
(function (LinkedSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestack.v20200601preview.LinkedSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LinkedSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestack-v20200601preview-linkedsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestack.v20200601preview.LinkedSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200601preview_2.LinkedSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LinkedSubscription shouldn't use an unstable API (azurestack.v20200601preview.LinkedSubscription). A compatible replacement can be found at 'azurestack.LinkedSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestack"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LinkedSubscription || (LinkedSubscription = {}));
exports.LinkedSubscription = LinkedSubscription;
var Registration;
(function (Registration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestack.v20200601preview.Registration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Registration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestack-v20200601preview-registration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestack.v20200601preview.Registration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200601preview_3.Registration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Registration shouldn't use an unstable API (azurestack.v20200601preview.Registration). A compatible replacement can be found at 'azurestack.Registration'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestack"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Registration || (Registration = {}));
exports.Registration = Registration;
