// @pulumi/azure/containerservice/fleetUpdateRun.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FleetUpdateRun = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Kubernetes Fleet Update Run.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-rg",
 *     location: "westeurope",
 * });
 * const exampleKubernetesFleetManager = new azure.containerservice.KubernetesFleetManager("example", {
 *     location: example.location,
 *     name: "example",
 *     resourceGroupName: example.name,
 * });
 * const exampleKubernetesCluster = new azure.containerservice.KubernetesCluster("example", {
 *     name: "example",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     dnsPrefix: "example",
 *     defaultNodePool: {
 *         name: "default",
 *         nodeCount: 1,
 *         vmSize: "Standard_DS2_v2",
 *     },
 *     identity: {
 *         type: "SystemAssigned",
 *     },
 * });
 * const exampleFleetMember = new azure.containerservice.FleetMember("example", {
 *     name: "example",
 *     kubernetesFleetId: exampleKubernetesFleetManager.id,
 *     kubernetesClusterId: exampleKubernetesCluster.id,
 *     group: "example-group",
 * });
 * const exampleFleetUpdateRun = new azure.containerservice.FleetUpdateRun("example", {
 *     name: "example",
 *     kubernetesFleetManagerId: exampleKubernetesFleetManager.id,
 *     managedClusterUpdate: {
 *         upgrade: {
 *             type: "Full",
 *             kubernetesVersion: "1.27",
 *         },
 *         nodeImageSelection: {
 *             type: "Latest",
 *         },
 *     },
 *     stages: [{
 *         name: "example",
 *         groups: [{
 *             name: "example-group",
 *         }],
 *         afterStageWaitInSeconds: 21,
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * Kubernetes Fleet Update Runs can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:containerservice/fleetUpdateRun:FleetUpdateRun example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resourceGroup1/providers/Microsoft.ContainerService/fleets/fleet1/updateRuns/updateRun1
 * ```
 */
class FleetUpdateRun extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["fleetUpdateStrategyId"] = state ? state.fleetUpdateStrategyId : undefined;
            resourceInputs["kubernetesFleetManagerId"] = state ? state.kubernetesFleetManagerId : undefined;
            resourceInputs["managedClusterUpdate"] = state ? state.managedClusterUpdate : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["stages"] = state ? state.stages : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.kubernetesFleetManagerId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kubernetesFleetManagerId'");
            }
            if ((!args || args.managedClusterUpdate === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managedClusterUpdate'");
            }
            resourceInputs["fleetUpdateStrategyId"] = args ? args.fleetUpdateStrategyId : undefined;
            resourceInputs["kubernetesFleetManagerId"] = args ? args.kubernetesFleetManagerId : undefined;
            resourceInputs["managedClusterUpdate"] = args ? args.managedClusterUpdate : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["stages"] = args ? args.stages : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(FleetUpdateRun.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing FleetUpdateRun resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new FleetUpdateRun(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FleetUpdateRun.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FleetUpdateRun.__pulumiType;
    }
}
exports.FleetUpdateRun = FleetUpdateRun;
/** @internal */
FleetUpdateRun.__pulumiType = 'azure:containerservice/fleetUpdateRun:FleetUpdateRun';
