// @pulumi/azure-compliance-policies/azurenative/integrationspaces/v20231114preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Space = exports.InfrastructureResource = exports.BusinessProcess = exports.ApplicationResource = exports.Application = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20231114preview_1 = require("@pulumi/azure-native/integrationspaces/v20231114preview");
const v20231114preview_2 = require("@pulumi/azure-native/integrationspaces/v20231114preview");
const v20231114preview_3 = require("@pulumi/azure-native/integrationspaces/v20231114preview");
const v20231114preview_4 = require("@pulumi/azure-native/integrationspaces/v20231114preview");
const v20231114preview_5 = require("@pulumi/azure-native/integrationspaces/v20231114preview");
var Application;
(function (Application) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.Application).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Application.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-integrationspaces-v20231114preview-application-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.Application).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231114preview_1.Application, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Application shouldn't use an unstable API (integrationspaces.v20231114preview.Application). A compatible replacement can be found at 'integrationspaces.Application'.");
            }),
        },
        vendors: ["azure"],
        services: ["integrationspaces"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Application || (Application = {}));
exports.Application = Application;
var ApplicationResource;
(function (ApplicationResource) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.ApplicationResource).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApplicationResource.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-integrationspaces-v20231114preview-applicationresource-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.ApplicationResource).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231114preview_2.ApplicationResource, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApplicationResource shouldn't use an unstable API (integrationspaces.v20231114preview.ApplicationResource). A compatible replacement can be found at 'integrationspaces.ApplicationResource'.");
            }),
        },
        vendors: ["azure"],
        services: ["integrationspaces"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApplicationResource || (ApplicationResource = {}));
exports.ApplicationResource = ApplicationResource;
var BusinessProcess;
(function (BusinessProcess) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.BusinessProcess).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BusinessProcess.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-integrationspaces-v20231114preview-businessprocess-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.BusinessProcess).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231114preview_3.BusinessProcess, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BusinessProcess shouldn't use an unstable API (integrationspaces.v20231114preview.BusinessProcess). A compatible replacement can be found at 'integrationspaces.BusinessProcess'.");
            }),
        },
        vendors: ["azure"],
        services: ["integrationspaces"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BusinessProcess || (BusinessProcess = {}));
exports.BusinessProcess = BusinessProcess;
var InfrastructureResource;
(function (InfrastructureResource) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.InfrastructureResource).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    InfrastructureResource.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-integrationspaces-v20231114preview-infrastructureresource-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.InfrastructureResource).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231114preview_4.InfrastructureResource, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure InfrastructureResource shouldn't use an unstable API (integrationspaces.v20231114preview.InfrastructureResource). A compatible replacement can be found at 'integrationspaces.InfrastructureResource'.");
            }),
        },
        vendors: ["azure"],
        services: ["integrationspaces"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(InfrastructureResource || (InfrastructureResource = {}));
exports.InfrastructureResource = InfrastructureResource;
var Space;
(function (Space) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.Space).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Space.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-integrationspaces-v20231114preview-space-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (integrationspaces.v20231114preview.Space).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231114preview_5.Space, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Space shouldn't use an unstable API (integrationspaces.v20231114preview.Space). A compatible replacement can be found at 'integrationspaces.Space'.");
            }),
        },
        vendors: ["azure"],
        services: ["integrationspaces"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Space || (Space = {}));
exports.Space = Space;
