// @pulumi/azure/arcmachine/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getOutput = exports.get = exports.Extension = exports.AutomanageConfigurationAssignment = exports.ArcMachine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ArcMachine = null;
utilities.lazyLoad(exports, ["ArcMachine"], () => require("./arcMachine"));
exports.AutomanageConfigurationAssignment = null;
utilities.lazyLoad(exports, ["AutomanageConfigurationAssignment"], () => require("./automanageConfigurationAssignment"));
exports.Extension = null;
utilities.lazyLoad(exports, ["Extension"], () => require("./extension"));
exports.get = null;
exports.getOutput = null;
utilities.lazyLoad(exports, ["get", "getOutput"], () => require("./get"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:arcmachine/arcMachine:ArcMachine":
                return new exports.ArcMachine(name, undefined, { urn });
            case "azure:arcmachine/automanageConfigurationAssignment:AutomanageConfigurationAssignment":
                return new exports.AutomanageConfigurationAssignment(name, undefined, { urn });
            case "azure:arcmachine/extension:Extension":
                return new exports.Extension(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "arcmachine/arcMachine", _module);
pulumi.runtime.registerResourceModule("azure", "arcmachine/automanageConfigurationAssignment", _module);
pulumi.runtime.registerResourceModule("azure", "arcmachine/extension", _module);
