// @pulumi/azure-compliance-policies/azurenative/insights/v20210501preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionDiagnosticSetting = exports.ManagementGroupDiagnosticSetting = exports.DiagnosticSetting = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20210501preview_1 = require("@pulumi/azure-native/insights/v20210501preview");
const v20210501preview_2 = require("@pulumi/azure-native/insights/v20210501preview");
const v20210501preview_3 = require("@pulumi/azure-native/insights/v20210501preview");
var DiagnosticSetting;
(function (DiagnosticSetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (insights.v20210501preview.DiagnosticSetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DiagnosticSetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-insights-v20210501preview-diagnosticsetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (insights.v20210501preview.DiagnosticSetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210501preview_1.DiagnosticSetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DiagnosticSetting shouldn't use an unstable API (insights.v20210501preview.DiagnosticSetting). A compatible replacement can be found at 'insights.DiagnosticSetting'.");
            }),
        },
        vendors: ["azure"],
        services: ["insights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DiagnosticSetting || (DiagnosticSetting = {}));
exports.DiagnosticSetting = DiagnosticSetting;
var ManagementGroupDiagnosticSetting;
(function (ManagementGroupDiagnosticSetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (insights.v20210501preview.ManagementGroupDiagnosticSetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagementGroupDiagnosticSetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-insights-v20210501preview-managementgroupdiagnosticsetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (insights.v20210501preview.ManagementGroupDiagnosticSetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210501preview_2.ManagementGroupDiagnosticSetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagementGroupDiagnosticSetting shouldn't use an unstable API (insights.v20210501preview.ManagementGroupDiagnosticSetting). A compatible replacement can be found at 'insights.ManagementGroupDiagnosticSetting'.");
            }),
        },
        vendors: ["azure"],
        services: ["insights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagementGroupDiagnosticSetting || (ManagementGroupDiagnosticSetting = {}));
exports.ManagementGroupDiagnosticSetting = ManagementGroupDiagnosticSetting;
var SubscriptionDiagnosticSetting;
(function (SubscriptionDiagnosticSetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (insights.v20210501preview.SubscriptionDiagnosticSetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SubscriptionDiagnosticSetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-insights-v20210501preview-subscriptiondiagnosticsetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (insights.v20210501preview.SubscriptionDiagnosticSetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210501preview_3.SubscriptionDiagnosticSetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SubscriptionDiagnosticSetting shouldn't use an unstable API (insights.v20210501preview.SubscriptionDiagnosticSetting). A compatible replacement can be found at 'insights.SubscriptionDiagnosticSetting'.");
            }),
        },
        vendors: ["azure"],
        services: ["insights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SubscriptionDiagnosticSetting || (SubscriptionDiagnosticSetting = {}));
exports.SubscriptionDiagnosticSetting = SubscriptionDiagnosticSetting;
