// @pulumi/azure/appinsights/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkbookTemplate = exports.Workbook = exports.WebTest = exports.StandardWebTest = exports.SmartDetectionRule = exports.Insights = exports.getInsightsOutput = exports.getInsights = exports.ApiKey = exports.AnalyticsItem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AnalyticsItem = null;
utilities.lazyLoad(exports, ["AnalyticsItem"], () => require("./analyticsItem"));
exports.ApiKey = null;
utilities.lazyLoad(exports, ["ApiKey"], () => require("./apiKey"));
exports.getInsights = null;
exports.getInsightsOutput = null;
utilities.lazyLoad(exports, ["getInsights", "getInsightsOutput"], () => require("./getInsights"));
exports.Insights = null;
utilities.lazyLoad(exports, ["Insights"], () => require("./insights"));
exports.SmartDetectionRule = null;
utilities.lazyLoad(exports, ["SmartDetectionRule"], () => require("./smartDetectionRule"));
exports.StandardWebTest = null;
utilities.lazyLoad(exports, ["StandardWebTest"], () => require("./standardWebTest"));
exports.WebTest = null;
utilities.lazyLoad(exports, ["WebTest"], () => require("./webTest"));
exports.Workbook = null;
utilities.lazyLoad(exports, ["Workbook"], () => require("./workbook"));
exports.WorkbookTemplate = null;
utilities.lazyLoad(exports, ["WorkbookTemplate"], () => require("./workbookTemplate"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:appinsights/analyticsItem:AnalyticsItem":
                return new exports.AnalyticsItem(name, undefined, { urn });
            case "azure:appinsights/apiKey:ApiKey":
                return new exports.ApiKey(name, undefined, { urn });
            case "azure:appinsights/insights:Insights":
                return new exports.Insights(name, undefined, { urn });
            case "azure:appinsights/smartDetectionRule:SmartDetectionRule":
                return new exports.SmartDetectionRule(name, undefined, { urn });
            case "azure:appinsights/standardWebTest:StandardWebTest":
                return new exports.StandardWebTest(name, undefined, { urn });
            case "azure:appinsights/webTest:WebTest":
                return new exports.WebTest(name, undefined, { urn });
            case "azure:appinsights/workbook:Workbook":
                return new exports.Workbook(name, undefined, { urn });
            case "azure:appinsights/workbookTemplate:WorkbookTemplate":
                return new exports.WorkbookTemplate(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "appinsights/analyticsItem", _module);
pulumi.runtime.registerResourceModule("azure", "appinsights/apiKey", _module);
pulumi.runtime.registerResourceModule("azure", "appinsights/insights", _module);
pulumi.runtime.registerResourceModule("azure", "appinsights/smartDetectionRule", _module);
pulumi.runtime.registerResourceModule("azure", "appinsights/standardWebTest", _module);
pulumi.runtime.registerResourceModule("azure", "appinsights/webTest", _module);
pulumi.runtime.registerResourceModule("azure", "appinsights/workbook", _module);
pulumi.runtime.registerResourceModule("azure", "appinsights/workbookTemplate", _module);
