// @pulumi/azure-compliance-policies/azurenative/deviceregistry/v20240901preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaVersion = exports.SchemaRegistry = exports.Schema = exports.DiscoveredAssetEndpointProfile = exports.DiscoveredAsset = exports.AssetEndpointProfile = exports.Asset = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240901preview_1 = require("@pulumi/azure-native/deviceregistry/v20240901preview");
const v20240901preview_2 = require("@pulumi/azure-native/deviceregistry/v20240901preview");
const v20240901preview_3 = require("@pulumi/azure-native/deviceregistry/v20240901preview");
const v20240901preview_4 = require("@pulumi/azure-native/deviceregistry/v20240901preview");
const v20240901preview_5 = require("@pulumi/azure-native/deviceregistry/v20240901preview");
const v20240901preview_6 = require("@pulumi/azure-native/deviceregistry/v20240901preview");
const v20240901preview_7 = require("@pulumi/azure-native/deviceregistry/v20240901preview");
var Asset;
(function (Asset) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.Asset).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Asset.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-deviceregistry-v20240901preview-asset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.Asset).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_1.Asset, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Asset shouldn't use an unstable API (deviceregistry.v20240901preview.Asset). A compatible replacement can be found at 'deviceregistry.Asset'.");
            }),
        },
        vendors: ["azure"],
        services: ["deviceregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Asset || (Asset = {}));
exports.Asset = Asset;
var AssetEndpointProfile;
(function (AssetEndpointProfile) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.AssetEndpointProfile).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AssetEndpointProfile.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-deviceregistry-v20240901preview-assetendpointprofile-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.AssetEndpointProfile).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_2.AssetEndpointProfile, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AssetEndpointProfile shouldn't use an unstable API (deviceregistry.v20240901preview.AssetEndpointProfile). A compatible replacement can be found at 'deviceregistry.AssetEndpointProfile'.");
            }),
        },
        vendors: ["azure"],
        services: ["deviceregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AssetEndpointProfile || (AssetEndpointProfile = {}));
exports.AssetEndpointProfile = AssetEndpointProfile;
var DiscoveredAsset;
(function (DiscoveredAsset) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.DiscoveredAsset).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DiscoveredAsset.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-deviceregistry-v20240901preview-discoveredasset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.DiscoveredAsset).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_3.DiscoveredAsset, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DiscoveredAsset shouldn't use an unstable API (deviceregistry.v20240901preview.DiscoveredAsset). A compatible replacement can be found at 'deviceregistry.DiscoveredAsset'.");
            }),
        },
        vendors: ["azure"],
        services: ["deviceregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DiscoveredAsset || (DiscoveredAsset = {}));
exports.DiscoveredAsset = DiscoveredAsset;
var DiscoveredAssetEndpointProfile;
(function (DiscoveredAssetEndpointProfile) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.DiscoveredAssetEndpointProfile).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DiscoveredAssetEndpointProfile.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-deviceregistry-v20240901preview-discoveredassetendpointprofile-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.DiscoveredAssetEndpointProfile).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_4.DiscoveredAssetEndpointProfile, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DiscoveredAssetEndpointProfile shouldn't use an unstable API (deviceregistry.v20240901preview.DiscoveredAssetEndpointProfile). A compatible replacement can be found at 'deviceregistry.DiscoveredAssetEndpointProfile'.");
            }),
        },
        vendors: ["azure"],
        services: ["deviceregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DiscoveredAssetEndpointProfile || (DiscoveredAssetEndpointProfile = {}));
exports.DiscoveredAssetEndpointProfile = DiscoveredAssetEndpointProfile;
var Schema;
(function (Schema) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.Schema).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Schema.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-deviceregistry-v20240901preview-schema-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.Schema).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_5.Schema, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Schema shouldn't use an unstable API (deviceregistry.v20240901preview.Schema). A compatible replacement can be found at 'deviceregistry.Schema'.");
            }),
        },
        vendors: ["azure"],
        services: ["deviceregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Schema || (Schema = {}));
exports.Schema = Schema;
var SchemaRegistry;
(function (SchemaRegistry) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.SchemaRegistry).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SchemaRegistry.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-deviceregistry-v20240901preview-schemaregistry-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.SchemaRegistry).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_6.SchemaRegistry, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SchemaRegistry shouldn't use an unstable API (deviceregistry.v20240901preview.SchemaRegistry). A compatible replacement can be found at 'deviceregistry.SchemaRegistry'.");
            }),
        },
        vendors: ["azure"],
        services: ["deviceregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SchemaRegistry || (SchemaRegistry = {}));
exports.SchemaRegistry = SchemaRegistry;
var SchemaVersion;
(function (SchemaVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.SchemaVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SchemaVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-deviceregistry-v20240901preview-schemaversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20240901preview.SchemaVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240901preview_7.SchemaVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SchemaVersion shouldn't use an unstable API (deviceregistry.v20240901preview.SchemaVersion). A compatible replacement can be found at 'deviceregistry.SchemaVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["deviceregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SchemaVersion || (SchemaVersion = {}));
exports.SchemaVersion = SchemaVersion;
