// @pulumi/azure/elasticcloud/elasticsearch.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Elasticsearch = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Elasticsearch in Elastic Cloud.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const test = new azure.core.ResourceGroup("test", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const testElasticsearch = new azure.elasticcloud.Elasticsearch("test", {
 *     name: "example-elasticsearch",
 *     resourceGroupName: test.name,
 *     location: test.location,
 *     skuName: "ess-consumption-2024_Monthly",
 *     elasticCloudEmailAddress: "user@example.com",
 * });
 * ```
 *
 * ## Import
 *
 * Elasticsearch's can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:elasticcloud/elasticsearch:Elasticsearch example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Elastic/monitors/monitor1
 * ```
 */
class Elasticsearch extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["elasticCloudDeploymentId"] = state ? state.elasticCloudDeploymentId : undefined;
            resourceInputs["elasticCloudEmailAddress"] = state ? state.elasticCloudEmailAddress : undefined;
            resourceInputs["elasticCloudSsoDefaultUrl"] = state ? state.elasticCloudSsoDefaultUrl : undefined;
            resourceInputs["elasticCloudUserId"] = state ? state.elasticCloudUserId : undefined;
            resourceInputs["elasticsearchServiceUrl"] = state ? state.elasticsearchServiceUrl : undefined;
            resourceInputs["kibanaServiceUrl"] = state ? state.kibanaServiceUrl : undefined;
            resourceInputs["kibanaSsoUri"] = state ? state.kibanaSsoUri : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["logs"] = state ? state.logs : undefined;
            resourceInputs["monitoringEnabled"] = state ? state.monitoringEnabled : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["skuName"] = state ? state.skuName : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.elasticCloudEmailAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'elasticCloudEmailAddress'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.skuName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'skuName'");
            }
            resourceInputs["elasticCloudEmailAddress"] = args ? args.elasticCloudEmailAddress : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["logs"] = args ? args.logs : undefined;
            resourceInputs["monitoringEnabled"] = args ? args.monitoringEnabled : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["skuName"] = args ? args.skuName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["elasticCloudDeploymentId"] = undefined /*out*/;
            resourceInputs["elasticCloudSsoDefaultUrl"] = undefined /*out*/;
            resourceInputs["elasticCloudUserId"] = undefined /*out*/;
            resourceInputs["elasticsearchServiceUrl"] = undefined /*out*/;
            resourceInputs["kibanaServiceUrl"] = undefined /*out*/;
            resourceInputs["kibanaSsoUri"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Elasticsearch.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Elasticsearch resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Elasticsearch(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Elasticsearch.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Elasticsearch.__pulumiType;
    }
}
exports.Elasticsearch = Elasticsearch;
/** @internal */
Elasticsearch.__pulumiType = 'azure:elasticcloud/elasticsearch:Elasticsearch';
