// @pulumi/azure-compliance-policies/azurenative/contoso/v20211001preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Employee = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20211001preview_1 = require("@pulumi/azure-native/contoso/v20211001preview");
var Employee;
(function (Employee) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (contoso.v20211001preview.Employee).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Employee.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-contoso-v20211001preview-employee-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (contoso.v20211001preview.Employee).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20211001preview_1.Employee, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Employee shouldn't use an unstable API (contoso.v20211001preview.Employee). A compatible replacement can be found at 'contoso.Employee'.");
            }),
        },
        vendors: ["azure"],
        services: ["contoso"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Employee || (Employee = {}));
exports.Employee = Employee;
