// @pulumi/azure-native/cloudngfw/fqdnListGlobalRulestack.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FqdnListGlobalRulestack = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * GlobalRulestack fqdnList
 * Azure REST API version: 2023-09-01.
 *
 * Other available API versions: 2022-08-29, 2022-08-29-preview, 2023-09-01-preview, 2023-10-10-preview, 2024-01-19-preview, 2024-02-07-preview, 2025-02-06-preview.
 */
class FqdnListGlobalRulestack extends pulumi.CustomResource {
    /**
     * Get an existing FqdnListGlobalRulestack resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FqdnListGlobalRulestack(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FqdnListGlobalRulestack.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FqdnListGlobalRulestack.__pulumiType;
    }
    /**
     * Create a FqdnListGlobalRulestack resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fqdnList === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fqdnList'");
            }
            if ((!args || args.globalRulestackName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'globalRulestackName'");
            }
            resourceInputs["auditComment"] = args ? args.auditComment : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["fqdnList"] = args ? args.fqdnList : undefined;
            resourceInputs["globalRulestackName"] = args ? args.globalRulestackName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["auditComment"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fqdnList"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cloudngfw/v20220829:FqdnListGlobalRulestack" }, { type: "azure-native:cloudngfw/v20220829preview:FqdnListGlobalRulestack" }, { type: "azure-native:cloudngfw/v20230901:FqdnListGlobalRulestack" }, { type: "azure-native:cloudngfw/v20230901preview:FqdnListGlobalRulestack" }, { type: "azure-native:cloudngfw/v20231010preview:FqdnListGlobalRulestack" }, { type: "azure-native:cloudngfw/v20240119preview:FqdnListGlobalRulestack" }, { type: "azure-native:cloudngfw/v20240207preview:FqdnListGlobalRulestack" }, { type: "azure-native:cloudngfw/v20250206preview:FqdnListGlobalRulestack" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FqdnListGlobalRulestack.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FqdnListGlobalRulestack = FqdnListGlobalRulestack;
/** @internal */
FqdnListGlobalRulestack.__pulumiType = 'azure-native:cloudngfw:FqdnListGlobalRulestack';
