// @pulumi/azure/avs/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateCloud = exports.NetappVolumeAttachment = exports.getPrivateCloudOutput = exports.getPrivateCloud = exports.ExpressRouteAuthorization = exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ExpressRouteAuthorization = null;
utilities.lazyLoad(exports, ["ExpressRouteAuthorization"], () => require("./expressRouteAuthorization"));
exports.getPrivateCloud = null;
exports.getPrivateCloudOutput = null;
utilities.lazyLoad(exports, ["getPrivateCloud", "getPrivateCloudOutput"], () => require("./getPrivateCloud"));
exports.NetappVolumeAttachment = null;
utilities.lazyLoad(exports, ["NetappVolumeAttachment"], () => require("./netappVolumeAttachment"));
exports.PrivateCloud = null;
utilities.lazyLoad(exports, ["PrivateCloud"], () => require("./privateCloud"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:avs/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure:avs/expressRouteAuthorization:ExpressRouteAuthorization":
                return new exports.ExpressRouteAuthorization(name, undefined, { urn });
            case "azure:avs/netappVolumeAttachment:NetappVolumeAttachment":
                return new exports.NetappVolumeAttachment(name, undefined, { urn });
            case "azure:avs/privateCloud:PrivateCloud":
                return new exports.PrivateCloud(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "avs/cluster", _module);
pulumi.runtime.registerResourceModule("azure", "avs/expressRouteAuthorization", _module);
pulumi.runtime.registerResourceModule("azure", "avs/netappVolumeAttachment", _module);
pulumi.runtime.registerResourceModule("azure", "avs/privateCloud", _module);
