// @pulumi/azure/databricks/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceRootDbfsCustomerManagedKey = exports.WorkspaceCustomerManagedKey = exports.Workspace = exports.VirtualNetworkPeering = exports.getWorkspacePrivateEndpointConnectionOutput = exports.getWorkspacePrivateEndpointConnection = exports.getWorkspaceOutput = exports.getWorkspace = exports.getAccessConnectorOutput = exports.getAccessConnector = exports.AccessConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccessConnector = null;
utilities.lazyLoad(exports, ["AccessConnector"], () => require("./accessConnector"));
exports.getAccessConnector = null;
exports.getAccessConnectorOutput = null;
utilities.lazyLoad(exports, ["getAccessConnector", "getAccessConnectorOutput"], () => require("./getAccessConnector"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.getWorkspacePrivateEndpointConnection = null;
exports.getWorkspacePrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getWorkspacePrivateEndpointConnection", "getWorkspacePrivateEndpointConnectionOutput"], () => require("./getWorkspacePrivateEndpointConnection"));
exports.VirtualNetworkPeering = null;
utilities.lazyLoad(exports, ["VirtualNetworkPeering"], () => require("./virtualNetworkPeering"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
exports.WorkspaceCustomerManagedKey = null;
utilities.lazyLoad(exports, ["WorkspaceCustomerManagedKey"], () => require("./workspaceCustomerManagedKey"));
exports.WorkspaceRootDbfsCustomerManagedKey = null;
utilities.lazyLoad(exports, ["WorkspaceRootDbfsCustomerManagedKey"], () => require("./workspaceRootDbfsCustomerManagedKey"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:databricks/accessConnector:AccessConnector":
                return new exports.AccessConnector(name, undefined, { urn });
            case "azure:databricks/virtualNetworkPeering:VirtualNetworkPeering":
                return new exports.VirtualNetworkPeering(name, undefined, { urn });
            case "azure:databricks/workspace:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            case "azure:databricks/workspaceCustomerManagedKey:WorkspaceCustomerManagedKey":
                return new exports.WorkspaceCustomerManagedKey(name, undefined, { urn });
            case "azure:databricks/workspaceRootDbfsCustomerManagedKey:WorkspaceRootDbfsCustomerManagedKey":
                return new exports.WorkspaceRootDbfsCustomerManagedKey(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "databricks/accessConnector", _module);
pulumi.runtime.registerResourceModule("azure", "databricks/virtualNetworkPeering", _module);
pulumi.runtime.registerResourceModule("azure", "databricks/workspace", _module);
pulumi.runtime.registerResourceModule("azure", "databricks/workspaceCustomerManagedKey", _module);
pulumi.runtime.registerResourceModule("azure", "databricks/workspaceRootDbfsCustomerManagedKey", _module);
