// @pulumi/azure/databricks/virtualNetworkPeering.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkPeering = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Databricks Virtual Network Peering
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const remote = new azure.network.VirtualNetwork("remote", {
 *     name: "remote-vnet",
 *     resourceGroupName: example.name,
 *     addressSpaces: ["10.0.1.0/24"],
 *     location: example.location,
 * });
 * const exampleWorkspace = new azure.databricks.Workspace("example", {
 *     name: "example-workspace",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     sku: "standard",
 * });
 * const exampleVirtualNetworkPeering = new azure.databricks.VirtualNetworkPeering("example", {
 *     name: "databricks-vnet-peer",
 *     resourceGroupName: example.name,
 *     workspaceId: exampleWorkspace.id,
 *     remoteAddressSpacePrefixes: remote.addressSpaces,
 *     remoteVirtualNetworkId: remote.id,
 *     allowVirtualNetworkAccess: true,
 * });
 * const remoteVirtualNetworkPeering = new azure.network.VirtualNetworkPeering("remote", {
 *     name: "peer-to-databricks",
 *     resourceGroupName: example.name,
 *     virtualNetworkName: remote.name,
 *     remoteVirtualNetworkId: exampleVirtualNetworkPeering.virtualNetworkId,
 *     allowVirtualNetworkAccess: true,
 * });
 * ```
 *
 * ## Import
 *
 * Databrick Virtual Network Peerings can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:databricks/virtualNetworkPeering:VirtualNetworkPeering example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Databricks/workspaces/workspace1/virtualNetworkPeerings/peering1
 * ```
 */
class VirtualNetworkPeering extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["addressSpacePrefixes"] = state ? state.addressSpacePrefixes : undefined;
            resourceInputs["allowForwardedTraffic"] = state ? state.allowForwardedTraffic : undefined;
            resourceInputs["allowGatewayTransit"] = state ? state.allowGatewayTransit : undefined;
            resourceInputs["allowVirtualNetworkAccess"] = state ? state.allowVirtualNetworkAccess : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["remoteAddressSpacePrefixes"] = state ? state.remoteAddressSpacePrefixes : undefined;
            resourceInputs["remoteVirtualNetworkId"] = state ? state.remoteVirtualNetworkId : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["useRemoteGateways"] = state ? state.useRemoteGateways : undefined;
            resourceInputs["virtualNetworkId"] = state ? state.virtualNetworkId : undefined;
            resourceInputs["workspaceId"] = state ? state.workspaceId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.remoteAddressSpacePrefixes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'remoteAddressSpacePrefixes'");
            }
            if ((!args || args.remoteVirtualNetworkId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'remoteVirtualNetworkId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceId'");
            }
            resourceInputs["allowForwardedTraffic"] = args ? args.allowForwardedTraffic : undefined;
            resourceInputs["allowGatewayTransit"] = args ? args.allowGatewayTransit : undefined;
            resourceInputs["allowVirtualNetworkAccess"] = args ? args.allowVirtualNetworkAccess : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["remoteAddressSpacePrefixes"] = args ? args.remoteAddressSpacePrefixes : undefined;
            resourceInputs["remoteVirtualNetworkId"] = args ? args.remoteVirtualNetworkId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["useRemoteGateways"] = args ? args.useRemoteGateways : undefined;
            resourceInputs["workspaceId"] = args ? args.workspaceId : undefined;
            resourceInputs["addressSpacePrefixes"] = undefined /*out*/;
            resourceInputs["virtualNetworkId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VirtualNetworkPeering.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing VirtualNetworkPeering resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VirtualNetworkPeering(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualNetworkPeering.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualNetworkPeering.__pulumiType;
    }
}
exports.VirtualNetworkPeering = VirtualNetworkPeering;
/** @internal */
VirtualNetworkPeering.__pulumiType = 'azure:databricks/virtualNetworkPeering:VirtualNetworkPeering';
