// @pulumi/azure/redis/cacheAccessPolicyAssignment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CacheAccessPolicyAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Redis Cache Access Policy Assignment
 *
 * ## Import
 *
 * Redis Cache Policy Assignment can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:redis/cacheAccessPolicyAssignment:CacheAccessPolicyAssignment example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Cache/redis/cache1/accessPolicyAssignments/assignment1
 * ```
 */
class CacheAccessPolicyAssignment extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accessPolicyName"] = state ? state.accessPolicyName : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["objectId"] = state ? state.objectId : undefined;
            resourceInputs["objectIdAlias"] = state ? state.objectIdAlias : undefined;
            resourceInputs["redisCacheId"] = state ? state.redisCacheId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accessPolicyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accessPolicyName'");
            }
            if ((!args || args.objectId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'objectId'");
            }
            if ((!args || args.objectIdAlias === undefined) && !opts.urn) {
                throw new Error("Missing required property 'objectIdAlias'");
            }
            if ((!args || args.redisCacheId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'redisCacheId'");
            }
            resourceInputs["accessPolicyName"] = args ? args.accessPolicyName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["objectId"] = args ? args.objectId : undefined;
            resourceInputs["objectIdAlias"] = args ? args.objectIdAlias : undefined;
            resourceInputs["redisCacheId"] = args ? args.redisCacheId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CacheAccessPolicyAssignment.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing CacheAccessPolicyAssignment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CacheAccessPolicyAssignment(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CacheAccessPolicyAssignment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CacheAccessPolicyAssignment.__pulumiType;
    }
}
exports.CacheAccessPolicyAssignment = CacheAccessPolicyAssignment;
/** @internal */
CacheAccessPolicyAssignment.__pulumiType = 'azure:redis/cacheAccessPolicyAssignment:CacheAccessPolicyAssignment';
