// @pulumi/azure/redis/enterpriseDatabase.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnterpriseDatabase = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Redis Enterprise Database.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-redisenterprise",
 *     location: "West Europe",
 * });
 * const exampleEnterpriseCluster = new azure.redis.EnterpriseCluster("example", {
 *     name: "example-redisenterprise",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     skuName: "Enterprise_E20-4",
 * });
 * const example1 = new azure.redis.EnterpriseCluster("example1", {
 *     name: "example-redisenterprise1",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     skuName: "Enterprise_E20-4",
 * });
 * const exampleEnterpriseDatabase = new azure.redis.EnterpriseDatabase("example", {
 *     name: "default",
 *     clusterId: exampleEnterpriseCluster.id,
 *     clientProtocol: "Encrypted",
 *     clusteringPolicy: "EnterpriseCluster",
 *     evictionPolicy: "NoEviction",
 *     port: 10000,
 *     linkedDatabaseIds: [
 *         pulumi.interpolate`${exampleEnterpriseCluster.id}/databases/default`,
 *         pulumi.interpolate`${example1.id}/databases/default`,
 *     ],
 *     linkedDatabaseGroupNickname: "tftestGeoGroup",
 * });
 * ```
 *
 * ## Import
 *
 * Redis Enterprise Databases can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:redis/enterpriseDatabase:EnterpriseDatabase example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Cache/redisEnterprise/cluster1/databases/database1
 * ```
 */
class EnterpriseDatabase extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["clientProtocol"] = state ? state.clientProtocol : undefined;
            resourceInputs["clusterId"] = state ? state.clusterId : undefined;
            resourceInputs["clusteringPolicy"] = state ? state.clusteringPolicy : undefined;
            resourceInputs["evictionPolicy"] = state ? state.evictionPolicy : undefined;
            resourceInputs["linkedDatabaseGroupNickname"] = state ? state.linkedDatabaseGroupNickname : undefined;
            resourceInputs["linkedDatabaseIds"] = state ? state.linkedDatabaseIds : undefined;
            resourceInputs["modules"] = state ? state.modules : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["port"] = state ? state.port : undefined;
            resourceInputs["primaryAccessKey"] = state ? state.primaryAccessKey : undefined;
            resourceInputs["secondaryAccessKey"] = state ? state.secondaryAccessKey : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.clusterId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterId'");
            }
            resourceInputs["clientProtocol"] = args ? args.clientProtocol : undefined;
            resourceInputs["clusterId"] = args ? args.clusterId : undefined;
            resourceInputs["clusteringPolicy"] = args ? args.clusteringPolicy : undefined;
            resourceInputs["evictionPolicy"] = args ? args.evictionPolicy : undefined;
            resourceInputs["linkedDatabaseGroupNickname"] = args ? args.linkedDatabaseGroupNickname : undefined;
            resourceInputs["linkedDatabaseIds"] = args ? args.linkedDatabaseIds : undefined;
            resourceInputs["modules"] = args ? args.modules : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["port"] = args ? args.port : undefined;
            resourceInputs["primaryAccessKey"] = undefined /*out*/;
            resourceInputs["secondaryAccessKey"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["primaryAccessKey", "secondaryAccessKey"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(EnterpriseDatabase.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing EnterpriseDatabase resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EnterpriseDatabase(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EnterpriseDatabase.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnterpriseDatabase.__pulumiType;
    }
}
exports.EnterpriseDatabase = EnterpriseDatabase;
/** @internal */
EnterpriseDatabase.__pulumiType = 'azure:redis/enterpriseDatabase:EnterpriseDatabase';
