// @pulumi/azure/automation/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook = exports.Watcher = exports.VariableObject = exports.StringVariable = exports.SourceControl = exports.SoftwareUpdateConfiguration = exports.Schedule = exports.RunBook = exports.Python3Package = exports.Powershell72Module = exports.Module = exports.JobSchedule = exports.IntVariable = exports.HybridRunbookWorkerGroup = exports.HybridRunbookWorker = exports.getVariablesOutput = exports.getVariables = exports.getVariableObjectOutput = exports.getVariableObject = exports.getStringVariableOutput = exports.getStringVariable = exports.getRunbookOutput = exports.getRunbook = exports.getIntVariableOutput = exports.getIntVariable = exports.getDateTimeVariableOutput = exports.getDateTimeVariable = exports.getBoolVariableOutput = exports.getBoolVariable = exports.getAccountOutput = exports.getAccount = exports.DscNodeConfiguration = exports.DscConfiguration = exports.DateTimeVariable = exports.Credential = exports.ConnectionType = exports.ConnectionServicePrincipal = exports.ConnectionClassicCertificate = exports.ConnectionCertificate = exports.Connection = exports.Certificate = exports.BoolVariable = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.BoolVariable = null;
utilities.lazyLoad(exports, ["BoolVariable"], () => require("./boolVariable"));
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.ConnectionCertificate = null;
utilities.lazyLoad(exports, ["ConnectionCertificate"], () => require("./connectionCertificate"));
exports.ConnectionClassicCertificate = null;
utilities.lazyLoad(exports, ["ConnectionClassicCertificate"], () => require("./connectionClassicCertificate"));
exports.ConnectionServicePrincipal = null;
utilities.lazyLoad(exports, ["ConnectionServicePrincipal"], () => require("./connectionServicePrincipal"));
exports.ConnectionType = null;
utilities.lazyLoad(exports, ["ConnectionType"], () => require("./connectionType"));
exports.Credential = null;
utilities.lazyLoad(exports, ["Credential"], () => require("./credential"));
exports.DateTimeVariable = null;
utilities.lazyLoad(exports, ["DateTimeVariable"], () => require("./dateTimeVariable"));
exports.DscConfiguration = null;
utilities.lazyLoad(exports, ["DscConfiguration"], () => require("./dscConfiguration"));
exports.DscNodeConfiguration = null;
utilities.lazyLoad(exports, ["DscNodeConfiguration"], () => require("./dscNodeConfiguration"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getBoolVariable = null;
exports.getBoolVariableOutput = null;
utilities.lazyLoad(exports, ["getBoolVariable", "getBoolVariableOutput"], () => require("./getBoolVariable"));
exports.getDateTimeVariable = null;
exports.getDateTimeVariableOutput = null;
utilities.lazyLoad(exports, ["getDateTimeVariable", "getDateTimeVariableOutput"], () => require("./getDateTimeVariable"));
exports.getIntVariable = null;
exports.getIntVariableOutput = null;
utilities.lazyLoad(exports, ["getIntVariable", "getIntVariableOutput"], () => require("./getIntVariable"));
exports.getRunbook = null;
exports.getRunbookOutput = null;
utilities.lazyLoad(exports, ["getRunbook", "getRunbookOutput"], () => require("./getRunbook"));
exports.getStringVariable = null;
exports.getStringVariableOutput = null;
utilities.lazyLoad(exports, ["getStringVariable", "getStringVariableOutput"], () => require("./getStringVariable"));
exports.getVariableObject = null;
exports.getVariableObjectOutput = null;
utilities.lazyLoad(exports, ["getVariableObject", "getVariableObjectOutput"], () => require("./getVariableObject"));
exports.getVariables = null;
exports.getVariablesOutput = null;
utilities.lazyLoad(exports, ["getVariables", "getVariablesOutput"], () => require("./getVariables"));
exports.HybridRunbookWorker = null;
utilities.lazyLoad(exports, ["HybridRunbookWorker"], () => require("./hybridRunbookWorker"));
exports.HybridRunbookWorkerGroup = null;
utilities.lazyLoad(exports, ["HybridRunbookWorkerGroup"], () => require("./hybridRunbookWorkerGroup"));
exports.IntVariable = null;
utilities.lazyLoad(exports, ["IntVariable"], () => require("./intVariable"));
exports.JobSchedule = null;
utilities.lazyLoad(exports, ["JobSchedule"], () => require("./jobSchedule"));
exports.Module = null;
utilities.lazyLoad(exports, ["Module"], () => require("./module"));
exports.Powershell72Module = null;
utilities.lazyLoad(exports, ["Powershell72Module"], () => require("./powershell72Module"));
exports.Python3Package = null;
utilities.lazyLoad(exports, ["Python3Package"], () => require("./python3Package"));
exports.RunBook = null;
utilities.lazyLoad(exports, ["RunBook"], () => require("./runBook"));
exports.Schedule = null;
utilities.lazyLoad(exports, ["Schedule"], () => require("./schedule"));
exports.SoftwareUpdateConfiguration = null;
utilities.lazyLoad(exports, ["SoftwareUpdateConfiguration"], () => require("./softwareUpdateConfiguration"));
exports.SourceControl = null;
utilities.lazyLoad(exports, ["SourceControl"], () => require("./sourceControl"));
exports.StringVariable = null;
utilities.lazyLoad(exports, ["StringVariable"], () => require("./stringVariable"));
exports.VariableObject = null;
utilities.lazyLoad(exports, ["VariableObject"], () => require("./variableObject"));
exports.Watcher = null;
utilities.lazyLoad(exports, ["Watcher"], () => require("./watcher"));
exports.Webhook = null;
utilities.lazyLoad(exports, ["Webhook"], () => require("./webhook"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:automation/account:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure:automation/boolVariable:BoolVariable":
                return new exports.BoolVariable(name, undefined, { urn });
            case "azure:automation/certificate:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "azure:automation/connection:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "azure:automation/connectionCertificate:ConnectionCertificate":
                return new exports.ConnectionCertificate(name, undefined, { urn });
            case "azure:automation/connectionClassicCertificate:ConnectionClassicCertificate":
                return new exports.ConnectionClassicCertificate(name, undefined, { urn });
            case "azure:automation/connectionServicePrincipal:ConnectionServicePrincipal":
                return new exports.ConnectionServicePrincipal(name, undefined, { urn });
            case "azure:automation/connectionType:ConnectionType":
                return new exports.ConnectionType(name, undefined, { urn });
            case "azure:automation/credential:Credential":
                return new exports.Credential(name, undefined, { urn });
            case "azure:automation/dateTimeVariable:DateTimeVariable":
                return new exports.DateTimeVariable(name, undefined, { urn });
            case "azure:automation/dscConfiguration:DscConfiguration":
                return new exports.DscConfiguration(name, undefined, { urn });
            case "azure:automation/dscNodeConfiguration:DscNodeConfiguration":
                return new exports.DscNodeConfiguration(name, undefined, { urn });
            case "azure:automation/hybridRunbookWorker:HybridRunbookWorker":
                return new exports.HybridRunbookWorker(name, undefined, { urn });
            case "azure:automation/hybridRunbookWorkerGroup:HybridRunbookWorkerGroup":
                return new exports.HybridRunbookWorkerGroup(name, undefined, { urn });
            case "azure:automation/intVariable:IntVariable":
                return new exports.IntVariable(name, undefined, { urn });
            case "azure:automation/jobSchedule:JobSchedule":
                return new exports.JobSchedule(name, undefined, { urn });
            case "azure:automation/module:Module":
                return new exports.Module(name, undefined, { urn });
            case "azure:automation/powershell72Module:Powershell72Module":
                return new exports.Powershell72Module(name, undefined, { urn });
            case "azure:automation/python3Package:Python3Package":
                return new exports.Python3Package(name, undefined, { urn });
            case "azure:automation/runBook:RunBook":
                return new exports.RunBook(name, undefined, { urn });
            case "azure:automation/schedule:Schedule":
                return new exports.Schedule(name, undefined, { urn });
            case "azure:automation/softwareUpdateConfiguration:SoftwareUpdateConfiguration":
                return new exports.SoftwareUpdateConfiguration(name, undefined, { urn });
            case "azure:automation/sourceControl:SourceControl":
                return new exports.SourceControl(name, undefined, { urn });
            case "azure:automation/stringVariable:StringVariable":
                return new exports.StringVariable(name, undefined, { urn });
            case "azure:automation/variableObject:VariableObject":
                return new exports.VariableObject(name, undefined, { urn });
            case "azure:automation/watcher:Watcher":
                return new exports.Watcher(name, undefined, { urn });
            case "azure:automation/webhook:Webhook":
                return new exports.Webhook(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "automation/account", _module);
pulumi.runtime.registerResourceModule("azure", "automation/boolVariable", _module);
pulumi.runtime.registerResourceModule("azure", "automation/certificate", _module);
pulumi.runtime.registerResourceModule("azure", "automation/connection", _module);
pulumi.runtime.registerResourceModule("azure", "automation/connectionCertificate", _module);
pulumi.runtime.registerResourceModule("azure", "automation/connectionClassicCertificate", _module);
pulumi.runtime.registerResourceModule("azure", "automation/connectionServicePrincipal", _module);
pulumi.runtime.registerResourceModule("azure", "automation/connectionType", _module);
pulumi.runtime.registerResourceModule("azure", "automation/credential", _module);
pulumi.runtime.registerResourceModule("azure", "automation/dateTimeVariable", _module);
pulumi.runtime.registerResourceModule("azure", "automation/dscConfiguration", _module);
pulumi.runtime.registerResourceModule("azure", "automation/dscNodeConfiguration", _module);
pulumi.runtime.registerResourceModule("azure", "automation/hybridRunbookWorker", _module);
pulumi.runtime.registerResourceModule("azure", "automation/hybridRunbookWorkerGroup", _module);
pulumi.runtime.registerResourceModule("azure", "automation/intVariable", _module);
pulumi.runtime.registerResourceModule("azure", "automation/jobSchedule", _module);
pulumi.runtime.registerResourceModule("azure", "automation/module", _module);
pulumi.runtime.registerResourceModule("azure", "automation/powershell72Module", _module);
pulumi.runtime.registerResourceModule("azure", "automation/python3Package", _module);
pulumi.runtime.registerResourceModule("azure", "automation/runBook", _module);
pulumi.runtime.registerResourceModule("azure", "automation/schedule", _module);
pulumi.runtime.registerResourceModule("azure", "automation/softwareUpdateConfiguration", _module);
pulumi.runtime.registerResourceModule("azure", "automation/sourceControl", _module);
pulumi.runtime.registerResourceModule("azure", "automation/stringVariable", _module);
pulumi.runtime.registerResourceModule("azure", "automation/variableObject", _module);
pulumi.runtime.registerResourceModule("azure", "automation/watcher", _module);
pulumi.runtime.registerResourceModule("azure", "automation/webhook", _module);
