// @pulumi/azure-compliance-policies/azurenative/servicefabricmesh/v20180901preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Volume = exports.SecretValue = exports.Secret = exports.Network = exports.Gateway = exports.Application = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20180901preview_1 = require("@pulumi/azure-native/servicefabricmesh/v20180901preview");
const v20180901preview_2 = require("@pulumi/azure-native/servicefabricmesh/v20180901preview");
const v20180901preview_3 = require("@pulumi/azure-native/servicefabricmesh/v20180901preview");
const v20180901preview_4 = require("@pulumi/azure-native/servicefabricmesh/v20180901preview");
const v20180901preview_5 = require("@pulumi/azure-native/servicefabricmesh/v20180901preview");
const v20180901preview_6 = require("@pulumi/azure-native/servicefabricmesh/v20180901preview");
var Application;
(function (Application) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Application).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Application.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-servicefabricmesh-v20180901preview-application-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Application).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_1.Application, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Application shouldn't use an unstable API (servicefabricmesh.v20180901preview.Application). A compatible replacement can be found at 'servicefabricmesh.Application'.");
            }),
        },
        vendors: ["azure"],
        services: ["servicefabricmesh"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Application || (Application = {}));
exports.Application = Application;
var Gateway;
(function (Gateway) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Gateway).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Gateway.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-servicefabricmesh-v20180901preview-gateway-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Gateway).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_2.Gateway, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Gateway shouldn't use an unstable API (servicefabricmesh.v20180901preview.Gateway). A compatible replacement can be found at 'servicefabricmesh.Gateway'.");
            }),
        },
        vendors: ["azure"],
        services: ["servicefabricmesh"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Gateway || (Gateway = {}));
exports.Gateway = Gateway;
var Network;
(function (Network) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Network).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Network.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-servicefabricmesh-v20180901preview-network-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Network).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_3.Network, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Network shouldn't use an unstable API (servicefabricmesh.v20180901preview.Network). A compatible replacement can be found at 'servicefabricmesh.Network'.");
            }),
        },
        vendors: ["azure"],
        services: ["servicefabricmesh"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Network || (Network = {}));
exports.Network = Network;
var Secret;
(function (Secret) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Secret).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Secret.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-servicefabricmesh-v20180901preview-secret-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Secret).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_4.Secret, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Secret shouldn't use an unstable API (servicefabricmesh.v20180901preview.Secret). A compatible replacement can be found at 'servicefabricmesh.Secret'.");
            }),
        },
        vendors: ["azure"],
        services: ["servicefabricmesh"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Secret || (Secret = {}));
exports.Secret = Secret;
var SecretValue;
(function (SecretValue) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.SecretValue).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SecretValue.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-servicefabricmesh-v20180901preview-secretvalue-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.SecretValue).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_5.SecretValue, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SecretValue shouldn't use an unstable API (servicefabricmesh.v20180901preview.SecretValue). A compatible replacement can be found at 'servicefabricmesh.SecretValue'.");
            }),
        },
        vendors: ["azure"],
        services: ["servicefabricmesh"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SecretValue || (SecretValue = {}));
exports.SecretValue = SecretValue;
var Volume;
(function (Volume) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Volume).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Volume.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-servicefabricmesh-v20180901preview-volume-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (servicefabricmesh.v20180901preview.Volume).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_6.Volume, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Volume shouldn't use an unstable API (servicefabricmesh.v20180901preview.Volume). A compatible replacement can be found at 'servicefabricmesh.Volume'.");
            }),
        },
        vendors: ["azure"],
        services: ["servicefabricmesh"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Volume || (Volume = {}));
exports.Volume = Volume;
