// @pulumi/azure-compliance-policies/azurenative/customproviders/v20180901preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomResourceProvider = exports.Association = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20180901preview_1 = require("@pulumi/azure-native/customproviders/v20180901preview");
const v20180901preview_2 = require("@pulumi/azure-native/customproviders/v20180901preview");
var Association;
(function (Association) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (customproviders.v20180901preview.Association).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Association.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-customproviders-v20180901preview-association-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (customproviders.v20180901preview.Association).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_1.Association, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Association shouldn't use an unstable API (customproviders.v20180901preview.Association). A compatible replacement can be found at 'customproviders.Association'.");
            }),
        },
        vendors: ["azure"],
        services: ["customproviders"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Association || (Association = {}));
exports.Association = Association;
var CustomResourceProvider;
(function (CustomResourceProvider) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (customproviders.v20180901preview.CustomResourceProvider).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomResourceProvider.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-customproviders-v20180901preview-customresourceprovider-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (customproviders.v20180901preview.CustomResourceProvider).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180901preview_2.CustomResourceProvider, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomResourceProvider shouldn't use an unstable API (customproviders.v20180901preview.CustomResourceProvider). A compatible replacement can be found at 'customproviders.CustomResourceProvider'.");
            }),
        },
        vendors: ["azure"],
        services: ["customproviders"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomResourceProvider || (CustomResourceProvider = {}));
exports.CustomResourceProvider = CustomResourceProvider;
