// @pulumi/azure-native/documentdb/cassandraResourceCassandraKeyspace.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CassandraResourceCassandraKeyspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Azure Cosmos DB Cassandra keyspace.
 * Azure REST API version: 2023-04-15. Prior API version in Azure Native 1.x: 2021-03-15.
 *
 * Other available API versions: 2023-03-15-preview, 2023-09-15, 2023-09-15-preview, 2023-11-15, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15, 2024-05-15-preview, 2024-08-15, 2024-09-01-preview, 2024-11-15, 2024-12-01-preview.
 */
class CassandraResourceCassandraKeyspace extends pulumi.CustomResource {
    /**
     * Get an existing CassandraResourceCassandraKeyspace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CassandraResourceCassandraKeyspace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CassandraResourceCassandraKeyspace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CassandraResourceCassandraKeyspace.__pulumiType;
    }
    /**
     * Create a CassandraResourceCassandraKeyspace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["keyspaceName"] = args ? args.keyspaceName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resource"] = args ? args.resource : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["options"] = undefined /*out*/;
            resourceInputs["resource"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb/v20150401:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20150408:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20151106:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20160319:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20160331:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20190801:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20191212:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20200301:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20200401:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20200601preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20200901:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210115:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210301preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210315:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210401preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210415:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210515:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210615:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210701preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20211015:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20211015preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20211115preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20220215preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20220515:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20220515preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20220815:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20220815preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20221115:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20221115preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20230301preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20230315:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20230315preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20230415:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20230915:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20230915preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20231115:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20231115preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20240215preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20240515:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20240515preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20240815:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20240901preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20241115:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20241201preview:CassandraResourceCassandraKeyspace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CassandraResourceCassandraKeyspace.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CassandraResourceCassandraKeyspace = CassandraResourceCassandraKeyspace;
/** @internal */
CassandraResourceCassandraKeyspace.__pulumiType = 'azure-native:documentdb:CassandraResourceCassandraKeyspace';
