// @pulumi/azure-native/documentdb/cassandraDataCenter.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CassandraDataCenter = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A managed Cassandra data center.
 * Azure REST API version: 2023-04-15. Prior API version in Azure Native 1.x: 2021-03-01-preview.
 *
 * Other available API versions: 2023-09-15, 2023-09-15-preview, 2023-11-15, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15, 2024-05-15-preview, 2024-08-15, 2024-09-01-preview, 2024-11-15, 2024-12-01-preview.
 */
class CassandraDataCenter extends pulumi.CustomResource {
    /**
     * Get an existing CassandraDataCenter resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CassandraDataCenter(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CassandraDataCenter.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CassandraDataCenter.__pulumiType;
    }
    /**
     * Create a CassandraDataCenter resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["dataCenterName"] = args ? args.dataCenterName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb/v20210301preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20210401preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20210701preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20211015:CassandraDataCenter" }, { type: "azure-native:documentdb/v20211015preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20211115preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20220215preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20220515:CassandraDataCenter" }, { type: "azure-native:documentdb/v20220515preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20220815:CassandraDataCenter" }, { type: "azure-native:documentdb/v20220815preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20221115:CassandraDataCenter" }, { type: "azure-native:documentdb/v20221115preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20230301preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20230315:CassandraDataCenter" }, { type: "azure-native:documentdb/v20230315preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20230415:CassandraDataCenter" }, { type: "azure-native:documentdb/v20230915:CassandraDataCenter" }, { type: "azure-native:documentdb/v20230915preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20231115:CassandraDataCenter" }, { type: "azure-native:documentdb/v20231115preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20240215preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20240515:CassandraDataCenter" }, { type: "azure-native:documentdb/v20240515preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20240815:CassandraDataCenter" }, { type: "azure-native:documentdb/v20240901preview:CassandraDataCenter" }, { type: "azure-native:documentdb/v20241115:CassandraDataCenter" }, { type: "azure-native:documentdb/v20241201preview:CassandraDataCenter" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CassandraDataCenter.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CassandraDataCenter = CassandraDataCenter;
/** @internal */
CassandraDataCenter.__pulumiType = 'azure-native:documentdb:CassandraDataCenter';
