// @pulumi/azure-native/documentdb/notebookWorkspace.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotebookWorkspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A notebook workspace resource
 * Azure REST API version: 2023-04-15. Prior API version in Azure Native 1.x: 2021-03-15.
 *
 * Other available API versions: 2023-09-15, 2023-09-15-preview, 2023-11-15, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15, 2024-05-15-preview, 2024-08-15, 2024-09-01-preview, 2024-11-15, 2024-12-01-preview.
 */
class NotebookWorkspace extends pulumi.CustomResource {
    /**
     * Get an existing NotebookWorkspace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NotebookWorkspace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NotebookWorkspace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NotebookWorkspace.__pulumiType;
    }
    /**
     * Create a NotebookWorkspace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["notebookWorkspaceName"] = args ? args.notebookWorkspaceName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["notebookServerEndpoint"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["notebookServerEndpoint"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb/v20190801:NotebookWorkspace" }, { type: "azure-native:documentdb/v20191212:NotebookWorkspace" }, { type: "azure-native:documentdb/v20200301:NotebookWorkspace" }, { type: "azure-native:documentdb/v20200401:NotebookWorkspace" }, { type: "azure-native:documentdb/v20200601preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20200901:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210115:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210301preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210315:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210401preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210415:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210515:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210615:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210701preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20211015:NotebookWorkspace" }, { type: "azure-native:documentdb/v20211015preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20211115preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20220215preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20220515:NotebookWorkspace" }, { type: "azure-native:documentdb/v20220515preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20220815:NotebookWorkspace" }, { type: "azure-native:documentdb/v20220815preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20221115:NotebookWorkspace" }, { type: "azure-native:documentdb/v20221115preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20230301preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20230315:NotebookWorkspace" }, { type: "azure-native:documentdb/v20230315preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20230415:NotebookWorkspace" }, { type: "azure-native:documentdb/v20230915:NotebookWorkspace" }, { type: "azure-native:documentdb/v20230915preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20231115:NotebookWorkspace" }, { type: "azure-native:documentdb/v20231115preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240215preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240515:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240515preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240815:NotebookWorkspace" }, { type: "azure-native:documentdb/v20240901preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20241115:NotebookWorkspace" }, { type: "azure-native:documentdb/v20241201preview:NotebookWorkspace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NotebookWorkspace.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NotebookWorkspace = NotebookWorkspace;
/** @internal */
NotebookWorkspace.__pulumiType = 'azure-native:documentdb:NotebookWorkspace';
