// @pulumi/azure-native/documentdb/sqlResourceSqlStoredProcedure.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlResourceSqlStoredProcedure = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Azure Cosmos DB storedProcedure.
 * Azure REST API version: 2023-04-15. Prior API version in Azure Native 1.x: 2021-03-15.
 *
 * Other available API versions: 2023-03-15-preview, 2023-09-15, 2023-09-15-preview, 2023-11-15, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15, 2024-05-15-preview, 2024-08-15, 2024-09-01-preview, 2024-11-15, 2024-12-01-preview.
 */
class SqlResourceSqlStoredProcedure extends pulumi.CustomResource {
    /**
     * Get an existing SqlResourceSqlStoredProcedure resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlResourceSqlStoredProcedure(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlResourceSqlStoredProcedure.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlResourceSqlStoredProcedure.__pulumiType;
    }
    /**
     * Create a SqlResourceSqlStoredProcedure resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.containerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerName'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["containerName"] = args ? args.containerName : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resource"] = args ? args.resource : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["storedProcedureName"] = args ? args.storedProcedureName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["resource"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb/v20190801:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20191212:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20200301:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20200401:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20200601preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20200901:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20210115:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20210301preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20210315:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20210401preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20210415:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20210515:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20210615:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20210701preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20211015:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20211015preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20211115preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20220215preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20220515:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20220515preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20220815:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20220815preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20221115:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20221115preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20230301preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20230315:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20230315preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20230415:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20230915:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20230915preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20231115:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20231115preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20240215preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20240515:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20240515preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20240815:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20240901preview:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20241115:SqlResourceSqlStoredProcedure" }, { type: "azure-native:documentdb/v20241201preview:SqlResourceSqlStoredProcedure" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlResourceSqlStoredProcedure.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SqlResourceSqlStoredProcedure = SqlResourceSqlStoredProcedure;
/** @internal */
SqlResourceSqlStoredProcedure.__pulumiType = 'azure-native:documentdb:SqlResourceSqlStoredProcedure';
