// @pulumi/azure-compliance-policies/azurenative/mobilenetwork/v20220301preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sim = exports.PacketCoreControlPlane = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20220301preview_1 = require("@pulumi/azure-native/mobilenetwork/v20220301preview");
const v20220301preview_2 = require("@pulumi/azure-native/mobilenetwork/v20220301preview");
var PacketCoreControlPlane;
(function (PacketCoreControlPlane) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (mobilenetwork.v20220301preview.PacketCoreControlPlane).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PacketCoreControlPlane.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-mobilenetwork-v20220301preview-packetcorecontrolplane-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (mobilenetwork.v20220301preview.PacketCoreControlPlane).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220301preview_1.PacketCoreControlPlane, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PacketCoreControlPlane shouldn't use an unstable API (mobilenetwork.v20220301preview.PacketCoreControlPlane). A compatible replacement can be found at 'mobilenetwork.PacketCoreControlPlane'.");
            }),
        },
        vendors: ["azure"],
        services: ["mobilenetwork"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PacketCoreControlPlane || (PacketCoreControlPlane = {}));
exports.PacketCoreControlPlane = PacketCoreControlPlane;
var Sim;
(function (Sim) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (mobilenetwork.v20220301preview.Sim).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Sim.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-mobilenetwork-v20220301preview-sim-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (mobilenetwork.v20220301preview.Sim).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220301preview_2.Sim, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Sim shouldn't use an unstable API (mobilenetwork.v20220301preview.Sim).");
            }),
        },
        vendors: ["azure"],
        services: ["mobilenetwork"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Sim || (Sim = {}));
exports.Sim = Sim;
