// @pulumi/azure-compliance-policies/azurenative/m365securityandcompliance/v20210325preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkServicesForSCCPowershell = exports.PrivateLinkServicesForO365ManagementActivityAPI = exports.PrivateLinkServicesForMIPPolicySync = exports.PrivateLinkServicesForM365SecurityCenter = exports.PrivateLinkServicesForM365ComplianceCenter = exports.PrivateLinkServicesForEDMUpload = exports.PrivateEndpointConnectionsSec = exports.PrivateEndpointConnectionsForSCCPowershell = exports.PrivateEndpointConnectionsForMIPPolicySync = exports.PrivateEndpointConnectionsForEDM = exports.PrivateEndpointConnectionsComp = exports.PrivateEndpointConnectionsAdtAPI = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20210325preview_1 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_2 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_3 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_4 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_5 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_6 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_7 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_8 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_9 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_10 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_11 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
const v20210325preview_12 = require("@pulumi/azure-native/m365securityandcompliance/v20210325preview");
var PrivateEndpointConnectionsAdtAPI;
(function (PrivateEndpointConnectionsAdtAPI) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsAdtAPI).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnectionsAdtAPI.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privateendpointconnectionsadtapi-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsAdtAPI).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_1.PrivateEndpointConnectionsAdtAPI, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnectionsAdtAPI shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsAdtAPI). A compatible replacement can be found at 'm365securityandcompliance.PrivateEndpointConnectionsAdtAPI'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnectionsAdtAPI || (PrivateEndpointConnectionsAdtAPI = {}));
exports.PrivateEndpointConnectionsAdtAPI = PrivateEndpointConnectionsAdtAPI;
var PrivateEndpointConnectionsComp;
(function (PrivateEndpointConnectionsComp) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsComp).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnectionsComp.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privateendpointconnectionscomp-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsComp).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_2.PrivateEndpointConnectionsComp, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnectionsComp shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsComp). A compatible replacement can be found at 'm365securityandcompliance.PrivateEndpointConnectionsComp'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnectionsComp || (PrivateEndpointConnectionsComp = {}));
exports.PrivateEndpointConnectionsComp = PrivateEndpointConnectionsComp;
var PrivateEndpointConnectionsForEDM;
(function (PrivateEndpointConnectionsForEDM) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsForEDM).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnectionsForEDM.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privateendpointconnectionsforedm-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsForEDM).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_3.PrivateEndpointConnectionsForEDM, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnectionsForEDM shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsForEDM). A compatible replacement can be found at 'm365securityandcompliance.PrivateEndpointConnectionsForEDM'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnectionsForEDM || (PrivateEndpointConnectionsForEDM = {}));
exports.PrivateEndpointConnectionsForEDM = PrivateEndpointConnectionsForEDM;
var PrivateEndpointConnectionsForMIPPolicySync;
(function (PrivateEndpointConnectionsForMIPPolicySync) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsForMIPPolicySync).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnectionsForMIPPolicySync.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privateendpointconnectionsformippolicysync-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsForMIPPolicySync).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_4.PrivateEndpointConnectionsForMIPPolicySync, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnectionsForMIPPolicySync shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsForMIPPolicySync). A compatible replacement can be found at 'm365securityandcompliance.PrivateEndpointConnectionsForMIPPolicySync'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnectionsForMIPPolicySync || (PrivateEndpointConnectionsForMIPPolicySync = {}));
exports.PrivateEndpointConnectionsForMIPPolicySync = PrivateEndpointConnectionsForMIPPolicySync;
var PrivateEndpointConnectionsForSCCPowershell;
(function (PrivateEndpointConnectionsForSCCPowershell) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsForSCCPowershell).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnectionsForSCCPowershell.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privateendpointconnectionsforsccpowershell-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsForSCCPowershell).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_5.PrivateEndpointConnectionsForSCCPowershell, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnectionsForSCCPowershell shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsForSCCPowershell). A compatible replacement can be found at 'm365securityandcompliance.PrivateEndpointConnectionsForSCCPowershell'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnectionsForSCCPowershell || (PrivateEndpointConnectionsForSCCPowershell = {}));
exports.PrivateEndpointConnectionsForSCCPowershell = PrivateEndpointConnectionsForSCCPowershell;
var PrivateEndpointConnectionsSec;
(function (PrivateEndpointConnectionsSec) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsSec).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnectionsSec.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privateendpointconnectionssec-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsSec).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_6.PrivateEndpointConnectionsSec, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnectionsSec shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateEndpointConnectionsSec). A compatible replacement can be found at 'm365securityandcompliance.PrivateEndpointConnectionsSec'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnectionsSec || (PrivateEndpointConnectionsSec = {}));
exports.PrivateEndpointConnectionsSec = PrivateEndpointConnectionsSec;
var PrivateLinkServicesForEDMUpload;
(function (PrivateLinkServicesForEDMUpload) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForEDMUpload).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateLinkServicesForEDMUpload.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privatelinkservicesforedmupload-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForEDMUpload).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_7.PrivateLinkServicesForEDMUpload, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateLinkServicesForEDMUpload shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateLinkServicesForEDMUpload). A compatible replacement can be found at 'm365securityandcompliance.PrivateLinkServicesForEDMUpload'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateLinkServicesForEDMUpload || (PrivateLinkServicesForEDMUpload = {}));
exports.PrivateLinkServicesForEDMUpload = PrivateLinkServicesForEDMUpload;
var PrivateLinkServicesForM365ComplianceCenter;
(function (PrivateLinkServicesForM365ComplianceCenter) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForM365ComplianceCenter).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateLinkServicesForM365ComplianceCenter.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privatelinkservicesform365compliancecenter-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForM365ComplianceCenter).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_8.PrivateLinkServicesForM365ComplianceCenter, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateLinkServicesForM365ComplianceCenter shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateLinkServicesForM365ComplianceCenter). A compatible replacement can be found at 'm365securityandcompliance.PrivateLinkServicesForM365ComplianceCenter'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateLinkServicesForM365ComplianceCenter || (PrivateLinkServicesForM365ComplianceCenter = {}));
exports.PrivateLinkServicesForM365ComplianceCenter = PrivateLinkServicesForM365ComplianceCenter;
var PrivateLinkServicesForM365SecurityCenter;
(function (PrivateLinkServicesForM365SecurityCenter) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForM365SecurityCenter).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateLinkServicesForM365SecurityCenter.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privatelinkservicesform365securitycenter-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForM365SecurityCenter).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_9.PrivateLinkServicesForM365SecurityCenter, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateLinkServicesForM365SecurityCenter shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateLinkServicesForM365SecurityCenter). A compatible replacement can be found at 'm365securityandcompliance.PrivateLinkServicesForM365SecurityCenter'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateLinkServicesForM365SecurityCenter || (PrivateLinkServicesForM365SecurityCenter = {}));
exports.PrivateLinkServicesForM365SecurityCenter = PrivateLinkServicesForM365SecurityCenter;
var PrivateLinkServicesForMIPPolicySync;
(function (PrivateLinkServicesForMIPPolicySync) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForMIPPolicySync).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateLinkServicesForMIPPolicySync.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privatelinkservicesformippolicysync-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForMIPPolicySync).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_10.PrivateLinkServicesForMIPPolicySync, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateLinkServicesForMIPPolicySync shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateLinkServicesForMIPPolicySync). A compatible replacement can be found at 'm365securityandcompliance.PrivateLinkServicesForMIPPolicySync'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateLinkServicesForMIPPolicySync || (PrivateLinkServicesForMIPPolicySync = {}));
exports.PrivateLinkServicesForMIPPolicySync = PrivateLinkServicesForMIPPolicySync;
var PrivateLinkServicesForO365ManagementActivityAPI;
(function (PrivateLinkServicesForO365ManagementActivityAPI) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForO365ManagementActivityAPI).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateLinkServicesForO365ManagementActivityAPI.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privatelinkservicesforo365managementactivityapi-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForO365ManagementActivityAPI).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_11.PrivateLinkServicesForO365ManagementActivityAPI, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateLinkServicesForO365ManagementActivityAPI shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateLinkServicesForO365ManagementActivityAPI). A compatible replacement can be found at 'm365securityandcompliance.PrivateLinkServicesForO365ManagementActivityAPI'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateLinkServicesForO365ManagementActivityAPI || (PrivateLinkServicesForO365ManagementActivityAPI = {}));
exports.PrivateLinkServicesForO365ManagementActivityAPI = PrivateLinkServicesForO365ManagementActivityAPI;
var PrivateLinkServicesForSCCPowershell;
(function (PrivateLinkServicesForSCCPowershell) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForSCCPowershell).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateLinkServicesForSCCPowershell.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-m365securityandcompliance-v20210325preview-privatelinkservicesforsccpowershell-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (m365securityandcompliance.v20210325preview.PrivateLinkServicesForSCCPowershell).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210325preview_12.PrivateLinkServicesForSCCPowershell, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateLinkServicesForSCCPowershell shouldn't use an unstable API (m365securityandcompliance.v20210325preview.PrivateLinkServicesForSCCPowershell). A compatible replacement can be found at 'm365securityandcompliance.PrivateLinkServicesForSCCPowershell'.");
            }),
        },
        vendors: ["azure"],
        services: ["m365securityandcompliance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateLinkServicesForSCCPowershell || (PrivateLinkServicesForSCCPowershell = {}));
exports.PrivateLinkServicesForSCCPowershell = PrivateLinkServicesForSCCPowershell;
