// @pulumi/azure-compliance-policies/azurenative/machinelearningservices/v20230601preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceConnection = exports.Workspace = exports.Schedule = exports.RegistryModelVersion = exports.RegistryModelContainer = exports.RegistryEnvironmentVersion = exports.RegistryEnvironmentContainer = exports.RegistryDataVersion = exports.RegistryDataContainer = exports.RegistryComponentVersion = exports.RegistryComponentContainer = exports.RegistryCodeVersion = exports.RegistryCodeContainer = exports.Registry = exports.PrivateEndpointConnection = exports.OnlineEndpoint = exports.OnlineDeployment = exports.ModelVersion = exports.ModelContainer = exports.ManagedNetworkSettingsRule = exports.LabelingJob = exports.Job = exports.FeaturestoreEntityVersion = exports.FeaturestoreEntityContainerEntity = exports.FeaturesetVersion = exports.FeaturesetContainerEntity = exports.EnvironmentVersion = exports.EnvironmentContainer = exports.Datastore = exports.DataVersion = exports.DataContainer = exports.Compute = exports.ComponentVersion = exports.ComponentContainer = exports.CodeVersion = exports.CodeContainer = exports.BatchEndpoint = exports.BatchDeployment = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230601preview_1 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_2 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_3 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_4 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_5 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_6 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_7 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_8 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_9 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_10 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_11 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_12 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_13 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_14 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_15 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_16 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_17 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_18 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_19 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_20 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_21 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_22 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_23 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_24 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_25 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_26 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_27 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_28 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_29 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_30 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_31 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_32 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_33 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_34 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_35 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_36 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_37 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
const v20230601preview_38 = require("@pulumi/azure-native/machinelearningservices/v20230601preview");
var BatchDeployment;
(function (BatchDeployment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.BatchDeployment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BatchDeployment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-batchdeployment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.BatchDeployment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_1.BatchDeployment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BatchDeployment shouldn't use an unstable API (machinelearningservices.v20230601preview.BatchDeployment). A compatible replacement can be found at 'machinelearningservices.BatchDeployment'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BatchDeployment || (BatchDeployment = {}));
exports.BatchDeployment = BatchDeployment;
var BatchEndpoint;
(function (BatchEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.BatchEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BatchEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-batchendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.BatchEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_2.BatchEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BatchEndpoint shouldn't use an unstable API (machinelearningservices.v20230601preview.BatchEndpoint). A compatible replacement can be found at 'machinelearningservices.BatchEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BatchEndpoint || (BatchEndpoint = {}));
exports.BatchEndpoint = BatchEndpoint;
var CodeContainer;
(function (CodeContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.CodeContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CodeContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-codecontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.CodeContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_3.CodeContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CodeContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.CodeContainer). A compatible replacement can be found at 'machinelearningservices.CodeContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CodeContainer || (CodeContainer = {}));
exports.CodeContainer = CodeContainer;
var CodeVersion;
(function (CodeVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.CodeVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CodeVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-codeversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.CodeVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_4.CodeVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CodeVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.CodeVersion). A compatible replacement can be found at 'machinelearningservices.CodeVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CodeVersion || (CodeVersion = {}));
exports.CodeVersion = CodeVersion;
var ComponentContainer;
(function (ComponentContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ComponentContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ComponentContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-componentcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ComponentContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_5.ComponentContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ComponentContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.ComponentContainer). A compatible replacement can be found at 'machinelearningservices.ComponentContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ComponentContainer || (ComponentContainer = {}));
exports.ComponentContainer = ComponentContainer;
var ComponentVersion;
(function (ComponentVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ComponentVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ComponentVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-componentversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ComponentVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_6.ComponentVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ComponentVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.ComponentVersion). A compatible replacement can be found at 'machinelearningservices.ComponentVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ComponentVersion || (ComponentVersion = {}));
exports.ComponentVersion = ComponentVersion;
var Compute;
(function (Compute) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Compute).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Compute.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-compute-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Compute).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_7.Compute, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Compute shouldn't use an unstable API (machinelearningservices.v20230601preview.Compute). A compatible replacement can be found at 'machinelearningservices.Compute'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Compute || (Compute = {}));
exports.Compute = Compute;
var DataContainer;
(function (DataContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.DataContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DataContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-datacontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.DataContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_8.DataContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DataContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.DataContainer). A compatible replacement can be found at 'machinelearningservices.DataContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DataContainer || (DataContainer = {}));
exports.DataContainer = DataContainer;
var DataVersion;
(function (DataVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.DataVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DataVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-dataversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.DataVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_9.DataVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DataVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.DataVersion). A compatible replacement can be found at 'machinelearningservices.DataVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DataVersion || (DataVersion = {}));
exports.DataVersion = DataVersion;
var Datastore;
(function (Datastore) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Datastore).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Datastore.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-datastore-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Datastore).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_10.Datastore, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Datastore shouldn't use an unstable API (machinelearningservices.v20230601preview.Datastore). A compatible replacement can be found at 'machinelearningservices.Datastore'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Datastore || (Datastore = {}));
exports.Datastore = Datastore;
var EnvironmentContainer;
(function (EnvironmentContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.EnvironmentContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EnvironmentContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-environmentcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.EnvironmentContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_11.EnvironmentContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EnvironmentContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.EnvironmentContainer). A compatible replacement can be found at 'machinelearningservices.EnvironmentContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EnvironmentContainer || (EnvironmentContainer = {}));
exports.EnvironmentContainer = EnvironmentContainer;
var EnvironmentVersion;
(function (EnvironmentVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.EnvironmentVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EnvironmentVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-environmentversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.EnvironmentVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_12.EnvironmentVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EnvironmentVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.EnvironmentVersion). A compatible replacement can be found at 'machinelearningservices.EnvironmentVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EnvironmentVersion || (EnvironmentVersion = {}));
exports.EnvironmentVersion = EnvironmentVersion;
var FeaturesetContainerEntity;
(function (FeaturesetContainerEntity) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.FeaturesetContainerEntity).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FeaturesetContainerEntity.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-featuresetcontainerentity-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.FeaturesetContainerEntity).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_13.FeaturesetContainerEntity, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FeaturesetContainerEntity shouldn't use an unstable API (machinelearningservices.v20230601preview.FeaturesetContainerEntity). A compatible replacement can be found at 'machinelearningservices.FeaturesetContainerEntity'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FeaturesetContainerEntity || (FeaturesetContainerEntity = {}));
exports.FeaturesetContainerEntity = FeaturesetContainerEntity;
var FeaturesetVersion;
(function (FeaturesetVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.FeaturesetVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FeaturesetVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-featuresetversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.FeaturesetVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_14.FeaturesetVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FeaturesetVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.FeaturesetVersion). A compatible replacement can be found at 'machinelearningservices.FeaturesetVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FeaturesetVersion || (FeaturesetVersion = {}));
exports.FeaturesetVersion = FeaturesetVersion;
var FeaturestoreEntityContainerEntity;
(function (FeaturestoreEntityContainerEntity) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.FeaturestoreEntityContainerEntity).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FeaturestoreEntityContainerEntity.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-featurestoreentitycontainerentity-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.FeaturestoreEntityContainerEntity).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_15.FeaturestoreEntityContainerEntity, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FeaturestoreEntityContainerEntity shouldn't use an unstable API (machinelearningservices.v20230601preview.FeaturestoreEntityContainerEntity). A compatible replacement can be found at 'machinelearningservices.FeaturestoreEntityContainerEntity'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FeaturestoreEntityContainerEntity || (FeaturestoreEntityContainerEntity = {}));
exports.FeaturestoreEntityContainerEntity = FeaturestoreEntityContainerEntity;
var FeaturestoreEntityVersion;
(function (FeaturestoreEntityVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.FeaturestoreEntityVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FeaturestoreEntityVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-featurestoreentityversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.FeaturestoreEntityVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_16.FeaturestoreEntityVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FeaturestoreEntityVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.FeaturestoreEntityVersion). A compatible replacement can be found at 'machinelearningservices.FeaturestoreEntityVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FeaturestoreEntityVersion || (FeaturestoreEntityVersion = {}));
exports.FeaturestoreEntityVersion = FeaturestoreEntityVersion;
var Job;
(function (Job) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Job).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Job.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-job-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Job).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_17.Job, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Job shouldn't use an unstable API (machinelearningservices.v20230601preview.Job). A compatible replacement can be found at 'machinelearningservices.Job'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Job || (Job = {}));
exports.Job = Job;
var LabelingJob;
(function (LabelingJob) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.LabelingJob).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LabelingJob.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-labelingjob-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.LabelingJob).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_18.LabelingJob, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LabelingJob shouldn't use an unstable API (machinelearningservices.v20230601preview.LabelingJob). A compatible replacement can be found at 'machinelearningservices.LabelingJob'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LabelingJob || (LabelingJob = {}));
exports.LabelingJob = LabelingJob;
var ManagedNetworkSettingsRule;
(function (ManagedNetworkSettingsRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ManagedNetworkSettingsRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagedNetworkSettingsRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-managednetworksettingsrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ManagedNetworkSettingsRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_19.ManagedNetworkSettingsRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagedNetworkSettingsRule shouldn't use an unstable API (machinelearningservices.v20230601preview.ManagedNetworkSettingsRule). A compatible replacement can be found at 'machinelearningservices.ManagedNetworkSettingsRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagedNetworkSettingsRule || (ManagedNetworkSettingsRule = {}));
exports.ManagedNetworkSettingsRule = ManagedNetworkSettingsRule;
var ModelContainer;
(function (ModelContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ModelContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ModelContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-modelcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ModelContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_20.ModelContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ModelContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.ModelContainer). A compatible replacement can be found at 'machinelearningservices.ModelContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ModelContainer || (ModelContainer = {}));
exports.ModelContainer = ModelContainer;
var ModelVersion;
(function (ModelVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ModelVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ModelVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-modelversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.ModelVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_21.ModelVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ModelVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.ModelVersion). A compatible replacement can be found at 'machinelearningservices.ModelVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ModelVersion || (ModelVersion = {}));
exports.ModelVersion = ModelVersion;
var OnlineDeployment;
(function (OnlineDeployment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.OnlineDeployment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OnlineDeployment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-onlinedeployment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.OnlineDeployment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_22.OnlineDeployment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OnlineDeployment shouldn't use an unstable API (machinelearningservices.v20230601preview.OnlineDeployment). A compatible replacement can be found at 'machinelearningservices.OnlineDeployment'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OnlineDeployment || (OnlineDeployment = {}));
exports.OnlineDeployment = OnlineDeployment;
var OnlineEndpoint;
(function (OnlineEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.OnlineEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OnlineEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-onlineendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.OnlineEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_23.OnlineEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OnlineEndpoint shouldn't use an unstable API (machinelearningservices.v20230601preview.OnlineEndpoint). A compatible replacement can be found at 'machinelearningservices.OnlineEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OnlineEndpoint || (OnlineEndpoint = {}));
exports.OnlineEndpoint = OnlineEndpoint;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_24.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (machinelearningservices.v20230601preview.PrivateEndpointConnection). A compatible replacement can be found at 'machinelearningservices.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
var Registry;
(function (Registry) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Registry).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Registry.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registry-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Registry).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_25.Registry, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Registry shouldn't use an unstable API (machinelearningservices.v20230601preview.Registry). A compatible replacement can be found at 'machinelearningservices.Registry'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Registry || (Registry = {}));
exports.Registry = Registry;
var RegistryCodeContainer;
(function (RegistryCodeContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryCodeContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryCodeContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registrycodecontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryCodeContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_26.RegistryCodeContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryCodeContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryCodeContainer). A compatible replacement can be found at 'machinelearningservices.RegistryCodeContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryCodeContainer || (RegistryCodeContainer = {}));
exports.RegistryCodeContainer = RegistryCodeContainer;
var RegistryCodeVersion;
(function (RegistryCodeVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryCodeVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryCodeVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registrycodeversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryCodeVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_27.RegistryCodeVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryCodeVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryCodeVersion). A compatible replacement can be found at 'machinelearningservices.RegistryCodeVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryCodeVersion || (RegistryCodeVersion = {}));
exports.RegistryCodeVersion = RegistryCodeVersion;
var RegistryComponentContainer;
(function (RegistryComponentContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryComponentContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryComponentContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registrycomponentcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryComponentContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_28.RegistryComponentContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryComponentContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryComponentContainer). A compatible replacement can be found at 'machinelearningservices.RegistryComponentContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryComponentContainer || (RegistryComponentContainer = {}));
exports.RegistryComponentContainer = RegistryComponentContainer;
var RegistryComponentVersion;
(function (RegistryComponentVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryComponentVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryComponentVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registrycomponentversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryComponentVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_29.RegistryComponentVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryComponentVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryComponentVersion). A compatible replacement can be found at 'machinelearningservices.RegistryComponentVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryComponentVersion || (RegistryComponentVersion = {}));
exports.RegistryComponentVersion = RegistryComponentVersion;
var RegistryDataContainer;
(function (RegistryDataContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryDataContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryDataContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registrydatacontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryDataContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_30.RegistryDataContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryDataContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryDataContainer). A compatible replacement can be found at 'machinelearningservices.RegistryDataContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryDataContainer || (RegistryDataContainer = {}));
exports.RegistryDataContainer = RegistryDataContainer;
var RegistryDataVersion;
(function (RegistryDataVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryDataVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryDataVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registrydataversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryDataVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_31.RegistryDataVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryDataVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryDataVersion). A compatible replacement can be found at 'machinelearningservices.RegistryDataVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryDataVersion || (RegistryDataVersion = {}));
exports.RegistryDataVersion = RegistryDataVersion;
var RegistryEnvironmentContainer;
(function (RegistryEnvironmentContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryEnvironmentContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryEnvironmentContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registryenvironmentcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryEnvironmentContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_32.RegistryEnvironmentContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryEnvironmentContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryEnvironmentContainer). A compatible replacement can be found at 'machinelearningservices.RegistryEnvironmentContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryEnvironmentContainer || (RegistryEnvironmentContainer = {}));
exports.RegistryEnvironmentContainer = RegistryEnvironmentContainer;
var RegistryEnvironmentVersion;
(function (RegistryEnvironmentVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryEnvironmentVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryEnvironmentVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registryenvironmentversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryEnvironmentVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_33.RegistryEnvironmentVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryEnvironmentVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryEnvironmentVersion). A compatible replacement can be found at 'machinelearningservices.RegistryEnvironmentVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryEnvironmentVersion || (RegistryEnvironmentVersion = {}));
exports.RegistryEnvironmentVersion = RegistryEnvironmentVersion;
var RegistryModelContainer;
(function (RegistryModelContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryModelContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryModelContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registrymodelcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryModelContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_34.RegistryModelContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryModelContainer shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryModelContainer). A compatible replacement can be found at 'machinelearningservices.RegistryModelContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryModelContainer || (RegistryModelContainer = {}));
exports.RegistryModelContainer = RegistryModelContainer;
var RegistryModelVersion;
(function (RegistryModelVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryModelVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryModelVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-registrymodelversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.RegistryModelVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_35.RegistryModelVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryModelVersion shouldn't use an unstable API (machinelearningservices.v20230601preview.RegistryModelVersion). A compatible replacement can be found at 'machinelearningservices.RegistryModelVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryModelVersion || (RegistryModelVersion = {}));
exports.RegistryModelVersion = RegistryModelVersion;
var Schedule;
(function (Schedule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Schedule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Schedule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-schedule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Schedule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_36.Schedule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Schedule shouldn't use an unstable API (machinelearningservices.v20230601preview.Schedule). A compatible replacement can be found at 'machinelearningservices.Schedule'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Schedule || (Schedule = {}));
exports.Schedule = Schedule;
var Workspace;
(function (Workspace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Workspace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Workspace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-workspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.Workspace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_37.Workspace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Workspace shouldn't use an unstable API (machinelearningservices.v20230601preview.Workspace). A compatible replacement can be found at 'machinelearningservices.Workspace'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Workspace || (Workspace = {}));
exports.Workspace = Workspace;
var WorkspaceConnection;
(function (WorkspaceConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.WorkspaceConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20230601preview-workspaceconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20230601preview.WorkspaceConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230601preview_38.WorkspaceConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceConnection shouldn't use an unstable API (machinelearningservices.v20230601preview.WorkspaceConnection). A compatible replacement can be found at 'machinelearningservices.WorkspaceConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceConnection || (WorkspaceConnection = {}));
exports.WorkspaceConnection = WorkspaceConnection;
