// @pulumi/azure/operationalinsights/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryPackQuery = exports.getAnalyticsWorkspaceOutput = exports.getAnalyticsWorkspace = exports.AnalyticsWorkspace = exports.AnalyticsSolution = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AnalyticsSolution = null;
utilities.lazyLoad(exports, ["AnalyticsSolution"], () => require("./analyticsSolution"));
exports.AnalyticsWorkspace = null;
utilities.lazyLoad(exports, ["AnalyticsWorkspace"], () => require("./analyticsWorkspace"));
exports.getAnalyticsWorkspace = null;
exports.getAnalyticsWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getAnalyticsWorkspace", "getAnalyticsWorkspaceOutput"], () => require("./getAnalyticsWorkspace"));
exports.QueryPackQuery = null;
utilities.lazyLoad(exports, ["QueryPackQuery"], () => require("./queryPackQuery"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:operationalinsights/analyticsSolution:AnalyticsSolution":
                return new exports.AnalyticsSolution(name, undefined, { urn });
            case "azure:operationalinsights/analyticsWorkspace:AnalyticsWorkspace":
                return new exports.AnalyticsWorkspace(name, undefined, { urn });
            case "azure:operationalinsights/queryPackQuery:QueryPackQuery":
                return new exports.QueryPackQuery(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "operationalinsights/analyticsSolution", _module);
pulumi.runtime.registerResourceModule("azure", "operationalinsights/analyticsWorkspace", _module);
pulumi.runtime.registerResourceModule("azure", "operationalinsights/queryPackQuery", _module);
