// @pulumi/azure-compliance-policies/azurenative/streamanalytics/v20200301preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpoint = exports.Cluster = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20200301preview_1 = require("@pulumi/azure-native/streamanalytics/v20200301preview");
const v20200301preview_2 = require("@pulumi/azure-native/streamanalytics/v20200301preview");
var Cluster;
(function (Cluster) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (streamanalytics.v20200301preview.Cluster).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Cluster.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-streamanalytics-v20200301preview-cluster-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (streamanalytics.v20200301preview.Cluster).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200301preview_1.Cluster, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Cluster shouldn't use an unstable API (streamanalytics.v20200301preview.Cluster). A compatible replacement can be found at 'streamanalytics.Cluster'.");
            }),
        },
        vendors: ["azure"],
        services: ["streamanalytics"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Cluster || (Cluster = {}));
exports.Cluster = Cluster;
var PrivateEndpoint;
(function (PrivateEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (streamanalytics.v20200301preview.PrivateEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-streamanalytics-v20200301preview-privateendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (streamanalytics.v20200301preview.PrivateEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200301preview_2.PrivateEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpoint shouldn't use an unstable API (streamanalytics.v20200301preview.PrivateEndpoint). A compatible replacement can be found at 'streamanalytics.PrivateEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["streamanalytics"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpoint || (PrivateEndpoint = {}));
exports.PrivateEndpoint = PrivateEndpoint;
