// @pulumi/azure/nginx/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDeploymentOutput = exports.getDeployment = exports.getConfigurationOutput = exports.getConfiguration = exports.getCertificateOutput = exports.getCertificate = exports.Deployment = exports.Configuration = exports.Certificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.Configuration = null;
utilities.lazyLoad(exports, ["Configuration"], () => require("./configuration"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getConfiguration = null;
exports.getConfigurationOutput = null;
utilities.lazyLoad(exports, ["getConfiguration", "getConfigurationOutput"], () => require("./getConfiguration"));
exports.getDeployment = null;
exports.getDeploymentOutput = null;
utilities.lazyLoad(exports, ["getDeployment", "getDeploymentOutput"], () => require("./getDeployment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:nginx/certificate:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "azure:nginx/configuration:Configuration":
                return new exports.Configuration(name, undefined, { urn });
            case "azure:nginx/deployment:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "nginx/certificate", _module);
pulumi.runtime.registerResourceModule("azure", "nginx/configuration", _module);
pulumi.runtime.registerResourceModule("azure", "nginx/deployment", _module);
