// @pulumi/azure/signalr/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedPrivateLinkResource = exports.ServiceNetworkAcl = exports.ServiceCustomDomain = exports.ServiceCustomCertificate = exports.Service = exports.getServiceOutput = exports.getService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceCustomCertificate = null;
utilities.lazyLoad(exports, ["ServiceCustomCertificate"], () => require("./serviceCustomCertificate"));
exports.ServiceCustomDomain = null;
utilities.lazyLoad(exports, ["ServiceCustomDomain"], () => require("./serviceCustomDomain"));
exports.ServiceNetworkAcl = null;
utilities.lazyLoad(exports, ["ServiceNetworkAcl"], () => require("./serviceNetworkAcl"));
exports.SharedPrivateLinkResource = null;
utilities.lazyLoad(exports, ["SharedPrivateLinkResource"], () => require("./sharedPrivateLinkResource"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:signalr/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure:signalr/serviceCustomCertificate:ServiceCustomCertificate":
                return new exports.ServiceCustomCertificate(name, undefined, { urn });
            case "azure:signalr/serviceCustomDomain:ServiceCustomDomain":
                return new exports.ServiceCustomDomain(name, undefined, { urn });
            case "azure:signalr/serviceNetworkAcl:ServiceNetworkAcl":
                return new exports.ServiceNetworkAcl(name, undefined, { urn });
            case "azure:signalr/sharedPrivateLinkResource:SharedPrivateLinkResource":
                return new exports.SharedPrivateLinkResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "signalr/service", _module);
pulumi.runtime.registerResourceModule("azure", "signalr/serviceCustomCertificate", _module);
pulumi.runtime.registerResourceModule("azure", "signalr/serviceCustomDomain", _module);
pulumi.runtime.registerResourceModule("azure", "signalr/serviceNetworkAcl", _module);
pulumi.runtime.registerResourceModule("azure", "signalr/sharedPrivateLinkResource", _module);
