// @pulumi/azure/signalr/sharedPrivateLinkResource.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedPrivateLinkResource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages the Shared Private Link Resource for a Signalr service.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const current = azure.core.getClientConfig({});
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "terraform-signalr",
 *     location: "east us",
 * });
 * const exampleKeyVault = new azure.keyvault.KeyVault("example", {
 *     name: "examplekeyvault",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     tenantId: current.then(current => current.tenantId),
 *     skuName: "standard",
 *     softDeleteRetentionDays: 7,
 *     accessPolicies: [{
 *         tenantId: current.then(current => current.tenantId),
 *         objectId: current.then(current => current.objectId),
 *         certificatePermissions: ["ManageContacts"],
 *         keyPermissions: ["Create"],
 *         secretPermissions: ["Set"],
 *     }],
 * });
 * const test = new azure.signalr.Service("test", {
 *     name: "tfex-signalr",
 *     location: testAzurermResourceGroup.location,
 *     resourceGroupName: testAzurermResourceGroup.name,
 *     sku: {
 *         name: "Standard_S1",
 *         capacity: 1,
 *     },
 * });
 * const exampleSharedPrivateLinkResource = new azure.signalr.SharedPrivateLinkResource("example", {
 *     name: "tfex-signalr-splr",
 *     signalrServiceId: exampleAzurermSignalrService.id,
 *     subResourceName: "vault",
 *     targetResourceId: exampleKeyVault.id,
 * });
 * ```
 *
 * ## Import
 *
 * Signalr Shared Private Link Resource can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:signalr/sharedPrivateLinkResource:SharedPrivateLinkResource example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.SignalRService/signalR/signalr1/sharedPrivateLinkResources/resource1
 * ```
 */
class SharedPrivateLinkResource extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["requestMessage"] = state ? state.requestMessage : undefined;
            resourceInputs["signalrServiceId"] = state ? state.signalrServiceId : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["subResourceName"] = state ? state.subResourceName : undefined;
            resourceInputs["targetResourceId"] = state ? state.targetResourceId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.signalrServiceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'signalrServiceId'");
            }
            if ((!args || args.subResourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subResourceName'");
            }
            if ((!args || args.targetResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetResourceId'");
            }
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["requestMessage"] = args ? args.requestMessage : undefined;
            resourceInputs["signalrServiceId"] = args ? args.signalrServiceId : undefined;
            resourceInputs["subResourceName"] = args ? args.subResourceName : undefined;
            resourceInputs["targetResourceId"] = args ? args.targetResourceId : undefined;
            resourceInputs["status"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SharedPrivateLinkResource.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SharedPrivateLinkResource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SharedPrivateLinkResource(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SharedPrivateLinkResource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SharedPrivateLinkResource.__pulumiType;
    }
}
exports.SharedPrivateLinkResource = SharedPrivateLinkResource;
/** @internal */
SharedPrivateLinkResource.__pulumiType = 'azure:signalr/sharedPrivateLinkResource:SharedPrivateLinkResource';
