// @pulumi/azure/mobile/networkService.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Mobile Network Service.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "east us",
 * });
 * const exampleNetwork = new azure.mobile.Network("example", {
 *     name: "example-mn",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     mobileCountryCode: "001",
 *     mobileNetworkCode: "01",
 * });
 * const exampleNetworkService = new azure.mobile.NetworkService("example", {
 *     name: "example-mns",
 *     mobileNetworkId: exampleNetwork.id,
 *     location: example.location,
 *     servicePrecedence: 0,
 *     pccRules: [{
 *         name: "default-rule",
 *         precedence: 1,
 *         trafficControlEnabled: true,
 *         qosPolicy: {
 *             allocationAndRetentionPriorityLevel: 9,
 *             qosIndicator: 9,
 *             preemptionCapability: "NotPreempt",
 *             preemptionVulnerability: "Preemptable",
 *             guaranteedBitRate: {
 *                 downlink: "100 Mbps",
 *                 uplink: "10 Mbps",
 *             },
 *             maximumBitRate: {
 *                 downlink: "1 Gbps",
 *                 uplink: "100 Mbps",
 *             },
 *         },
 *         serviceDataFlowTemplates: [{
 *             direction: "Uplink",
 *             name: "IP-to-server",
 *             ports: [],
 *             protocols: ["ip"],
 *             remoteIpLists: ["10.3.4.0/24"],
 *         }],
 *     }],
 *     serviceQosPolicy: {
 *         allocationAndRetentionPriorityLevel: 9,
 *         qosIndicator: 9,
 *         preemptionCapability: "NotPreempt",
 *         preemptionVulnerability: "Preemptable",
 *         maximumBitRate: {
 *             downlink: "1 Gbps",
 *             uplink: "100 Mbps",
 *         },
 *     },
 *     tags: {
 *         key: "value",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Mobile Network Service can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:mobile/networkService:NetworkService example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.MobileNetwork/mobileNetworks/mobileNetwork1/services/service1
 * ```
 */
class NetworkService extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["mobileNetworkId"] = state ? state.mobileNetworkId : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["pccRules"] = state ? state.pccRules : undefined;
            resourceInputs["servicePrecedence"] = state ? state.servicePrecedence : undefined;
            resourceInputs["serviceQosPolicy"] = state ? state.serviceQosPolicy : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.mobileNetworkId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mobileNetworkId'");
            }
            if ((!args || args.pccRules === undefined) && !opts.urn) {
                throw new Error("Missing required property 'pccRules'");
            }
            if ((!args || args.servicePrecedence === undefined) && !opts.urn) {
                throw new Error("Missing required property 'servicePrecedence'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["mobileNetworkId"] = args ? args.mobileNetworkId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["pccRules"] = args ? args.pccRules : undefined;
            resourceInputs["servicePrecedence"] = args ? args.servicePrecedence : undefined;
            resourceInputs["serviceQosPolicy"] = args ? args.serviceQosPolicy : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkService.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NetworkService resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NetworkService(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkService.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkService.__pulumiType;
    }
}
exports.NetworkService = NetworkService;
/** @internal */
NetworkService.__pulumiType = 'azure:mobile/networkService:NetworkService';
