// @pulumi/azure/mobile/networkPacketCoreDataPlane.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkPacketCoreDataPlane = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Mobile Network Packet Core Data Plane.
 *
 * ## Import
 *
 * Mobile Network Packet Core Data Plane can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:mobile/networkPacketCoreDataPlane:NetworkPacketCoreDataPlane example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.MobileNetwork/packetCoreControlPlanes/packetCoreControlPlane1/packetCoreDataPlanes/packetCoreDataPlane1
 * ```
 */
class NetworkPacketCoreDataPlane extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["mobileNetworkPacketCoreControlPlaneId"] = state ? state.mobileNetworkPacketCoreControlPlaneId : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["userPlaneAccessIpv4Address"] = state ? state.userPlaneAccessIpv4Address : undefined;
            resourceInputs["userPlaneAccessIpv4Gateway"] = state ? state.userPlaneAccessIpv4Gateway : undefined;
            resourceInputs["userPlaneAccessIpv4Subnet"] = state ? state.userPlaneAccessIpv4Subnet : undefined;
            resourceInputs["userPlaneAccessName"] = state ? state.userPlaneAccessName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.mobileNetworkPacketCoreControlPlaneId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mobileNetworkPacketCoreControlPlaneId'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["mobileNetworkPacketCoreControlPlaneId"] = args ? args.mobileNetworkPacketCoreControlPlaneId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["userPlaneAccessIpv4Address"] = args ? args.userPlaneAccessIpv4Address : undefined;
            resourceInputs["userPlaneAccessIpv4Gateway"] = args ? args.userPlaneAccessIpv4Gateway : undefined;
            resourceInputs["userPlaneAccessIpv4Subnet"] = args ? args.userPlaneAccessIpv4Subnet : undefined;
            resourceInputs["userPlaneAccessName"] = args ? args.userPlaneAccessName : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkPacketCoreDataPlane.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NetworkPacketCoreDataPlane resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NetworkPacketCoreDataPlane(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkPacketCoreDataPlane.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkPacketCoreDataPlane.__pulumiType;
    }
}
exports.NetworkPacketCoreDataPlane = NetworkPacketCoreDataPlane;
/** @internal */
NetworkPacketCoreDataPlane.__pulumiType = 'azure:mobile/networkPacketCoreDataPlane:NetworkPacketCoreDataPlane';
