// @pulumi/azure-native/devcenter/devBoxDefinition.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DevBoxDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a definition for a Developer Machine.
 * Azure REST API version: 2023-04-01. Prior API version in Azure Native 1.x: 2022-09-01-preview.
 *
 * Other available API versions: 2022-11-11-preview, 2023-08-01-preview, 2023-10-01-preview, 2024-02-01, 2024-05-01-preview, 2024-06-01-preview, 2024-07-01-preview, 2024-08-01-preview, 2024-10-01-preview.
 */
class DevBoxDefinition extends pulumi.CustomResource {
    /**
     * Get an existing DevBoxDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DevBoxDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DevBoxDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DevBoxDefinition.__pulumiType;
    }
    /**
     * Create a DevBoxDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.devCenterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'devCenterName'");
            }
            if ((!args || args.imageReference === undefined) && !opts.urn) {
                throw new Error("Missing required property 'imageReference'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            resourceInputs["devBoxDefinitionName"] = args ? args.devBoxDefinitionName : undefined;
            resourceInputs["devCenterName"] = args ? args.devCenterName : undefined;
            resourceInputs["hibernateSupport"] = args ? args.hibernateSupport : undefined;
            resourceInputs["imageReference"] = args ? args.imageReference : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["osStorageType"] = args ? args.osStorageType : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["activeImageReference"] = undefined /*out*/;
            resourceInputs["imageValidationErrorDetails"] = undefined /*out*/;
            resourceInputs["imageValidationStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["activeImageReference"] = undefined /*out*/;
            resourceInputs["hibernateSupport"] = undefined /*out*/;
            resourceInputs["imageReference"] = undefined /*out*/;
            resourceInputs["imageValidationErrorDetails"] = undefined /*out*/;
            resourceInputs["imageValidationStatus"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["osStorageType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devcenter/v20220801preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20220901preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20221012preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20221111preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20230101preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20230401:DevBoxDefinition" }, { type: "azure-native:devcenter/v20230801preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20231001preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20240201:DevBoxDefinition" }, { type: "azure-native:devcenter/v20240501preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20240601preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20240701preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20240801preview:DevBoxDefinition" }, { type: "azure-native:devcenter/v20241001preview:DevBoxDefinition" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DevBoxDefinition.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DevBoxDefinition = DevBoxDefinition;
/** @internal */
DevBoxDefinition.__pulumiType = 'azure-native:devcenter:DevBoxDefinition';
