// @pulumi/azure-compliance-policies/azurenative/standbypool/v20240301preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.StandbyVirtualMachinePool = exports.StandbyContainerGroupPool = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240301preview_1 = require("@pulumi/azure-native/standbypool/v20240301preview");
const v20240301preview_2 = require("@pulumi/azure-native/standbypool/v20240301preview");
var StandbyContainerGroupPool;
(function (StandbyContainerGroupPool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (standbypool.v20240301preview.StandbyContainerGroupPool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    StandbyContainerGroupPool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-standbypool-v20240301preview-standbycontainergrouppool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (standbypool.v20240301preview.StandbyContainerGroupPool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240301preview_1.StandbyContainerGroupPool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure StandbyContainerGroupPool shouldn't use an unstable API (standbypool.v20240301preview.StandbyContainerGroupPool). A compatible replacement can be found at 'standbypool.StandbyContainerGroupPool'.");
            }),
        },
        vendors: ["azure"],
        services: ["standbypool"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(StandbyContainerGroupPool || (StandbyContainerGroupPool = {}));
exports.StandbyContainerGroupPool = StandbyContainerGroupPool;
var StandbyVirtualMachinePool;
(function (StandbyVirtualMachinePool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (standbypool.v20240301preview.StandbyVirtualMachinePool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    StandbyVirtualMachinePool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-standbypool-v20240301preview-standbyvirtualmachinepool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (standbypool.v20240301preview.StandbyVirtualMachinePool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240301preview_2.StandbyVirtualMachinePool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure StandbyVirtualMachinePool shouldn't use an unstable API (standbypool.v20240301preview.StandbyVirtualMachinePool). A compatible replacement can be found at 'standbypool.StandbyVirtualMachinePool'.");
            }),
        },
        vendors: ["azure"],
        services: ["standbypool"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(StandbyVirtualMachinePool || (StandbyVirtualMachinePool = {}));
exports.StandbyVirtualMachinePool = StandbyVirtualMachinePool;
