// @pulumi/azure/network/expressRoutePortAuthorization.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressRoutePortAuthorization = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an ExpressRoute Port Authorization.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "exprtTest",
 *     location: "West Europe",
 * });
 * const exampleExpressRoutePort = new azure.network.ExpressRoutePort("example", {
 *     name: "port1",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     peeringLocation: "Airtel-Chennai-CLS",
 *     bandwidthInGbps: 10,
 *     encapsulation: "Dot1Q",
 * });
 * const exampleExpressRoutePortAuthorization = new azure.network.ExpressRoutePortAuthorization("example", {
 *     name: "exampleERCAuth",
 *     expressRoutePortName: exampleExpressRoutePort.name,
 *     resourceGroupName: example.name,
 * });
 * ```
 *
 * ## Import
 *
 * ExpressRoute Port Authorizations can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:network/expressRoutePortAuthorization:ExpressRoutePortAuthorization auth1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Network/expressRoutePorts/myExpressPort/authorizations/auth1
 * ```
 */
class ExpressRoutePortAuthorization extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["authorizationKey"] = state ? state.authorizationKey : undefined;
            resourceInputs["authorizationUseStatus"] = state ? state.authorizationUseStatus : undefined;
            resourceInputs["expressRoutePortName"] = state ? state.expressRoutePortName : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.expressRoutePortName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'expressRoutePortName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["expressRoutePortName"] = args ? args.expressRoutePortName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["authorizationKey"] = undefined /*out*/;
            resourceInputs["authorizationUseStatus"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["authorizationKey"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(ExpressRoutePortAuthorization.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ExpressRoutePortAuthorization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ExpressRoutePortAuthorization(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExpressRoutePortAuthorization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExpressRoutePortAuthorization.__pulumiType;
    }
}
exports.ExpressRoutePortAuthorization = ExpressRoutePortAuthorization;
/** @internal */
ExpressRoutePortAuthorization.__pulumiType = 'azure:network/expressRoutePortAuthorization:ExpressRoutePortAuthorization';
