// @pulumi/azure/network/networkManagerStaticMember.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkManagerStaticMember = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Network Manager Static Member.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const current = azure.core.getSubscription({});
 * const exampleNetworkManager = new azure.network.NetworkManager("example", {
 *     name: "example-network-manager",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     scope: {
 *         subscriptionIds: [current.then(current => current.id)],
 *     },
 *     scopeAccesses: [
 *         "Connectivity",
 *         "SecurityAdmin",
 *     ],
 *     description: "example network manager",
 * });
 * const exampleNetworkManagerNetworkGroup = new azure.network.NetworkManagerNetworkGroup("example", {
 *     name: "example-group",
 *     networkManagerId: exampleNetworkManager.id,
 *     description: "example network group",
 * });
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("example", {
 *     name: "example-vnet",
 *     resourceGroupName: example.name,
 *     addressSpaces: ["192.168.1.0/24"],
 *     location: example.location,
 * });
 * const exampleNetworkManagerStaticMember = new azure.network.NetworkManagerStaticMember("example", {
 *     name: "example-nmsm",
 *     networkGroupId: exampleNetworkManagerNetworkGroup.id,
 *     targetVirtualNetworkId: exampleVirtualNetwork.id,
 * });
 * ```
 *
 * ## Import
 *
 * Network Manager Static Member can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:network/networkManagerStaticMember:NetworkManagerStaticMember example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.Network/networkManagers/networkManager1/networkGroups/networkGroup1/staticMembers/staticMember1
 * ```
 */
class NetworkManagerStaticMember extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["networkGroupId"] = state ? state.networkGroupId : undefined;
            resourceInputs["region"] = state ? state.region : undefined;
            resourceInputs["targetVirtualNetworkId"] = state ? state.targetVirtualNetworkId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.networkGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkGroupId'");
            }
            if ((!args || args.targetVirtualNetworkId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetVirtualNetworkId'");
            }
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["networkGroupId"] = args ? args.networkGroupId : undefined;
            resourceInputs["targetVirtualNetworkId"] = args ? args.targetVirtualNetworkId : undefined;
            resourceInputs["region"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkManagerStaticMember.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NetworkManagerStaticMember resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NetworkManagerStaticMember(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkManagerStaticMember.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkManagerStaticMember.__pulumiType;
    }
}
exports.NetworkManagerStaticMember = NetworkManagerStaticMember;
/** @internal */
NetworkManagerStaticMember.__pulumiType = 'azure:network/networkManagerStaticMember:NetworkManagerStaticMember';
