// @pulumi/azure/network/networkManagerConnectivityConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkManagerConnectivityConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Network Manager Connectivity Configuration.
 *
 * > **Note:** The `azure.network.NetworkManagerConnectivityConfiguration` deployment may modify or delete existing Network Peering resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const current = azure.core.getSubscription({});
 * const exampleNetworkManager = new azure.network.NetworkManager("example", {
 *     name: "example-network-manager",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     scope: {
 *         subscriptionIds: [current.then(current => current.id)],
 *     },
 *     scopeAccesses: [
 *         "Connectivity",
 *         "SecurityAdmin",
 *     ],
 *     description: "example network manager",
 * });
 * const exampleNetworkManagerNetworkGroup = new azure.network.NetworkManagerNetworkGroup("example", {
 *     name: "example-group",
 *     networkManagerId: exampleNetworkManager.id,
 * });
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("example", {
 *     name: "example-net",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     addressSpaces: ["10.0.0.0/16"],
 *     flowTimeoutInMinutes: 10,
 * });
 * const example2 = new azure.network.NetworkManagerNetworkGroup("example2", {
 *     name: "example-group2",
 *     networkManagerId: exampleNetworkManager.id,
 * });
 * const exampleNetworkManagerConnectivityConfiguration = new azure.network.NetworkManagerConnectivityConfiguration("example", {
 *     name: "example-connectivity-conf",
 *     networkManagerId: exampleNetworkManager.id,
 *     connectivityTopology: "HubAndSpoke",
 *     appliesToGroups: [
 *         {
 *             groupConnectivity: "DirectlyConnected",
 *             networkGroupId: exampleNetworkManagerNetworkGroup.id,
 *         },
 *         {
 *             groupConnectivity: "DirectlyConnected",
 *             networkGroupId: example2.id,
 *         },
 *     ],
 *     hub: {
 *         resourceId: exampleVirtualNetwork.id,
 *         resourceType: "Microsoft.Network/virtualNetworks",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Network Manager Connectivity Configuration can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:network/networkManagerConnectivityConfiguration:NetworkManagerConnectivityConfiguration example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.Network/networkManagers/networkManager1/connectivityConfigurations/configuration1
 * ```
 */
class NetworkManagerConnectivityConfiguration extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appliesToGroups"] = state ? state.appliesToGroups : undefined;
            resourceInputs["connectivityTopology"] = state ? state.connectivityTopology : undefined;
            resourceInputs["deleteExistingPeeringEnabled"] = state ? state.deleteExistingPeeringEnabled : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["globalMeshEnabled"] = state ? state.globalMeshEnabled : undefined;
            resourceInputs["hub"] = state ? state.hub : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["networkManagerId"] = state ? state.networkManagerId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.appliesToGroups === undefined) && !opts.urn) {
                throw new Error("Missing required property 'appliesToGroups'");
            }
            if ((!args || args.connectivityTopology === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectivityTopology'");
            }
            if ((!args || args.networkManagerId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkManagerId'");
            }
            resourceInputs["appliesToGroups"] = args ? args.appliesToGroups : undefined;
            resourceInputs["connectivityTopology"] = args ? args.connectivityTopology : undefined;
            resourceInputs["deleteExistingPeeringEnabled"] = args ? args.deleteExistingPeeringEnabled : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["globalMeshEnabled"] = args ? args.globalMeshEnabled : undefined;
            resourceInputs["hub"] = args ? args.hub : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["networkManagerId"] = args ? args.networkManagerId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkManagerConnectivityConfiguration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NetworkManagerConnectivityConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NetworkManagerConnectivityConfiguration(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkManagerConnectivityConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkManagerConnectivityConfiguration.__pulumiType;
    }
}
exports.NetworkManagerConnectivityConfiguration = NetworkManagerConnectivityConfiguration;
/** @internal */
NetworkManagerConnectivityConfiguration.__pulumiType = 'azure:network/networkManagerConnectivityConfiguration:NetworkManagerConnectivityConfiguration';
