// @pulumi/azure/network/routeServerBgpConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RouteServerBgpConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Bgp Connection for a Route Server
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("example", {
 *     name: "example-vn",
 *     addressSpaces: ["10.0.0.0/16"],
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     tags: {
 *         environment: "Production",
 *     },
 * });
 * const exampleSubnet = new azure.network.Subnet("example", {
 *     name: "RouteServerSubnet",
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     resourceGroupName: example.name,
 *     addressPrefixes: ["10.0.1.0/24"],
 * });
 * const examplePublicIp = new azure.network.PublicIp("example", {
 *     name: "example-pip",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     allocationMethod: "Static",
 *     sku: "Standard",
 * });
 * const exampleRouteServer = new azure.network.RouteServer("example", {
 *     name: "example-routerserver",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     sku: "Standard",
 *     publicIpAddressId: examplePublicIp.id,
 *     subnetId: exampleSubnet.id,
 *     branchToBranchTrafficEnabled: true,
 * });
 * const exampleRouteServerBgpConnection = new azure.network.RouteServerBgpConnection("example", {
 *     name: "example-rs-bgpconnection",
 *     routeServerId: exampleRouteServer.id,
 *     peerAsn: 65501,
 *     peerIp: "169.254.21.5",
 * });
 * ```
 *
 * ## Import
 *
 * Route Server Bgp Connections can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:network/routeServerBgpConnection:RouteServerBgpConnection example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Network/virtualHubs/routeServer1/bgpConnections/connection1
 * ```
 */
class RouteServerBgpConnection extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["peerAsn"] = state ? state.peerAsn : undefined;
            resourceInputs["peerIp"] = state ? state.peerIp : undefined;
            resourceInputs["routeServerId"] = state ? state.routeServerId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.peerAsn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peerAsn'");
            }
            if ((!args || args.peerIp === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peerIp'");
            }
            if ((!args || args.routeServerId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'routeServerId'");
            }
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["peerAsn"] = args ? args.peerAsn : undefined;
            resourceInputs["peerIp"] = args ? args.peerIp : undefined;
            resourceInputs["routeServerId"] = args ? args.routeServerId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RouteServerBgpConnection.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing RouteServerBgpConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RouteServerBgpConnection(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RouteServerBgpConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RouteServerBgpConnection.__pulumiType;
    }
}
exports.RouteServerBgpConnection = RouteServerBgpConnection;
/** @internal */
RouteServerBgpConnection.__pulumiType = 'azure:network/routeServerBgpConnection:RouteServerBgpConnection';
