// @pulumi/azure/netapp/volumeGroupSapHana.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeGroupSapHana = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as random from "@pulumi/random";
 *
 * const example = new random.RandomString("example", {
 *     length: 12,
 *     special: true,
 * });
 * const adminUsername = "exampleadmin";
 * const adminPassword = example.result;
 * const exampleResourceGroup = new azure.core.ResourceGroup("example", {
 *     name: `${prefix}-resources`,
 *     location: location,
 * });
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("example", {
 *     name: `${prefix}-vnet`,
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     addressSpaces: ["10.6.0.0/16"],
 * });
 * const exampleSubnet = new azure.network.Subnet("example", {
 *     name: `${prefix}-delegated-subnet`,
 *     resourceGroupName: exampleResourceGroup.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.6.2.0/24"],
 *     delegations: [{
 *         name: "testdelegation",
 *         serviceDelegation: {
 *             name: "Microsoft.Netapp/volumes",
 *             actions: [
 *                 "Microsoft.Network/networkinterfaces/*",
 *                 "Microsoft.Network/virtualNetworks/subnets/join/action",
 *             ],
 *         },
 *     }],
 * });
 * const example1 = new azure.network.Subnet("example1", {
 *     name: `${prefix}-hosts-subnet`,
 *     resourceGroupName: exampleResourceGroup.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.6.1.0/24"],
 * });
 * const examplePlacementGroup = new azure.proximity.PlacementGroup("example", {
 *     name: `${prefix}-ppg`,
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 * });
 * const exampleAvailabilitySet = new azure.compute.AvailabilitySet("example", {
 *     name: `${prefix}-avset`,
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     proximityPlacementGroupId: examplePlacementGroup.id,
 * });
 * const exampleNetworkInterface = new azure.network.NetworkInterface("example", {
 *     name: `${prefix}-nic`,
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     ipConfigurations: [{
 *         name: "internal",
 *         subnetId: example1.id,
 *         privateIpAddressAllocation: "Dynamic",
 *     }],
 * });
 * const exampleLinuxVirtualMachine = new azure.compute.LinuxVirtualMachine("example", {
 *     name: `${prefix}-vm`,
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     size: "Standard_M8ms",
 *     adminUsername: adminUsername,
 *     adminPassword: adminPassword,
 *     disablePasswordAuthentication: false,
 *     proximityPlacementGroupId: examplePlacementGroup.id,
 *     availabilitySetId: exampleAvailabilitySet.id,
 *     networkInterfaceIds: [exampleNetworkInterface.id],
 *     sourceImageReference: {
 *         publisher: "Canonical",
 *         offer: "0001-com-ubuntu-server-jammy",
 *         sku: "22_04-lts",
 *         version: "latest",
 *     },
 *     osDisk: {
 *         storageAccountType: "Standard_LRS",
 *         caching: "ReadWrite",
 *     },
 * });
 * const exampleAccount = new azure.netapp.Account("example", {
 *     name: `${prefix}-netapp-account`,
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 * }, {
 *     dependsOn: [
 *         exampleSubnet,
 *         example1,
 *     ],
 * });
 * const examplePool = new azure.netapp.Pool("example", {
 *     name: `${prefix}-netapp-pool`,
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     accountName: exampleAccount.name,
 *     serviceLevel: "Standard",
 *     sizeInTb: 8,
 *     qosType: "Manual",
 * });
 * const exampleVolumeGroupSapHana = new azure.netapp.VolumeGroupSapHana("example", {
 *     name: `${prefix}-netapp-volumegroup`,
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     accountName: exampleAccount.name,
 *     groupDescription: "Test volume group",
 *     applicationIdentifier: "TST",
 *     volumes: [
 *         {
 *             name: `${prefix}-netapp-volume-1`,
 *             volumePath: "my-unique-file-path-1",
 *             serviceLevel: "Standard",
 *             capacityPoolId: examplePool.id,
 *             subnetId: exampleSubnet.id,
 *             proximityPlacementGroupId: examplePlacementGroup.id,
 *             volumeSpecName: "data",
 *             storageQuotaInGb: 1024,
 *             throughputInMibps: 24,
 *             protocols: "NFSv4.1",
 *             securityStyle: "unix",
 *             snapshotDirectoryVisible: false,
 *             exportPolicyRules: [{
 *                 ruleIndex: 1,
 *                 allowedClients: "0.0.0.0/0",
 *                 nfsv3Enabled: false,
 *                 nfsv41Enabled: true,
 *                 unixReadOnly: false,
 *                 unixReadWrite: true,
 *                 rootAccessEnabled: false,
 *             }],
 *             tags: {
 *                 foo: "bar",
 *             },
 *         },
 *         {
 *             name: `${prefix}-netapp-volume-2`,
 *             volumePath: "my-unique-file-path-2",
 *             serviceLevel: "Standard",
 *             capacityPoolId: examplePool.id,
 *             subnetId: exampleSubnet.id,
 *             proximityPlacementGroupId: examplePlacementGroup.id,
 *             volumeSpecName: "log",
 *             storageQuotaInGb: 1024,
 *             throughputInMibps: 24,
 *             protocols: "NFSv4.1",
 *             securityStyle: "unix",
 *             snapshotDirectoryVisible: false,
 *             exportPolicyRules: [{
 *                 ruleIndex: 1,
 *                 allowedClients: "0.0.0.0/0",
 *                 nfsv3Enabled: false,
 *                 nfsv41Enabled: true,
 *                 unixReadOnly: false,
 *                 unixReadWrite: true,
 *                 rootAccessEnabled: false,
 *             }],
 *             tags: {
 *                 foo: "bar",
 *             },
 *         },
 *         {
 *             name: `${prefix}-netapp-volume-3`,
 *             volumePath: "my-unique-file-path-3",
 *             serviceLevel: "Standard",
 *             capacityPoolId: examplePool.id,
 *             subnetId: exampleSubnet.id,
 *             proximityPlacementGroupId: examplePlacementGroup.id,
 *             volumeSpecName: "shared",
 *             storageQuotaInGb: 1024,
 *             throughputInMibps: 24,
 *             protocols: "NFSv4.1",
 *             securityStyle: "unix",
 *             snapshotDirectoryVisible: false,
 *             exportPolicyRules: [{
 *                 ruleIndex: 1,
 *                 allowedClients: "0.0.0.0/0",
 *                 nfsv3Enabled: false,
 *                 nfsv41Enabled: true,
 *                 unixReadOnly: false,
 *                 unixReadWrite: true,
 *                 rootAccessEnabled: false,
 *             }],
 *         },
 *     ],
 * }, {
 *     dependsOn: [
 *         exampleLinuxVirtualMachine,
 *         examplePlacementGroup,
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * Application Volume Groups can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:netapp/volumeGroupSapHana:VolumeGroupSapHana example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mytest-rg/providers/Microsoft.NetApp/netAppAccounts/netapp-account-test/volumeGroups/netapp-volumegroup-test
 * ```
 */
class VolumeGroupSapHana extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountName"] = state ? state.accountName : undefined;
            resourceInputs["applicationIdentifier"] = state ? state.applicationIdentifier : undefined;
            resourceInputs["groupDescription"] = state ? state.groupDescription : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["volumes"] = state ? state.volumes : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.applicationIdentifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'applicationIdentifier'");
            }
            if ((!args || args.groupDescription === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupDescription'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.volumes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'volumes'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["applicationIdentifier"] = args ? args.applicationIdentifier : undefined;
            resourceInputs["groupDescription"] = args ? args.groupDescription : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["volumes"] = args ? args.volumes : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VolumeGroupSapHana.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing VolumeGroupSapHana resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VolumeGroupSapHana(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VolumeGroupSapHana.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VolumeGroupSapHana.__pulumiType;
    }
}
exports.VolumeGroupSapHana = VolumeGroupSapHana;
/** @internal */
VolumeGroupSapHana.__pulumiType = 'azure:netapp/volumeGroupSapHana:VolumeGroupSapHana';
