// @pulumi/azure/privatedns/resolverForwardingRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResolverForwardingRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Private DNS Resolver Forwarding Rule.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "west europe",
 * });
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("example", {
 *     name: "example-vnet",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     addressSpaces: ["10.0.0.0/16"],
 * });
 * const exampleSubnet = new azure.network.Subnet("example", {
 *     name: "outbounddns",
 *     resourceGroupName: example.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.0.0.64/28"],
 *     delegations: [{
 *         name: "Microsoft.Network.dnsResolvers",
 *         serviceDelegation: {
 *             actions: ["Microsoft.Network/virtualNetworks/subnets/join/action"],
 *             name: "Microsoft.Network/dnsResolvers",
 *         },
 *     }],
 * });
 * const exampleResolver = new azure.privatedns.Resolver("example", {
 *     name: "example-resolver",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     virtualNetworkId: exampleVirtualNetwork.id,
 * });
 * const exampleResolverOutboundEndpoint = new azure.privatedns.ResolverOutboundEndpoint("example", {
 *     name: "example-endpoint",
 *     privateDnsResolverId: exampleResolver.id,
 *     location: exampleResolver.location,
 *     subnetId: exampleSubnet.id,
 *     tags: {
 *         key: "value",
 *     },
 * });
 * const exampleResolverDnsForwardingRuleset = new azure.privatedns.ResolverDnsForwardingRuleset("example", {
 *     name: "example-drdfr",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     privateDnsResolverOutboundEndpointIds: [exampleResolverOutboundEndpoint.id],
 * });
 * const exampleResolverForwardingRule = new azure.privatedns.ResolverForwardingRule("example", {
 *     name: "example-rule",
 *     dnsForwardingRulesetId: exampleResolverDnsForwardingRuleset.id,
 *     domainName: "onprem.local.",
 *     enabled: true,
 *     targetDnsServers: [{
 *         ipAddress: "10.10.0.1",
 *         port: 53,
 *     }],
 *     metadata: {
 *         key: "value",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Private DNS Resolver Forwarding Rules can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:privatedns/resolverForwardingRule:ResolverForwardingRule example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.Network/dnsForwardingRulesets/dnsForwardingRuleset1/forwardingRules/forwardingRule1
 * ```
 */
class ResolverForwardingRule extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["dnsForwardingRulesetId"] = state ? state.dnsForwardingRulesetId : undefined;
            resourceInputs["domainName"] = state ? state.domainName : undefined;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["metadata"] = state ? state.metadata : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["targetDnsServers"] = state ? state.targetDnsServers : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.dnsForwardingRulesetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dnsForwardingRulesetId'");
            }
            if ((!args || args.domainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            if ((!args || args.targetDnsServers === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetDnsServers'");
            }
            resourceInputs["dnsForwardingRulesetId"] = args ? args.dnsForwardingRulesetId : undefined;
            resourceInputs["domainName"] = args ? args.domainName : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["metadata"] = args ? args.metadata : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["targetDnsServers"] = args ? args.targetDnsServers : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ResolverForwardingRule.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ResolverForwardingRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ResolverForwardingRule(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ResolverForwardingRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResolverForwardingRule.__pulumiType;
    }
}
exports.ResolverForwardingRule = ResolverForwardingRule;
/** @internal */
ResolverForwardingRule.__pulumiType = 'azure:privatedns/resolverForwardingRule:ResolverForwardingRule';
