// @pulumi/azure-native/storage/objectReplicationPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectReplicationPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The replication policy between two storage accounts. Multiple rules can be defined in one policy.
 * Azure REST API version: 2022-09-01. Prior API version in Azure Native 1.x: 2021-02-01.
 *
 * Other available API versions: 2023-01-01, 2023-04-01, 2023-05-01, 2024-01-01.
 */
class ObjectReplicationPolicy extends pulumi.CustomResource {
    /**
     * Get an existing ObjectReplicationPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ObjectReplicationPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ObjectReplicationPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ObjectReplicationPolicy.__pulumiType;
    }
    /**
     * Create a ObjectReplicationPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.destinationAccount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'destinationAccount'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sourceAccount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceAccount'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["destinationAccount"] = args ? args.destinationAccount : undefined;
            resourceInputs["objectReplicationPolicyId"] = args ? args.objectReplicationPolicyId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["rules"] = args ? args.rules : undefined;
            resourceInputs["sourceAccount"] = args ? args.sourceAccount : undefined;
            resourceInputs["enabledTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["policyId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["destinationAccount"] = undefined /*out*/;
            resourceInputs["enabledTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["policyId"] = undefined /*out*/;
            resourceInputs["rules"] = undefined /*out*/;
            resourceInputs["sourceAccount"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storage/v20190601:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20200801preview:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20210101:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20210201:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20210401:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20210601:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20210801:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20210901:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20220501:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20220901:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20230101:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20230401:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20230501:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20240101:ObjectReplicationPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ObjectReplicationPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ObjectReplicationPolicy = ObjectReplicationPolicy;
/** @internal */
ObjectReplicationPolicy.__pulumiType = 'azure-native:storage:ObjectReplicationPolicy';
