// @pulumi/azure-compliance-policies/azurenative/loadtestservice/v20231201preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadTestProfileMapping = exports.LoadTestMapping = exports.LoadTest = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20231201preview_1 = require("@pulumi/azure-native/loadtestservice/v20231201preview");
const v20231201preview_2 = require("@pulumi/azure-native/loadtestservice/v20231201preview");
const v20231201preview_3 = require("@pulumi/azure-native/loadtestservice/v20231201preview");
var LoadTest;
(function (LoadTest) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (loadtestservice.v20231201preview.LoadTest).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LoadTest.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-loadtestservice-v20231201preview-loadtest-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (loadtestservice.v20231201preview.LoadTest).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231201preview_1.LoadTest, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LoadTest shouldn't use an unstable API (loadtestservice.v20231201preview.LoadTest). A compatible replacement can be found at 'loadtestservice.LoadTest'.");
            }),
        },
        vendors: ["azure"],
        services: ["loadtestservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LoadTest || (LoadTest = {}));
exports.LoadTest = LoadTest;
var LoadTestMapping;
(function (LoadTestMapping) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (loadtestservice.v20231201preview.LoadTestMapping).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LoadTestMapping.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-loadtestservice-v20231201preview-loadtestmapping-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (loadtestservice.v20231201preview.LoadTestMapping).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231201preview_2.LoadTestMapping, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LoadTestMapping shouldn't use an unstable API (loadtestservice.v20231201preview.LoadTestMapping). A compatible replacement can be found at 'loadtestservice.LoadTestMapping'.");
            }),
        },
        vendors: ["azure"],
        services: ["loadtestservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LoadTestMapping || (LoadTestMapping = {}));
exports.LoadTestMapping = LoadTestMapping;
var LoadTestProfileMapping;
(function (LoadTestProfileMapping) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (loadtestservice.v20231201preview.LoadTestProfileMapping).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LoadTestProfileMapping.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-loadtestservice-v20231201preview-loadtestprofilemapping-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (loadtestservice.v20231201preview.LoadTestProfileMapping).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231201preview_3.LoadTestProfileMapping, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LoadTestProfileMapping shouldn't use an unstable API (loadtestservice.v20231201preview.LoadTestProfileMapping). A compatible replacement can be found at 'loadtestservice.LoadTestProfileMapping'.");
            }),
        },
        vendors: ["azure"],
        services: ["loadtestservice"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LoadTestProfileMapping || (LoadTestProfileMapping = {}));
exports.LoadTestProfileMapping = LoadTestProfileMapping;
