// @pulumi/azure/digitaltwins/endpointEventHub.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointEventHub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Digital Twins Event Hub Endpoint.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example_resources",
 *     location: "West Europe",
 * });
 * const exampleInstance = new azure.digitaltwins.Instance("example", {
 *     name: "example-DT",
 *     resourceGroupName: example.name,
 *     location: example.location,
 * });
 * const exampleEventHubNamespace = new azure.eventhub.EventHubNamespace("example", {
 *     name: "example-eh-ns",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     sku: "Standard",
 * });
 * const exampleEventHub = new azure.eventhub.EventHub("example", {
 *     name: "example-eh",
 *     namespaceName: exampleEventHubNamespace.name,
 *     resourceGroupName: example.name,
 *     partitionCount: 2,
 *     messageRetention: 1,
 * });
 * const exampleAuthorizationRule = new azure.eventhub.AuthorizationRule("example", {
 *     name: "example-ar",
 *     namespaceName: exampleEventHubNamespace.name,
 *     eventhubName: exampleEventHub.name,
 *     resourceGroupName: example.name,
 *     listen: false,
 *     send: true,
 *     manage: false,
 * });
 * const exampleEndpointEventHub = new azure.digitaltwins.EndpointEventHub("example", {
 *     name: "example-EH",
 *     digitalTwinsId: exampleInstance.id,
 *     eventhubPrimaryConnectionString: exampleAuthorizationRule.primaryConnectionString,
 *     eventhubSecondaryConnectionString: exampleAuthorizationRule.secondaryConnectionString,
 * });
 * ```
 *
 * ## Import
 *
 * Digital Twins Eventhub Endpoints can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:digitaltwins/endpointEventHub:EndpointEventHub example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.DigitalTwins/digitalTwinsInstances/dt1/endpoints/ep1
 * ```
 */
class EndpointEventHub extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deadLetterStorageSecret"] = state ? state.deadLetterStorageSecret : undefined;
            resourceInputs["digitalTwinsId"] = state ? state.digitalTwinsId : undefined;
            resourceInputs["eventhubPrimaryConnectionString"] = state ? state.eventhubPrimaryConnectionString : undefined;
            resourceInputs["eventhubSecondaryConnectionString"] = state ? state.eventhubSecondaryConnectionString : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.digitalTwinsId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'digitalTwinsId'");
            }
            if ((!args || args.eventhubPrimaryConnectionString === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventhubPrimaryConnectionString'");
            }
            if ((!args || args.eventhubSecondaryConnectionString === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventhubSecondaryConnectionString'");
            }
            resourceInputs["deadLetterStorageSecret"] = (args === null || args === void 0 ? void 0 : args.deadLetterStorageSecret) ? pulumi.secret(args.deadLetterStorageSecret) : undefined;
            resourceInputs["digitalTwinsId"] = args ? args.digitalTwinsId : undefined;
            resourceInputs["eventhubPrimaryConnectionString"] = (args === null || args === void 0 ? void 0 : args.eventhubPrimaryConnectionString) ? pulumi.secret(args.eventhubPrimaryConnectionString) : undefined;
            resourceInputs["eventhubSecondaryConnectionString"] = (args === null || args === void 0 ? void 0 : args.eventhubSecondaryConnectionString) ? pulumi.secret(args.eventhubSecondaryConnectionString) : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["deadLetterStorageSecret", "eventhubPrimaryConnectionString", "eventhubSecondaryConnectionString"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(EndpointEventHub.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing EndpointEventHub resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EndpointEventHub(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EndpointEventHub.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointEventHub.__pulumiType;
    }
}
exports.EndpointEventHub = EndpointEventHub;
/** @internal */
EndpointEventHub.__pulumiType = 'azure:digitaltwins/endpointEventHub:EndpointEventHub';
