// @pulumi/azure/digitaltwins/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeSeriesDatabaseConnection = exports.Instance = exports.getInstanceOutput = exports.getInstance = exports.EndpointServicebus = exports.EndpointEventHub = exports.EndpointEventGrid = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.EndpointEventGrid = null;
utilities.lazyLoad(exports, ["EndpointEventGrid"], () => require("./endpointEventGrid"));
exports.EndpointEventHub = null;
utilities.lazyLoad(exports, ["EndpointEventHub"], () => require("./endpointEventHub"));
exports.EndpointServicebus = null;
utilities.lazyLoad(exports, ["EndpointServicebus"], () => require("./endpointServicebus"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.TimeSeriesDatabaseConnection = null;
utilities.lazyLoad(exports, ["TimeSeriesDatabaseConnection"], () => require("./timeSeriesDatabaseConnection"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:digitaltwins/endpointEventGrid:EndpointEventGrid":
                return new exports.EndpointEventGrid(name, undefined, { urn });
            case "azure:digitaltwins/endpointEventHub:EndpointEventHub":
                return new exports.EndpointEventHub(name, undefined, { urn });
            case "azure:digitaltwins/endpointServicebus:EndpointServicebus":
                return new exports.EndpointServicebus(name, undefined, { urn });
            case "azure:digitaltwins/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "azure:digitaltwins/timeSeriesDatabaseConnection:TimeSeriesDatabaseConnection":
                return new exports.TimeSeriesDatabaseConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "digitaltwins/endpointEventGrid", _module);
pulumi.runtime.registerResourceModule("azure", "digitaltwins/endpointEventHub", _module);
pulumi.runtime.registerResourceModule("azure", "digitaltwins/endpointServicebus", _module);
pulumi.runtime.registerResourceModule("azure", "digitaltwins/instance", _module);
pulumi.runtime.registerResourceModule("azure", "digitaltwins/timeSeriesDatabaseConnection", _module);
