// @pulumi/azure/iotcentral/applicationNetworkRuleSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationNetworkRuleSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an IoT Central Application Network Rule Set.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resource",
 *     location: "West Europe",
 * });
 * const exampleApplication = new azure.iotcentral.Application("example", {
 *     name: "example-iotcentral-app",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     subDomain: "example-iotcentral-app-subdomain",
 *     displayName: "example-iotcentral-app-display-name",
 *     sku: "ST1",
 *     tags: {
 *         Foo: "Bar",
 *     },
 * });
 * const exampleApplicationNetworkRuleSet = new azure.iotcentral.ApplicationNetworkRuleSet("example", {
 *     iotcentralApplicationId: exampleApplication.id,
 *     ipRules: [
 *         {
 *             name: "rule1",
 *             ipMask: "10.0.1.0/24",
 *         },
 *         {
 *             name: "rule2",
 *             ipMask: "10.1.1.0/24",
 *         },
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * IoT Central Application Network Rule Sets can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:iotcentral/applicationNetworkRuleSet:ApplicationNetworkRuleSet example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.IoTCentral/iotApps/app1
 * ```
 */
class ApplicationNetworkRuleSet extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["applyToDevice"] = state ? state.applyToDevice : undefined;
            resourceInputs["defaultAction"] = state ? state.defaultAction : undefined;
            resourceInputs["iotcentralApplicationId"] = state ? state.iotcentralApplicationId : undefined;
            resourceInputs["ipRules"] = state ? state.ipRules : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.iotcentralApplicationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iotcentralApplicationId'");
            }
            resourceInputs["applyToDevice"] = args ? args.applyToDevice : undefined;
            resourceInputs["defaultAction"] = args ? args.defaultAction : undefined;
            resourceInputs["iotcentralApplicationId"] = args ? args.iotcentralApplicationId : undefined;
            resourceInputs["ipRules"] = args ? args.ipRules : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ApplicationNetworkRuleSet.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ApplicationNetworkRuleSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ApplicationNetworkRuleSet(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApplicationNetworkRuleSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApplicationNetworkRuleSet.__pulumiType;
    }
}
exports.ApplicationNetworkRuleSet = ApplicationNetworkRuleSet;
/** @internal */
ApplicationNetworkRuleSet.__pulumiType = 'azure:iotcentral/applicationNetworkRuleSet:ApplicationNetworkRuleSet';
