// @pulumi/azure-compliance-policies/azurenative/dbforpostgresql/v20171201preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkRule = exports.ServerSecurityAlertPolicy = exports.ServerAdministrator = exports.Server = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20171201preview_1 = require("@pulumi/azure-native/dbforpostgresql/v20171201preview");
const v20171201preview_2 = require("@pulumi/azure-native/dbforpostgresql/v20171201preview");
const v20171201preview_3 = require("@pulumi/azure-native/dbforpostgresql/v20171201preview");
const v20171201preview_4 = require("@pulumi/azure-native/dbforpostgresql/v20171201preview");
var Server;
(function (Server) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dbforpostgresql.v20171201preview.Server).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Server.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dbforpostgresql-v20171201preview-server-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dbforpostgresql.v20171201preview.Server).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20171201preview_1.Server, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Server shouldn't use an unstable API (dbforpostgresql.v20171201preview.Server). A compatible replacement can be found at 'dbforpostgresql.Server'.");
            }),
        },
        vendors: ["azure"],
        services: ["dbforpostgresql"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Server || (Server = {}));
exports.Server = Server;
var ServerAdministrator;
(function (ServerAdministrator) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dbforpostgresql.v20171201preview.ServerAdministrator).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ServerAdministrator.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dbforpostgresql-v20171201preview-serveradministrator-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dbforpostgresql.v20171201preview.ServerAdministrator).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20171201preview_2.ServerAdministrator, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ServerAdministrator shouldn't use an unstable API (dbforpostgresql.v20171201preview.ServerAdministrator). A compatible replacement can be found at 'dbforpostgresql.ServerAdministrator'.");
            }),
        },
        vendors: ["azure"],
        services: ["dbforpostgresql"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ServerAdministrator || (ServerAdministrator = {}));
exports.ServerAdministrator = ServerAdministrator;
var ServerSecurityAlertPolicy;
(function (ServerSecurityAlertPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dbforpostgresql.v20171201preview.ServerSecurityAlertPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ServerSecurityAlertPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dbforpostgresql-v20171201preview-serversecurityalertpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dbforpostgresql.v20171201preview.ServerSecurityAlertPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20171201preview_3.ServerSecurityAlertPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ServerSecurityAlertPolicy shouldn't use an unstable API (dbforpostgresql.v20171201preview.ServerSecurityAlertPolicy). A compatible replacement can be found at 'dbforpostgresql.ServerSecurityAlertPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["dbforpostgresql"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ServerSecurityAlertPolicy || (ServerSecurityAlertPolicy = {}));
exports.ServerSecurityAlertPolicy = ServerSecurityAlertPolicy;
var VirtualNetworkRule;
(function (VirtualNetworkRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dbforpostgresql.v20171201preview.VirtualNetworkRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VirtualNetworkRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dbforpostgresql-v20171201preview-virtualnetworkrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dbforpostgresql.v20171201preview.VirtualNetworkRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20171201preview_4.VirtualNetworkRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VirtualNetworkRule shouldn't use an unstable API (dbforpostgresql.v20171201preview.VirtualNetworkRule). A compatible replacement can be found at 'dbforpostgresql.VirtualNetworkRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["dbforpostgresql"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VirtualNetworkRule || (VirtualNetworkRule = {}));
exports.VirtualNetworkRule = VirtualNetworkRule;
