// @pulumi/azure/servicebus/namespaceDisasterRecoveryConfig.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NamespaceDisasterRecoveryConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Disaster Recovery Config for a Service Bus Namespace.
 *
 * > **NOTE:** Disaster Recovery Config is a Premium SKU only capability.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "servicebus-replication",
 *     location: "West Europe",
 * });
 * const primary = new azure.servicebus.Namespace("primary", {
 *     name: "servicebus-primary",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     sku: "Premium",
 *     capacity: 1,
 * });
 * const secondary = new azure.servicebus.Namespace("secondary", {
 *     name: "servicebus-secondary",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     sku: "Premium",
 *     capacity: 1,
 * });
 * const exampleNamespaceAuthorizationRule = new azure.servicebus.NamespaceAuthorizationRule("example", {
 *     name: "examplerule",
 *     namespaceId: exampleAzurermServicebusNamespace.id,
 *     listen: true,
 *     send: true,
 *     manage: false,
 * });
 * const exampleNamespaceDisasterRecoveryConfig = new azure.servicebus.NamespaceDisasterRecoveryConfig("example", {
 *     name: "servicebus-alias-name",
 *     primaryNamespaceId: primary.id,
 *     partnerNamespaceId: secondary.id,
 *     aliasAuthorizationRuleId: exampleNamespaceAuthorizationRule.id,
 * });
 * ```
 *
 * ## Import
 *
 * Service Bus DR configs can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:servicebus/namespaceDisasterRecoveryConfig:NamespaceDisasterRecoveryConfig config1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.ServiceBus/namespaces/namespace1/disasterRecoveryConfigs/config1
 * ```
 */
class NamespaceDisasterRecoveryConfig extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["aliasAuthorizationRuleId"] = state ? state.aliasAuthorizationRuleId : undefined;
            resourceInputs["defaultPrimaryKey"] = state ? state.defaultPrimaryKey : undefined;
            resourceInputs["defaultSecondaryKey"] = state ? state.defaultSecondaryKey : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["partnerNamespaceId"] = state ? state.partnerNamespaceId : undefined;
            resourceInputs["primaryConnectionStringAlias"] = state ? state.primaryConnectionStringAlias : undefined;
            resourceInputs["primaryNamespaceId"] = state ? state.primaryNamespaceId : undefined;
            resourceInputs["secondaryConnectionStringAlias"] = state ? state.secondaryConnectionStringAlias : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.partnerNamespaceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'partnerNamespaceId'");
            }
            if ((!args || args.primaryNamespaceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'primaryNamespaceId'");
            }
            resourceInputs["aliasAuthorizationRuleId"] = args ? args.aliasAuthorizationRuleId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["partnerNamespaceId"] = args ? args.partnerNamespaceId : undefined;
            resourceInputs["primaryNamespaceId"] = args ? args.primaryNamespaceId : undefined;
            resourceInputs["defaultPrimaryKey"] = undefined /*out*/;
            resourceInputs["defaultSecondaryKey"] = undefined /*out*/;
            resourceInputs["primaryConnectionStringAlias"] = undefined /*out*/;
            resourceInputs["secondaryConnectionStringAlias"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["defaultPrimaryKey", "defaultSecondaryKey", "primaryConnectionStringAlias", "secondaryConnectionStringAlias"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(NamespaceDisasterRecoveryConfig.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NamespaceDisasterRecoveryConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NamespaceDisasterRecoveryConfig(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NamespaceDisasterRecoveryConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NamespaceDisasterRecoveryConfig.__pulumiType;
    }
}
exports.NamespaceDisasterRecoveryConfig = NamespaceDisasterRecoveryConfig;
/** @internal */
NamespaceDisasterRecoveryConfig.__pulumiType = 'azure:servicebus/namespaceDisasterRecoveryConfig:NamespaceDisasterRecoveryConfig';
