// @pulumi/azure-compliance-policies/azurenative/elasticsan/v20211120preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeGroup = exports.Volume = exports.ElasticSan = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20211120preview_1 = require("@pulumi/azure-native/elasticsan/v20211120preview");
const v20211120preview_2 = require("@pulumi/azure-native/elasticsan/v20211120preview");
const v20211120preview_3 = require("@pulumi/azure-native/elasticsan/v20211120preview");
var ElasticSan;
(function (ElasticSan) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (elasticsan.v20211120preview.ElasticSan).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ElasticSan.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-elasticsan-v20211120preview-elasticsan-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (elasticsan.v20211120preview.ElasticSan).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20211120preview_1.ElasticSan, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ElasticSan shouldn't use an unstable API (elasticsan.v20211120preview.ElasticSan). A compatible replacement can be found at 'elasticsan.ElasticSan'.");
            }),
        },
        vendors: ["azure"],
        services: ["elasticsan"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ElasticSan || (ElasticSan = {}));
exports.ElasticSan = ElasticSan;
var Volume;
(function (Volume) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (elasticsan.v20211120preview.Volume).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Volume.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-elasticsan-v20211120preview-volume-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (elasticsan.v20211120preview.Volume).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20211120preview_2.Volume, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Volume shouldn't use an unstable API (elasticsan.v20211120preview.Volume). A compatible replacement can be found at 'elasticsan.Volume'.");
            }),
        },
        vendors: ["azure"],
        services: ["elasticsan"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Volume || (Volume = {}));
exports.Volume = Volume;
var VolumeGroup;
(function (VolumeGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (elasticsan.v20211120preview.VolumeGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VolumeGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-elasticsan-v20211120preview-volumegroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (elasticsan.v20211120preview.VolumeGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20211120preview_3.VolumeGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VolumeGroup shouldn't use an unstable API (elasticsan.v20211120preview.VolumeGroup). A compatible replacement can be found at 'elasticsan.VolumeGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["elasticsan"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VolumeGroup || (VolumeGroup = {}));
exports.VolumeGroup = VolumeGroup;
