// @pulumi/azure-native/authorization/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRoleManagementPolicyAssignmentOutput = exports.getRoleManagementPolicyAssignment = exports.getRoleManagementPolicyOutput = exports.getRoleManagementPolicy = exports.getRoleDefinitionOutput = exports.getRoleDefinition = exports.getRoleAssignmentOutput = exports.getRoleAssignment = exports.getResourceManagementPrivateLinkOutput = exports.getResourceManagementPrivateLink = exports.getPrivateLinkAssociationOutput = exports.getPrivateLinkAssociation = exports.getPolicySetDefinitionVersionAtManagementGroupOutput = exports.getPolicySetDefinitionVersionAtManagementGroup = exports.getPolicySetDefinitionVersionOutput = exports.getPolicySetDefinitionVersion = exports.getPolicySetDefinitionAtManagementGroupOutput = exports.getPolicySetDefinitionAtManagementGroup = exports.getPolicySetDefinitionOutput = exports.getPolicySetDefinition = exports.getPolicyExemptionOutput = exports.getPolicyExemption = exports.getPolicyDefinitionVersionAtManagementGroupOutput = exports.getPolicyDefinitionVersionAtManagementGroup = exports.getPolicyDefinitionVersionOutput = exports.getPolicyDefinitionVersion = exports.getPolicyDefinitionAtManagementGroupOutput = exports.getPolicyDefinitionAtManagementGroup = exports.getPolicyDefinitionOutput = exports.getPolicyDefinition = exports.getPolicyAssignmentOutput = exports.getPolicyAssignment = exports.getManagementLockByScopeOutput = exports.getManagementLockByScope = exports.getManagementLockAtSubscriptionLevelOutput = exports.getManagementLockAtSubscriptionLevel = exports.getManagementLockAtResourceLevelOutput = exports.getManagementLockAtResourceLevel = exports.getManagementLockAtResourceGroupLevelOutput = exports.getManagementLockAtResourceGroupLevel = exports.getClientTokenOutput = exports.getClientToken = exports.getClientConfigOutput = exports.getClientConfig = exports.getAccessReviewScheduleDefinitionByIdOutput = exports.getAccessReviewScheduleDefinitionById = exports.getAccessReviewHistoryDefinitionByIdOutput = exports.getAccessReviewHistoryDefinitionById = exports.AccessReviewScheduleDefinitionById = exports.AccessReviewHistoryDefinitionById = void 0;
exports.VariableValueAtManagementGroup = exports.VariableValue = exports.VariableAtManagementGroup = exports.Variable = exports.ScopeAccessReviewScheduleDefinitionById = exports.ScopeAccessReviewHistoryDefinitionById = exports.RoleManagementPolicyAssignment = exports.RoleManagementPolicy = exports.RoleDefinition = exports.RoleAssignment = exports.ResourceManagementPrivateLink = exports.PrivateLinkAssociation = exports.PolicySetDefinitionVersionAtManagementGroup = exports.PolicySetDefinitionVersion = exports.PolicySetDefinitionAtManagementGroup = exports.PolicySetDefinition = exports.PolicyExemption = exports.PolicyDefinitionVersionAtManagementGroup = exports.PolicyDefinitionVersion = exports.PolicyDefinitionAtManagementGroup = exports.PolicyDefinition = exports.PolicyAssignment = exports.ManagementLockByScope = exports.ManagementLockAtSubscriptionLevel = exports.ManagementLockAtResourceLevel = exports.ManagementLockAtResourceGroupLevel = exports.listPolicySetDefinitionVersionAllBuiltinsOutput = exports.listPolicySetDefinitionVersionAllBuiltins = exports.listPolicySetDefinitionVersionAllAtManagementGroupOutput = exports.listPolicySetDefinitionVersionAllAtManagementGroup = exports.listPolicySetDefinitionVersionAllOutput = exports.listPolicySetDefinitionVersionAll = exports.listPolicyDefinitionVersionAllBuiltinsOutput = exports.listPolicyDefinitionVersionAllBuiltins = exports.listPolicyDefinitionVersionAllAtManagementGroupOutput = exports.listPolicyDefinitionVersionAllAtManagementGroup = exports.listPolicyDefinitionVersionAllOutput = exports.listPolicyDefinitionVersionAll = exports.getVariableValueAtManagementGroupOutput = exports.getVariableValueAtManagementGroup = exports.getVariableValueOutput = exports.getVariableValue = exports.getVariableAtManagementGroupOutput = exports.getVariableAtManagementGroup = exports.getVariableOutput = exports.getVariable = exports.getScopeAccessReviewScheduleDefinitionByIdOutput = exports.getScopeAccessReviewScheduleDefinitionById = exports.getScopeAccessReviewHistoryDefinitionByIdOutput = exports.getScopeAccessReviewHistoryDefinitionById = void 0;
exports.v20250101 = exports.v20240901preview = exports.v20240501 = exports.v20240401 = exports.v20240201preview = exports.v20230401 = exports.v20220801preview = exports.v20220701preview = exports.v20220601 = exports.v20220501preview = exports.v20220401 = exports.v20211201preview = exports.v20210601 = exports.v20201001preview = exports.v20201001 = exports.v20200501 = exports.v20200401preview = exports.v20200301preview = exports.v20200301 = exports.v20190601 = exports.v20180501 = exports.v20171001preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccessReviewHistoryDefinitionById = null;
utilities.lazyLoad(exports, ["AccessReviewHistoryDefinitionById"], () => require("./accessReviewHistoryDefinitionById"));
exports.AccessReviewScheduleDefinitionById = null;
utilities.lazyLoad(exports, ["AccessReviewScheduleDefinitionById"], () => require("./accessReviewScheduleDefinitionById"));
exports.getAccessReviewHistoryDefinitionById = null;
exports.getAccessReviewHistoryDefinitionByIdOutput = null;
utilities.lazyLoad(exports, ["getAccessReviewHistoryDefinitionById", "getAccessReviewHistoryDefinitionByIdOutput"], () => require("./getAccessReviewHistoryDefinitionById"));
exports.getAccessReviewScheduleDefinitionById = null;
exports.getAccessReviewScheduleDefinitionByIdOutput = null;
utilities.lazyLoad(exports, ["getAccessReviewScheduleDefinitionById", "getAccessReviewScheduleDefinitionByIdOutput"], () => require("./getAccessReviewScheduleDefinitionById"));
exports.getClientConfig = null;
exports.getClientConfigOutput = null;
utilities.lazyLoad(exports, ["getClientConfig", "getClientConfigOutput"], () => require("./getClientConfig"));
exports.getClientToken = null;
exports.getClientTokenOutput = null;
utilities.lazyLoad(exports, ["getClientToken", "getClientTokenOutput"], () => require("./getClientToken"));
exports.getManagementLockAtResourceGroupLevel = null;
exports.getManagementLockAtResourceGroupLevelOutput = null;
utilities.lazyLoad(exports, ["getManagementLockAtResourceGroupLevel", "getManagementLockAtResourceGroupLevelOutput"], () => require("./getManagementLockAtResourceGroupLevel"));
exports.getManagementLockAtResourceLevel = null;
exports.getManagementLockAtResourceLevelOutput = null;
utilities.lazyLoad(exports, ["getManagementLockAtResourceLevel", "getManagementLockAtResourceLevelOutput"], () => require("./getManagementLockAtResourceLevel"));
exports.getManagementLockAtSubscriptionLevel = null;
exports.getManagementLockAtSubscriptionLevelOutput = null;
utilities.lazyLoad(exports, ["getManagementLockAtSubscriptionLevel", "getManagementLockAtSubscriptionLevelOutput"], () => require("./getManagementLockAtSubscriptionLevel"));
exports.getManagementLockByScope = null;
exports.getManagementLockByScopeOutput = null;
utilities.lazyLoad(exports, ["getManagementLockByScope", "getManagementLockByScopeOutput"], () => require("./getManagementLockByScope"));
exports.getPolicyAssignment = null;
exports.getPolicyAssignmentOutput = null;
utilities.lazyLoad(exports, ["getPolicyAssignment", "getPolicyAssignmentOutput"], () => require("./getPolicyAssignment"));
exports.getPolicyDefinition = null;
exports.getPolicyDefinitionOutput = null;
utilities.lazyLoad(exports, ["getPolicyDefinition", "getPolicyDefinitionOutput"], () => require("./getPolicyDefinition"));
exports.getPolicyDefinitionAtManagementGroup = null;
exports.getPolicyDefinitionAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getPolicyDefinitionAtManagementGroup", "getPolicyDefinitionAtManagementGroupOutput"], () => require("./getPolicyDefinitionAtManagementGroup"));
exports.getPolicyDefinitionVersion = null;
exports.getPolicyDefinitionVersionOutput = null;
utilities.lazyLoad(exports, ["getPolicyDefinitionVersion", "getPolicyDefinitionVersionOutput"], () => require("./getPolicyDefinitionVersion"));
exports.getPolicyDefinitionVersionAtManagementGroup = null;
exports.getPolicyDefinitionVersionAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getPolicyDefinitionVersionAtManagementGroup", "getPolicyDefinitionVersionAtManagementGroupOutput"], () => require("./getPolicyDefinitionVersionAtManagementGroup"));
exports.getPolicyExemption = null;
exports.getPolicyExemptionOutput = null;
utilities.lazyLoad(exports, ["getPolicyExemption", "getPolicyExemptionOutput"], () => require("./getPolicyExemption"));
exports.getPolicySetDefinition = null;
exports.getPolicySetDefinitionOutput = null;
utilities.lazyLoad(exports, ["getPolicySetDefinition", "getPolicySetDefinitionOutput"], () => require("./getPolicySetDefinition"));
exports.getPolicySetDefinitionAtManagementGroup = null;
exports.getPolicySetDefinitionAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getPolicySetDefinitionAtManagementGroup", "getPolicySetDefinitionAtManagementGroupOutput"], () => require("./getPolicySetDefinitionAtManagementGroup"));
exports.getPolicySetDefinitionVersion = null;
exports.getPolicySetDefinitionVersionOutput = null;
utilities.lazyLoad(exports, ["getPolicySetDefinitionVersion", "getPolicySetDefinitionVersionOutput"], () => require("./getPolicySetDefinitionVersion"));
exports.getPolicySetDefinitionVersionAtManagementGroup = null;
exports.getPolicySetDefinitionVersionAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getPolicySetDefinitionVersionAtManagementGroup", "getPolicySetDefinitionVersionAtManagementGroupOutput"], () => require("./getPolicySetDefinitionVersionAtManagementGroup"));
exports.getPrivateLinkAssociation = null;
exports.getPrivateLinkAssociationOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkAssociation", "getPrivateLinkAssociationOutput"], () => require("./getPrivateLinkAssociation"));
exports.getResourceManagementPrivateLink = null;
exports.getResourceManagementPrivateLinkOutput = null;
utilities.lazyLoad(exports, ["getResourceManagementPrivateLink", "getResourceManagementPrivateLinkOutput"], () => require("./getResourceManagementPrivateLink"));
exports.getRoleAssignment = null;
exports.getRoleAssignmentOutput = null;
utilities.lazyLoad(exports, ["getRoleAssignment", "getRoleAssignmentOutput"], () => require("./getRoleAssignment"));
exports.getRoleDefinition = null;
exports.getRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getRoleDefinition", "getRoleDefinitionOutput"], () => require("./getRoleDefinition"));
exports.getRoleManagementPolicy = null;
exports.getRoleManagementPolicyOutput = null;
utilities.lazyLoad(exports, ["getRoleManagementPolicy", "getRoleManagementPolicyOutput"], () => require("./getRoleManagementPolicy"));
exports.getRoleManagementPolicyAssignment = null;
exports.getRoleManagementPolicyAssignmentOutput = null;
utilities.lazyLoad(exports, ["getRoleManagementPolicyAssignment", "getRoleManagementPolicyAssignmentOutput"], () => require("./getRoleManagementPolicyAssignment"));
exports.getScopeAccessReviewHistoryDefinitionById = null;
exports.getScopeAccessReviewHistoryDefinitionByIdOutput = null;
utilities.lazyLoad(exports, ["getScopeAccessReviewHistoryDefinitionById", "getScopeAccessReviewHistoryDefinitionByIdOutput"], () => require("./getScopeAccessReviewHistoryDefinitionById"));
exports.getScopeAccessReviewScheduleDefinitionById = null;
exports.getScopeAccessReviewScheduleDefinitionByIdOutput = null;
utilities.lazyLoad(exports, ["getScopeAccessReviewScheduleDefinitionById", "getScopeAccessReviewScheduleDefinitionByIdOutput"], () => require("./getScopeAccessReviewScheduleDefinitionById"));
exports.getVariable = null;
exports.getVariableOutput = null;
utilities.lazyLoad(exports, ["getVariable", "getVariableOutput"], () => require("./getVariable"));
exports.getVariableAtManagementGroup = null;
exports.getVariableAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getVariableAtManagementGroup", "getVariableAtManagementGroupOutput"], () => require("./getVariableAtManagementGroup"));
exports.getVariableValue = null;
exports.getVariableValueOutput = null;
utilities.lazyLoad(exports, ["getVariableValue", "getVariableValueOutput"], () => require("./getVariableValue"));
exports.getVariableValueAtManagementGroup = null;
exports.getVariableValueAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getVariableValueAtManagementGroup", "getVariableValueAtManagementGroupOutput"], () => require("./getVariableValueAtManagementGroup"));
exports.listPolicyDefinitionVersionAll = null;
exports.listPolicyDefinitionVersionAllOutput = null;
utilities.lazyLoad(exports, ["listPolicyDefinitionVersionAll", "listPolicyDefinitionVersionAllOutput"], () => require("./listPolicyDefinitionVersionAll"));
exports.listPolicyDefinitionVersionAllAtManagementGroup = null;
exports.listPolicyDefinitionVersionAllAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["listPolicyDefinitionVersionAllAtManagementGroup", "listPolicyDefinitionVersionAllAtManagementGroupOutput"], () => require("./listPolicyDefinitionVersionAllAtManagementGroup"));
exports.listPolicyDefinitionVersionAllBuiltins = null;
exports.listPolicyDefinitionVersionAllBuiltinsOutput = null;
utilities.lazyLoad(exports, ["listPolicyDefinitionVersionAllBuiltins", "listPolicyDefinitionVersionAllBuiltinsOutput"], () => require("./listPolicyDefinitionVersionAllBuiltins"));
exports.listPolicySetDefinitionVersionAll = null;
exports.listPolicySetDefinitionVersionAllOutput = null;
utilities.lazyLoad(exports, ["listPolicySetDefinitionVersionAll", "listPolicySetDefinitionVersionAllOutput"], () => require("./listPolicySetDefinitionVersionAll"));
exports.listPolicySetDefinitionVersionAllAtManagementGroup = null;
exports.listPolicySetDefinitionVersionAllAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["listPolicySetDefinitionVersionAllAtManagementGroup", "listPolicySetDefinitionVersionAllAtManagementGroupOutput"], () => require("./listPolicySetDefinitionVersionAllAtManagementGroup"));
exports.listPolicySetDefinitionVersionAllBuiltins = null;
exports.listPolicySetDefinitionVersionAllBuiltinsOutput = null;
utilities.lazyLoad(exports, ["listPolicySetDefinitionVersionAllBuiltins", "listPolicySetDefinitionVersionAllBuiltinsOutput"], () => require("./listPolicySetDefinitionVersionAllBuiltins"));
exports.ManagementLockAtResourceGroupLevel = null;
utilities.lazyLoad(exports, ["ManagementLockAtResourceGroupLevel"], () => require("./managementLockAtResourceGroupLevel"));
exports.ManagementLockAtResourceLevel = null;
utilities.lazyLoad(exports, ["ManagementLockAtResourceLevel"], () => require("./managementLockAtResourceLevel"));
exports.ManagementLockAtSubscriptionLevel = null;
utilities.lazyLoad(exports, ["ManagementLockAtSubscriptionLevel"], () => require("./managementLockAtSubscriptionLevel"));
exports.ManagementLockByScope = null;
utilities.lazyLoad(exports, ["ManagementLockByScope"], () => require("./managementLockByScope"));
exports.PolicyAssignment = null;
utilities.lazyLoad(exports, ["PolicyAssignment"], () => require("./policyAssignment"));
exports.PolicyDefinition = null;
utilities.lazyLoad(exports, ["PolicyDefinition"], () => require("./policyDefinition"));
exports.PolicyDefinitionAtManagementGroup = null;
utilities.lazyLoad(exports, ["PolicyDefinitionAtManagementGroup"], () => require("./policyDefinitionAtManagementGroup"));
exports.PolicyDefinitionVersion = null;
utilities.lazyLoad(exports, ["PolicyDefinitionVersion"], () => require("./policyDefinitionVersion"));
exports.PolicyDefinitionVersionAtManagementGroup = null;
utilities.lazyLoad(exports, ["PolicyDefinitionVersionAtManagementGroup"], () => require("./policyDefinitionVersionAtManagementGroup"));
exports.PolicyExemption = null;
utilities.lazyLoad(exports, ["PolicyExemption"], () => require("./policyExemption"));
exports.PolicySetDefinition = null;
utilities.lazyLoad(exports, ["PolicySetDefinition"], () => require("./policySetDefinition"));
exports.PolicySetDefinitionAtManagementGroup = null;
utilities.lazyLoad(exports, ["PolicySetDefinitionAtManagementGroup"], () => require("./policySetDefinitionAtManagementGroup"));
exports.PolicySetDefinitionVersion = null;
utilities.lazyLoad(exports, ["PolicySetDefinitionVersion"], () => require("./policySetDefinitionVersion"));
exports.PolicySetDefinitionVersionAtManagementGroup = null;
utilities.lazyLoad(exports, ["PolicySetDefinitionVersionAtManagementGroup"], () => require("./policySetDefinitionVersionAtManagementGroup"));
exports.PrivateLinkAssociation = null;
utilities.lazyLoad(exports, ["PrivateLinkAssociation"], () => require("./privateLinkAssociation"));
exports.ResourceManagementPrivateLink = null;
utilities.lazyLoad(exports, ["ResourceManagementPrivateLink"], () => require("./resourceManagementPrivateLink"));
exports.RoleAssignment = null;
utilities.lazyLoad(exports, ["RoleAssignment"], () => require("./roleAssignment"));
exports.RoleDefinition = null;
utilities.lazyLoad(exports, ["RoleDefinition"], () => require("./roleDefinition"));
exports.RoleManagementPolicy = null;
utilities.lazyLoad(exports, ["RoleManagementPolicy"], () => require("./roleManagementPolicy"));
exports.RoleManagementPolicyAssignment = null;
utilities.lazyLoad(exports, ["RoleManagementPolicyAssignment"], () => require("./roleManagementPolicyAssignment"));
exports.ScopeAccessReviewHistoryDefinitionById = null;
utilities.lazyLoad(exports, ["ScopeAccessReviewHistoryDefinitionById"], () => require("./scopeAccessReviewHistoryDefinitionById"));
exports.ScopeAccessReviewScheduleDefinitionById = null;
utilities.lazyLoad(exports, ["ScopeAccessReviewScheduleDefinitionById"], () => require("./scopeAccessReviewScheduleDefinitionById"));
exports.Variable = null;
utilities.lazyLoad(exports, ["Variable"], () => require("./variable"));
exports.VariableAtManagementGroup = null;
utilities.lazyLoad(exports, ["VariableAtManagementGroup"], () => require("./variableAtManagementGroup"));
exports.VariableValue = null;
utilities.lazyLoad(exports, ["VariableValue"], () => require("./variableValue"));
exports.VariableValueAtManagementGroup = null;
utilities.lazyLoad(exports, ["VariableValueAtManagementGroup"], () => require("./variableValueAtManagementGroup"));
// Export enums:
__exportStar(require("../types/enums/authorization"), exports);
// Export sub-modules:
const v20171001preview = require("./v20171001preview");
exports.v20171001preview = v20171001preview;
const v20180501 = require("./v20180501");
exports.v20180501 = v20180501;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20200401preview = require("./v20200401preview");
exports.v20200401preview = v20200401preview;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20201001preview = require("./v20201001preview");
exports.v20201001preview = v20201001preview;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20211201preview = require("./v20211201preview");
exports.v20211201preview = v20211201preview;
const v20220401 = require("./v20220401");
exports.v20220401 = v20220401;
const v20220501preview = require("./v20220501preview");
exports.v20220501preview = v20220501preview;
const v20220601 = require("./v20220601");
exports.v20220601 = v20220601;
const v20220701preview = require("./v20220701preview");
exports.v20220701preview = v20220701preview;
const v20220801preview = require("./v20220801preview");
exports.v20220801preview = v20220801preview;
const v20230401 = require("./v20230401");
exports.v20230401 = v20230401;
const v20240201preview = require("./v20240201preview");
exports.v20240201preview = v20240201preview;
const v20240401 = require("./v20240401");
exports.v20240401 = v20240401;
const v20240501 = require("./v20240501");
exports.v20240501 = v20240501;
const v20240901preview = require("./v20240901preview");
exports.v20240901preview = v20240901preview;
const v20250101 = require("./v20250101");
exports.v20250101 = v20250101;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:authorization:AccessReviewHistoryDefinitionById":
                return new exports.AccessReviewHistoryDefinitionById(name, undefined, { urn });
            case "azure-native:authorization:AccessReviewScheduleDefinitionById":
                return new exports.AccessReviewScheduleDefinitionById(name, undefined, { urn });
            case "azure-native:authorization:ManagementLockAtResourceGroupLevel":
                return new exports.ManagementLockAtResourceGroupLevel(name, undefined, { urn });
            case "azure-native:authorization:ManagementLockAtResourceLevel":
                return new exports.ManagementLockAtResourceLevel(name, undefined, { urn });
            case "azure-native:authorization:ManagementLockAtSubscriptionLevel":
                return new exports.ManagementLockAtSubscriptionLevel(name, undefined, { urn });
            case "azure-native:authorization:ManagementLockByScope":
                return new exports.ManagementLockByScope(name, undefined, { urn });
            case "azure-native:authorization:PolicyAssignment":
                return new exports.PolicyAssignment(name, undefined, { urn });
            case "azure-native:authorization:PolicyDefinition":
                return new exports.PolicyDefinition(name, undefined, { urn });
            case "azure-native:authorization:PolicyDefinitionAtManagementGroup":
                return new exports.PolicyDefinitionAtManagementGroup(name, undefined, { urn });
            case "azure-native:authorization:PolicyDefinitionVersion":
                return new exports.PolicyDefinitionVersion(name, undefined, { urn });
            case "azure-native:authorization:PolicyDefinitionVersionAtManagementGroup":
                return new exports.PolicyDefinitionVersionAtManagementGroup(name, undefined, { urn });
            case "azure-native:authorization:PolicyExemption":
                return new exports.PolicyExemption(name, undefined, { urn });
            case "azure-native:authorization:PolicySetDefinition":
                return new exports.PolicySetDefinition(name, undefined, { urn });
            case "azure-native:authorization:PolicySetDefinitionAtManagementGroup":
                return new exports.PolicySetDefinitionAtManagementGroup(name, undefined, { urn });
            case "azure-native:authorization:PolicySetDefinitionVersion":
                return new exports.PolicySetDefinitionVersion(name, undefined, { urn });
            case "azure-native:authorization:PolicySetDefinitionVersionAtManagementGroup":
                return new exports.PolicySetDefinitionVersionAtManagementGroup(name, undefined, { urn });
            case "azure-native:authorization:PrivateLinkAssociation":
                return new exports.PrivateLinkAssociation(name, undefined, { urn });
            case "azure-native:authorization:ResourceManagementPrivateLink":
                return new exports.ResourceManagementPrivateLink(name, undefined, { urn });
            case "azure-native:authorization:RoleAssignment":
                return new exports.RoleAssignment(name, undefined, { urn });
            case "azure-native:authorization:RoleDefinition":
                return new exports.RoleDefinition(name, undefined, { urn });
            case "azure-native:authorization:RoleManagementPolicy":
                return new exports.RoleManagementPolicy(name, undefined, { urn });
            case "azure-native:authorization:RoleManagementPolicyAssignment":
                return new exports.RoleManagementPolicyAssignment(name, undefined, { urn });
            case "azure-native:authorization:ScopeAccessReviewHistoryDefinitionById":
                return new exports.ScopeAccessReviewHistoryDefinitionById(name, undefined, { urn });
            case "azure-native:authorization:ScopeAccessReviewScheduleDefinitionById":
                return new exports.ScopeAccessReviewScheduleDefinitionById(name, undefined, { urn });
            case "azure-native:authorization:Variable":
                return new exports.Variable(name, undefined, { urn });
            case "azure-native:authorization:VariableAtManagementGroup":
                return new exports.VariableAtManagementGroup(name, undefined, { urn });
            case "azure-native:authorization:VariableValue":
                return new exports.VariableValue(name, undefined, { urn });
            case "azure-native:authorization:VariableValueAtManagementGroup":
                return new exports.VariableValueAtManagementGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "authorization", _module);
