// @pulumi/azure-compliance-policies/azurenative/portal/v20200901preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantConfiguration = exports.Dashboard = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20200901preview_1 = require("@pulumi/azure-native/portal/v20200901preview");
const v20200901preview_2 = require("@pulumi/azure-native/portal/v20200901preview");
var Dashboard;
(function (Dashboard) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (portal.v20200901preview.Dashboard).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Dashboard.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-portal-v20200901preview-dashboard-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (portal.v20200901preview.Dashboard).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200901preview_1.Dashboard, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Dashboard shouldn't use an unstable API (portal.v20200901preview.Dashboard). A compatible replacement can be found at 'portal.Dashboard'.");
            }),
        },
        vendors: ["azure"],
        services: ["portal"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Dashboard || (Dashboard = {}));
exports.Dashboard = Dashboard;
var TenantConfiguration;
(function (TenantConfiguration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (portal.v20200901preview.TenantConfiguration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TenantConfiguration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-portal-v20200901preview-tenantconfiguration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (portal.v20200901preview.TenantConfiguration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200901preview_2.TenantConfiguration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TenantConfiguration shouldn't use an unstable API (portal.v20200901preview.TenantConfiguration). A compatible replacement can be found at 'portal.TenantConfiguration'.");
            }),
        },
        vendors: ["azure"],
        services: ["portal"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TenantConfiguration || (TenantConfiguration = {}));
exports.TenantConfiguration = TenantConfiguration;
