// @pulumi/azure-native/mobilenetwork/sim.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sim = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * SIM resource.
 * Azure REST API version: 2023-06-01. Prior API version in Azure Native 1.x: 2022-04-01-preview.
 *
 * Other available API versions: 2022-03-01-preview, 2022-04-01-preview, 2022-11-01, 2023-09-01, 2024-02-01, 2024-04-01.
 */
class Sim extends pulumi.CustomResource {
    /**
     * Get an existing Sim resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Sim(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Sim.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Sim.__pulumiType;
    }
    /**
     * Create a Sim resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.internationalMobileSubscriberIdentity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'internationalMobileSubscriberIdentity'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.simGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'simGroupName'");
            }
            resourceInputs["authenticationKey"] = args ? args.authenticationKey : undefined;
            resourceInputs["deviceType"] = args ? args.deviceType : undefined;
            resourceInputs["integratedCircuitCardIdentifier"] = args ? args.integratedCircuitCardIdentifier : undefined;
            resourceInputs["internationalMobileSubscriberIdentity"] = args ? args.internationalMobileSubscriberIdentity : undefined;
            resourceInputs["operatorKeyCode"] = args ? args.operatorKeyCode : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["simGroupName"] = args ? args.simGroupName : undefined;
            resourceInputs["simName"] = args ? args.simName : undefined;
            resourceInputs["simPolicy"] = args ? args.simPolicy : undefined;
            resourceInputs["staticIpConfiguration"] = args ? args.staticIpConfiguration : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["simState"] = undefined /*out*/;
            resourceInputs["siteProvisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vendorKeyFingerprint"] = undefined /*out*/;
            resourceInputs["vendorName"] = undefined /*out*/;
        }
        else {
            resourceInputs["deviceType"] = undefined /*out*/;
            resourceInputs["integratedCircuitCardIdentifier"] = undefined /*out*/;
            resourceInputs["internationalMobileSubscriberIdentity"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["simPolicy"] = undefined /*out*/;
            resourceInputs["simState"] = undefined /*out*/;
            resourceInputs["siteProvisioningState"] = undefined /*out*/;
            resourceInputs["staticIpConfiguration"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vendorKeyFingerprint"] = undefined /*out*/;
            resourceInputs["vendorName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:mobilenetwork/v20220401preview:Sim" }, { type: "azure-native:mobilenetwork/v20221101:Sim" }, { type: "azure-native:mobilenetwork/v20230601:Sim" }, { type: "azure-native:mobilenetwork/v20230901:Sim" }, { type: "azure-native:mobilenetwork/v20240201:Sim" }, { type: "azure-native:mobilenetwork/v20240401:Sim" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Sim.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Sim = Sim;
/** @internal */
Sim.__pulumiType = 'azure-native:mobilenetwork:Sim';
