// @pulumi/azure-compliance-policies/azurenative/apimanagement/v20191201preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Logger = exports.Cache = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20191201preview_1 = require("@pulumi/azure-native/apimanagement/v20191201preview");
const v20191201preview_2 = require("@pulumi/azure-native/apimanagement/v20191201preview");
var Cache;
(function (Cache) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20191201preview.Cache).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Cache.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20191201preview-cache-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20191201preview.Cache).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20191201preview_1.Cache, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Cache shouldn't use an unstable API (apimanagement.v20191201preview.Cache). A compatible replacement can be found at 'apimanagement.Cache'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Cache || (Cache = {}));
exports.Cache = Cache;
var Logger;
(function (Logger) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20191201preview.Logger).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Logger.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20191201preview-logger-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20191201preview.Logger).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20191201preview_2.Logger, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Logger shouldn't use an unstable API (apimanagement.v20191201preview.Logger). A compatible replacement can be found at 'apimanagement.Logger'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Logger || (Logger = {}));
exports.Logger = Logger;
