// @pulumi/azure-compliance-policies/azurenative/delegatednetwork/v20230627preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrchestratorInstanceServiceDetails = exports.DelegatedSubnetServiceDetails = exports.ControllerDetails = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230627preview_1 = require("@pulumi/azure-native/delegatednetwork/v20230627preview");
const v20230627preview_2 = require("@pulumi/azure-native/delegatednetwork/v20230627preview");
const v20230627preview_3 = require("@pulumi/azure-native/delegatednetwork/v20230627preview");
var ControllerDetails;
(function (ControllerDetails) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (delegatednetwork.v20230627preview.ControllerDetails).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ControllerDetails.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-delegatednetwork-v20230627preview-controllerdetails-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (delegatednetwork.v20230627preview.ControllerDetails).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230627preview_1.ControllerDetails, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ControllerDetails shouldn't use an unstable API (delegatednetwork.v20230627preview.ControllerDetails). A compatible replacement can be found at 'delegatednetwork.ControllerDetails'.");
            }),
        },
        vendors: ["azure"],
        services: ["delegatednetwork"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ControllerDetails || (ControllerDetails = {}));
exports.ControllerDetails = ControllerDetails;
var DelegatedSubnetServiceDetails;
(function (DelegatedSubnetServiceDetails) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (delegatednetwork.v20230627preview.DelegatedSubnetServiceDetails).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DelegatedSubnetServiceDetails.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-delegatednetwork-v20230627preview-delegatedsubnetservicedetails-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (delegatednetwork.v20230627preview.DelegatedSubnetServiceDetails).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230627preview_2.DelegatedSubnetServiceDetails, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DelegatedSubnetServiceDetails shouldn't use an unstable API (delegatednetwork.v20230627preview.DelegatedSubnetServiceDetails). A compatible replacement can be found at 'delegatednetwork.DelegatedSubnetServiceDetails'.");
            }),
        },
        vendors: ["azure"],
        services: ["delegatednetwork"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DelegatedSubnetServiceDetails || (DelegatedSubnetServiceDetails = {}));
exports.DelegatedSubnetServiceDetails = DelegatedSubnetServiceDetails;
var OrchestratorInstanceServiceDetails;
(function (OrchestratorInstanceServiceDetails) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (delegatednetwork.v20230627preview.OrchestratorInstanceServiceDetails).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OrchestratorInstanceServiceDetails.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-delegatednetwork-v20230627preview-orchestratorinstanceservicedetails-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (delegatednetwork.v20230627preview.OrchestratorInstanceServiceDetails).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230627preview_3.OrchestratorInstanceServiceDetails, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OrchestratorInstanceServiceDetails shouldn't use an unstable API (delegatednetwork.v20230627preview.OrchestratorInstanceServiceDetails). A compatible replacement can be found at 'delegatednetwork.OrchestratorInstanceServiceDetails'.");
            }),
        },
        vendors: ["azure"],
        services: ["delegatednetwork"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OrchestratorInstanceServiceDetails || (OrchestratorInstanceServiceDetails = {}));
exports.OrchestratorInstanceServiceDetails = OrchestratorInstanceServiceDetails;
