// @pulumi/azure/attestation/provider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Provider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Attestation Provider.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as std from "@pulumi/std";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleProvider = new azure.attestation.Provider("example", {
 *     name: "exampleprovider",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     policySigningCertificateData: std.file({
 *         input: "./example/cert.pem",
 *     }).then(invoke => invoke.result),
 * });
 * ```
 *
 * ## Import
 *
 * Attestation Providers can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:attestation/provider:Provider example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Attestation/attestationProviders/provider1
 * ```
 */
class Provider extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["attestationUri"] = state ? state.attestationUri : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["openEnclavePolicyBase64"] = state ? state.openEnclavePolicyBase64 : undefined;
            resourceInputs["policySigningCertificateData"] = state ? state.policySigningCertificateData : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["sevSnpPolicyBase64"] = state ? state.sevSnpPolicyBase64 : undefined;
            resourceInputs["sgxEnclavePolicyBase64"] = state ? state.sgxEnclavePolicyBase64 : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tpmPolicyBase64"] = state ? state.tpmPolicyBase64 : undefined;
            resourceInputs["trustModel"] = state ? state.trustModel : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["openEnclavePolicyBase64"] = args ? args.openEnclavePolicyBase64 : undefined;
            resourceInputs["policySigningCertificateData"] = args ? args.policySigningCertificateData : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sevSnpPolicyBase64"] = args ? args.sevSnpPolicyBase64 : undefined;
            resourceInputs["sgxEnclavePolicyBase64"] = args ? args.sgxEnclavePolicyBase64 : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tpmPolicyBase64"] = args ? args.tpmPolicyBase64 : undefined;
            resourceInputs["attestationUri"] = undefined /*out*/;
            resourceInputs["trustModel"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Provider.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Provider resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Provider(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Provider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Provider.__pulumiType;
    }
}
exports.Provider = Provider;
/** @internal */
Provider.__pulumiType = 'azure:attestation/provider:Provider';
