// @pulumi/azure-compliance-policies/azurenative/hybridcompute/v20230620preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkScope = exports.PrivateEndpointConnection = exports.MachineExtension = exports.Machine = exports.LicenseProfile = exports.License = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230620preview_1 = require("@pulumi/azure-native/hybridcompute/v20230620preview");
const v20230620preview_2 = require("@pulumi/azure-native/hybridcompute/v20230620preview");
const v20230620preview_3 = require("@pulumi/azure-native/hybridcompute/v20230620preview");
const v20230620preview_4 = require("@pulumi/azure-native/hybridcompute/v20230620preview");
const v20230620preview_5 = require("@pulumi/azure-native/hybridcompute/v20230620preview");
const v20230620preview_6 = require("@pulumi/azure-native/hybridcompute/v20230620preview");
var License;
(function (License) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.License).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    License.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-hybridcompute-v20230620preview-license-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.License).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230620preview_1.License, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure License shouldn't use an unstable API (hybridcompute.v20230620preview.License). A compatible replacement can be found at 'hybridcompute.License'.");
            }),
        },
        vendors: ["azure"],
        services: ["hybridcompute"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(License || (License = {}));
exports.License = License;
var LicenseProfile;
(function (LicenseProfile) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.LicenseProfile).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    LicenseProfile.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-hybridcompute-v20230620preview-licenseprofile-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.LicenseProfile).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230620preview_2.LicenseProfile, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure LicenseProfile shouldn't use an unstable API (hybridcompute.v20230620preview.LicenseProfile). A compatible replacement can be found at 'hybridcompute.LicenseProfile'.");
            }),
        },
        vendors: ["azure"],
        services: ["hybridcompute"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(LicenseProfile || (LicenseProfile = {}));
exports.LicenseProfile = LicenseProfile;
var Machine;
(function (Machine) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.Machine).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Machine.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-hybridcompute-v20230620preview-machine-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.Machine).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230620preview_3.Machine, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Machine shouldn't use an unstable API (hybridcompute.v20230620preview.Machine). A compatible replacement can be found at 'hybridcompute.Machine'.");
            }),
        },
        vendors: ["azure"],
        services: ["hybridcompute"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Machine || (Machine = {}));
exports.Machine = Machine;
var MachineExtension;
(function (MachineExtension) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.MachineExtension).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MachineExtension.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-hybridcompute-v20230620preview-machineextension-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.MachineExtension).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230620preview_4.MachineExtension, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MachineExtension shouldn't use an unstable API (hybridcompute.v20230620preview.MachineExtension). A compatible replacement can be found at 'hybridcompute.MachineExtension'.");
            }),
        },
        vendors: ["azure"],
        services: ["hybridcompute"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MachineExtension || (MachineExtension = {}));
exports.MachineExtension = MachineExtension;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-hybridcompute-v20230620preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230620preview_5.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (hybridcompute.v20230620preview.PrivateEndpointConnection). A compatible replacement can be found at 'hybridcompute.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["hybridcompute"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
var PrivateLinkScope;
(function (PrivateLinkScope) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.PrivateLinkScope).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateLinkScope.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-hybridcompute-v20230620preview-privatelinkscope-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (hybridcompute.v20230620preview.PrivateLinkScope).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230620preview_6.PrivateLinkScope, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateLinkScope shouldn't use an unstable API (hybridcompute.v20230620preview.PrivateLinkScope). A compatible replacement can be found at 'hybridcompute.PrivateLinkScope'.");
            }),
        },
        vendors: ["azure"],
        services: ["hybridcompute"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateLinkScope || (PrivateLinkScope = {}));
exports.PrivateLinkScope = PrivateLinkScope;
