// @pulumi/azure-compliance-policies/azurenative/elastic/v20240101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagRule = exports.OpenAI = exports.Monitor = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240101preview_1 = require("@pulumi/azure-native/elastic/v20240101preview");
const v20240101preview_2 = require("@pulumi/azure-native/elastic/v20240101preview");
const v20240101preview_3 = require("@pulumi/azure-native/elastic/v20240101preview");
var Monitor;
(function (Monitor) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (elastic.v20240101preview.Monitor).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Monitor.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-elastic-v20240101preview-monitor-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (elastic.v20240101preview.Monitor).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_1.Monitor, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Monitor shouldn't use an unstable API (elastic.v20240101preview.Monitor). A compatible replacement can be found at 'elastic.Monitor'.");
            }),
        },
        vendors: ["azure"],
        services: ["elastic"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Monitor || (Monitor = {}));
exports.Monitor = Monitor;
var OpenAI;
(function (OpenAI) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (elastic.v20240101preview.OpenAI).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OpenAI.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-elastic-v20240101preview-openai-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (elastic.v20240101preview.OpenAI).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_2.OpenAI, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OpenAI shouldn't use an unstable API (elastic.v20240101preview.OpenAI). A compatible replacement can be found at 'elastic.OpenAI'.");
            }),
        },
        vendors: ["azure"],
        services: ["elastic"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OpenAI || (OpenAI = {}));
exports.OpenAI = OpenAI;
var TagRule;
(function (TagRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (elastic.v20240101preview.TagRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TagRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-elastic-v20240101preview-tagrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (elastic.v20240101preview.TagRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240101preview_3.TagRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TagRule shouldn't use an unstable API (elastic.v20240101preview.TagRule). A compatible replacement can be found at 'elastic.TagRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["elastic"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TagRule || (TagRule = {}));
exports.TagRule = TagRule;
