// @pulumi/azure/role/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRoleDefinitionOutput = exports.getRoleDefinition = exports.Definition = exports.Assignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Assignment = null;
utilities.lazyLoad(exports, ["Assignment"], () => require("./assignment"));
exports.Definition = null;
utilities.lazyLoad(exports, ["Definition"], () => require("./definition"));
exports.getRoleDefinition = null;
exports.getRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getRoleDefinition", "getRoleDefinitionOutput"], () => require("./getRoleDefinition"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:role/assignment:Assignment":
                return new exports.Assignment(name, undefined, { urn });
            case "azure:role/definition:Definition":
                return new exports.Definition(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "role/assignment", _module);
pulumi.runtime.registerResourceModule("azure", "role/definition", _module);
