// @pulumi/azure-compliance-policies/azurenative/documentdb/v20230315preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableResourceTable = exports.SqlResourceSqlUserDefinedFunction = exports.SqlResourceSqlTrigger = exports.SqlResourceSqlStoredProcedure = exports.SqlResourceSqlDatabase = exports.SqlResourceSqlContainer = exports.MongoDBResourceMongoDBDatabase = exports.MongoDBResourceMongoDBCollection = exports.MongoClusterFirewallRule = exports.MongoCluster = exports.GremlinResourceGremlinGraph = exports.GremlinResourceGremlinDatabase = exports.GraphResourceGraph = exports.DatabaseAccount = exports.CassandraResourceCassandraView = exports.CassandraResourceCassandraTable = exports.CassandraResourceCassandraKeyspace = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230315preview_1 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_2 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_3 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_4 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_5 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_6 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_7 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_8 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_9 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_10 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_11 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_12 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_13 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_14 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_15 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_16 = require("@pulumi/azure-native/documentdb/v20230315preview");
const v20230315preview_17 = require("@pulumi/azure-native/documentdb/v20230315preview");
var CassandraResourceCassandraKeyspace;
(function (CassandraResourceCassandraKeyspace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.CassandraResourceCassandraKeyspace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CassandraResourceCassandraKeyspace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-cassandraresourcecassandrakeyspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.CassandraResourceCassandraKeyspace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_1.CassandraResourceCassandraKeyspace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CassandraResourceCassandraKeyspace shouldn't use an unstable API (documentdb.v20230315preview.CassandraResourceCassandraKeyspace). A compatible replacement can be found at 'documentdb.CassandraResourceCassandraKeyspace'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CassandraResourceCassandraKeyspace || (CassandraResourceCassandraKeyspace = {}));
exports.CassandraResourceCassandraKeyspace = CassandraResourceCassandraKeyspace;
var CassandraResourceCassandraTable;
(function (CassandraResourceCassandraTable) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.CassandraResourceCassandraTable).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CassandraResourceCassandraTable.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-cassandraresourcecassandratable-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.CassandraResourceCassandraTable).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_2.CassandraResourceCassandraTable, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CassandraResourceCassandraTable shouldn't use an unstable API (documentdb.v20230315preview.CassandraResourceCassandraTable). A compatible replacement can be found at 'documentdb.CassandraResourceCassandraTable'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CassandraResourceCassandraTable || (CassandraResourceCassandraTable = {}));
exports.CassandraResourceCassandraTable = CassandraResourceCassandraTable;
var CassandraResourceCassandraView;
(function (CassandraResourceCassandraView) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.CassandraResourceCassandraView).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CassandraResourceCassandraView.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-cassandraresourcecassandraview-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.CassandraResourceCassandraView).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_3.CassandraResourceCassandraView, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CassandraResourceCassandraView shouldn't use an unstable API (documentdb.v20230315preview.CassandraResourceCassandraView). A compatible replacement can be found at 'documentdb.CassandraResourceCassandraView'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CassandraResourceCassandraView || (CassandraResourceCassandraView = {}));
exports.CassandraResourceCassandraView = CassandraResourceCassandraView;
var DatabaseAccount;
(function (DatabaseAccount) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.DatabaseAccount).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DatabaseAccount.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-databaseaccount-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.DatabaseAccount).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_4.DatabaseAccount, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DatabaseAccount shouldn't use an unstable API (documentdb.v20230315preview.DatabaseAccount). A compatible replacement can be found at 'documentdb.DatabaseAccount'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DatabaseAccount || (DatabaseAccount = {}));
exports.DatabaseAccount = DatabaseAccount;
var GraphResourceGraph;
(function (GraphResourceGraph) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.GraphResourceGraph).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GraphResourceGraph.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-graphresourcegraph-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.GraphResourceGraph).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_5.GraphResourceGraph, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GraphResourceGraph shouldn't use an unstable API (documentdb.v20230315preview.GraphResourceGraph). A compatible replacement can be found at 'documentdb.GraphResourceGraph'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GraphResourceGraph || (GraphResourceGraph = {}));
exports.GraphResourceGraph = GraphResourceGraph;
var GremlinResourceGremlinDatabase;
(function (GremlinResourceGremlinDatabase) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.GremlinResourceGremlinDatabase).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GremlinResourceGremlinDatabase.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-gremlinresourcegremlindatabase-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.GremlinResourceGremlinDatabase).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_6.GremlinResourceGremlinDatabase, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GremlinResourceGremlinDatabase shouldn't use an unstable API (documentdb.v20230315preview.GremlinResourceGremlinDatabase). A compatible replacement can be found at 'documentdb.GremlinResourceGremlinDatabase'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GremlinResourceGremlinDatabase || (GremlinResourceGremlinDatabase = {}));
exports.GremlinResourceGremlinDatabase = GremlinResourceGremlinDatabase;
var GremlinResourceGremlinGraph;
(function (GremlinResourceGremlinGraph) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.GremlinResourceGremlinGraph).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GremlinResourceGremlinGraph.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-gremlinresourcegremlingraph-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.GremlinResourceGremlinGraph).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_7.GremlinResourceGremlinGraph, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GremlinResourceGremlinGraph shouldn't use an unstable API (documentdb.v20230315preview.GremlinResourceGremlinGraph). A compatible replacement can be found at 'documentdb.GremlinResourceGremlinGraph'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GremlinResourceGremlinGraph || (GremlinResourceGremlinGraph = {}));
exports.GremlinResourceGremlinGraph = GremlinResourceGremlinGraph;
var MongoCluster;
(function (MongoCluster) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.MongoCluster).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MongoCluster.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-mongocluster-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.MongoCluster).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_8.MongoCluster, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MongoCluster shouldn't use an unstable API (documentdb.v20230315preview.MongoCluster). A compatible replacement can be found at 'documentdb.MongoCluster'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MongoCluster || (MongoCluster = {}));
exports.MongoCluster = MongoCluster;
var MongoClusterFirewallRule;
(function (MongoClusterFirewallRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.MongoClusterFirewallRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MongoClusterFirewallRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-mongoclusterfirewallrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.MongoClusterFirewallRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_9.MongoClusterFirewallRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MongoClusterFirewallRule shouldn't use an unstable API (documentdb.v20230315preview.MongoClusterFirewallRule). A compatible replacement can be found at 'documentdb.MongoClusterFirewallRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MongoClusterFirewallRule || (MongoClusterFirewallRule = {}));
exports.MongoClusterFirewallRule = MongoClusterFirewallRule;
var MongoDBResourceMongoDBCollection;
(function (MongoDBResourceMongoDBCollection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.MongoDBResourceMongoDBCollection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MongoDBResourceMongoDBCollection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-mongodbresourcemongodbcollection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.MongoDBResourceMongoDBCollection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_10.MongoDBResourceMongoDBCollection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MongoDBResourceMongoDBCollection shouldn't use an unstable API (documentdb.v20230315preview.MongoDBResourceMongoDBCollection). A compatible replacement can be found at 'documentdb.MongoDBResourceMongoDBCollection'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MongoDBResourceMongoDBCollection || (MongoDBResourceMongoDBCollection = {}));
exports.MongoDBResourceMongoDBCollection = MongoDBResourceMongoDBCollection;
var MongoDBResourceMongoDBDatabase;
(function (MongoDBResourceMongoDBDatabase) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.MongoDBResourceMongoDBDatabase).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MongoDBResourceMongoDBDatabase.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-mongodbresourcemongodbdatabase-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.MongoDBResourceMongoDBDatabase).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_11.MongoDBResourceMongoDBDatabase, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MongoDBResourceMongoDBDatabase shouldn't use an unstable API (documentdb.v20230315preview.MongoDBResourceMongoDBDatabase). A compatible replacement can be found at 'documentdb.MongoDBResourceMongoDBDatabase'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MongoDBResourceMongoDBDatabase || (MongoDBResourceMongoDBDatabase = {}));
exports.MongoDBResourceMongoDBDatabase = MongoDBResourceMongoDBDatabase;
var SqlResourceSqlContainer;
(function (SqlResourceSqlContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SqlResourceSqlContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-sqlresourcesqlcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_12.SqlResourceSqlContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SqlResourceSqlContainer shouldn't use an unstable API (documentdb.v20230315preview.SqlResourceSqlContainer). A compatible replacement can be found at 'documentdb.SqlResourceSqlContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SqlResourceSqlContainer || (SqlResourceSqlContainer = {}));
exports.SqlResourceSqlContainer = SqlResourceSqlContainer;
var SqlResourceSqlDatabase;
(function (SqlResourceSqlDatabase) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlDatabase).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SqlResourceSqlDatabase.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-sqlresourcesqldatabase-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlDatabase).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_13.SqlResourceSqlDatabase, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SqlResourceSqlDatabase shouldn't use an unstable API (documentdb.v20230315preview.SqlResourceSqlDatabase). A compatible replacement can be found at 'documentdb.SqlResourceSqlDatabase'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SqlResourceSqlDatabase || (SqlResourceSqlDatabase = {}));
exports.SqlResourceSqlDatabase = SqlResourceSqlDatabase;
var SqlResourceSqlStoredProcedure;
(function (SqlResourceSqlStoredProcedure) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlStoredProcedure).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SqlResourceSqlStoredProcedure.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-sqlresourcesqlstoredprocedure-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlStoredProcedure).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_14.SqlResourceSqlStoredProcedure, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SqlResourceSqlStoredProcedure shouldn't use an unstable API (documentdb.v20230315preview.SqlResourceSqlStoredProcedure). A compatible replacement can be found at 'documentdb.SqlResourceSqlStoredProcedure'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SqlResourceSqlStoredProcedure || (SqlResourceSqlStoredProcedure = {}));
exports.SqlResourceSqlStoredProcedure = SqlResourceSqlStoredProcedure;
var SqlResourceSqlTrigger;
(function (SqlResourceSqlTrigger) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlTrigger).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SqlResourceSqlTrigger.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-sqlresourcesqltrigger-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlTrigger).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_15.SqlResourceSqlTrigger, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SqlResourceSqlTrigger shouldn't use an unstable API (documentdb.v20230315preview.SqlResourceSqlTrigger). A compatible replacement can be found at 'documentdb.SqlResourceSqlTrigger'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SqlResourceSqlTrigger || (SqlResourceSqlTrigger = {}));
exports.SqlResourceSqlTrigger = SqlResourceSqlTrigger;
var SqlResourceSqlUserDefinedFunction;
(function (SqlResourceSqlUserDefinedFunction) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlUserDefinedFunction).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SqlResourceSqlUserDefinedFunction.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-sqlresourcesqluserdefinedfunction-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.SqlResourceSqlUserDefinedFunction).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_16.SqlResourceSqlUserDefinedFunction, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SqlResourceSqlUserDefinedFunction shouldn't use an unstable API (documentdb.v20230315preview.SqlResourceSqlUserDefinedFunction). A compatible replacement can be found at 'documentdb.SqlResourceSqlUserDefinedFunction'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SqlResourceSqlUserDefinedFunction || (SqlResourceSqlUserDefinedFunction = {}));
exports.SqlResourceSqlUserDefinedFunction = SqlResourceSqlUserDefinedFunction;
var TableResourceTable;
(function (TableResourceTable) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.TableResourceTable).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TableResourceTable.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-documentdb-v20230315preview-tableresourcetable-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (documentdb.v20230315preview.TableResourceTable).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230315preview_17.TableResourceTable, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TableResourceTable shouldn't use an unstable API (documentdb.v20230315preview.TableResourceTable). A compatible replacement can be found at 'documentdb.TableResourceTable'.");
            }),
        },
        vendors: ["azure"],
        services: ["documentdb"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TableResourceTable || (TableResourceTable = {}));
exports.TableResourceTable = TableResourceTable;
