// @pulumi/azure-compliance-policies/azurenative/devices/v20250201preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotDpsResourcePrivateEndpointConnection = exports.IotDpsResource = exports.DpsCertificate = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20250201preview_1 = require("@pulumi/azure-native/devices/v20250201preview");
const v20250201preview_2 = require("@pulumi/azure-native/devices/v20250201preview");
const v20250201preview_3 = require("@pulumi/azure-native/devices/v20250201preview");
var DpsCertificate;
(function (DpsCertificate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devices.v20250201preview.DpsCertificate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DpsCertificate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devices-v20250201preview-dpscertificate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devices.v20250201preview.DpsCertificate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20250201preview_1.DpsCertificate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DpsCertificate shouldn't use an unstable API (devices.v20250201preview.DpsCertificate). A compatible replacement can be found at 'devices.DpsCertificate'.");
            }),
        },
        vendors: ["azure"],
        services: ["devices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DpsCertificate || (DpsCertificate = {}));
exports.DpsCertificate = DpsCertificate;
var IotDpsResource;
(function (IotDpsResource) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devices.v20250201preview.IotDpsResource).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IotDpsResource.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devices-v20250201preview-iotdpsresource-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devices.v20250201preview.IotDpsResource).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20250201preview_2.IotDpsResource, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IotDpsResource shouldn't use an unstable API (devices.v20250201preview.IotDpsResource). A compatible replacement can be found at 'devices.IotDpsResource'.");
            }),
        },
        vendors: ["azure"],
        services: ["devices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IotDpsResource || (IotDpsResource = {}));
exports.IotDpsResource = IotDpsResource;
var IotDpsResourcePrivateEndpointConnection;
(function (IotDpsResourcePrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (devices.v20250201preview.IotDpsResourcePrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IotDpsResourcePrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-devices-v20250201preview-iotdpsresourceprivateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (devices.v20250201preview.IotDpsResourcePrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20250201preview_3.IotDpsResourcePrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IotDpsResourcePrivateEndpointConnection shouldn't use an unstable API (devices.v20250201preview.IotDpsResourcePrivateEndpointConnection). A compatible replacement can be found at 'devices.IotDpsResourcePrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["devices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IotDpsResourcePrivateEndpointConnection || (IotDpsResourcePrivateEndpointConnection = {}));
exports.IotDpsResourcePrivateEndpointConnection = IotDpsResourcePrivateEndpointConnection;
