// @pulumi/azure-compliance-policies/azurenative/securityinsights/v20210301preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Watchlist = exports.Ueba = exports.TiTaxiiDataConnector = exports.ThreatIntelligenceAlertRule = exports.TIDataConnector = exports.SourceControl = exports.SentinelOnboardingState = exports.ScheduledAlertRule = exports.OfficeDataConnector = exports.OfficeATPDataConnector = exports.MicrosoftSecurityIncidentCreationAlertRule = exports.Metadata = exports.MTPDataConnector = exports.MSTIDataConnector = exports.MLBehaviorAnalyticsAlertRule = exports.MDATPDataConnector = exports.MCASDataConnector = exports.IncidentRelation = exports.IncidentComment = exports.Incident = exports.FusionAlertRule = exports.EyesOn = exports.EntityAnalytics = exports.Dynamics365DataConnector = exports.CodelessUiDataConnector = exports.AwsCloudTrailDataConnector = exports.Anomalies = exports.ActivityCustomEntityQuery = exports.Action = exports.ASCDataConnector = exports.AATPDataConnector = exports.AADDataConnector = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20210301preview_1 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_2 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_3 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_4 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_5 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_6 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_7 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_8 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_9 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_10 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_11 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_12 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_13 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_14 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_15 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_16 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_17 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_18 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_19 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_20 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_21 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_22 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_23 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_24 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_25 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_26 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_27 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_28 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_29 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_30 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_31 = require("@pulumi/azure-native/securityinsights/v20210301preview");
const v20210301preview_32 = require("@pulumi/azure-native/securityinsights/v20210301preview");
var AADDataConnector;
(function (AADDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.AADDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AADDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-aaddataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.AADDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_1.AADDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AADDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.AADDataConnector). A compatible replacement can be found at 'securityinsights.AADDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AADDataConnector || (AADDataConnector = {}));
exports.AADDataConnector = AADDataConnector;
var AATPDataConnector;
(function (AATPDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.AATPDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AATPDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-aatpdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.AATPDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_2.AATPDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AATPDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.AATPDataConnector). A compatible replacement can be found at 'securityinsights.AATPDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AATPDataConnector || (AATPDataConnector = {}));
exports.AATPDataConnector = AATPDataConnector;
var ASCDataConnector;
(function (ASCDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.ASCDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ASCDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-ascdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.ASCDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_3.ASCDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ASCDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.ASCDataConnector). A compatible replacement can be found at 'securityinsights.ASCDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ASCDataConnector || (ASCDataConnector = {}));
exports.ASCDataConnector = ASCDataConnector;
var Action;
(function (Action) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Action).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Action.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-action-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Action).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_4.Action, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Action shouldn't use an unstable API (securityinsights.v20210301preview.Action). A compatible replacement can be found at 'securityinsights.Action'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Action || (Action = {}));
exports.Action = Action;
var ActivityCustomEntityQuery;
(function (ActivityCustomEntityQuery) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.ActivityCustomEntityQuery).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ActivityCustomEntityQuery.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-activitycustomentityquery-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.ActivityCustomEntityQuery).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_5.ActivityCustomEntityQuery, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ActivityCustomEntityQuery shouldn't use an unstable API (securityinsights.v20210301preview.ActivityCustomEntityQuery). A compatible replacement can be found at 'securityinsights.ActivityCustomEntityQuery'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ActivityCustomEntityQuery || (ActivityCustomEntityQuery = {}));
exports.ActivityCustomEntityQuery = ActivityCustomEntityQuery;
var Anomalies;
(function (Anomalies) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Anomalies).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Anomalies.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-anomalies-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Anomalies).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_6.Anomalies, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Anomalies shouldn't use an unstable API (securityinsights.v20210301preview.Anomalies). A compatible replacement can be found at 'securityinsights.Anomalies'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Anomalies || (Anomalies = {}));
exports.Anomalies = Anomalies;
var AwsCloudTrailDataConnector;
(function (AwsCloudTrailDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.AwsCloudTrailDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AwsCloudTrailDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-awscloudtraildataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.AwsCloudTrailDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_7.AwsCloudTrailDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AwsCloudTrailDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.AwsCloudTrailDataConnector). A compatible replacement can be found at 'securityinsights.AwsCloudTrailDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AwsCloudTrailDataConnector || (AwsCloudTrailDataConnector = {}));
exports.AwsCloudTrailDataConnector = AwsCloudTrailDataConnector;
var CodelessUiDataConnector;
(function (CodelessUiDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.CodelessUiDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CodelessUiDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-codelessuidataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.CodelessUiDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_8.CodelessUiDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CodelessUiDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.CodelessUiDataConnector). A compatible replacement can be found at 'securityinsights.CodelessUiDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CodelessUiDataConnector || (CodelessUiDataConnector = {}));
exports.CodelessUiDataConnector = CodelessUiDataConnector;
var Dynamics365DataConnector;
(function (Dynamics365DataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Dynamics365DataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Dynamics365DataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-dynamics365dataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Dynamics365DataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_9.Dynamics365DataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Dynamics365DataConnector shouldn't use an unstable API (securityinsights.v20210301preview.Dynamics365DataConnector). A compatible replacement can be found at 'securityinsights.Dynamics365DataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Dynamics365DataConnector || (Dynamics365DataConnector = {}));
exports.Dynamics365DataConnector = Dynamics365DataConnector;
var EntityAnalytics;
(function (EntityAnalytics) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.EntityAnalytics).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EntityAnalytics.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-entityanalytics-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.EntityAnalytics).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_10.EntityAnalytics, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EntityAnalytics shouldn't use an unstable API (securityinsights.v20210301preview.EntityAnalytics). A compatible replacement can be found at 'securityinsights.EntityAnalytics'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EntityAnalytics || (EntityAnalytics = {}));
exports.EntityAnalytics = EntityAnalytics;
var EyesOn;
(function (EyesOn) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.EyesOn).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EyesOn.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-eyeson-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.EyesOn).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_11.EyesOn, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EyesOn shouldn't use an unstable API (securityinsights.v20210301preview.EyesOn). A compatible replacement can be found at 'securityinsights.EyesOn'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EyesOn || (EyesOn = {}));
exports.EyesOn = EyesOn;
var FusionAlertRule;
(function (FusionAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.FusionAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FusionAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-fusionalertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.FusionAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_12.FusionAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FusionAlertRule shouldn't use an unstable API (securityinsights.v20210301preview.FusionAlertRule). A compatible replacement can be found at 'securityinsights.FusionAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FusionAlertRule || (FusionAlertRule = {}));
exports.FusionAlertRule = FusionAlertRule;
var Incident;
(function (Incident) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Incident).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Incident.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-incident-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Incident).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_13.Incident, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Incident shouldn't use an unstable API (securityinsights.v20210301preview.Incident). A compatible replacement can be found at 'securityinsights.Incident'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Incident || (Incident = {}));
exports.Incident = Incident;
var IncidentComment;
(function (IncidentComment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.IncidentComment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IncidentComment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-incidentcomment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.IncidentComment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_14.IncidentComment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IncidentComment shouldn't use an unstable API (securityinsights.v20210301preview.IncidentComment). A compatible replacement can be found at 'securityinsights.IncidentComment'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IncidentComment || (IncidentComment = {}));
exports.IncidentComment = IncidentComment;
var IncidentRelation;
(function (IncidentRelation) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.IncidentRelation).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    IncidentRelation.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-incidentrelation-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.IncidentRelation).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_15.IncidentRelation, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure IncidentRelation shouldn't use an unstable API (securityinsights.v20210301preview.IncidentRelation). A compatible replacement can be found at 'securityinsights.IncidentRelation'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(IncidentRelation || (IncidentRelation = {}));
exports.IncidentRelation = IncidentRelation;
var MCASDataConnector;
(function (MCASDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MCASDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MCASDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-mcasdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MCASDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_16.MCASDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MCASDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.MCASDataConnector). A compatible replacement can be found at 'securityinsights.MCASDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MCASDataConnector || (MCASDataConnector = {}));
exports.MCASDataConnector = MCASDataConnector;
var MDATPDataConnector;
(function (MDATPDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MDATPDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MDATPDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-mdatpdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MDATPDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_17.MDATPDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MDATPDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.MDATPDataConnector). A compatible replacement can be found at 'securityinsights.MDATPDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MDATPDataConnector || (MDATPDataConnector = {}));
exports.MDATPDataConnector = MDATPDataConnector;
var MLBehaviorAnalyticsAlertRule;
(function (MLBehaviorAnalyticsAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MLBehaviorAnalyticsAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MLBehaviorAnalyticsAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-mlbehavioranalyticsalertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MLBehaviorAnalyticsAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_18.MLBehaviorAnalyticsAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MLBehaviorAnalyticsAlertRule shouldn't use an unstable API (securityinsights.v20210301preview.MLBehaviorAnalyticsAlertRule). A compatible replacement can be found at 'securityinsights.MLBehaviorAnalyticsAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MLBehaviorAnalyticsAlertRule || (MLBehaviorAnalyticsAlertRule = {}));
exports.MLBehaviorAnalyticsAlertRule = MLBehaviorAnalyticsAlertRule;
var MSTIDataConnector;
(function (MSTIDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MSTIDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MSTIDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-mstidataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MSTIDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_19.MSTIDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MSTIDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.MSTIDataConnector). A compatible replacement can be found at 'securityinsights.MSTIDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MSTIDataConnector || (MSTIDataConnector = {}));
exports.MSTIDataConnector = MSTIDataConnector;
var MTPDataConnector;
(function (MTPDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MTPDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MTPDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-mtpdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MTPDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_20.MTPDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MTPDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.MTPDataConnector). A compatible replacement can be found at 'securityinsights.MTPDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MTPDataConnector || (MTPDataConnector = {}));
exports.MTPDataConnector = MTPDataConnector;
var Metadata;
(function (Metadata) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Metadata).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Metadata.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-metadata-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Metadata).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_21.Metadata, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Metadata shouldn't use an unstable API (securityinsights.v20210301preview.Metadata). A compatible replacement can be found at 'securityinsights.Metadata'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Metadata || (Metadata = {}));
exports.Metadata = Metadata;
var MicrosoftSecurityIncidentCreationAlertRule;
(function (MicrosoftSecurityIncidentCreationAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MicrosoftSecurityIncidentCreationAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MicrosoftSecurityIncidentCreationAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-microsoftsecurityincidentcreationalertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.MicrosoftSecurityIncidentCreationAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_22.MicrosoftSecurityIncidentCreationAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MicrosoftSecurityIncidentCreationAlertRule shouldn't use an unstable API (securityinsights.v20210301preview.MicrosoftSecurityIncidentCreationAlertRule). A compatible replacement can be found at 'securityinsights.MicrosoftSecurityIncidentCreationAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MicrosoftSecurityIncidentCreationAlertRule || (MicrosoftSecurityIncidentCreationAlertRule = {}));
exports.MicrosoftSecurityIncidentCreationAlertRule = MicrosoftSecurityIncidentCreationAlertRule;
var OfficeATPDataConnector;
(function (OfficeATPDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.OfficeATPDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OfficeATPDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-officeatpdataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.OfficeATPDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_23.OfficeATPDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OfficeATPDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.OfficeATPDataConnector). A compatible replacement can be found at 'securityinsights.OfficeATPDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OfficeATPDataConnector || (OfficeATPDataConnector = {}));
exports.OfficeATPDataConnector = OfficeATPDataConnector;
var OfficeDataConnector;
(function (OfficeDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.OfficeDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OfficeDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-officedataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.OfficeDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_24.OfficeDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OfficeDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.OfficeDataConnector). A compatible replacement can be found at 'securityinsights.OfficeDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OfficeDataConnector || (OfficeDataConnector = {}));
exports.OfficeDataConnector = OfficeDataConnector;
var ScheduledAlertRule;
(function (ScheduledAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.ScheduledAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ScheduledAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-scheduledalertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.ScheduledAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_25.ScheduledAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ScheduledAlertRule shouldn't use an unstable API (securityinsights.v20210301preview.ScheduledAlertRule). A compatible replacement can be found at 'securityinsights.ScheduledAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ScheduledAlertRule || (ScheduledAlertRule = {}));
exports.ScheduledAlertRule = ScheduledAlertRule;
var SentinelOnboardingState;
(function (SentinelOnboardingState) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.SentinelOnboardingState).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SentinelOnboardingState.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-sentinelonboardingstate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.SentinelOnboardingState).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_26.SentinelOnboardingState, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SentinelOnboardingState shouldn't use an unstable API (securityinsights.v20210301preview.SentinelOnboardingState). A compatible replacement can be found at 'securityinsights.SentinelOnboardingState'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SentinelOnboardingState || (SentinelOnboardingState = {}));
exports.SentinelOnboardingState = SentinelOnboardingState;
var SourceControl;
(function (SourceControl) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.SourceControl).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SourceControl.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-sourcecontrol-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.SourceControl).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_27.SourceControl, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SourceControl shouldn't use an unstable API (securityinsights.v20210301preview.SourceControl). A compatible replacement can be found at 'securityinsights.SourceControl'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SourceControl || (SourceControl = {}));
exports.SourceControl = SourceControl;
var TIDataConnector;
(function (TIDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.TIDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TIDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-tidataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.TIDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_28.TIDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TIDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.TIDataConnector). A compatible replacement can be found at 'securityinsights.TIDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TIDataConnector || (TIDataConnector = {}));
exports.TIDataConnector = TIDataConnector;
var ThreatIntelligenceAlertRule;
(function (ThreatIntelligenceAlertRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.ThreatIntelligenceAlertRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ThreatIntelligenceAlertRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-threatintelligencealertrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.ThreatIntelligenceAlertRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_29.ThreatIntelligenceAlertRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ThreatIntelligenceAlertRule shouldn't use an unstable API (securityinsights.v20210301preview.ThreatIntelligenceAlertRule). A compatible replacement can be found at 'securityinsights.ThreatIntelligenceAlertRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ThreatIntelligenceAlertRule || (ThreatIntelligenceAlertRule = {}));
exports.ThreatIntelligenceAlertRule = ThreatIntelligenceAlertRule;
var TiTaxiiDataConnector;
(function (TiTaxiiDataConnector) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.TiTaxiiDataConnector).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TiTaxiiDataConnector.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-titaxiidataconnector-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.TiTaxiiDataConnector).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_30.TiTaxiiDataConnector, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TiTaxiiDataConnector shouldn't use an unstable API (securityinsights.v20210301preview.TiTaxiiDataConnector). A compatible replacement can be found at 'securityinsights.TiTaxiiDataConnector'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TiTaxiiDataConnector || (TiTaxiiDataConnector = {}));
exports.TiTaxiiDataConnector = TiTaxiiDataConnector;
var Ueba;
(function (Ueba) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Ueba).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Ueba.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-ueba-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Ueba).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_31.Ueba, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Ueba shouldn't use an unstable API (securityinsights.v20210301preview.Ueba). A compatible replacement can be found at 'securityinsights.Ueba'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Ueba || (Ueba = {}));
exports.Ueba = Ueba;
var Watchlist;
(function (Watchlist) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Watchlist).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Watchlist.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-securityinsights-v20210301preview-watchlist-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (securityinsights.v20210301preview.Watchlist).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210301preview_32.Watchlist, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Watchlist shouldn't use an unstable API (securityinsights.v20210301preview.Watchlist). A compatible replacement can be found at 'securityinsights.Watchlist'.");
            }),
        },
        vendors: ["azure"],
        services: ["securityinsights"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Watchlist || (Watchlist = {}));
exports.Watchlist = Watchlist;
