// @pulumi/azure-native/web/appServiceEnvironmentAseCustomDnsSuffixConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppServiceEnvironmentAseCustomDnsSuffixConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Full view of the custom domain suffix configuration for ASEv3.
 * Azure REST API version: 2022-09-01. Prior API version in Azure Native 1.x: 2022-03-01.
 *
 * Other available API versions: 2023-01-01, 2023-12-01, 2024-04-01.
 */
class AppServiceEnvironmentAseCustomDnsSuffixConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing AppServiceEnvironmentAseCustomDnsSuffixConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AppServiceEnvironmentAseCustomDnsSuffixConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AppServiceEnvironmentAseCustomDnsSuffixConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AppServiceEnvironmentAseCustomDnsSuffixConfiguration.__pulumiType;
    }
    /**
     * Create a AppServiceEnvironmentAseCustomDnsSuffixConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["certificateUrl"] = args ? args.certificateUrl : undefined;
            resourceInputs["dnsSuffix"] = args ? args.dnsSuffix : undefined;
            resourceInputs["keyVaultReferenceIdentity"] = args ? args.keyVaultReferenceIdentity : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["provisioningDetails"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["certificateUrl"] = undefined /*out*/;
            resourceInputs["dnsSuffix"] = undefined /*out*/;
            resourceInputs["keyVaultReferenceIdentity"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningDetails"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20220301:AppServiceEnvironmentAseCustomDnsSuffixConfiguration" }, { type: "azure-native:web/v20220901:AppServiceEnvironmentAseCustomDnsSuffixConfiguration" }, { type: "azure-native:web/v20230101:AppServiceEnvironmentAseCustomDnsSuffixConfiguration" }, { type: "azure-native:web/v20231201:AppServiceEnvironmentAseCustomDnsSuffixConfiguration" }, { type: "azure-native:web/v20240401:AppServiceEnvironmentAseCustomDnsSuffixConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AppServiceEnvironmentAseCustomDnsSuffixConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AppServiceEnvironmentAseCustomDnsSuffixConfiguration = AppServiceEnvironmentAseCustomDnsSuffixConfiguration;
/** @internal */
AppServiceEnvironmentAseCustomDnsSuffixConfiguration.__pulumiType = 'azure-native:web:AppServiceEnvironmentAseCustomDnsSuffixConfiguration';
