// @pulumi/azure-native/web/staticSiteUserProvidedFunctionAppForStaticSiteBuild.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticSiteUserProvidedFunctionAppForStaticSiteBuild = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Static Site User Provided Function App ARM resource.
 * Azure REST API version: 2022-09-01. Prior API version in Azure Native 1.x: 2020-12-01.
 *
 * Other available API versions: 2023-01-01, 2023-12-01, 2024-04-01.
 */
class StaticSiteUserProvidedFunctionAppForStaticSiteBuild extends pulumi.CustomResource {
    /**
     * Get an existing StaticSiteUserProvidedFunctionAppForStaticSiteBuild resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StaticSiteUserProvidedFunctionAppForStaticSiteBuild(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StaticSiteUserProvidedFunctionAppForStaticSiteBuild.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StaticSiteUserProvidedFunctionAppForStaticSiteBuild.__pulumiType;
    }
    /**
     * Create a StaticSiteUserProvidedFunctionAppForStaticSiteBuild resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.environmentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environmentName'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["environmentName"] = args ? args.environmentName : undefined;
            resourceInputs["functionAppName"] = args ? args.functionAppName : undefined;
            resourceInputs["functionAppRegion"] = args ? args.functionAppRegion : undefined;
            resourceInputs["functionAppResourceId"] = args ? args.functionAppResourceId : undefined;
            resourceInputs["isForced"] = args ? args.isForced : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["functionAppRegion"] = undefined /*out*/;
            resourceInputs["functionAppResourceId"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20201201:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }, { type: "azure-native:web/v20210101:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }, { type: "azure-native:web/v20210115:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }, { type: "azure-native:web/v20210201:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }, { type: "azure-native:web/v20210301:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }, { type: "azure-native:web/v20220301:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }, { type: "azure-native:web/v20220901:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }, { type: "azure-native:web/v20230101:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }, { type: "azure-native:web/v20231201:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }, { type: "azure-native:web/v20240401:StaticSiteUserProvidedFunctionAppForStaticSiteBuild" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StaticSiteUserProvidedFunctionAppForStaticSiteBuild.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StaticSiteUserProvidedFunctionAppForStaticSiteBuild = StaticSiteUserProvidedFunctionAppForStaticSiteBuild;
/** @internal */
StaticSiteUserProvidedFunctionAppForStaticSiteBuild.__pulumiType = 'azure-native:web:StaticSiteUserProvidedFunctionAppForStaticSiteBuild';
