// @pulumi/azure-native/web/webAppSlotConfigurationNames.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppSlotConfigurationNames = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Slot Config names azure resource.
 * Azure REST API version: 2022-09-01. Prior API version in Azure Native 1.x: 2020-12-01.
 *
 * Other available API versions: 2020-10-01, 2023-01-01, 2023-12-01, 2024-04-01.
 */
class WebAppSlotConfigurationNames extends pulumi.CustomResource {
    /**
     * Get an existing WebAppSlotConfigurationNames resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppSlotConfigurationNames(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppSlotConfigurationNames.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppSlotConfigurationNames.__pulumiType;
    }
    /**
     * Create a WebAppSlotConfigurationNames resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["appSettingNames"] = args ? args.appSettingNames : undefined;
            resourceInputs["azureStorageConfigNames"] = args ? args.azureStorageConfigNames : undefined;
            resourceInputs["connectionStringNames"] = args ? args.connectionStringNames : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["appSettingNames"] = undefined /*out*/;
            resourceInputs["azureStorageConfigNames"] = undefined /*out*/;
            resourceInputs["connectionStringNames"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20150801:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20160801:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20180201:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20181101:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20190801:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20200601:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20200901:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20201001:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20201201:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20210101:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20210115:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20210201:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20210301:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20220301:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20220901:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20230101:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20231201:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20240401:WebAppSlotConfigurationNames" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppSlotConfigurationNames.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppSlotConfigurationNames = WebAppSlotConfigurationNames;
/** @internal */
WebAppSlotConfigurationNames.__pulumiType = 'azure-native:web:WebAppSlotConfigurationNames';
