// @pulumi/azure-native/web/webAppFunction.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppFunction = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Function information.
 * Azure REST API version: 2022-09-01. Prior API version in Azure Native 1.x: 2020-12-01.
 *
 * Other available API versions: 2016-08-01, 2020-10-01, 2023-01-01, 2023-12-01, 2024-04-01.
 */
class WebAppFunction extends pulumi.CustomResource {
    /**
     * Get an existing WebAppFunction resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppFunction(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppFunction.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppFunction.__pulumiType;
    }
    /**
     * Create a WebAppFunction resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["config"] = args ? args.config : undefined;
            resourceInputs["configHref"] = args ? args.configHref : undefined;
            resourceInputs["files"] = args ? args.files : undefined;
            resourceInputs["functionAppId"] = args ? args.functionAppId : undefined;
            resourceInputs["functionName"] = args ? args.functionName : undefined;
            resourceInputs["href"] = args ? args.href : undefined;
            resourceInputs["invokeUrlTemplate"] = args ? args.invokeUrlTemplate : undefined;
            resourceInputs["isDisabled"] = args ? args.isDisabled : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["language"] = args ? args.language : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scriptHref"] = args ? args.scriptHref : undefined;
            resourceInputs["scriptRootPathHref"] = args ? args.scriptRootPathHref : undefined;
            resourceInputs["secretsFileHref"] = args ? args.secretsFileHref : undefined;
            resourceInputs["testData"] = args ? args.testData : undefined;
            resourceInputs["testDataHref"] = args ? args.testDataHref : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["config"] = undefined /*out*/;
            resourceInputs["configHref"] = undefined /*out*/;
            resourceInputs["files"] = undefined /*out*/;
            resourceInputs["functionAppId"] = undefined /*out*/;
            resourceInputs["href"] = undefined /*out*/;
            resourceInputs["invokeUrlTemplate"] = undefined /*out*/;
            resourceInputs["isDisabled"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["language"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["scriptHref"] = undefined /*out*/;
            resourceInputs["scriptRootPathHref"] = undefined /*out*/;
            resourceInputs["secretsFileHref"] = undefined /*out*/;
            resourceInputs["testData"] = undefined /*out*/;
            resourceInputs["testDataHref"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20160801:WebAppFunction" }, { type: "azure-native:web/v20180201:WebAppFunction" }, { type: "azure-native:web/v20181101:WebAppFunction" }, { type: "azure-native:web/v20190801:WebAppFunction" }, { type: "azure-native:web/v20200601:WebAppFunction" }, { type: "azure-native:web/v20200901:WebAppFunction" }, { type: "azure-native:web/v20201001:WebAppFunction" }, { type: "azure-native:web/v20201201:WebAppFunction" }, { type: "azure-native:web/v20210101:WebAppFunction" }, { type: "azure-native:web/v20210115:WebAppFunction" }, { type: "azure-native:web/v20210201:WebAppFunction" }, { type: "azure-native:web/v20210301:WebAppFunction" }, { type: "azure-native:web/v20220301:WebAppFunction" }, { type: "azure-native:web/v20220901:WebAppFunction" }, { type: "azure-native:web/v20230101:WebAppFunction" }, { type: "azure-native:web/v20231201:WebAppFunction" }, { type: "azure-native:web/v20240401:WebAppFunction" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppFunction.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppFunction = WebAppFunction;
/** @internal */
WebAppFunction.__pulumiType = 'azure-native:web:WebAppFunction';
