// @pulumi/azure-native/web/webAppPremierAddOnSlot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppPremierAddOnSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Premier add-on.
 * Azure REST API version: 2022-09-01. Prior API version in Azure Native 1.x: 2020-12-01.
 *
 * Other available API versions: 2016-08-01, 2020-10-01, 2023-01-01, 2023-12-01, 2024-04-01.
 */
class WebAppPremierAddOnSlot extends pulumi.CustomResource {
    /**
     * Get an existing WebAppPremierAddOnSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppPremierAddOnSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppPremierAddOnSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppPremierAddOnSlot.__pulumiType;
    }
    /**
     * Create a WebAppPremierAddOnSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["marketplaceOffer"] = args ? args.marketplaceOffer : undefined;
            resourceInputs["marketplacePublisher"] = args ? args.marketplacePublisher : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["premierAddOnName"] = args ? args.premierAddOnName : undefined;
            resourceInputs["product"] = args ? args.product : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["slot"] = args ? args.slot : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["vendor"] = args ? args.vendor : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["marketplaceOffer"] = undefined /*out*/;
            resourceInputs["marketplacePublisher"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["product"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vendor"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20150801:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20160801:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20180201:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20181101:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20190801:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20200601:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20200901:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20201001:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20201201:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20210101:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20210115:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20210201:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20210301:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20220301:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20220901:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20230101:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20231201:WebAppPremierAddOnSlot" }, { type: "azure-native:web/v20240401:WebAppPremierAddOnSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppPremierAddOnSlot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppPremierAddOnSlot = WebAppPremierAddOnSlot;
/** @internal */
WebAppPremierAddOnSlot.__pulumiType = 'azure-native:web:WebAppPremierAddOnSlot';
