// @pulumi/azure-native/web/webAppFtpAllowedSlot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppFtpAllowedSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Publishing Credentials Policies parameters.
 * Azure REST API version: 2023-12-01.
 *
 * Other available API versions: 2020-12-01, 2021-01-01, 2021-01-15, 2021-02-01, 2021-03-01, 2022-03-01, 2022-09-01, 2023-01-01, 2024-04-01.
 */
class WebAppFtpAllowedSlot extends pulumi.CustomResource {
    /**
     * Get an existing WebAppFtpAllowedSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppFtpAllowedSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppFtpAllowedSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppFtpAllowedSlot.__pulumiType;
    }
    /**
     * Create a WebAppFtpAllowedSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.allow === undefined) && !opts.urn) {
                throw new Error("Missing required property 'allow'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["allow"] = args ? args.allow : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["slot"] = args ? args.slot : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["allow"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20201201:WebAppFtpAllowedSlot" }, { type: "azure-native:web/v20210101:WebAppFtpAllowedSlot" }, { type: "azure-native:web/v20210115:WebAppFtpAllowedSlot" }, { type: "azure-native:web/v20210201:WebAppFtpAllowedSlot" }, { type: "azure-native:web/v20210301:WebAppFtpAllowedSlot" }, { type: "azure-native:web/v20220301:WebAppFtpAllowedSlot" }, { type: "azure-native:web/v20220901:WebAppFtpAllowedSlot" }, { type: "azure-native:web/v20230101:WebAppFtpAllowedSlot" }, { type: "azure-native:web/v20231201:WebAppFtpAllowedSlot" }, { type: "azure-native:web/v20240401:WebAppFtpAllowedSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppFtpAllowedSlot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppFtpAllowedSlot = WebAppFtpAllowedSlot;
/** @internal */
WebAppFtpAllowedSlot.__pulumiType = 'azure-native:web:WebAppFtpAllowedSlot';
