// @pulumi/azure/siterecovery/hypervNetworkMapping.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HypervNetworkMapping = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a HyperV site recovery network mapping on Azure. A HyperV network mapping decides how to translate connected networks when a VM is migrated from HyperV VMM Center to Azure.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const target = new azure.core.ResourceGroup("target", {
 *     name: "tfex-network-mapping",
 *     location: "East US",
 * });
 * const vault = new azure.recoveryservices.Vault("vault", {
 *     name: "example-recovery-vault",
 *     location: target.location,
 *     resourceGroupName: target.name,
 *     sku: "Standard",
 * });
 * const targetVirtualNetwork = new azure.network.VirtualNetwork("target", {
 *     name: "network",
 *     resourceGroupName: target.name,
 *     addressSpaces: ["192.168.2.0/24"],
 *     location: target.location,
 * });
 * const recovery_mapping = new azure.siterecovery.HypervNetworkMapping("recovery-mapping", {
 *     name: "recovery-network-mapping",
 *     recoveryVaultId: vault.id,
 *     sourceSystemCenterVirtualMachineManagerName: "my-vmm-server",
 *     sourceNetworkName: "my-vmm-network",
 *     targetNetworkId: targetVirtualNetwork.id,
 * });
 * ```
 *
 * ## Import
 *
 * Site Recovery Network Mapping can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:siterecovery/hypervNetworkMapping:HypervNetworkMapping azurerm_site_recovery_hyperv_network_mapping.mymapping /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resource-group-name/providers/Microsoft.RecoveryServices/vaults/recovery-vault-name/replicationFabrics/primary-fabric-name/replicationNetworks/azureNetwork/replicationNetworkMappings/mapping-name
 * ```
 */
class HypervNetworkMapping extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["recoveryVaultId"] = state ? state.recoveryVaultId : undefined;
            resourceInputs["sourceNetworkName"] = state ? state.sourceNetworkName : undefined;
            resourceInputs["sourceSystemCenterVirtualMachineManagerName"] = state ? state.sourceSystemCenterVirtualMachineManagerName : undefined;
            resourceInputs["targetNetworkId"] = state ? state.targetNetworkId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.recoveryVaultId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'recoveryVaultId'");
            }
            if ((!args || args.sourceNetworkName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceNetworkName'");
            }
            if ((!args || args.sourceSystemCenterVirtualMachineManagerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceSystemCenterVirtualMachineManagerName'");
            }
            if ((!args || args.targetNetworkId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetNetworkId'");
            }
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["recoveryVaultId"] = args ? args.recoveryVaultId : undefined;
            resourceInputs["sourceNetworkName"] = args ? args.sourceNetworkName : undefined;
            resourceInputs["sourceSystemCenterVirtualMachineManagerName"] = args ? args.sourceSystemCenterVirtualMachineManagerName : undefined;
            resourceInputs["targetNetworkId"] = args ? args.targetNetworkId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(HypervNetworkMapping.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing HypervNetworkMapping resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HypervNetworkMapping(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HypervNetworkMapping.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HypervNetworkMapping.__pulumiType;
    }
}
exports.HypervNetworkMapping = HypervNetworkMapping;
/** @internal */
HypervNetworkMapping.__pulumiType = 'azure:siterecovery/hypervNetworkMapping:HypervNetworkMapping';
