// @pulumi/azure-compliance-policies/azurenative/deviceregistry/v20231101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetEndpointProfile = exports.Asset = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20231101preview_1 = require("@pulumi/azure-native/deviceregistry/v20231101preview");
const v20231101preview_2 = require("@pulumi/azure-native/deviceregistry/v20231101preview");
var Asset;
(function (Asset) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20231101preview.Asset).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Asset.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-deviceregistry-v20231101preview-asset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20231101preview.Asset).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_1.Asset, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Asset shouldn't use an unstable API (deviceregistry.v20231101preview.Asset). A compatible replacement can be found at 'deviceregistry.Asset'.");
            }),
        },
        vendors: ["azure"],
        services: ["deviceregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Asset || (Asset = {}));
exports.Asset = Asset;
var AssetEndpointProfile;
(function (AssetEndpointProfile) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20231101preview.AssetEndpointProfile).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AssetEndpointProfile.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-deviceregistry-v20231101preview-assetendpointprofile-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (deviceregistry.v20231101preview.AssetEndpointProfile).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_2.AssetEndpointProfile, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AssetEndpointProfile shouldn't use an unstable API (deviceregistry.v20231101preview.AssetEndpointProfile). A compatible replacement can be found at 'deviceregistry.AssetEndpointProfile'.");
            }),
        },
        vendors: ["azure"],
        services: ["deviceregistry"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AssetEndpointProfile || (AssetEndpointProfile = {}));
exports.AssetEndpointProfile = AssetEndpointProfile;
