// @pulumi/azure-native/cache/enterprisePrivateEndpointConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnterprisePrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The Private Endpoint Connection resource.
 * Azure REST API version: 2023-03-01-preview.
 *
 * Other available API versions: 2023-07-01, 2023-08-01-preview, 2023-10-01-preview, 2023-11-01, 2024-02-01, 2024-03-01-preview, 2024-06-01-preview, 2024-09-01-preview, 2024-10-01.
 */
class EnterprisePrivateEndpointConnection extends pulumi.CustomResource {
    /**
     * Get an existing EnterprisePrivateEndpointConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EnterprisePrivateEndpointConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EnterprisePrivateEndpointConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnterprisePrivateEndpointConnection.__pulumiType;
    }
    /**
     * Create a EnterprisePrivateEndpointConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.privateLinkServiceConnectionState === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateLinkServiceConnectionState'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["privateEndpointConnectionName"] = args ? args.privateEndpointConnectionName : undefined;
            resourceInputs["privateLinkServiceConnectionState"] = args ? args.privateLinkServiceConnectionState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpoint"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpoint"] = undefined /*out*/;
            resourceInputs["privateLinkServiceConnectionState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cache/v20201001preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20210201preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20210301:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20210801:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20220101:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20221101preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20230301preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20230701:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20230801preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20231001preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20231101:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20240201:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20240301preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20240601preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20240901preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20241001:EnterprisePrivateEndpointConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EnterprisePrivateEndpointConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EnterprisePrivateEndpointConnection = EnterprisePrivateEndpointConnection;
/** @internal */
EnterprisePrivateEndpointConnection.__pulumiType = 'azure-native:cache:EnterprisePrivateEndpointConnection';
