// @pulumi/azure-compliance-policies/azurenative/kusto/v20180907preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventHubConnection = exports.Database = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20180907preview_1 = require("@pulumi/azure-native/kusto/v20180907preview");
const v20180907preview_2 = require("@pulumi/azure-native/kusto/v20180907preview");
var Database;
(function (Database) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (kusto.v20180907preview.Database).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Database.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-kusto-v20180907preview-database-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (kusto.v20180907preview.Database).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180907preview_1.Database, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Database shouldn't use an unstable API (kusto.v20180907preview.Database). A compatible replacement can be found at 'kusto.Database'.");
            }),
        },
        vendors: ["azure"],
        services: ["kusto"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Database || (Database = {}));
exports.Database = Database;
var EventHubConnection;
(function (EventHubConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (kusto.v20180907preview.EventHubConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EventHubConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-kusto-v20180907preview-eventhubconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (kusto.v20180907preview.EventHubConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180907preview_2.EventHubConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EventHubConnection shouldn't use an unstable API (kusto.v20180907preview.EventHubConnection). A compatible replacement can be found at 'kusto.EventHubConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["kusto"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EventHubConnection || (EventHubConnection = {}));
exports.EventHubConnection = EventHubConnection;
