// @pulumi/azure-native/network/configurationPolicyGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationPolicyGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * VpnServerConfigurationPolicyGroup Resource.
 * Azure REST API version: 2023-02-01. Prior API version in Azure Native 1.x: 2022-01-01.
 *
 * Other available API versions: 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-05-01.
 */
class ConfigurationPolicyGroup extends pulumi.CustomResource {
    /**
     * Get an existing ConfigurationPolicyGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ConfigurationPolicyGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConfigurationPolicyGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConfigurationPolicyGroup.__pulumiType;
    }
    /**
     * Create a ConfigurationPolicyGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vpnServerConfigurationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpnServerConfigurationName'");
            }
            resourceInputs["configurationPolicyGroupName"] = args ? args.configurationPolicyGroupName : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["isDefault"] = args ? args.isDefault : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["policyMembers"] = args ? args.policyMembers : undefined;
            resourceInputs["priority"] = args ? args.priority : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["vpnServerConfigurationName"] = args ? args.vpnServerConfigurationName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["p2SConnectionConfigurations"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["isDefault"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["p2SConnectionConfigurations"] = undefined /*out*/;
            resourceInputs["policyMembers"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20210801:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20220101:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20220501:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20220701:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20220901:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20221101:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20230201:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20230401:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20230501:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20230601:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20230901:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20231101:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20240101:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20240301:ConfigurationPolicyGroup" }, { type: "azure-native:network/v20240501:ConfigurationPolicyGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ConfigurationPolicyGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ConfigurationPolicyGroup = ConfigurationPolicyGroup;
/** @internal */
ConfigurationPolicyGroup.__pulumiType = 'azure-native:network:ConfigurationPolicyGroup';
