// @pulumi/azure-native/network/expressRouteCircuitPeering.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressRouteCircuitPeering = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Peering in an ExpressRouteCircuit resource.
 * Azure REST API version: 2023-02-01. Prior API version in Azure Native 1.x: 2020-11-01.
 *
 * Other available API versions: 2019-02-01, 2019-06-01, 2019-08-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-05-01.
 */
class ExpressRouteCircuitPeering extends pulumi.CustomResource {
    /**
     * Get an existing ExpressRouteCircuitPeering resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExpressRouteCircuitPeering(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExpressRouteCircuitPeering.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExpressRouteCircuitPeering.__pulumiType;
    }
    /**
     * Create a ExpressRouteCircuitPeering resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.circuitName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'circuitName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["azureASN"] = args ? args.azureASN : undefined;
            resourceInputs["circuitName"] = args ? args.circuitName : undefined;
            resourceInputs["connections"] = args ? args.connections : undefined;
            resourceInputs["gatewayManagerEtag"] = args ? args.gatewayManagerEtag : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["ipv6PeeringConfig"] = args ? args.ipv6PeeringConfig : undefined;
            resourceInputs["microsoftPeeringConfig"] = args ? args.microsoftPeeringConfig : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["peerASN"] = args ? args.peerASN : undefined;
            resourceInputs["peeringName"] = args ? args.peeringName : undefined;
            resourceInputs["peeringType"] = args ? args.peeringType : undefined;
            resourceInputs["primaryAzurePort"] = args ? args.primaryAzurePort : undefined;
            resourceInputs["primaryPeerAddressPrefix"] = args ? args.primaryPeerAddressPrefix : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["routeFilter"] = args ? args.routeFilter : undefined;
            resourceInputs["secondaryAzurePort"] = args ? args.secondaryAzurePort : undefined;
            resourceInputs["secondaryPeerAddressPrefix"] = args ? args.secondaryPeerAddressPrefix : undefined;
            resourceInputs["sharedKey"] = args ? args.sharedKey : undefined;
            resourceInputs["state"] = args ? args.state : undefined;
            resourceInputs["stats"] = args ? args.stats : undefined;
            resourceInputs["vlanId"] = args ? args.vlanId : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["expressRouteConnection"] = undefined /*out*/;
            resourceInputs["lastModifiedBy"] = undefined /*out*/;
            resourceInputs["peeredConnections"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureASN"] = undefined /*out*/;
            resourceInputs["connections"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["expressRouteConnection"] = undefined /*out*/;
            resourceInputs["gatewayManagerEtag"] = undefined /*out*/;
            resourceInputs["ipv6PeeringConfig"] = undefined /*out*/;
            resourceInputs["lastModifiedBy"] = undefined /*out*/;
            resourceInputs["microsoftPeeringConfig"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["peerASN"] = undefined /*out*/;
            resourceInputs["peeredConnections"] = undefined /*out*/;
            resourceInputs["peeringType"] = undefined /*out*/;
            resourceInputs["primaryAzurePort"] = undefined /*out*/;
            resourceInputs["primaryPeerAddressPrefix"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["routeFilter"] = undefined /*out*/;
            resourceInputs["secondaryAzurePort"] = undefined /*out*/;
            resourceInputs["secondaryPeerAddressPrefix"] = undefined /*out*/;
            resourceInputs["sharedKey"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["stats"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vlanId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20150501preview:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20150615:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20160330:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20160601:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20160901:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20161201:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20170301:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20170601:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20170801:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20170901:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20171001:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20171101:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20180101:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20180201:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20180401:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20180601:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20180701:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20180801:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20181001:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20181101:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20181201:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20190201:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20190401:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20190601:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20190701:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20190801:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20190901:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20191101:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20191201:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20200301:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20200401:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20200501:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20200601:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20200701:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20200801:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20201101:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20210201:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20210301:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20210501:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20210801:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20220101:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20220501:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20220701:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20220901:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20221101:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20230201:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20230401:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20230501:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20230601:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20230901:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20231101:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20240101:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20240301:ExpressRouteCircuitPeering" }, { type: "azure-native:network/v20240501:ExpressRouteCircuitPeering" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExpressRouteCircuitPeering.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ExpressRouteCircuitPeering = ExpressRouteCircuitPeering;
/** @internal */
ExpressRouteCircuitPeering.__pulumiType = 'azure-native:network:ExpressRouteCircuitPeering';
