// @pulumi/azure-native/network/firewallPolicyDraft.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirewallPolicyDraft = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * FirewallPolicy Resource.
 * Azure REST API version: 2023-11-01.
 *
 * Other available API versions: 2024-01-01, 2024-03-01, 2024-05-01.
 */
class FirewallPolicyDraft extends pulumi.CustomResource {
    /**
     * Get an existing FirewallPolicyDraft resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FirewallPolicyDraft(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FirewallPolicyDraft.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FirewallPolicyDraft.__pulumiType;
    }
    /**
     * Create a FirewallPolicyDraft resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.firewallPolicyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'firewallPolicyName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["basePolicy"] = args ? args.basePolicy : undefined;
            resourceInputs["dnsSettings"] = args ? args.dnsSettings : undefined;
            resourceInputs["explicitProxy"] = args ? args.explicitProxy : undefined;
            resourceInputs["firewallPolicyName"] = args ? args.firewallPolicyName : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["insights"] = args ? args.insights : undefined;
            resourceInputs["intrusionDetection"] = args ? args.intrusionDetection : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["snat"] = args ? args.snat : undefined;
            resourceInputs["sql"] = args ? args.sql : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["threatIntelMode"] = args ? args.threatIntelMode : undefined;
            resourceInputs["threatIntelWhitelist"] = args ? args.threatIntelWhitelist : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["basePolicy"] = undefined /*out*/;
            resourceInputs["dnsSettings"] = undefined /*out*/;
            resourceInputs["explicitProxy"] = undefined /*out*/;
            resourceInputs["insights"] = undefined /*out*/;
            resourceInputs["intrusionDetection"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["snat"] = undefined /*out*/;
            resourceInputs["sql"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["threatIntelMode"] = undefined /*out*/;
            resourceInputs["threatIntelWhitelist"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20231101:FirewallPolicyDraft" }, { type: "azure-native:network/v20240101:FirewallPolicyDraft" }, { type: "azure-native:network/v20240301:FirewallPolicyDraft" }, { type: "azure-native:network/v20240501:FirewallPolicyDraft" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FirewallPolicyDraft.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FirewallPolicyDraft = FirewallPolicyDraft;
/** @internal */
FirewallPolicyDraft.__pulumiType = 'azure-native:network:FirewallPolicyDraft';
