// @pulumi/azure-native/network/localNetworkGateway.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalNetworkGateway = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A common class for general resource information.
 * Azure REST API version: 2023-02-01. Prior API version in Azure Native 1.x: 2020-11-01.
 *
 * Other available API versions: 2019-08-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-05-01.
 */
class LocalNetworkGateway extends pulumi.CustomResource {
    /**
     * Get an existing LocalNetworkGateway resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LocalNetworkGateway(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocalNetworkGateway.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocalNetworkGateway.__pulumiType;
    }
    /**
     * Create a LocalNetworkGateway resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["bgpSettings"] = args ? args.bgpSettings : undefined;
            resourceInputs["fqdn"] = args ? args.fqdn : undefined;
            resourceInputs["gatewayIpAddress"] = args ? args.gatewayIpAddress : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["localNetworkAddressSpace"] = args ? args.localNetworkAddressSpace : undefined;
            resourceInputs["localNetworkGatewayName"] = args ? args.localNetworkGatewayName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["bgpSettings"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["gatewayIpAddress"] = undefined /*out*/;
            resourceInputs["localNetworkAddressSpace"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20150615:LocalNetworkGateway" }, { type: "azure-native:network/v20160330:LocalNetworkGateway" }, { type: "azure-native:network/v20160601:LocalNetworkGateway" }, { type: "azure-native:network/v20160901:LocalNetworkGateway" }, { type: "azure-native:network/v20161201:LocalNetworkGateway" }, { type: "azure-native:network/v20170301:LocalNetworkGateway" }, { type: "azure-native:network/v20170601:LocalNetworkGateway" }, { type: "azure-native:network/v20170801:LocalNetworkGateway" }, { type: "azure-native:network/v20170901:LocalNetworkGateway" }, { type: "azure-native:network/v20171001:LocalNetworkGateway" }, { type: "azure-native:network/v20171101:LocalNetworkGateway" }, { type: "azure-native:network/v20180101:LocalNetworkGateway" }, { type: "azure-native:network/v20180201:LocalNetworkGateway" }, { type: "azure-native:network/v20180401:LocalNetworkGateway" }, { type: "azure-native:network/v20180601:LocalNetworkGateway" }, { type: "azure-native:network/v20180701:LocalNetworkGateway" }, { type: "azure-native:network/v20180801:LocalNetworkGateway" }, { type: "azure-native:network/v20181001:LocalNetworkGateway" }, { type: "azure-native:network/v20181101:LocalNetworkGateway" }, { type: "azure-native:network/v20181201:LocalNetworkGateway" }, { type: "azure-native:network/v20190201:LocalNetworkGateway" }, { type: "azure-native:network/v20190401:LocalNetworkGateway" }, { type: "azure-native:network/v20190601:LocalNetworkGateway" }, { type: "azure-native:network/v20190701:LocalNetworkGateway" }, { type: "azure-native:network/v20190801:LocalNetworkGateway" }, { type: "azure-native:network/v20190901:LocalNetworkGateway" }, { type: "azure-native:network/v20191101:LocalNetworkGateway" }, { type: "azure-native:network/v20191201:LocalNetworkGateway" }, { type: "azure-native:network/v20200301:LocalNetworkGateway" }, { type: "azure-native:network/v20200401:LocalNetworkGateway" }, { type: "azure-native:network/v20200501:LocalNetworkGateway" }, { type: "azure-native:network/v20200601:LocalNetworkGateway" }, { type: "azure-native:network/v20200701:LocalNetworkGateway" }, { type: "azure-native:network/v20200801:LocalNetworkGateway" }, { type: "azure-native:network/v20201101:LocalNetworkGateway" }, { type: "azure-native:network/v20210201:LocalNetworkGateway" }, { type: "azure-native:network/v20210301:LocalNetworkGateway" }, { type: "azure-native:network/v20210501:LocalNetworkGateway" }, { type: "azure-native:network/v20210801:LocalNetworkGateway" }, { type: "azure-native:network/v20220101:LocalNetworkGateway" }, { type: "azure-native:network/v20220501:LocalNetworkGateway" }, { type: "azure-native:network/v20220701:LocalNetworkGateway" }, { type: "azure-native:network/v20220901:LocalNetworkGateway" }, { type: "azure-native:network/v20221101:LocalNetworkGateway" }, { type: "azure-native:network/v20230201:LocalNetworkGateway" }, { type: "azure-native:network/v20230401:LocalNetworkGateway" }, { type: "azure-native:network/v20230501:LocalNetworkGateway" }, { type: "azure-native:network/v20230601:LocalNetworkGateway" }, { type: "azure-native:network/v20230901:LocalNetworkGateway" }, { type: "azure-native:network/v20231101:LocalNetworkGateway" }, { type: "azure-native:network/v20240101:LocalNetworkGateway" }, { type: "azure-native:network/v20240301:LocalNetworkGateway" }, { type: "azure-native:network/v20240501:LocalNetworkGateway" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(LocalNetworkGateway.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LocalNetworkGateway = LocalNetworkGateway;
/** @internal */
LocalNetworkGateway.__pulumiType = 'azure-native:network:LocalNetworkGateway';
