// @pulumi/azure-native/network/networkSecurityGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkSecurityGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * NetworkSecurityGroup resource.
 * Azure REST API version: 2023-02-01. Prior API version in Azure Native 1.x: 2020-11-01.
 *
 * Other available API versions: 2019-06-01, 2019-08-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-05-01.
 */
class NetworkSecurityGroup extends pulumi.CustomResource {
    /**
     * Get an existing NetworkSecurityGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkSecurityGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkSecurityGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkSecurityGroup.__pulumiType;
    }
    /**
     * Create a NetworkSecurityGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["flushConnection"] = args ? args.flushConnection : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["networkSecurityGroupName"] = args ? args.networkSecurityGroupName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["securityRules"] = args ? args.securityRules : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["defaultSecurityRules"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["flowLogs"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkInterfaces"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["subnets"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["defaultSecurityRules"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["flowLogs"] = undefined /*out*/;
            resourceInputs["flushConnection"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkInterfaces"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["securityRules"] = undefined /*out*/;
            resourceInputs["subnets"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20150501preview:NetworkSecurityGroup" }, { type: "azure-native:network/v20150615:NetworkSecurityGroup" }, { type: "azure-native:network/v20160330:NetworkSecurityGroup" }, { type: "azure-native:network/v20160601:NetworkSecurityGroup" }, { type: "azure-native:network/v20160901:NetworkSecurityGroup" }, { type: "azure-native:network/v20161201:NetworkSecurityGroup" }, { type: "azure-native:network/v20170301:NetworkSecurityGroup" }, { type: "azure-native:network/v20170601:NetworkSecurityGroup" }, { type: "azure-native:network/v20170801:NetworkSecurityGroup" }, { type: "azure-native:network/v20170901:NetworkSecurityGroup" }, { type: "azure-native:network/v20171001:NetworkSecurityGroup" }, { type: "azure-native:network/v20171101:NetworkSecurityGroup" }, { type: "azure-native:network/v20180101:NetworkSecurityGroup" }, { type: "azure-native:network/v20180201:NetworkSecurityGroup" }, { type: "azure-native:network/v20180401:NetworkSecurityGroup" }, { type: "azure-native:network/v20180601:NetworkSecurityGroup" }, { type: "azure-native:network/v20180701:NetworkSecurityGroup" }, { type: "azure-native:network/v20180801:NetworkSecurityGroup" }, { type: "azure-native:network/v20181001:NetworkSecurityGroup" }, { type: "azure-native:network/v20181101:NetworkSecurityGroup" }, { type: "azure-native:network/v20181201:NetworkSecurityGroup" }, { type: "azure-native:network/v20190201:NetworkSecurityGroup" }, { type: "azure-native:network/v20190401:NetworkSecurityGroup" }, { type: "azure-native:network/v20190601:NetworkSecurityGroup" }, { type: "azure-native:network/v20190701:NetworkSecurityGroup" }, { type: "azure-native:network/v20190801:NetworkSecurityGroup" }, { type: "azure-native:network/v20190901:NetworkSecurityGroup" }, { type: "azure-native:network/v20191101:NetworkSecurityGroup" }, { type: "azure-native:network/v20191201:NetworkSecurityGroup" }, { type: "azure-native:network/v20200301:NetworkSecurityGroup" }, { type: "azure-native:network/v20200401:NetworkSecurityGroup" }, { type: "azure-native:network/v20200501:NetworkSecurityGroup" }, { type: "azure-native:network/v20200601:NetworkSecurityGroup" }, { type: "azure-native:network/v20200701:NetworkSecurityGroup" }, { type: "azure-native:network/v20200801:NetworkSecurityGroup" }, { type: "azure-native:network/v20201101:NetworkSecurityGroup" }, { type: "azure-native:network/v20210201:NetworkSecurityGroup" }, { type: "azure-native:network/v20210301:NetworkSecurityGroup" }, { type: "azure-native:network/v20210501:NetworkSecurityGroup" }, { type: "azure-native:network/v20210801:NetworkSecurityGroup" }, { type: "azure-native:network/v20220101:NetworkSecurityGroup" }, { type: "azure-native:network/v20220501:NetworkSecurityGroup" }, { type: "azure-native:network/v20220701:NetworkSecurityGroup" }, { type: "azure-native:network/v20220901:NetworkSecurityGroup" }, { type: "azure-native:network/v20221101:NetworkSecurityGroup" }, { type: "azure-native:network/v20230201:NetworkSecurityGroup" }, { type: "azure-native:network/v20230401:NetworkSecurityGroup" }, { type: "azure-native:network/v20230501:NetworkSecurityGroup" }, { type: "azure-native:network/v20230601:NetworkSecurityGroup" }, { type: "azure-native:network/v20230901:NetworkSecurityGroup" }, { type: "azure-native:network/v20231101:NetworkSecurityGroup" }, { type: "azure-native:network/v20240101:NetworkSecurityGroup" }, { type: "azure-native:network/v20240301:NetworkSecurityGroup" }, { type: "azure-native:network/v20240501:NetworkSecurityGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NetworkSecurityGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NetworkSecurityGroup = NetworkSecurityGroup;
/** @internal */
NetworkSecurityGroup.__pulumiType = 'azure-native:network:NetworkSecurityGroup';
