// @pulumi/azure-native/network/networkWatcher.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkWatcher = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Network watcher in a resource group.
 * Azure REST API version: 2023-02-01. Prior API version in Azure Native 1.x: 2020-11-01.
 *
 * Other available API versions: 2022-05-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-05-01.
 */
class NetworkWatcher extends pulumi.CustomResource {
    /**
     * Get an existing NetworkWatcher resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkWatcher(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkWatcher.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkWatcher.__pulumiType;
    }
    /**
     * Create a NetworkWatcher resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["networkWatcherName"] = args ? args.networkWatcherName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20160901:NetworkWatcher" }, { type: "azure-native:network/v20161201:NetworkWatcher" }, { type: "azure-native:network/v20170301:NetworkWatcher" }, { type: "azure-native:network/v20170601:NetworkWatcher" }, { type: "azure-native:network/v20170801:NetworkWatcher" }, { type: "azure-native:network/v20170901:NetworkWatcher" }, { type: "azure-native:network/v20171001:NetworkWatcher" }, { type: "azure-native:network/v20171101:NetworkWatcher" }, { type: "azure-native:network/v20180101:NetworkWatcher" }, { type: "azure-native:network/v20180201:NetworkWatcher" }, { type: "azure-native:network/v20180401:NetworkWatcher" }, { type: "azure-native:network/v20180601:NetworkWatcher" }, { type: "azure-native:network/v20180701:NetworkWatcher" }, { type: "azure-native:network/v20180801:NetworkWatcher" }, { type: "azure-native:network/v20181001:NetworkWatcher" }, { type: "azure-native:network/v20181101:NetworkWatcher" }, { type: "azure-native:network/v20181201:NetworkWatcher" }, { type: "azure-native:network/v20190201:NetworkWatcher" }, { type: "azure-native:network/v20190401:NetworkWatcher" }, { type: "azure-native:network/v20190601:NetworkWatcher" }, { type: "azure-native:network/v20190701:NetworkWatcher" }, { type: "azure-native:network/v20190801:NetworkWatcher" }, { type: "azure-native:network/v20190901:NetworkWatcher" }, { type: "azure-native:network/v20191101:NetworkWatcher" }, { type: "azure-native:network/v20191201:NetworkWatcher" }, { type: "azure-native:network/v20200301:NetworkWatcher" }, { type: "azure-native:network/v20200401:NetworkWatcher" }, { type: "azure-native:network/v20200501:NetworkWatcher" }, { type: "azure-native:network/v20200601:NetworkWatcher" }, { type: "azure-native:network/v20200701:NetworkWatcher" }, { type: "azure-native:network/v20200801:NetworkWatcher" }, { type: "azure-native:network/v20201101:NetworkWatcher" }, { type: "azure-native:network/v20210201:NetworkWatcher" }, { type: "azure-native:network/v20210301:NetworkWatcher" }, { type: "azure-native:network/v20210501:NetworkWatcher" }, { type: "azure-native:network/v20210801:NetworkWatcher" }, { type: "azure-native:network/v20220101:NetworkWatcher" }, { type: "azure-native:network/v20220501:NetworkWatcher" }, { type: "azure-native:network/v20220701:NetworkWatcher" }, { type: "azure-native:network/v20220901:NetworkWatcher" }, { type: "azure-native:network/v20221101:NetworkWatcher" }, { type: "azure-native:network/v20230201:NetworkWatcher" }, { type: "azure-native:network/v20230401:NetworkWatcher" }, { type: "azure-native:network/v20230501:NetworkWatcher" }, { type: "azure-native:network/v20230601:NetworkWatcher" }, { type: "azure-native:network/v20230901:NetworkWatcher" }, { type: "azure-native:network/v20231101:NetworkWatcher" }, { type: "azure-native:network/v20240101:NetworkWatcher" }, { type: "azure-native:network/v20240301:NetworkWatcher" }, { type: "azure-native:network/v20240501:NetworkWatcher" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NetworkWatcher.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NetworkWatcher = NetworkWatcher;
/** @internal */
NetworkWatcher.__pulumiType = 'azure-native:network:NetworkWatcher';
