// @pulumi/azure-native/network/virtualWan.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualWan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * VirtualWAN Resource.
 * Azure REST API version: 2023-02-01. Prior API version in Azure Native 1.x: 2020-11-01.
 *
 * Other available API versions: 2019-07-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-05-01.
 */
class VirtualWan extends pulumi.CustomResource {
    /**
     * Get an existing VirtualWan resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualWan(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualWan.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualWan.__pulumiType;
    }
    /**
     * Create a VirtualWan resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["allowBranchToBranchTraffic"] = args ? args.allowBranchToBranchTraffic : undefined;
            resourceInputs["allowVnetToVnetTraffic"] = args ? args.allowVnetToVnetTraffic : undefined;
            resourceInputs["disableVpnEncryption"] = args ? args.disableVpnEncryption : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["virtualWANName"] = args ? args.virtualWANName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["office365LocalBreakoutCategory"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["virtualHubs"] = undefined /*out*/;
            resourceInputs["vpnSites"] = undefined /*out*/;
        }
        else {
            resourceInputs["allowBranchToBranchTraffic"] = undefined /*out*/;
            resourceInputs["allowVnetToVnetTraffic"] = undefined /*out*/;
            resourceInputs["disableVpnEncryption"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["office365LocalBreakoutCategory"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualHubs"] = undefined /*out*/;
            resourceInputs["vpnSites"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20180401:VirtualWan" }, { type: "azure-native:network/v20180601:VirtualWan" }, { type: "azure-native:network/v20180701:VirtualWan" }, { type: "azure-native:network/v20180801:VirtualWan" }, { type: "azure-native:network/v20181001:VirtualWan" }, { type: "azure-native:network/v20181101:VirtualWan" }, { type: "azure-native:network/v20181201:VirtualWan" }, { type: "azure-native:network/v20190201:VirtualWan" }, { type: "azure-native:network/v20190401:VirtualWan" }, { type: "azure-native:network/v20190601:VirtualWan" }, { type: "azure-native:network/v20190701:VirtualWan" }, { type: "azure-native:network/v20190801:VirtualWan" }, { type: "azure-native:network/v20190901:VirtualWan" }, { type: "azure-native:network/v20191101:VirtualWan" }, { type: "azure-native:network/v20191201:VirtualWan" }, { type: "azure-native:network/v20200301:VirtualWan" }, { type: "azure-native:network/v20200401:VirtualWan" }, { type: "azure-native:network/v20200501:VirtualWan" }, { type: "azure-native:network/v20200601:VirtualWan" }, { type: "azure-native:network/v20200701:VirtualWan" }, { type: "azure-native:network/v20200801:VirtualWan" }, { type: "azure-native:network/v20201101:VirtualWan" }, { type: "azure-native:network/v20210201:VirtualWan" }, { type: "azure-native:network/v20210301:VirtualWan" }, { type: "azure-native:network/v20210501:VirtualWan" }, { type: "azure-native:network/v20210801:VirtualWan" }, { type: "azure-native:network/v20220101:VirtualWan" }, { type: "azure-native:network/v20220501:VirtualWan" }, { type: "azure-native:network/v20220701:VirtualWan" }, { type: "azure-native:network/v20220901:VirtualWan" }, { type: "azure-native:network/v20221101:VirtualWan" }, { type: "azure-native:network/v20230201:VirtualWan" }, { type: "azure-native:network/v20230401:VirtualWan" }, { type: "azure-native:network/v20230501:VirtualWan" }, { type: "azure-native:network/v20230601:VirtualWan" }, { type: "azure-native:network/v20230901:VirtualWan" }, { type: "azure-native:network/v20231101:VirtualWan" }, { type: "azure-native:network/v20240101:VirtualWan" }, { type: "azure-native:network/v20240301:VirtualWan" }, { type: "azure-native:network/v20240501:VirtualWan" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualWan.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VirtualWan = VirtualWan;
/** @internal */
VirtualWan.__pulumiType = 'azure-native:network:VirtualWan';
