// @pulumi/azure-native/network/virtualApplianceSite.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualApplianceSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Virtual Appliance Site resource.
 * Azure REST API version: 2023-02-01. Prior API version in Azure Native 1.x: 2020-11-01.
 *
 * Other available API versions: 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-05-01.
 */
class VirtualApplianceSite extends pulumi.CustomResource {
    /**
     * Get an existing VirtualApplianceSite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualApplianceSite(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualApplianceSite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualApplianceSite.__pulumiType;
    }
    /**
     * Create a VirtualApplianceSite resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.networkVirtualApplianceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkVirtualApplianceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["addressPrefix"] = args ? args.addressPrefix : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["networkVirtualApplianceName"] = args ? args.networkVirtualApplianceName : undefined;
            resourceInputs["o365Policy"] = args ? args.o365Policy : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["siteName"] = args ? args.siteName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["addressPrefix"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["o365Policy"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20200501:VirtualApplianceSite" }, { type: "azure-native:network/v20200601:VirtualApplianceSite" }, { type: "azure-native:network/v20200701:VirtualApplianceSite" }, { type: "azure-native:network/v20200801:VirtualApplianceSite" }, { type: "azure-native:network/v20201101:VirtualApplianceSite" }, { type: "azure-native:network/v20210201:VirtualApplianceSite" }, { type: "azure-native:network/v20210301:VirtualApplianceSite" }, { type: "azure-native:network/v20210501:VirtualApplianceSite" }, { type: "azure-native:network/v20210801:VirtualApplianceSite" }, { type: "azure-native:network/v20220101:VirtualApplianceSite" }, { type: "azure-native:network/v20220501:VirtualApplianceSite" }, { type: "azure-native:network/v20220701:VirtualApplianceSite" }, { type: "azure-native:network/v20220901:VirtualApplianceSite" }, { type: "azure-native:network/v20221101:VirtualApplianceSite" }, { type: "azure-native:network/v20230201:VirtualApplianceSite" }, { type: "azure-native:network/v20230401:VirtualApplianceSite" }, { type: "azure-native:network/v20230501:VirtualApplianceSite" }, { type: "azure-native:network/v20230601:VirtualApplianceSite" }, { type: "azure-native:network/v20230901:VirtualApplianceSite" }, { type: "azure-native:network/v20231101:VirtualApplianceSite" }, { type: "azure-native:network/v20240101:VirtualApplianceSite" }, { type: "azure-native:network/v20240301:VirtualApplianceSite" }, { type: "azure-native:network/v20240501:VirtualApplianceSite" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualApplianceSite.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VirtualApplianceSite = VirtualApplianceSite;
/** @internal */
VirtualApplianceSite.__pulumiType = 'azure-native:network:VirtualApplianceSite';
