// @pulumi/azure-compliance-policies/azurenative/network/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240601preview = exports.v20240101preview = exports.v20230801preview = exports.v20230701preview = exports.v20220401preview = exports.v20210501preview = exports.v20210301preview = exports.v20210201preview = exports.v20200401preview = void 0;
exports.v20200401preview = require("./v20200401preview");
exports.v20210201preview = require("./v20210201preview");
exports.v20210301preview = require("./v20210301preview");
exports.v20210501preview = require("./v20210501preview");
exports.v20220401preview = require("./v20220401preview");
exports.v20230701preview = require("./v20230701preview");
exports.v20230801preview = require("./v20230801preview");
exports.v20240101preview = require("./v20240101preview");
exports.v20240601preview = require("./v20240601preview");
