// @pulumi/azure/loganalytics/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceTable = exports.StorageInsights = exports.SavedSearch = exports.QueryPack = exports.LinkedStorageAccount = exports.LinkedService = exports.DataSourceWindowsPerformanceCounter = exports.DataSourceWindowsEvent = exports.DataExportRule = exports.ClusterCustomerManagedKey = exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ClusterCustomerManagedKey = null;
utilities.lazyLoad(exports, ["ClusterCustomerManagedKey"], () => require("./clusterCustomerManagedKey"));
exports.DataExportRule = null;
utilities.lazyLoad(exports, ["DataExportRule"], () => require("./dataExportRule"));
exports.DataSourceWindowsEvent = null;
utilities.lazyLoad(exports, ["DataSourceWindowsEvent"], () => require("./dataSourceWindowsEvent"));
exports.DataSourceWindowsPerformanceCounter = null;
utilities.lazyLoad(exports, ["DataSourceWindowsPerformanceCounter"], () => require("./dataSourceWindowsPerformanceCounter"));
exports.LinkedService = null;
utilities.lazyLoad(exports, ["LinkedService"], () => require("./linkedService"));
exports.LinkedStorageAccount = null;
utilities.lazyLoad(exports, ["LinkedStorageAccount"], () => require("./linkedStorageAccount"));
exports.QueryPack = null;
utilities.lazyLoad(exports, ["QueryPack"], () => require("./queryPack"));
exports.SavedSearch = null;
utilities.lazyLoad(exports, ["SavedSearch"], () => require("./savedSearch"));
exports.StorageInsights = null;
utilities.lazyLoad(exports, ["StorageInsights"], () => require("./storageInsights"));
exports.WorkspaceTable = null;
utilities.lazyLoad(exports, ["WorkspaceTable"], () => require("./workspaceTable"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:loganalytics/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure:loganalytics/clusterCustomerManagedKey:ClusterCustomerManagedKey":
                return new exports.ClusterCustomerManagedKey(name, undefined, { urn });
            case "azure:loganalytics/dataExportRule:DataExportRule":
                return new exports.DataExportRule(name, undefined, { urn });
            case "azure:loganalytics/dataSourceWindowsEvent:DataSourceWindowsEvent":
                return new exports.DataSourceWindowsEvent(name, undefined, { urn });
            case "azure:loganalytics/dataSourceWindowsPerformanceCounter:DataSourceWindowsPerformanceCounter":
                return new exports.DataSourceWindowsPerformanceCounter(name, undefined, { urn });
            case "azure:loganalytics/linkedService:LinkedService":
                return new exports.LinkedService(name, undefined, { urn });
            case "azure:loganalytics/linkedStorageAccount:LinkedStorageAccount":
                return new exports.LinkedStorageAccount(name, undefined, { urn });
            case "azure:loganalytics/queryPack:QueryPack":
                return new exports.QueryPack(name, undefined, { urn });
            case "azure:loganalytics/savedSearch:SavedSearch":
                return new exports.SavedSearch(name, undefined, { urn });
            case "azure:loganalytics/storageInsights:StorageInsights":
                return new exports.StorageInsights(name, undefined, { urn });
            case "azure:loganalytics/workspaceTable:WorkspaceTable":
                return new exports.WorkspaceTable(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "loganalytics/cluster", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/clusterCustomerManagedKey", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/dataExportRule", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/dataSourceWindowsEvent", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/dataSourceWindowsPerformanceCounter", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/linkedService", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/linkedStorageAccount", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/queryPack", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/savedSearch", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/storageInsights", _module);
pulumi.runtime.registerResourceModule("azure", "loganalytics/workspaceTable", _module);
