// @pulumi/azure/securitycenter/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.SubscriptionPricing = exports.StorageDefender = exports.Setting = exports.ServerVulnerabilityAssessmentsSetting = exports.ServerVulnerabilityAssessmentVirtualMachine = exports.Contact = exports.Automation = exports.AutoProvisioning = exports.AssessmentPolicy = exports.Assessment = exports.AdvancedThreatProtection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AdvancedThreatProtection = null;
utilities.lazyLoad(exports, ["AdvancedThreatProtection"], () => require("./advancedThreatProtection"));
exports.Assessment = null;
utilities.lazyLoad(exports, ["Assessment"], () => require("./assessment"));
exports.AssessmentPolicy = null;
utilities.lazyLoad(exports, ["AssessmentPolicy"], () => require("./assessmentPolicy"));
exports.AutoProvisioning = null;
utilities.lazyLoad(exports, ["AutoProvisioning"], () => require("./autoProvisioning"));
exports.Automation = null;
utilities.lazyLoad(exports, ["Automation"], () => require("./automation"));
exports.Contact = null;
utilities.lazyLoad(exports, ["Contact"], () => require("./contact"));
exports.ServerVulnerabilityAssessmentVirtualMachine = null;
utilities.lazyLoad(exports, ["ServerVulnerabilityAssessmentVirtualMachine"], () => require("./serverVulnerabilityAssessmentVirtualMachine"));
exports.ServerVulnerabilityAssessmentsSetting = null;
utilities.lazyLoad(exports, ["ServerVulnerabilityAssessmentsSetting"], () => require("./serverVulnerabilityAssessmentsSetting"));
exports.Setting = null;
utilities.lazyLoad(exports, ["Setting"], () => require("./setting"));
exports.StorageDefender = null;
utilities.lazyLoad(exports, ["StorageDefender"], () => require("./storageDefender"));
exports.SubscriptionPricing = null;
utilities.lazyLoad(exports, ["SubscriptionPricing"], () => require("./subscriptionPricing"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:securitycenter/advancedThreatProtection:AdvancedThreatProtection":
                return new exports.AdvancedThreatProtection(name, undefined, { urn });
            case "azure:securitycenter/assessment:Assessment":
                return new exports.Assessment(name, undefined, { urn });
            case "azure:securitycenter/assessmentPolicy:AssessmentPolicy":
                return new exports.AssessmentPolicy(name, undefined, { urn });
            case "azure:securitycenter/autoProvisioning:AutoProvisioning":
                return new exports.AutoProvisioning(name, undefined, { urn });
            case "azure:securitycenter/automation:Automation":
                return new exports.Automation(name, undefined, { urn });
            case "azure:securitycenter/contact:Contact":
                return new exports.Contact(name, undefined, { urn });
            case "azure:securitycenter/serverVulnerabilityAssessmentVirtualMachine:ServerVulnerabilityAssessmentVirtualMachine":
                return new exports.ServerVulnerabilityAssessmentVirtualMachine(name, undefined, { urn });
            case "azure:securitycenter/serverVulnerabilityAssessmentsSetting:ServerVulnerabilityAssessmentsSetting":
                return new exports.ServerVulnerabilityAssessmentsSetting(name, undefined, { urn });
            case "azure:securitycenter/setting:Setting":
                return new exports.Setting(name, undefined, { urn });
            case "azure:securitycenter/storageDefender:StorageDefender":
                return new exports.StorageDefender(name, undefined, { urn });
            case "azure:securitycenter/subscriptionPricing:SubscriptionPricing":
                return new exports.SubscriptionPricing(name, undefined, { urn });
            case "azure:securitycenter/workspace:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "securitycenter/advancedThreatProtection", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/assessment", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/assessmentPolicy", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/autoProvisioning", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/automation", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/contact", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/serverVulnerabilityAssessmentVirtualMachine", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/serverVulnerabilityAssessmentsSetting", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/setting", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/storageDefender", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/subscriptionPricing", _module);
pulumi.runtime.registerResourceModule("azure", "securitycenter/workspace", _module);
