// @pulumi/azure-compliance-policies/azurenative/azurestackhci/v20231101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSummary = exports.UpdateRun = exports.Update = exports.SecuritySetting = exports.Extension = exports.EdgeDevice = exports.DeploymentSetting = exports.Cluster = exports.ArcSetting = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20231101preview_1 = require("@pulumi/azure-native/azurestackhci/v20231101preview");
const v20231101preview_2 = require("@pulumi/azure-native/azurestackhci/v20231101preview");
const v20231101preview_3 = require("@pulumi/azure-native/azurestackhci/v20231101preview");
const v20231101preview_4 = require("@pulumi/azure-native/azurestackhci/v20231101preview");
const v20231101preview_5 = require("@pulumi/azure-native/azurestackhci/v20231101preview");
const v20231101preview_6 = require("@pulumi/azure-native/azurestackhci/v20231101preview");
const v20231101preview_7 = require("@pulumi/azure-native/azurestackhci/v20231101preview");
const v20231101preview_8 = require("@pulumi/azure-native/azurestackhci/v20231101preview");
const v20231101preview_9 = require("@pulumi/azure-native/azurestackhci/v20231101preview");
var ArcSetting;
(function (ArcSetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.ArcSetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ArcSetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20231101preview-arcsetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.ArcSetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_1.ArcSetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ArcSetting shouldn't use an unstable API (azurestackhci.v20231101preview.ArcSetting). A compatible replacement can be found at 'azurestackhci.ArcSetting'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ArcSetting || (ArcSetting = {}));
exports.ArcSetting = ArcSetting;
var Cluster;
(function (Cluster) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.Cluster).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Cluster.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20231101preview-cluster-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.Cluster).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_2.Cluster, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Cluster shouldn't use an unstable API (azurestackhci.v20231101preview.Cluster). A compatible replacement can be found at 'azurestackhci.Cluster'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Cluster || (Cluster = {}));
exports.Cluster = Cluster;
var DeploymentSetting;
(function (DeploymentSetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.DeploymentSetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DeploymentSetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20231101preview-deploymentsetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.DeploymentSetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_3.DeploymentSetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DeploymentSetting shouldn't use an unstable API (azurestackhci.v20231101preview.DeploymentSetting). A compatible replacement can be found at 'azurestackhci.DeploymentSetting'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DeploymentSetting || (DeploymentSetting = {}));
exports.DeploymentSetting = DeploymentSetting;
var EdgeDevice;
(function (EdgeDevice) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.EdgeDevice).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EdgeDevice.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20231101preview-edgedevice-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.EdgeDevice).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_4.EdgeDevice, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EdgeDevice shouldn't use an unstable API (azurestackhci.v20231101preview.EdgeDevice). A compatible replacement can be found at 'azurestackhci.EdgeDevice'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EdgeDevice || (EdgeDevice = {}));
exports.EdgeDevice = EdgeDevice;
var Extension;
(function (Extension) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.Extension).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Extension.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20231101preview-extension-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.Extension).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_5.Extension, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Extension shouldn't use an unstable API (azurestackhci.v20231101preview.Extension). A compatible replacement can be found at 'azurestackhci.Extension'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Extension || (Extension = {}));
exports.Extension = Extension;
var SecuritySetting;
(function (SecuritySetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.SecuritySetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SecuritySetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20231101preview-securitysetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.SecuritySetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_6.SecuritySetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SecuritySetting shouldn't use an unstable API (azurestackhci.v20231101preview.SecuritySetting). A compatible replacement can be found at 'azurestackhci.SecuritySetting'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SecuritySetting || (SecuritySetting = {}));
exports.SecuritySetting = SecuritySetting;
var Update;
(function (Update) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.Update).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Update.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20231101preview-update-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.Update).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_7.Update, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Update shouldn't use an unstable API (azurestackhci.v20231101preview.Update). A compatible replacement can be found at 'azurestackhci.Update'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Update || (Update = {}));
exports.Update = Update;
var UpdateRun;
(function (UpdateRun) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.UpdateRun).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    UpdateRun.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20231101preview-updaterun-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.UpdateRun).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_8.UpdateRun, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure UpdateRun shouldn't use an unstable API (azurestackhci.v20231101preview.UpdateRun). A compatible replacement can be found at 'azurestackhci.UpdateRun'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(UpdateRun || (UpdateRun = {}));
exports.UpdateRun = UpdateRun;
var UpdateSummary;
(function (UpdateSummary) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.UpdateSummary).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    UpdateSummary.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20231101preview-updatesummary-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20231101preview.UpdateSummary).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20231101preview_9.UpdateSummary, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure UpdateSummary shouldn't use an unstable API (azurestackhci.v20231101preview.UpdateSummary). A compatible replacement can be found at 'azurestackhci.UpdateSummary'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(UpdateSummary || (UpdateSummary = {}));
exports.UpdateSummary = UpdateSummary;
