// @pulumi/azure/communication/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceEmailDomainAssociation = exports.Service = exports.getServiceOutput = exports.getService = exports.EmailServiceDomain = exports.EmailService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.EmailService = null;
utilities.lazyLoad(exports, ["EmailService"], () => require("./emailService"));
exports.EmailServiceDomain = null;
utilities.lazyLoad(exports, ["EmailServiceDomain"], () => require("./emailServiceDomain"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceEmailDomainAssociation = null;
utilities.lazyLoad(exports, ["ServiceEmailDomainAssociation"], () => require("./serviceEmailDomainAssociation"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:communication/emailService:EmailService":
                return new exports.EmailService(name, undefined, { urn });
            case "azure:communication/emailServiceDomain:EmailServiceDomain":
                return new exports.EmailServiceDomain(name, undefined, { urn });
            case "azure:communication/service:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure:communication/serviceEmailDomainAssociation:ServiceEmailDomainAssociation":
                return new exports.ServiceEmailDomainAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "communication/emailService", _module);
pulumi.runtime.registerResourceModule("azure", "communication/emailServiceDomain", _module);
pulumi.runtime.registerResourceModule("azure", "communication/service", _module);
pulumi.runtime.registerResourceModule("azure", "communication/serviceEmailDomainAssociation", _module);
