// @pulumi/azure/appservice/environmentV3.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentV3 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a 3rd Generation (v3) App Service Environment.
 *
 * ## Example Usage
 *
 * This example provisions an App Service Environment V3. Additional examples of how to use the `azure.appservice.EnvironmentV3` resource can be found in the `./examples/app-service-environment-v3` directory within the GitHub Repository.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "exampleRG1",
 *     location: "West Europe",
 * });
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("example", {
 *     name: "example-vnet",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     addressSpaces: ["10.0.0.0/16"],
 * });
 * const exampleSubnet = new azure.network.Subnet("example", {
 *     name: "example-subnet",
 *     resourceGroupName: example.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.0.2.0/24"],
 *     delegations: [{
 *         name: "Microsoft.Web.hostingEnvironments",
 *         serviceDelegation: {
 *             name: "Microsoft.Web/hostingEnvironments",
 *             actions: ["Microsoft.Network/virtualNetworks/subnets/action"],
 *         },
 *     }],
 * });
 * const exampleEnvironmentV3 = new azure.appservice.EnvironmentV3("example", {
 *     name: "example-asev3",
 *     resourceGroupName: example.name,
 *     subnetId: exampleSubnet.id,
 *     internalLoadBalancingMode: "Web, Publishing",
 *     clusterSettings: [
 *         {
 *             name: "DisableTls1.0",
 *             value: "1",
 *         },
 *         {
 *             name: "InternalEncryption",
 *             value: "true",
 *         },
 *         {
 *             name: "FrontEndSSLCipherSuiteOrder",
 *             value: "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
 *         },
 *     ],
 *     tags: {
 *         env: "production",
 *         terraformed: "true",
 *     },
 * });
 * const exampleServicePlan = new azure.appservice.ServicePlan("example", {
 *     name: "example",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     osType: "Linux",
 *     skuName: "I1v2",
 *     appServiceEnvironmentId: exampleEnvironmentV3.id,
 * });
 * ```
 *
 * ## Import
 *
 * A 3rd Generation (v3) App Service Environment can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:appservice/environmentV3:EnvironmentV3 myAppServiceEnv /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.Web/hostingEnvironments/myAppServiceEnv
 * ```
 */
class EnvironmentV3 extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["allowNewPrivateEndpointConnections"] = state ? state.allowNewPrivateEndpointConnections : undefined;
            resourceInputs["clusterSettings"] = state ? state.clusterSettings : undefined;
            resourceInputs["dedicatedHostCount"] = state ? state.dedicatedHostCount : undefined;
            resourceInputs["dnsSuffix"] = state ? state.dnsSuffix : undefined;
            resourceInputs["externalInboundIpAddresses"] = state ? state.externalInboundIpAddresses : undefined;
            resourceInputs["inboundNetworkDependencies"] = state ? state.inboundNetworkDependencies : undefined;
            resourceInputs["internalInboundIpAddresses"] = state ? state.internalInboundIpAddresses : undefined;
            resourceInputs["internalLoadBalancingMode"] = state ? state.internalLoadBalancingMode : undefined;
            resourceInputs["ipSslAddressCount"] = state ? state.ipSslAddressCount : undefined;
            resourceInputs["linuxOutboundIpAddresses"] = state ? state.linuxOutboundIpAddresses : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["pricingTier"] = state ? state.pricingTier : undefined;
            resourceInputs["remoteDebuggingEnabled"] = state ? state.remoteDebuggingEnabled : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["subnetId"] = state ? state.subnetId : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["windowsOutboundIpAddresses"] = state ? state.windowsOutboundIpAddresses : undefined;
            resourceInputs["zoneRedundant"] = state ? state.zoneRedundant : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.subnetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnetId'");
            }
            resourceInputs["allowNewPrivateEndpointConnections"] = args ? args.allowNewPrivateEndpointConnections : undefined;
            resourceInputs["clusterSettings"] = args ? args.clusterSettings : undefined;
            resourceInputs["dedicatedHostCount"] = args ? args.dedicatedHostCount : undefined;
            resourceInputs["internalLoadBalancingMode"] = args ? args.internalLoadBalancingMode : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["remoteDebuggingEnabled"] = args ? args.remoteDebuggingEnabled : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["subnetId"] = args ? args.subnetId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["zoneRedundant"] = args ? args.zoneRedundant : undefined;
            resourceInputs["dnsSuffix"] = undefined /*out*/;
            resourceInputs["externalInboundIpAddresses"] = undefined /*out*/;
            resourceInputs["inboundNetworkDependencies"] = undefined /*out*/;
            resourceInputs["internalInboundIpAddresses"] = undefined /*out*/;
            resourceInputs["ipSslAddressCount"] = undefined /*out*/;
            resourceInputs["linuxOutboundIpAddresses"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["pricingTier"] = undefined /*out*/;
            resourceInputs["windowsOutboundIpAddresses"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EnvironmentV3.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing EnvironmentV3 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EnvironmentV3(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EnvironmentV3.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnvironmentV3.__pulumiType;
    }
}
exports.EnvironmentV3 = EnvironmentV3;
/** @internal */
EnvironmentV3.__pulumiType = 'azure:appservice/environmentV3:EnvironmentV3';
