// @pulumi/azure/appservice/managedCertificate.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedCertificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * This certificate can be used to secure custom domains on App Services (Windows and Linux) hosted on an App Service Plan of Basic and above (free and shared tiers are not supported).
 *
 * > NOTE: A certificate is valid for six months, and about a month before the certificate’s expiration date, App Services renews/rotates the certificate. This is managed by Azure and doesn't require this resource to be changed or reprovisioned. It will change the `thumbprint` computed attribute the next time the resource is refreshed after rotation occurs, so keep that in mind if you have any dependencies on this attribute directly.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as std from "@pulumi/std";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const example = azure.dns.getZoneOutput({
 *     name: "mydomain.com",
 *     resourceGroupName: exampleResourceGroup.name,
 * });
 * const examplePlan = new azure.appservice.Plan("example", {
 *     name: "example-plan",
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     kind: "Linux",
 *     reserved: true,
 *     sku: {
 *         tier: "Basic",
 *         size: "B1",
 *     },
 * });
 * const exampleAppService = new azure.appservice.AppService("example", {
 *     name: "example-app",
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     appServicePlanId: examplePlan.id,
 * });
 * const exampleTxtRecord = new azure.dns.TxtRecord("example", {
 *     name: "asuid.mycustomhost.contoso.com",
 *     zoneName: example.apply(example => example.name),
 *     resourceGroupName: example.apply(example => example.resourceGroupName),
 *     ttl: 300,
 *     records: [{
 *         value: exampleAppService.customDomainVerificationId,
 *     }],
 * });
 * const exampleCNameRecord = new azure.dns.CNameRecord("example", {
 *     name: "example-adcr",
 *     zoneName: example.apply(example => example.name),
 *     resourceGroupName: example.apply(example => example.resourceGroupName),
 *     ttl: 300,
 *     record: exampleAppService.defaultSiteHostname,
 * });
 * const exampleCustomHostnameBinding = new azure.appservice.CustomHostnameBinding("example", {
 *     hostname: std.joinOutput({
 *         separator: ".",
 *         input: [
 *             exampleCNameRecord.name,
 *             exampleCNameRecord.zoneName,
 *         ],
 *     }).apply(invoke => invoke.result),
 *     appServiceName: exampleAppService.name,
 *     resourceGroupName: exampleResourceGroup.name,
 * });
 * const exampleManagedCertificate = new azure.appservice.ManagedCertificate("example", {customHostnameBindingId: exampleCustomHostnameBinding.id});
 * const exampleCertificateBinding = new azure.appservice.CertificateBinding("example", {
 *     hostnameBindingId: exampleCustomHostnameBinding.id,
 *     certificateId: exampleManagedCertificate.id,
 *     sslState: "SniEnabled",
 * });
 * ```
 *
 * ## Import
 *
 * App Service Managed Certificates can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:appservice/managedCertificate:ManagedCertificate example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.Web/certificates/customhost.contoso.com
 * ```
 */
class ManagedCertificate extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["canonicalName"] = state ? state.canonicalName : undefined;
            resourceInputs["customHostnameBindingId"] = state ? state.customHostnameBindingId : undefined;
            resourceInputs["expirationDate"] = state ? state.expirationDate : undefined;
            resourceInputs["friendlyName"] = state ? state.friendlyName : undefined;
            resourceInputs["hostNames"] = state ? state.hostNames : undefined;
            resourceInputs["issueDate"] = state ? state.issueDate : undefined;
            resourceInputs["issuer"] = state ? state.issuer : undefined;
            resourceInputs["subjectName"] = state ? state.subjectName : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["thumbprint"] = state ? state.thumbprint : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.customHostnameBindingId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'customHostnameBindingId'");
            }
            resourceInputs["customHostnameBindingId"] = args ? args.customHostnameBindingId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["canonicalName"] = undefined /*out*/;
            resourceInputs["expirationDate"] = undefined /*out*/;
            resourceInputs["friendlyName"] = undefined /*out*/;
            resourceInputs["hostNames"] = undefined /*out*/;
            resourceInputs["issueDate"] = undefined /*out*/;
            resourceInputs["issuer"] = undefined /*out*/;
            resourceInputs["subjectName"] = undefined /*out*/;
            resourceInputs["thumbprint"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ManagedCertificate.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ManagedCertificate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ManagedCertificate(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagedCertificate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedCertificate.__pulumiType;
    }
}
exports.ManagedCertificate = ManagedCertificate;
/** @internal */
ManagedCertificate.__pulumiType = 'azure:appservice/managedCertificate:ManagedCertificate';
