// @pulumi/azure/appservice/staticWebAppCustomDomain.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticWebAppCustomDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * ### CNAME validation
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleStaticWebApp = new azure.appservice.StaticWebApp("example", {
 *     name: "example",
 *     resourceGroupName: example.name,
 *     location: example.location,
 * });
 * const exampleCNameRecord = new azure.dns.CNameRecord("example", {
 *     name: "my-domain",
 *     zoneName: "contoso.com",
 *     resourceGroupName: example.name,
 *     ttl: 300,
 *     record: exampleStaticWebApp.defaultHostName,
 * });
 * const exampleStaticWebAppCustomDomain = new azure.appservice.StaticWebAppCustomDomain("example", {
 *     staticWebAppId: exampleStaticWebApp.id,
 *     domainName: pulumi.interpolate`${exampleCNameRecord.name}.${exampleCNameRecord.zoneName}`,
 *     validationType: "cname-delegation",
 * });
 * ```
 *
 * ### TXT validation
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleStaticWebApp = new azure.appservice.StaticWebApp("example", {
 *     name: "example",
 *     resourceGroupName: example.name,
 *     location: example.location,
 * });
 * const exampleStaticWebAppCustomDomain = new azure.appservice.StaticWebAppCustomDomain("example", {
 *     staticWebAppId: exampleStaticWebApp.id,
 *     domainName: "my-domain.contoso.com",
 *     validationType: "dns-txt-token",
 * });
 * const exampleTxtRecord = new azure.dns.TxtRecord("example", {
 *     name: "_dnsauth.my-domain",
 *     zoneName: "contoso.com",
 *     resourceGroupName: example.name,
 *     ttl: 300,
 *     records: [{
 *         value: exampleStaticWebAppCustomDomain.validationToken,
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * Static Site Custom Domains can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:appservice/staticWebAppCustomDomain:StaticWebAppCustomDomain example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/group1/providers/Microsoft.Web/staticSites/my-static-site1/customDomains/name.contoso.com
 * ```
 */
class StaticWebAppCustomDomain extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["domainName"] = state ? state.domainName : undefined;
            resourceInputs["staticWebAppId"] = state ? state.staticWebAppId : undefined;
            resourceInputs["validationToken"] = state ? state.validationToken : undefined;
            resourceInputs["validationType"] = state ? state.validationType : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.domainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            if ((!args || args.staticWebAppId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'staticWebAppId'");
            }
            if ((!args || args.validationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'validationType'");
            }
            resourceInputs["domainName"] = args ? args.domainName : undefined;
            resourceInputs["staticWebAppId"] = args ? args.staticWebAppId : undefined;
            resourceInputs["validationType"] = args ? args.validationType : undefined;
            resourceInputs["validationToken"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["validationToken"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(StaticWebAppCustomDomain.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing StaticWebAppCustomDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new StaticWebAppCustomDomain(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StaticWebAppCustomDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StaticWebAppCustomDomain.__pulumiType;
    }
}
exports.StaticWebAppCustomDomain = StaticWebAppCustomDomain;
/** @internal */
StaticWebAppCustomDomain.__pulumiType = 'azure:appservice/staticWebAppCustomDomain:StaticWebAppCustomDomain';
