// @pulumi/azure/core/subscriptionPolicyExemption.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionPolicyExemption = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Subscription Policy Exemption.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = azure.core.getSubscription({});
 * const exampleGetPolicySetDefinition = azure.policy.getPolicySetDefinition({
 *     displayName: "Audit machines with insecure password security settings",
 * });
 * const exampleSubscriptionPolicyAssignment = new azure.core.SubscriptionPolicyAssignment("example", {
 *     name: "exampleAssignment",
 *     subscriptionId: example.then(example => example.id),
 *     policyDefinitionId: exampleGetPolicySetDefinition.then(exampleGetPolicySetDefinition => exampleGetPolicySetDefinition.id),
 *     location: "westus",
 *     identity: {
 *         type: "SystemAssigned",
 *     },
 * });
 * const exampleSubscriptionPolicyExemption = new azure.core.SubscriptionPolicyExemption("example", {
 *     name: "exampleExemption",
 *     subscriptionId: example.then(example => example.id),
 *     policyAssignmentId: exampleSubscriptionPolicyAssignment.id,
 *     exemptionCategory: "Mitigated",
 * });
 * ```
 *
 * ## Import
 *
 * Policy Exemptions can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:core/subscriptionPolicyExemption:SubscriptionPolicyExemption exemption1 /subscriptions/00000000-0000-0000-000000000000/providers/Microsoft.Authorization/policyExemptions/exemption1
 * ```
 */
class SubscriptionPolicyExemption extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["exemptionCategory"] = state ? state.exemptionCategory : undefined;
            resourceInputs["expiresOn"] = state ? state.expiresOn : undefined;
            resourceInputs["metadata"] = state ? state.metadata : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["policyAssignmentId"] = state ? state.policyAssignmentId : undefined;
            resourceInputs["policyDefinitionReferenceIds"] = state ? state.policyDefinitionReferenceIds : undefined;
            resourceInputs["subscriptionId"] = state ? state.subscriptionId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.exemptionCategory === undefined) && !opts.urn) {
                throw new Error("Missing required property 'exemptionCategory'");
            }
            if ((!args || args.policyAssignmentId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'policyAssignmentId'");
            }
            if ((!args || args.subscriptionId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subscriptionId'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["exemptionCategory"] = args ? args.exemptionCategory : undefined;
            resourceInputs["expiresOn"] = args ? args.expiresOn : undefined;
            resourceInputs["metadata"] = args ? args.metadata : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["policyAssignmentId"] = args ? args.policyAssignmentId : undefined;
            resourceInputs["policyDefinitionReferenceIds"] = args ? args.policyDefinitionReferenceIds : undefined;
            resourceInputs["subscriptionId"] = args ? args.subscriptionId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SubscriptionPolicyExemption.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SubscriptionPolicyExemption resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SubscriptionPolicyExemption(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SubscriptionPolicyExemption.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SubscriptionPolicyExemption.__pulumiType;
    }
}
exports.SubscriptionPolicyExemption = SubscriptionPolicyExemption;
/** @internal */
SubscriptionPolicyExemption.__pulumiType = 'azure:core/subscriptionPolicyExemption:SubscriptionPolicyExemption';
