// @pulumi/azure/privatelink/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getServiceEndpointConnectionsOutput = exports.getServiceEndpointConnections = exports.getServiceOutput = exports.getService = exports.getEndpointConnectionOutput = exports.getEndpointConnection = exports.Endpoint = exports.ApplicationSecurityGroupAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ApplicationSecurityGroupAssociation = null;
utilities.lazyLoad(exports, ["ApplicationSecurityGroupAssociation"], () => require("./applicationSecurityGroupAssociation"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.getEndpointConnection = null;
exports.getEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getEndpointConnection", "getEndpointConnectionOutput"], () => require("./getEndpointConnection"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServiceEndpointConnections = null;
exports.getServiceEndpointConnectionsOutput = null;
utilities.lazyLoad(exports, ["getServiceEndpointConnections", "getServiceEndpointConnectionsOutput"], () => require("./getServiceEndpointConnections"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:privatelink/applicationSecurityGroupAssociation:ApplicationSecurityGroupAssociation":
                return new exports.ApplicationSecurityGroupAssociation(name, undefined, { urn });
            case "azure:privatelink/endpoint:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "privatelink/applicationSecurityGroupAssociation", _module);
pulumi.runtime.registerResourceModule("azure", "privatelink/endpoint", _module);
