// @pulumi/azure-compliance-policies/azurenative/machinelearningservices/v20200501preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.MachineLearningDatastore = exports.MachineLearningDataset = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20200501preview_1 = require("@pulumi/azure-native/machinelearningservices/v20200501preview");
const v20200501preview_2 = require("@pulumi/azure-native/machinelearningservices/v20200501preview");
var MachineLearningDataset;
(function (MachineLearningDataset) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20200501preview.MachineLearningDataset).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MachineLearningDataset.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20200501preview-machinelearningdataset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20200501preview.MachineLearningDataset).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200501preview_1.MachineLearningDataset, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MachineLearningDataset shouldn't use an unstable API (machinelearningservices.v20200501preview.MachineLearningDataset). A compatible replacement can be found at 'machinelearningservices.MachineLearningDataset'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MachineLearningDataset || (MachineLearningDataset = {}));
exports.MachineLearningDataset = MachineLearningDataset;
var MachineLearningDatastore;
(function (MachineLearningDatastore) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20200501preview.MachineLearningDatastore).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MachineLearningDatastore.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20200501preview-machinelearningdatastore-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20200501preview.MachineLearningDatastore).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20200501preview_2.MachineLearningDatastore, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MachineLearningDatastore shouldn't use an unstable API (machinelearningservices.v20200501preview.MachineLearningDatastore). A compatible replacement can be found at 'machinelearningservices.MachineLearningDatastore'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MachineLearningDatastore || (MachineLearningDatastore = {}));
exports.MachineLearningDatastore = MachineLearningDatastore;
