// @pulumi/azure-compliance-policies/azurenative/azurelargeinstance/v20240801preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureLargeStorageInstance = exports.AzureLargeInstance = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240801preview_1 = require("@pulumi/azure-native/azurelargeinstance/v20240801preview");
const v20240801preview_2 = require("@pulumi/azure-native/azurelargeinstance/v20240801preview");
var AzureLargeInstance;
(function (AzureLargeInstance) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurelargeinstance.v20240801preview.AzureLargeInstance).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AzureLargeInstance.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurelargeinstance-v20240801preview-azurelargeinstance-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurelargeinstance.v20240801preview.AzureLargeInstance).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240801preview_1.AzureLargeInstance, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AzureLargeInstance shouldn't use an unstable API (azurelargeinstance.v20240801preview.AzureLargeInstance). A compatible replacement can be found at 'azurelargeinstance.AzureLargeInstance'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurelargeinstance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AzureLargeInstance || (AzureLargeInstance = {}));
exports.AzureLargeInstance = AzureLargeInstance;
var AzureLargeStorageInstance;
(function (AzureLargeStorageInstance) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurelargeinstance.v20240801preview.AzureLargeStorageInstance).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AzureLargeStorageInstance.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurelargeinstance-v20240801preview-azurelargestorageinstance-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurelargeinstance.v20240801preview.AzureLargeStorageInstance).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240801preview_2.AzureLargeStorageInstance, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AzureLargeStorageInstance shouldn't use an unstable API (azurelargeinstance.v20240801preview.AzureLargeStorageInstance). A compatible replacement can be found at 'azurelargeinstance.AzureLargeStorageInstance'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurelargeinstance"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AzureLargeStorageInstance || (AzureLargeStorageInstance = {}));
exports.AzureLargeStorageInstance = AzureLargeStorageInstance;
