// @pulumi/azure-compliance-policies/azurenative/apimanagement/v20180601preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagDescription = exports.ProductPolicy = exports.ProductApi = exports.Policy = exports.ApiPolicy = exports.ApiOperationPolicy = exports.Api = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20180601preview_1 = require("@pulumi/azure-native/apimanagement/v20180601preview");
const v20180601preview_2 = require("@pulumi/azure-native/apimanagement/v20180601preview");
const v20180601preview_3 = require("@pulumi/azure-native/apimanagement/v20180601preview");
const v20180601preview_4 = require("@pulumi/azure-native/apimanagement/v20180601preview");
const v20180601preview_5 = require("@pulumi/azure-native/apimanagement/v20180601preview");
const v20180601preview_6 = require("@pulumi/azure-native/apimanagement/v20180601preview");
const v20180601preview_7 = require("@pulumi/azure-native/apimanagement/v20180601preview");
var Api;
(function (Api) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.Api).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Api.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20180601preview-api-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.Api).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180601preview_1.Api, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Api shouldn't use an unstable API (apimanagement.v20180601preview.Api). A compatible replacement can be found at 'apimanagement.Api'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Api || (Api = {}));
exports.Api = Api;
var ApiOperationPolicy;
(function (ApiOperationPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.ApiOperationPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiOperationPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20180601preview-apioperationpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.ApiOperationPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180601preview_2.ApiOperationPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiOperationPolicy shouldn't use an unstable API (apimanagement.v20180601preview.ApiOperationPolicy). A compatible replacement can be found at 'apimanagement.ApiOperationPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiOperationPolicy || (ApiOperationPolicy = {}));
exports.ApiOperationPolicy = ApiOperationPolicy;
var ApiPolicy;
(function (ApiPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.ApiPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApiPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20180601preview-apipolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.ApiPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180601preview_3.ApiPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApiPolicy shouldn't use an unstable API (apimanagement.v20180601preview.ApiPolicy). A compatible replacement can be found at 'apimanagement.ApiPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApiPolicy || (ApiPolicy = {}));
exports.ApiPolicy = ApiPolicy;
var Policy;
(function (Policy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.Policy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Policy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20180601preview-policy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.Policy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180601preview_4.Policy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Policy shouldn't use an unstable API (apimanagement.v20180601preview.Policy). A compatible replacement can be found at 'apimanagement.Policy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Policy || (Policy = {}));
exports.Policy = Policy;
var ProductApi;
(function (ProductApi) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.ProductApi).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProductApi.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20180601preview-productapi-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.ProductApi).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180601preview_5.ProductApi, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProductApi shouldn't use an unstable API (apimanagement.v20180601preview.ProductApi). A compatible replacement can be found at 'apimanagement.ProductApi'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProductApi || (ProductApi = {}));
exports.ProductApi = ProductApi;
var ProductPolicy;
(function (ProductPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.ProductPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProductPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20180601preview-productpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.ProductPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180601preview_6.ProductPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProductPolicy shouldn't use an unstable API (apimanagement.v20180601preview.ProductPolicy). A compatible replacement can be found at 'apimanagement.ProductPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProductPolicy || (ProductPolicy = {}));
exports.ProductPolicy = ProductPolicy;
var TagDescription;
(function (TagDescription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.TagDescription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TagDescription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-apimanagement-v20180601preview-tagdescription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (apimanagement.v20180601preview.TagDescription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20180601preview_7.TagDescription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TagDescription shouldn't use an unstable API (apimanagement.v20180601preview.TagDescription). A compatible replacement can be found at 'apimanagement.TagDescription'.");
            }),
        },
        vendors: ["azure"],
        services: ["apimanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TagDescription || (TagDescription = {}));
exports.TagDescription = TagDescription;
