// @pulumi/azure-native/recoveryservices/protectionPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProtectionPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Base class for backup policy. Workload-specific backup policies are derived from this class.
 * Azure REST API version: 2023-04-01. Prior API version in Azure Native 1.x: 2021-02-01.
 *
 * Other available API versions: 2023-06-01, 2023-08-01, 2024-01-01, 2024-02-01, 2024-04-01, 2024-04-30-preview, 2024-07-30-preview, 2024-10-01.
 */
class ProtectionPolicy extends pulumi.CustomResource {
    /**
     * Get an existing ProtectionPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ProtectionPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ProtectionPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ProtectionPolicy.__pulumiType;
    }
    /**
     * Create a ProtectionPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vaultName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vaultName'");
            }
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["policyName"] = args ? args.policyName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["vaultName"] = args ? args.vaultName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices/v20160601:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20201001:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20201201:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20210101:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20210201:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20210201preview:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20210210:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20210301:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20210401:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20210601:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20210701:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20210801:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20211001:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20211201:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20220101:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20220201:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20220301:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20220401:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20220601preview:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20220901preview:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20220930preview:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20221001:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20230101:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20230201:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20230401:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20230601:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20230801:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20240101:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20240201:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20240401:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20240430preview:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20240730preview:ProtectionPolicy" }, { type: "azure-native:recoveryservices/v20241001:ProtectionPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ProtectionPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ProtectionPolicy = ProtectionPolicy;
/** @internal */
ProtectionPolicy.__pulumiType = 'azure-native:recoveryservices:ProtectionPolicy';
