// @pulumi/azure-native/recoveryservices/replicationMigrationItem.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationMigrationItem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Migration item.
 * Azure REST API version: 2023-04-01. Prior API version in Azure Native 1.x: 2018-07-10.
 *
 * Other available API versions: 2023-06-01, 2023-08-01, 2024-01-01, 2024-02-01, 2024-04-01, 2024-10-01.
 */
class ReplicationMigrationItem extends pulumi.CustomResource {
    /**
     * Get an existing ReplicationMigrationItem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ReplicationMigrationItem(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationMigrationItem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationMigrationItem.__pulumiType;
    }
    /**
     * Create a ReplicationMigrationItem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if ((!args || args.properties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            if ((!args || args.protectionContainerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'protectionContainerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["fabricName"] = args ? args.fabricName : undefined;
            resourceInputs["migrationItemName"] = args ? args.migrationItemName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["protectionContainerName"] = args ? args.protectionContainerName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices/v20180110:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20180710:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20210210:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20210301:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20210401:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20210601:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20210701:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20210801:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20211001:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20211101:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20211201:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20220101:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20220201:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20220301:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20220401:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20220501:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20220801:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20220910:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20221001:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20230101:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20230201:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20230401:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20230601:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20230801:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20240101:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20240201:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20240401:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20241001:ReplicationMigrationItem" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ReplicationMigrationItem.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ReplicationMigrationItem = ReplicationMigrationItem;
/** @internal */
ReplicationMigrationItem.__pulumiType = 'azure-native:recoveryservices:ReplicationMigrationItem';
