// @pulumi/azure/relay/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NamespaceAuthorizationRule = exports.Namespace = exports.HybridConnectionAuthorizationRule = exports.HybridConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.HybridConnection = null;
utilities.lazyLoad(exports, ["HybridConnection"], () => require("./hybridConnection"));
exports.HybridConnectionAuthorizationRule = null;
utilities.lazyLoad(exports, ["HybridConnectionAuthorizationRule"], () => require("./hybridConnectionAuthorizationRule"));
exports.Namespace = null;
utilities.lazyLoad(exports, ["Namespace"], () => require("./namespace"));
exports.NamespaceAuthorizationRule = null;
utilities.lazyLoad(exports, ["NamespaceAuthorizationRule"], () => require("./namespaceAuthorizationRule"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:relay/hybridConnection:HybridConnection":
                return new exports.HybridConnection(name, undefined, { urn });
            case "azure:relay/hybridConnectionAuthorizationRule:HybridConnectionAuthorizationRule":
                return new exports.HybridConnectionAuthorizationRule(name, undefined, { urn });
            case "azure:relay/namespace:Namespace":
                return new exports.Namespace(name, undefined, { urn });
            case "azure:relay/namespaceAuthorizationRule:NamespaceAuthorizationRule":
                return new exports.NamespaceAuthorizationRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "relay/hybridConnection", _module);
pulumi.runtime.registerResourceModule("azure", "relay/hybridConnectionAuthorizationRule", _module);
pulumi.runtime.registerResourceModule("azure", "relay/namespace", _module);
pulumi.runtime.registerResourceModule("azure", "relay/namespaceAuthorizationRule", _module);
