// @pulumi/azure-compliance-policies/azurenative/sql/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240501preview = exports.v20230801preview = exports.v20230501preview = exports.v20230201preview = exports.v20221101preview = exports.v20210201preview = exports.v20200801preview = exports.v20200202preview = exports.v20190601preview = exports.v20180601preview = exports.v20170301preview = exports.v20150501preview = void 0;
exports.v20150501preview = require("./v20150501preview");
exports.v20170301preview = require("./v20170301preview");
exports.v20180601preview = require("./v20180601preview");
exports.v20190601preview = require("./v20190601preview");
exports.v20200202preview = require("./v20200202preview");
exports.v20200801preview = require("./v20200801preview");
exports.v20210201preview = require("./v20210201preview");
exports.v20221101preview = require("./v20221101preview");
exports.v20230201preview = require("./v20230201preview");
exports.v20230501preview = require("./v20230501preview");
exports.v20230801preview = require("./v20230801preview");
exports.v20240501preview = require("./v20240501preview");
