// @pulumi/azure-compliance-policies/azurenative/desktopvirtualization/v20220401preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostPool = exports.ApplicationGroup = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20220401preview_1 = require("@pulumi/azure-native/desktopvirtualization/v20220401preview");
const v20220401preview_2 = require("@pulumi/azure-native/desktopvirtualization/v20220401preview");
var ApplicationGroup;
(function (ApplicationGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20220401preview.ApplicationGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ApplicationGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20220401preview-applicationgroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20220401preview.ApplicationGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220401preview_1.ApplicationGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ApplicationGroup shouldn't use an unstable API (desktopvirtualization.v20220401preview.ApplicationGroup). A compatible replacement can be found at 'desktopvirtualization.ApplicationGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ApplicationGroup || (ApplicationGroup = {}));
exports.ApplicationGroup = ApplicationGroup;
var HostPool;
(function (HostPool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20220401preview.HostPool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    HostPool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-desktopvirtualization-v20220401preview-hostpool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (desktopvirtualization.v20220401preview.HostPool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220401preview_2.HostPool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure HostPool shouldn't use an unstable API (desktopvirtualization.v20220401preview.HostPool). A compatible replacement can be found at 'desktopvirtualization.HostPool'.");
            }),
        },
        vendors: ["azure"],
        services: ["desktopvirtualization"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(HostPool || (HostPool = {}));
exports.HostPool = HostPool;
