// @pulumi/azure/provider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Provider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The provider type for the azurerm package. By default, resources use package-wide configuration
 * settings, however an explicit `Provider` instance may be created and passed during resource
 * construction to achieve fine-grained programmatic control over provider settings. See the
 * [documentation](https://www.pulumi.com/docs/reference/programming-model/#providers) for more information.
 */
class Provider extends pulumi.ProviderResource {
    /**
     * Create a Provider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d, _e;
        let resourceInputs = {};
        opts = opts || {};
        {
            resourceInputs["adoPipelineServiceConnectionId"] = args ? args.adoPipelineServiceConnectionId : undefined;
            resourceInputs["auxiliaryTenantIds"] = pulumi.output((args === null || args === void 0 ? void 0 : args.auxiliaryTenantIds) ? pulumi.secret(args.auxiliaryTenantIds) : undefined).apply(JSON.stringify);
            resourceInputs["clientCertificate"] = (args === null || args === void 0 ? void 0 : args.clientCertificate) ? pulumi.secret(args.clientCertificate) : undefined;
            resourceInputs["clientCertificatePassword"] = (args === null || args === void 0 ? void 0 : args.clientCertificatePassword) ? pulumi.secret(args.clientCertificatePassword) : undefined;
            resourceInputs["clientCertificatePath"] = (args === null || args === void 0 ? void 0 : args.clientCertificatePath) ? pulumi.secret(args.clientCertificatePath) : undefined;
            resourceInputs["clientId"] = (args === null || args === void 0 ? void 0 : args.clientId) ? pulumi.secret(args.clientId) : undefined;
            resourceInputs["clientIdFilePath"] = (args === null || args === void 0 ? void 0 : args.clientIdFilePath) ? pulumi.secret(args.clientIdFilePath) : undefined;
            resourceInputs["clientSecret"] = (args === null || args === void 0 ? void 0 : args.clientSecret) ? pulumi.secret(args.clientSecret) : undefined;
            resourceInputs["clientSecretFilePath"] = (args === null || args === void 0 ? void 0 : args.clientSecretFilePath) ? pulumi.secret(args.clientSecretFilePath) : undefined;
            resourceInputs["disableCorrelationRequestId"] = pulumi.output(args ? args.disableCorrelationRequestId : undefined).apply(JSON.stringify);
            resourceInputs["disableTerraformPartnerId"] = pulumi.output(args ? args.disableTerraformPartnerId : undefined).apply(JSON.stringify);
            resourceInputs["environment"] = (_a = (args ? args.environment : undefined)) !== null && _a !== void 0 ? _a : (utilities.getEnv("AZURE_ENVIRONMENT", "ARM_ENVIRONMENT") || "public");
            resourceInputs["features"] = pulumi.output(args ? args.features : undefined).apply(JSON.stringify);
            resourceInputs["metadataHost"] = (_b = (args ? args.metadataHost : undefined)) !== null && _b !== void 0 ? _b : utilities.getEnv("ARM_METADATA_HOSTNAME");
            resourceInputs["msiEndpoint"] = args ? args.msiEndpoint : undefined;
            resourceInputs["oidcRequestToken"] = (args === null || args === void 0 ? void 0 : args.oidcRequestToken) ? pulumi.secret(args.oidcRequestToken) : undefined;
            resourceInputs["oidcRequestUrl"] = args ? args.oidcRequestUrl : undefined;
            resourceInputs["oidcToken"] = (args === null || args === void 0 ? void 0 : args.oidcToken) ? pulumi.secret(args.oidcToken) : undefined;
            resourceInputs["oidcTokenFilePath"] = (args === null || args === void 0 ? void 0 : args.oidcTokenFilePath) ? pulumi.secret(args.oidcTokenFilePath) : undefined;
            resourceInputs["partnerId"] = args ? args.partnerId : undefined;
            resourceInputs["resourceProviderRegistrations"] = args ? args.resourceProviderRegistrations : undefined;
            resourceInputs["resourceProvidersToRegisters"] = pulumi.output(args ? args.resourceProvidersToRegisters : undefined).apply(JSON.stringify);
            resourceInputs["skipProviderRegistration"] = pulumi.output((_c = (args ? args.skipProviderRegistration : undefined)) !== null && _c !== void 0 ? _c : (utilities.getEnvBoolean("ARM_SKIP_PROVIDER_REGISTRATION") || false)).apply(JSON.stringify);
            resourceInputs["storageUseAzuread"] = pulumi.output((_d = (args ? args.storageUseAzuread : undefined)) !== null && _d !== void 0 ? _d : (utilities.getEnvBoolean("ARM_STORAGE_USE_AZUREAD") || false)).apply(JSON.stringify);
            resourceInputs["subscriptionId"] = (_e = ((args === null || args === void 0 ? void 0 : args.subscriptionId) ? pulumi.secret(args.subscriptionId) : undefined)) !== null && _e !== void 0 ? _e : (utilities.getEnv("ARM_SUBSCRIPTION_ID") || "");
            resourceInputs["tenantId"] = (args === null || args === void 0 ? void 0 : args.tenantId) ? pulumi.secret(args.tenantId) : undefined;
            resourceInputs["useAksWorkloadIdentity"] = pulumi.output(args ? args.useAksWorkloadIdentity : undefined).apply(JSON.stringify);
            resourceInputs["useCli"] = pulumi.output(args ? args.useCli : undefined).apply(JSON.stringify);
            resourceInputs["useMsi"] = pulumi.output(args ? args.useMsi : undefined).apply(JSON.stringify);
            resourceInputs["useOidc"] = pulumi.output(args ? args.useOidc : undefined).apply(JSON.stringify);
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["clientCertificate", "clientCertificatePassword", "clientCertificatePath", "clientId", "clientIdFilePath", "clientSecret", "clientSecretFilePath", "oidcRequestToken", "oidcToken", "oidcTokenFilePath", "subscriptionId", "tenantId"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(Provider.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Returns true if the given object is an instance of Provider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === "pulumi:providers:" + Provider.__pulumiType;
    }
}
exports.Provider = Provider;
/** @internal */
Provider.__pulumiType = 'azure';
