// @pulumi/azure-compliance-policies/azurenative/costmanagement/v20230701preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewByScope = exports.View = exports.ScheduledActionByScope = exports.ScheduledAction = exports.Export = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230701preview_1 = require("@pulumi/azure-native/costmanagement/v20230701preview");
const v20230701preview_2 = require("@pulumi/azure-native/costmanagement/v20230701preview");
const v20230701preview_3 = require("@pulumi/azure-native/costmanagement/v20230701preview");
const v20230701preview_4 = require("@pulumi/azure-native/costmanagement/v20230701preview");
const v20230701preview_5 = require("@pulumi/azure-native/costmanagement/v20230701preview");
var Export;
(function (Export) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.Export).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Export.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-costmanagement-v20230701preview-export-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.Export).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230701preview_1.Export, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Export shouldn't use an unstable API (costmanagement.v20230701preview.Export). A compatible replacement can be found at 'costmanagement.Export'.");
            }),
        },
        vendors: ["azure"],
        services: ["costmanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Export || (Export = {}));
exports.Export = Export;
var ScheduledAction;
(function (ScheduledAction) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.ScheduledAction).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ScheduledAction.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-costmanagement-v20230701preview-scheduledaction-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.ScheduledAction).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230701preview_2.ScheduledAction, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ScheduledAction shouldn't use an unstable API (costmanagement.v20230701preview.ScheduledAction). A compatible replacement can be found at 'costmanagement.ScheduledAction'.");
            }),
        },
        vendors: ["azure"],
        services: ["costmanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ScheduledAction || (ScheduledAction = {}));
exports.ScheduledAction = ScheduledAction;
var ScheduledActionByScope;
(function (ScheduledActionByScope) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.ScheduledActionByScope).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ScheduledActionByScope.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-costmanagement-v20230701preview-scheduledactionbyscope-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.ScheduledActionByScope).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230701preview_3.ScheduledActionByScope, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ScheduledActionByScope shouldn't use an unstable API (costmanagement.v20230701preview.ScheduledActionByScope). A compatible replacement can be found at 'costmanagement.ScheduledActionByScope'.");
            }),
        },
        vendors: ["azure"],
        services: ["costmanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ScheduledActionByScope || (ScheduledActionByScope = {}));
exports.ScheduledActionByScope = ScheduledActionByScope;
var View;
(function (View) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.View).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    View.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-costmanagement-v20230701preview-view-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.View).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230701preview_4.View, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure View shouldn't use an unstable API (costmanagement.v20230701preview.View). A compatible replacement can be found at 'costmanagement.View'.");
            }),
        },
        vendors: ["azure"],
        services: ["costmanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(View || (View = {}));
exports.View = View;
var ViewByScope;
(function (ViewByScope) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.ViewByScope).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ViewByScope.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-costmanagement-v20230701preview-viewbyscope-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (costmanagement.v20230701preview.ViewByScope).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230701preview_5.ViewByScope, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ViewByScope shouldn't use an unstable API (costmanagement.v20230701preview.ViewByScope). A compatible replacement can be found at 'costmanagement.ViewByScope'.");
            }),
        },
        vendors: ["azure"],
        services: ["costmanagement"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ViewByScope || (ViewByScope = {}));
exports.ViewByScope = ViewByScope;
