// @pulumi/azure-compliance-policies/azurenative/app/v20230801preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedEnvironmentsStorage = exports.ManagedEnvironment = exports.ManagedCertificate = exports.Job = exports.DaprSubscription = exports.DaprComponentResiliencyPolicy = exports.DaprComponent = exports.ContainerAppsSourceControl = exports.ContainerAppsAuthConfig = exports.ContainerApp = exports.ConnectedEnvironmentsStorage = exports.ConnectedEnvironmentsDaprComponent = exports.ConnectedEnvironmentsCertificate = exports.ConnectedEnvironment = exports.Certificate = exports.Builder = exports.Build = exports.AppResiliency = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20230801preview_1 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_2 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_3 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_4 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_5 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_6 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_7 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_8 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_9 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_10 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_11 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_12 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_13 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_14 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_15 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_16 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_17 = require("@pulumi/azure-native/app/v20230801preview");
const v20230801preview_18 = require("@pulumi/azure-native/app/v20230801preview");
var AppResiliency;
(function (AppResiliency) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.AppResiliency).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AppResiliency.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-appresiliency-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.AppResiliency).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_1.AppResiliency, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AppResiliency shouldn't use an unstable API (app.v20230801preview.AppResiliency). A compatible replacement can be found at 'app.AppResiliency'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AppResiliency || (AppResiliency = {}));
exports.AppResiliency = AppResiliency;
var Build;
(function (Build) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.Build).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Build.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-build-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.Build).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_2.Build, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Build shouldn't use an unstable API (app.v20230801preview.Build). A compatible replacement can be found at 'app.Build'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Build || (Build = {}));
exports.Build = Build;
var Builder;
(function (Builder) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.Builder).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Builder.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-builder-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.Builder).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_3.Builder, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Builder shouldn't use an unstable API (app.v20230801preview.Builder). A compatible replacement can be found at 'app.Builder'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Builder || (Builder = {}));
exports.Builder = Builder;
var Certificate;
(function (Certificate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.Certificate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Certificate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-certificate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.Certificate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_4.Certificate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Certificate shouldn't use an unstable API (app.v20230801preview.Certificate). A compatible replacement can be found at 'app.Certificate'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Certificate || (Certificate = {}));
exports.Certificate = Certificate;
var ConnectedEnvironment;
(function (ConnectedEnvironment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ConnectedEnvironment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConnectedEnvironment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-connectedenvironment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ConnectedEnvironment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_5.ConnectedEnvironment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConnectedEnvironment shouldn't use an unstable API (app.v20230801preview.ConnectedEnvironment). A compatible replacement can be found at 'app.ConnectedEnvironment'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConnectedEnvironment || (ConnectedEnvironment = {}));
exports.ConnectedEnvironment = ConnectedEnvironment;
var ConnectedEnvironmentsCertificate;
(function (ConnectedEnvironmentsCertificate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ConnectedEnvironmentsCertificate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConnectedEnvironmentsCertificate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-connectedenvironmentscertificate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ConnectedEnvironmentsCertificate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_6.ConnectedEnvironmentsCertificate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConnectedEnvironmentsCertificate shouldn't use an unstable API (app.v20230801preview.ConnectedEnvironmentsCertificate). A compatible replacement can be found at 'app.ConnectedEnvironmentsCertificate'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConnectedEnvironmentsCertificate || (ConnectedEnvironmentsCertificate = {}));
exports.ConnectedEnvironmentsCertificate = ConnectedEnvironmentsCertificate;
var ConnectedEnvironmentsDaprComponent;
(function (ConnectedEnvironmentsDaprComponent) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ConnectedEnvironmentsDaprComponent).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConnectedEnvironmentsDaprComponent.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-connectedenvironmentsdaprcomponent-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ConnectedEnvironmentsDaprComponent).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_7.ConnectedEnvironmentsDaprComponent, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConnectedEnvironmentsDaprComponent shouldn't use an unstable API (app.v20230801preview.ConnectedEnvironmentsDaprComponent). A compatible replacement can be found at 'app.ConnectedEnvironmentsDaprComponent'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConnectedEnvironmentsDaprComponent || (ConnectedEnvironmentsDaprComponent = {}));
exports.ConnectedEnvironmentsDaprComponent = ConnectedEnvironmentsDaprComponent;
var ConnectedEnvironmentsStorage;
(function (ConnectedEnvironmentsStorage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ConnectedEnvironmentsStorage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConnectedEnvironmentsStorage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-connectedenvironmentsstorage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ConnectedEnvironmentsStorage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_8.ConnectedEnvironmentsStorage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConnectedEnvironmentsStorage shouldn't use an unstable API (app.v20230801preview.ConnectedEnvironmentsStorage). A compatible replacement can be found at 'app.ConnectedEnvironmentsStorage'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConnectedEnvironmentsStorage || (ConnectedEnvironmentsStorage = {}));
exports.ConnectedEnvironmentsStorage = ConnectedEnvironmentsStorage;
var ContainerApp;
(function (ContainerApp) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ContainerApp).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ContainerApp.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-containerapp-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ContainerApp).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_9.ContainerApp, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ContainerApp shouldn't use an unstable API (app.v20230801preview.ContainerApp). A compatible replacement can be found at 'app.ContainerApp'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ContainerApp || (ContainerApp = {}));
exports.ContainerApp = ContainerApp;
var ContainerAppsAuthConfig;
(function (ContainerAppsAuthConfig) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ContainerAppsAuthConfig).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ContainerAppsAuthConfig.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-containerappsauthconfig-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ContainerAppsAuthConfig).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_10.ContainerAppsAuthConfig, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ContainerAppsAuthConfig shouldn't use an unstable API (app.v20230801preview.ContainerAppsAuthConfig). A compatible replacement can be found at 'app.ContainerAppsAuthConfig'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ContainerAppsAuthConfig || (ContainerAppsAuthConfig = {}));
exports.ContainerAppsAuthConfig = ContainerAppsAuthConfig;
var ContainerAppsSourceControl;
(function (ContainerAppsSourceControl) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ContainerAppsSourceControl).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ContainerAppsSourceControl.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-containerappssourcecontrol-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ContainerAppsSourceControl).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_11.ContainerAppsSourceControl, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ContainerAppsSourceControl shouldn't use an unstable API (app.v20230801preview.ContainerAppsSourceControl). A compatible replacement can be found at 'app.ContainerAppsSourceControl'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ContainerAppsSourceControl || (ContainerAppsSourceControl = {}));
exports.ContainerAppsSourceControl = ContainerAppsSourceControl;
var DaprComponent;
(function (DaprComponent) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.DaprComponent).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DaprComponent.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-daprcomponent-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.DaprComponent).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_12.DaprComponent, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DaprComponent shouldn't use an unstable API (app.v20230801preview.DaprComponent). A compatible replacement can be found at 'app.DaprComponent'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DaprComponent || (DaprComponent = {}));
exports.DaprComponent = DaprComponent;
var DaprComponentResiliencyPolicy;
(function (DaprComponentResiliencyPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.DaprComponentResiliencyPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DaprComponentResiliencyPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-daprcomponentresiliencypolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.DaprComponentResiliencyPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_13.DaprComponentResiliencyPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DaprComponentResiliencyPolicy shouldn't use an unstable API (app.v20230801preview.DaprComponentResiliencyPolicy). A compatible replacement can be found at 'app.DaprComponentResiliencyPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DaprComponentResiliencyPolicy || (DaprComponentResiliencyPolicy = {}));
exports.DaprComponentResiliencyPolicy = DaprComponentResiliencyPolicy;
var DaprSubscription;
(function (DaprSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.DaprSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DaprSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-daprsubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.DaprSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_14.DaprSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DaprSubscription shouldn't use an unstable API (app.v20230801preview.DaprSubscription). A compatible replacement can be found at 'app.DaprSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DaprSubscription || (DaprSubscription = {}));
exports.DaprSubscription = DaprSubscription;
var Job;
(function (Job) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.Job).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Job.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-job-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.Job).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_15.Job, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Job shouldn't use an unstable API (app.v20230801preview.Job). A compatible replacement can be found at 'app.Job'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Job || (Job = {}));
exports.Job = Job;
var ManagedCertificate;
(function (ManagedCertificate) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ManagedCertificate).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagedCertificate.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-managedcertificate-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ManagedCertificate).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_16.ManagedCertificate, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagedCertificate shouldn't use an unstable API (app.v20230801preview.ManagedCertificate). A compatible replacement can be found at 'app.ManagedCertificate'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagedCertificate || (ManagedCertificate = {}));
exports.ManagedCertificate = ManagedCertificate;
var ManagedEnvironment;
(function (ManagedEnvironment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ManagedEnvironment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagedEnvironment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-managedenvironment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ManagedEnvironment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_17.ManagedEnvironment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagedEnvironment shouldn't use an unstable API (app.v20230801preview.ManagedEnvironment). A compatible replacement can be found at 'app.ManagedEnvironment'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagedEnvironment || (ManagedEnvironment = {}));
exports.ManagedEnvironment = ManagedEnvironment;
var ManagedEnvironmentsStorage;
(function (ManagedEnvironmentsStorage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ManagedEnvironmentsStorage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagedEnvironmentsStorage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-app-v20230801preview-managedenvironmentsstorage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (app.v20230801preview.ManagedEnvironmentsStorage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20230801preview_18.ManagedEnvironmentsStorage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagedEnvironmentsStorage shouldn't use an unstable API (app.v20230801preview.ManagedEnvironmentsStorage). A compatible replacement can be found at 'app.ManagedEnvironmentsStorage'.");
            }),
        },
        vendors: ["azure"],
        services: ["app"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagedEnvironmentsStorage || (ManagedEnvironmentsStorage = {}));
exports.ManagedEnvironmentsStorage = ManagedEnvironmentsStorage;
