// @pulumi/azure-native/workloads/sapapplicationServerInstance.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAPApplicationServerInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Define the SAP Application Server Instance resource.
 * Azure REST API version: 2023-04-01. Prior API version in Azure Native 1.x: 2021-12-01-preview.
 *
 * Other available API versions: 2021-12-01-preview, 2023-10-01-preview.
 */
class SAPApplicationServerInstance extends pulumi.CustomResource {
    /**
     * Get an existing SAPApplicationServerInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SAPApplicationServerInstance(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SAPApplicationServerInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SAPApplicationServerInstance.__pulumiType;
    }
    /**
     * Create a SAPApplicationServerInstance resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sapVirtualInstanceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sapVirtualInstanceName'");
            }
            resourceInputs["applicationInstanceName"] = args ? args.applicationInstanceName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sapVirtualInstanceName"] = args ? args.sapVirtualInstanceName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["gatewayPort"] = undefined /*out*/;
            resourceInputs["health"] = undefined /*out*/;
            resourceInputs["hostname"] = undefined /*out*/;
            resourceInputs["icmHttpPort"] = undefined /*out*/;
            resourceInputs["icmHttpsPort"] = undefined /*out*/;
            resourceInputs["instanceNo"] = undefined /*out*/;
            resourceInputs["ipAddress"] = undefined /*out*/;
            resourceInputs["kernelPatch"] = undefined /*out*/;
            resourceInputs["kernelVersion"] = undefined /*out*/;
            resourceInputs["loadBalancerDetails"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["subnet"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vmDetails"] = undefined /*out*/;
        }
        else {
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["gatewayPort"] = undefined /*out*/;
            resourceInputs["health"] = undefined /*out*/;
            resourceInputs["hostname"] = undefined /*out*/;
            resourceInputs["icmHttpPort"] = undefined /*out*/;
            resourceInputs["icmHttpsPort"] = undefined /*out*/;
            resourceInputs["instanceNo"] = undefined /*out*/;
            resourceInputs["ipAddress"] = undefined /*out*/;
            resourceInputs["kernelPatch"] = undefined /*out*/;
            resourceInputs["kernelVersion"] = undefined /*out*/;
            resourceInputs["loadBalancerDetails"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["subnet"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vmDetails"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:workloads/v20211201preview:SAPApplicationServerInstance" }, { type: "azure-native:workloads/v20221101preview:SAPApplicationServerInstance" }, { type: "azure-native:workloads/v20230401:SAPApplicationServerInstance" }, { type: "azure-native:workloads/v20231001preview:SAPApplicationServerInstance" }, { type: "azure-native:workloads/v20240901:SAPApplicationServerInstance" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SAPApplicationServerInstance.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SAPApplicationServerInstance = SAPApplicationServerInstance;
/** @internal */
SAPApplicationServerInstance.__pulumiType = 'azure-native:workloads:SAPApplicationServerInstance';
