// @pulumi/azure-compliance-policies/azurenative/billing/v20191001preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingRoleAssignmentByEnrollmentAccount = exports.BillingRoleAssignmentByDepartment = exports.BillingRoleAssignmentByBillingAccount = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20191001preview_1 = require("@pulumi/azure-native/billing/v20191001preview");
const v20191001preview_2 = require("@pulumi/azure-native/billing/v20191001preview");
const v20191001preview_3 = require("@pulumi/azure-native/billing/v20191001preview");
var BillingRoleAssignmentByBillingAccount;
(function (BillingRoleAssignmentByBillingAccount) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (billing.v20191001preview.BillingRoleAssignmentByBillingAccount).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BillingRoleAssignmentByBillingAccount.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-billing-v20191001preview-billingroleassignmentbybillingaccount-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (billing.v20191001preview.BillingRoleAssignmentByBillingAccount).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20191001preview_1.BillingRoleAssignmentByBillingAccount, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BillingRoleAssignmentByBillingAccount shouldn't use an unstable API (billing.v20191001preview.BillingRoleAssignmentByBillingAccount). A compatible replacement can be found at 'billing.BillingRoleAssignmentByBillingAccount'.");
            }),
        },
        vendors: ["azure"],
        services: ["billing"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BillingRoleAssignmentByBillingAccount || (BillingRoleAssignmentByBillingAccount = {}));
exports.BillingRoleAssignmentByBillingAccount = BillingRoleAssignmentByBillingAccount;
var BillingRoleAssignmentByDepartment;
(function (BillingRoleAssignmentByDepartment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (billing.v20191001preview.BillingRoleAssignmentByDepartment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BillingRoleAssignmentByDepartment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-billing-v20191001preview-billingroleassignmentbydepartment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (billing.v20191001preview.BillingRoleAssignmentByDepartment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20191001preview_2.BillingRoleAssignmentByDepartment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BillingRoleAssignmentByDepartment shouldn't use an unstable API (billing.v20191001preview.BillingRoleAssignmentByDepartment). A compatible replacement can be found at 'billing.BillingRoleAssignmentByDepartment'.");
            }),
        },
        vendors: ["azure"],
        services: ["billing"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BillingRoleAssignmentByDepartment || (BillingRoleAssignmentByDepartment = {}));
exports.BillingRoleAssignmentByDepartment = BillingRoleAssignmentByDepartment;
var BillingRoleAssignmentByEnrollmentAccount;
(function (BillingRoleAssignmentByEnrollmentAccount) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (billing.v20191001preview.BillingRoleAssignmentByEnrollmentAccount).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BillingRoleAssignmentByEnrollmentAccount.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-billing-v20191001preview-billingroleassignmentbyenrollmentaccount-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (billing.v20191001preview.BillingRoleAssignmentByEnrollmentAccount).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20191001preview_3.BillingRoleAssignmentByEnrollmentAccount, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BillingRoleAssignmentByEnrollmentAccount shouldn't use an unstable API (billing.v20191001preview.BillingRoleAssignmentByEnrollmentAccount). A compatible replacement can be found at 'billing.BillingRoleAssignmentByEnrollmentAccount'.");
            }),
        },
        vendors: ["azure"],
        services: ["billing"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BillingRoleAssignmentByEnrollmentAccount || (BillingRoleAssignmentByEnrollmentAccount = {}));
exports.BillingRoleAssignmentByEnrollmentAccount = BillingRoleAssignmentByEnrollmentAccount;
