// @pulumi/azure-compliance-policies/azurenative/databoxedge/v20220401preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Order = exports.Device = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20220401preview_1 = require("@pulumi/azure-native/databoxedge/v20220401preview");
const v20220401preview_2 = require("@pulumi/azure-native/databoxedge/v20220401preview");
var Device;
(function (Device) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (databoxedge.v20220401preview.Device).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Device.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-databoxedge-v20220401preview-device-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (databoxedge.v20220401preview.Device).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220401preview_1.Device, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Device shouldn't use an unstable API (databoxedge.v20220401preview.Device). A compatible replacement can be found at 'databoxedge.Device'.");
            }),
        },
        vendors: ["azure"],
        services: ["databoxedge"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Device || (Device = {}));
exports.Device = Device;
var Order;
(function (Order) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (databoxedge.v20220401preview.Order).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Order.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-databoxedge-v20220401preview-order-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (databoxedge.v20220401preview.Order).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20220401preview_2.Order, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Order shouldn't use an unstable API (databoxedge.v20220401preview.Order). A compatible replacement can be found at 'databoxedge.Order'.");
            }),
        },
        vendors: ["azure"],
        services: ["databoxedge"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Order || (Order = {}));
exports.Order = Order;
