// @pulumi/azure-compliance-policies/azurenative/visualstudio/v20171101preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.Extension = exports.Account = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20171101preview_1 = require("@pulumi/azure-native/visualstudio/v20171101preview");
const v20171101preview_2 = require("@pulumi/azure-native/visualstudio/v20171101preview");
var Account;
(function (Account) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (visualstudio.v20171101preview.Account).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Account.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-visualstudio-v20171101preview-account-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (visualstudio.v20171101preview.Account).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20171101preview_1.Account, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Account shouldn't use an unstable API (visualstudio.v20171101preview.Account). A compatible replacement can be found at 'visualstudio.Account'.");
            }),
        },
        vendors: ["azure"],
        services: ["visualstudio"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Account || (Account = {}));
exports.Account = Account;
var Extension;
(function (Extension) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (visualstudio.v20171101preview.Extension).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Extension.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-visualstudio-v20171101preview-extension-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (visualstudio.v20171101preview.Extension).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20171101preview_2.Extension, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Extension shouldn't use an unstable API (visualstudio.v20171101preview.Extension). A compatible replacement can be found at 'visualstudio.Extension'.");
            }),
        },
        vendors: ["azure"],
        services: ["visualstudio"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Extension || (Extension = {}));
exports.Extension = Extension;
