// @pulumi/azure-compliance-policies/azurenative/azurestackhci/v20221215preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetwork = exports.VirtualMachine = exports.VirtualHardDisk = exports.UpdateSummary = exports.UpdateRun = exports.Update = exports.StorageContainer = exports.NetworkInterface = exports.MarketplaceGalleryImage = exports.MachineExtension = exports.HybridIdentityMetadatum = exports.GuestAgent = exports.GalleryImage = exports.Extension = exports.Cluster = exports.ArcSetting = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20221215preview_1 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_2 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_3 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_4 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_5 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_6 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_7 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_8 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_9 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_10 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_11 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_12 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_13 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_14 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_15 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
const v20221215preview_16 = require("@pulumi/azure-native/azurestackhci/v20221215preview");
var ArcSetting;
(function (ArcSetting) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.ArcSetting).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ArcSetting.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-arcsetting-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.ArcSetting).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_1.ArcSetting, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ArcSetting shouldn't use an unstable API (azurestackhci.v20221215preview.ArcSetting). A compatible replacement can be found at 'azurestackhci.ArcSetting'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ArcSetting || (ArcSetting = {}));
exports.ArcSetting = ArcSetting;
var Cluster;
(function (Cluster) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.Cluster).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Cluster.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-cluster-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.Cluster).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_2.Cluster, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Cluster shouldn't use an unstable API (azurestackhci.v20221215preview.Cluster). A compatible replacement can be found at 'azurestackhci.Cluster'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Cluster || (Cluster = {}));
exports.Cluster = Cluster;
var Extension;
(function (Extension) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.Extension).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Extension.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-extension-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.Extension).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_3.Extension, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Extension shouldn't use an unstable API (azurestackhci.v20221215preview.Extension). A compatible replacement can be found at 'azurestackhci.Extension'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Extension || (Extension = {}));
exports.Extension = Extension;
var GalleryImage;
(function (GalleryImage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.GalleryImage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GalleryImage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-galleryimage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.GalleryImage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_4.GalleryImage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GalleryImage shouldn't use an unstable API (azurestackhci.v20221215preview.GalleryImage). A compatible replacement can be found at 'azurestackhci.GalleryImage'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GalleryImage || (GalleryImage = {}));
exports.GalleryImage = GalleryImage;
var GuestAgent;
(function (GuestAgent) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.GuestAgent).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    GuestAgent.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-guestagent-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.GuestAgent).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_5.GuestAgent, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure GuestAgent shouldn't use an unstable API (azurestackhci.v20221215preview.GuestAgent). A compatible replacement can be found at 'azurestackhci.GuestAgent'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(GuestAgent || (GuestAgent = {}));
exports.GuestAgent = GuestAgent;
var HybridIdentityMetadatum;
(function (HybridIdentityMetadatum) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.HybridIdentityMetadatum).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    HybridIdentityMetadatum.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-hybrididentitymetadatum-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.HybridIdentityMetadatum).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_6.HybridIdentityMetadatum, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure HybridIdentityMetadatum shouldn't use an unstable API (azurestackhci.v20221215preview.HybridIdentityMetadatum). A compatible replacement can be found at 'azurestackhci.HybridIdentityMetadatum'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(HybridIdentityMetadatum || (HybridIdentityMetadatum = {}));
exports.HybridIdentityMetadatum = HybridIdentityMetadatum;
var MachineExtension;
(function (MachineExtension) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.MachineExtension).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MachineExtension.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-machineextension-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.MachineExtension).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_7.MachineExtension, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MachineExtension shouldn't use an unstable API (azurestackhci.v20221215preview.MachineExtension). A compatible replacement can be found at 'azurestackhci.MachineExtension'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MachineExtension || (MachineExtension = {}));
exports.MachineExtension = MachineExtension;
var MarketplaceGalleryImage;
(function (MarketplaceGalleryImage) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.MarketplaceGalleryImage).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MarketplaceGalleryImage.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-marketplacegalleryimage-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.MarketplaceGalleryImage).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_8.MarketplaceGalleryImage, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MarketplaceGalleryImage shouldn't use an unstable API (azurestackhci.v20221215preview.MarketplaceGalleryImage). A compatible replacement can be found at 'azurestackhci.MarketplaceGalleryImage'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MarketplaceGalleryImage || (MarketplaceGalleryImage = {}));
exports.MarketplaceGalleryImage = MarketplaceGalleryImage;
var NetworkInterface;
(function (NetworkInterface) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.NetworkInterface).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    NetworkInterface.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-networkinterface-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.NetworkInterface).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_9.NetworkInterface, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure NetworkInterface shouldn't use an unstable API (azurestackhci.v20221215preview.NetworkInterface). A compatible replacement can be found at 'azurestackhci.NetworkInterface'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(NetworkInterface || (NetworkInterface = {}));
exports.NetworkInterface = NetworkInterface;
var StorageContainer;
(function (StorageContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.StorageContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    StorageContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-storagecontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.StorageContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_10.StorageContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure StorageContainer shouldn't use an unstable API (azurestackhci.v20221215preview.StorageContainer). A compatible replacement can be found at 'azurestackhci.StorageContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(StorageContainer || (StorageContainer = {}));
exports.StorageContainer = StorageContainer;
var Update;
(function (Update) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.Update).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Update.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-update-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.Update).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_11.Update, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Update shouldn't use an unstable API (azurestackhci.v20221215preview.Update). A compatible replacement can be found at 'azurestackhci.Update'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Update || (Update = {}));
exports.Update = Update;
var UpdateRun;
(function (UpdateRun) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.UpdateRun).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    UpdateRun.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-updaterun-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.UpdateRun).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_12.UpdateRun, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure UpdateRun shouldn't use an unstable API (azurestackhci.v20221215preview.UpdateRun). A compatible replacement can be found at 'azurestackhci.UpdateRun'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(UpdateRun || (UpdateRun = {}));
exports.UpdateRun = UpdateRun;
var UpdateSummary;
(function (UpdateSummary) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.UpdateSummary).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    UpdateSummary.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-updatesummary-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.UpdateSummary).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_13.UpdateSummary, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure UpdateSummary shouldn't use an unstable API (azurestackhci.v20221215preview.UpdateSummary). A compatible replacement can be found at 'azurestackhci.UpdateSummary'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(UpdateSummary || (UpdateSummary = {}));
exports.UpdateSummary = UpdateSummary;
var VirtualHardDisk;
(function (VirtualHardDisk) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.VirtualHardDisk).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VirtualHardDisk.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-virtualharddisk-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.VirtualHardDisk).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_14.VirtualHardDisk, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VirtualHardDisk shouldn't use an unstable API (azurestackhci.v20221215preview.VirtualHardDisk). A compatible replacement can be found at 'azurestackhci.VirtualHardDisk'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VirtualHardDisk || (VirtualHardDisk = {}));
exports.VirtualHardDisk = VirtualHardDisk;
var VirtualMachine;
(function (VirtualMachine) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.VirtualMachine).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VirtualMachine.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-virtualmachine-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.VirtualMachine).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_15.VirtualMachine, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VirtualMachine shouldn't use an unstable API (azurestackhci.v20221215preview.VirtualMachine). A compatible replacement can be found at 'azurestackhci.VirtualMachine'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VirtualMachine || (VirtualMachine = {}));
exports.VirtualMachine = VirtualMachine;
var VirtualNetwork;
(function (VirtualNetwork) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.VirtualNetwork).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    VirtualNetwork.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-azurestackhci-v20221215preview-virtualnetwork-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (azurestackhci.v20221215preview.VirtualNetwork).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20221215preview_16.VirtualNetwork, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure VirtualNetwork shouldn't use an unstable API (azurestackhci.v20221215preview.VirtualNetwork). A compatible replacement can be found at 'azurestackhci.VirtualNetwork'.");
            }),
        },
        vendors: ["azure"],
        services: ["azurestackhci"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(VirtualNetwork || (VirtualNetwork = {}));
exports.VirtualNetwork = VirtualNetwork;
