// @pulumi/azure-compliance-policies/azurenative/dataprotection/v20240201preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceGuard = exports.DppResourceGuardProxy = exports.BackupVault = exports.BackupPolicy = exports.BackupInstance = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240201preview_1 = require("@pulumi/azure-native/dataprotection/v20240201preview");
const v20240201preview_2 = require("@pulumi/azure-native/dataprotection/v20240201preview");
const v20240201preview_3 = require("@pulumi/azure-native/dataprotection/v20240201preview");
const v20240201preview_4 = require("@pulumi/azure-native/dataprotection/v20240201preview");
const v20240201preview_5 = require("@pulumi/azure-native/dataprotection/v20240201preview");
var BackupInstance;
(function (BackupInstance) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.BackupInstance).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BackupInstance.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dataprotection-v20240201preview-backupinstance-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.BackupInstance).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240201preview_1.BackupInstance, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BackupInstance shouldn't use an unstable API (dataprotection.v20240201preview.BackupInstance). A compatible replacement can be found at 'dataprotection.BackupInstance'.");
            }),
        },
        vendors: ["azure"],
        services: ["dataprotection"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BackupInstance || (BackupInstance = {}));
exports.BackupInstance = BackupInstance;
var BackupPolicy;
(function (BackupPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.BackupPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BackupPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dataprotection-v20240201preview-backuppolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.BackupPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240201preview_2.BackupPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BackupPolicy shouldn't use an unstable API (dataprotection.v20240201preview.BackupPolicy). A compatible replacement can be found at 'dataprotection.BackupPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["dataprotection"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BackupPolicy || (BackupPolicy = {}));
exports.BackupPolicy = BackupPolicy;
var BackupVault;
(function (BackupVault) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.BackupVault).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BackupVault.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dataprotection-v20240201preview-backupvault-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.BackupVault).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240201preview_3.BackupVault, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BackupVault shouldn't use an unstable API (dataprotection.v20240201preview.BackupVault). A compatible replacement can be found at 'dataprotection.BackupVault'.");
            }),
        },
        vendors: ["azure"],
        services: ["dataprotection"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BackupVault || (BackupVault = {}));
exports.BackupVault = BackupVault;
var DppResourceGuardProxy;
(function (DppResourceGuardProxy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.DppResourceGuardProxy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DppResourceGuardProxy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dataprotection-v20240201preview-dppresourceguardproxy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.DppResourceGuardProxy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240201preview_4.DppResourceGuardProxy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DppResourceGuardProxy shouldn't use an unstable API (dataprotection.v20240201preview.DppResourceGuardProxy). A compatible replacement can be found at 'dataprotection.DppResourceGuardProxy'.");
            }),
        },
        vendors: ["azure"],
        services: ["dataprotection"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DppResourceGuardProxy || (DppResourceGuardProxy = {}));
exports.DppResourceGuardProxy = DppResourceGuardProxy;
var ResourceGuard;
(function (ResourceGuard) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.ResourceGuard).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ResourceGuard.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-dataprotection-v20240201preview-resourceguard-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (dataprotection.v20240201preview.ResourceGuard).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240201preview_5.ResourceGuard, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ResourceGuard shouldn't use an unstable API (dataprotection.v20240201preview.ResourceGuard). A compatible replacement can be found at 'dataprotection.ResourceGuard'.");
            }),
        },
        vendors: ["azure"],
        services: ["dataprotection"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ResourceGuard || (ResourceGuard = {}));
exports.ResourceGuard = ResourceGuard;
