// @pulumi/azure-compliance-policies/azurenative/providerhub/v20210901preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkusNestedResourceTypeThird = exports.SkusNestedResourceTypeSecond = exports.SkusNestedResourceTypeFirst = exports.Skus = exports.ResourceTypeRegistration = exports.ProviderRegistration = exports.OperationByProviderRegistration = exports.NotificationRegistration = exports.DefaultRollout = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20210901preview_1 = require("@pulumi/azure-native/providerhub/v20210901preview");
const v20210901preview_2 = require("@pulumi/azure-native/providerhub/v20210901preview");
const v20210901preview_3 = require("@pulumi/azure-native/providerhub/v20210901preview");
const v20210901preview_4 = require("@pulumi/azure-native/providerhub/v20210901preview");
const v20210901preview_5 = require("@pulumi/azure-native/providerhub/v20210901preview");
const v20210901preview_6 = require("@pulumi/azure-native/providerhub/v20210901preview");
const v20210901preview_7 = require("@pulumi/azure-native/providerhub/v20210901preview");
const v20210901preview_8 = require("@pulumi/azure-native/providerhub/v20210901preview");
const v20210901preview_9 = require("@pulumi/azure-native/providerhub/v20210901preview");
var DefaultRollout;
(function (DefaultRollout) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.DefaultRollout).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DefaultRollout.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-providerhub-v20210901preview-defaultrollout-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.DefaultRollout).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210901preview_1.DefaultRollout, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DefaultRollout shouldn't use an unstable API (providerhub.v20210901preview.DefaultRollout). A compatible replacement can be found at 'providerhub.DefaultRollout'.");
            }),
        },
        vendors: ["azure"],
        services: ["providerhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DefaultRollout || (DefaultRollout = {}));
exports.DefaultRollout = DefaultRollout;
var NotificationRegistration;
(function (NotificationRegistration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.NotificationRegistration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    NotificationRegistration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-providerhub-v20210901preview-notificationregistration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.NotificationRegistration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210901preview_2.NotificationRegistration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure NotificationRegistration shouldn't use an unstable API (providerhub.v20210901preview.NotificationRegistration). A compatible replacement can be found at 'providerhub.NotificationRegistration'.");
            }),
        },
        vendors: ["azure"],
        services: ["providerhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(NotificationRegistration || (NotificationRegistration = {}));
exports.NotificationRegistration = NotificationRegistration;
var OperationByProviderRegistration;
(function (OperationByProviderRegistration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.OperationByProviderRegistration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OperationByProviderRegistration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-providerhub-v20210901preview-operationbyproviderregistration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.OperationByProviderRegistration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210901preview_3.OperationByProviderRegistration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OperationByProviderRegistration shouldn't use an unstable API (providerhub.v20210901preview.OperationByProviderRegistration). A compatible replacement can be found at 'providerhub.OperationByProviderRegistration'.");
            }),
        },
        vendors: ["azure"],
        services: ["providerhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OperationByProviderRegistration || (OperationByProviderRegistration = {}));
exports.OperationByProviderRegistration = OperationByProviderRegistration;
var ProviderRegistration;
(function (ProviderRegistration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.ProviderRegistration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ProviderRegistration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-providerhub-v20210901preview-providerregistration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.ProviderRegistration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210901preview_4.ProviderRegistration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ProviderRegistration shouldn't use an unstable API (providerhub.v20210901preview.ProviderRegistration). A compatible replacement can be found at 'providerhub.ProviderRegistration'.");
            }),
        },
        vendors: ["azure"],
        services: ["providerhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ProviderRegistration || (ProviderRegistration = {}));
exports.ProviderRegistration = ProviderRegistration;
var ResourceTypeRegistration;
(function (ResourceTypeRegistration) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.ResourceTypeRegistration).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ResourceTypeRegistration.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-providerhub-v20210901preview-resourcetyperegistration-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.ResourceTypeRegistration).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210901preview_5.ResourceTypeRegistration, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ResourceTypeRegistration shouldn't use an unstable API (providerhub.v20210901preview.ResourceTypeRegistration). A compatible replacement can be found at 'providerhub.ResourceTypeRegistration'.");
            }),
        },
        vendors: ["azure"],
        services: ["providerhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ResourceTypeRegistration || (ResourceTypeRegistration = {}));
exports.ResourceTypeRegistration = ResourceTypeRegistration;
var Skus;
(function (Skus) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.Skus).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Skus.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-providerhub-v20210901preview-skus-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.Skus).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210901preview_6.Skus, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Skus shouldn't use an unstable API (providerhub.v20210901preview.Skus). A compatible replacement can be found at 'providerhub.Skus'.");
            }),
        },
        vendors: ["azure"],
        services: ["providerhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Skus || (Skus = {}));
exports.Skus = Skus;
var SkusNestedResourceTypeFirst;
(function (SkusNestedResourceTypeFirst) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.SkusNestedResourceTypeFirst).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SkusNestedResourceTypeFirst.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-providerhub-v20210901preview-skusnestedresourcetypefirst-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.SkusNestedResourceTypeFirst).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210901preview_7.SkusNestedResourceTypeFirst, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SkusNestedResourceTypeFirst shouldn't use an unstable API (providerhub.v20210901preview.SkusNestedResourceTypeFirst). A compatible replacement can be found at 'providerhub.SkusNestedResourceTypeFirst'.");
            }),
        },
        vendors: ["azure"],
        services: ["providerhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SkusNestedResourceTypeFirst || (SkusNestedResourceTypeFirst = {}));
exports.SkusNestedResourceTypeFirst = SkusNestedResourceTypeFirst;
var SkusNestedResourceTypeSecond;
(function (SkusNestedResourceTypeSecond) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.SkusNestedResourceTypeSecond).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SkusNestedResourceTypeSecond.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-providerhub-v20210901preview-skusnestedresourcetypesecond-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.SkusNestedResourceTypeSecond).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210901preview_8.SkusNestedResourceTypeSecond, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SkusNestedResourceTypeSecond shouldn't use an unstable API (providerhub.v20210901preview.SkusNestedResourceTypeSecond). A compatible replacement can be found at 'providerhub.SkusNestedResourceTypeSecond'.");
            }),
        },
        vendors: ["azure"],
        services: ["providerhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SkusNestedResourceTypeSecond || (SkusNestedResourceTypeSecond = {}));
exports.SkusNestedResourceTypeSecond = SkusNestedResourceTypeSecond;
var SkusNestedResourceTypeThird;
(function (SkusNestedResourceTypeThird) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.SkusNestedResourceTypeThird).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SkusNestedResourceTypeThird.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-providerhub-v20210901preview-skusnestedresourcetypethird-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (providerhub.v20210901preview.SkusNestedResourceTypeThird).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20210901preview_9.SkusNestedResourceTypeThird, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SkusNestedResourceTypeThird shouldn't use an unstable API (providerhub.v20210901preview.SkusNestedResourceTypeThird). A compatible replacement can be found at 'providerhub.SkusNestedResourceTypeThird'.");
            }),
        },
        vendors: ["azure"],
        services: ["providerhub"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SkusNestedResourceTypeThird || (SkusNestedResourceTypeThird = {}));
exports.SkusNestedResourceTypeThird = SkusNestedResourceTypeThird;
