// @pulumi/azure/management/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkAssociation = exports.PrivateLink = exports.Lock = exports.GroupTemplateDeployment = exports.GroupSubscriptionAssociation = exports.GroupPolicyRemediation = exports.GroupPolicyExemption = exports.GroupPolicyAssignment = exports.Group = exports.getGroupTemplateDeploymentOutput = exports.getGroupTemplateDeployment = exports.getGroupOutput = exports.getGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGroup = null;
exports.getGroupOutput = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
exports.getGroupTemplateDeployment = null;
exports.getGroupTemplateDeploymentOutput = null;
utilities.lazyLoad(exports, ["getGroupTemplateDeployment", "getGroupTemplateDeploymentOutput"], () => require("./getGroupTemplateDeployment"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.GroupPolicyAssignment = null;
utilities.lazyLoad(exports, ["GroupPolicyAssignment"], () => require("./groupPolicyAssignment"));
exports.GroupPolicyExemption = null;
utilities.lazyLoad(exports, ["GroupPolicyExemption"], () => require("./groupPolicyExemption"));
exports.GroupPolicyRemediation = null;
utilities.lazyLoad(exports, ["GroupPolicyRemediation"], () => require("./groupPolicyRemediation"));
exports.GroupSubscriptionAssociation = null;
utilities.lazyLoad(exports, ["GroupSubscriptionAssociation"], () => require("./groupSubscriptionAssociation"));
exports.GroupTemplateDeployment = null;
utilities.lazyLoad(exports, ["GroupTemplateDeployment"], () => require("./groupTemplateDeployment"));
exports.Lock = null;
utilities.lazyLoad(exports, ["Lock"], () => require("./lock"));
exports.PrivateLink = null;
utilities.lazyLoad(exports, ["PrivateLink"], () => require("./privateLink"));
exports.PrivateLinkAssociation = null;
utilities.lazyLoad(exports, ["PrivateLinkAssociation"], () => require("./privateLinkAssociation"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:management/group:Group":
                return new exports.Group(name, undefined, { urn });
            case "azure:management/groupPolicyAssignment:GroupPolicyAssignment":
                return new exports.GroupPolicyAssignment(name, undefined, { urn });
            case "azure:management/groupPolicyExemption:GroupPolicyExemption":
                return new exports.GroupPolicyExemption(name, undefined, { urn });
            case "azure:management/groupPolicyRemediation:GroupPolicyRemediation":
                return new exports.GroupPolicyRemediation(name, undefined, { urn });
            case "azure:management/groupSubscriptionAssociation:GroupSubscriptionAssociation":
                return new exports.GroupSubscriptionAssociation(name, undefined, { urn });
            case "azure:management/groupTemplateDeployment:GroupTemplateDeployment":
                return new exports.GroupTemplateDeployment(name, undefined, { urn });
            case "azure:management/lock:Lock":
                return new exports.Lock(name, undefined, { urn });
            case "azure:management/privateLink:PrivateLink":
                return new exports.PrivateLink(name, undefined, { urn });
            case "azure:management/privateLinkAssociation:PrivateLinkAssociation":
                return new exports.PrivateLinkAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "management/group", _module);
pulumi.runtime.registerResourceModule("azure", "management/groupPolicyAssignment", _module);
pulumi.runtime.registerResourceModule("azure", "management/groupPolicyExemption", _module);
pulumi.runtime.registerResourceModule("azure", "management/groupPolicyRemediation", _module);
pulumi.runtime.registerResourceModule("azure", "management/groupSubscriptionAssociation", _module);
pulumi.runtime.registerResourceModule("azure", "management/groupTemplateDeployment", _module);
pulumi.runtime.registerResourceModule("azure", "management/lock", _module);
pulumi.runtime.registerResourceModule("azure", "management/privateLink", _module);
pulumi.runtime.registerResourceModule("azure", "management/privateLinkAssociation", _module);
