// @pulumi/azure-compliance-policies/azurenative/cdn/v20240601preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.TunnelPolicy = exports.SecurityPolicy = exports.Secret = exports.RuleSet = exports.Rule = exports.Route = exports.Profile = exports.Policy = exports.OriginGroup = exports.Origin = exports.KeyGroup = exports.Endpoint = exports.CustomDomain = exports.AFDTargetGroup = exports.AFDOriginGroup = exports.AFDOrigin = exports.AFDEndpoint = exports.AFDCustomDomain = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20240601preview_1 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_2 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_3 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_4 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_5 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_6 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_7 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_8 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_9 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_10 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_11 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_12 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_13 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_14 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_15 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_16 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_17 = require("@pulumi/azure-native/cdn/v20240601preview");
const v20240601preview_18 = require("@pulumi/azure-native/cdn/v20240601preview");
var AFDCustomDomain;
(function (AFDCustomDomain) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDCustomDomain).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AFDCustomDomain.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-afdcustomdomain-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDCustomDomain).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_1.AFDCustomDomain, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AFDCustomDomain shouldn't use an unstable API (cdn.v20240601preview.AFDCustomDomain). A compatible replacement can be found at 'cdn.AFDCustomDomain'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AFDCustomDomain || (AFDCustomDomain = {}));
exports.AFDCustomDomain = AFDCustomDomain;
var AFDEndpoint;
(function (AFDEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AFDEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-afdendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_2.AFDEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AFDEndpoint shouldn't use an unstable API (cdn.v20240601preview.AFDEndpoint). A compatible replacement can be found at 'cdn.AFDEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AFDEndpoint || (AFDEndpoint = {}));
exports.AFDEndpoint = AFDEndpoint;
var AFDOrigin;
(function (AFDOrigin) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDOrigin).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AFDOrigin.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-afdorigin-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDOrigin).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_3.AFDOrigin, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AFDOrigin shouldn't use an unstable API (cdn.v20240601preview.AFDOrigin). A compatible replacement can be found at 'cdn.AFDOrigin'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AFDOrigin || (AFDOrigin = {}));
exports.AFDOrigin = AFDOrigin;
var AFDOriginGroup;
(function (AFDOriginGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDOriginGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AFDOriginGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-afdorigingroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDOriginGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_4.AFDOriginGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AFDOriginGroup shouldn't use an unstable API (cdn.v20240601preview.AFDOriginGroup). A compatible replacement can be found at 'cdn.AFDOriginGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AFDOriginGroup || (AFDOriginGroup = {}));
exports.AFDOriginGroup = AFDOriginGroup;
var AFDTargetGroup;
(function (AFDTargetGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDTargetGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    AFDTargetGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-afdtargetgroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.AFDTargetGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_5.AFDTargetGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure AFDTargetGroup shouldn't use an unstable API (cdn.v20240601preview.AFDTargetGroup). A compatible replacement can be found at 'cdn.AFDTargetGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(AFDTargetGroup || (AFDTargetGroup = {}));
exports.AFDTargetGroup = AFDTargetGroup;
var CustomDomain;
(function (CustomDomain) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.CustomDomain).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CustomDomain.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-customdomain-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.CustomDomain).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_6.CustomDomain, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CustomDomain shouldn't use an unstable API (cdn.v20240601preview.CustomDomain). A compatible replacement can be found at 'cdn.CustomDomain'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CustomDomain || (CustomDomain = {}));
exports.CustomDomain = CustomDomain;
var Endpoint;
(function (Endpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Endpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Endpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-endpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Endpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_7.Endpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Endpoint shouldn't use an unstable API (cdn.v20240601preview.Endpoint). A compatible replacement can be found at 'cdn.Endpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Endpoint || (Endpoint = {}));
exports.Endpoint = Endpoint;
var KeyGroup;
(function (KeyGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.KeyGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    KeyGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-keygroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.KeyGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_8.KeyGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure KeyGroup shouldn't use an unstable API (cdn.v20240601preview.KeyGroup). A compatible replacement can be found at 'cdn.KeyGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(KeyGroup || (KeyGroup = {}));
exports.KeyGroup = KeyGroup;
var Origin;
(function (Origin) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Origin).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Origin.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-origin-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Origin).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_9.Origin, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Origin shouldn't use an unstable API (cdn.v20240601preview.Origin). A compatible replacement can be found at 'cdn.Origin'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Origin || (Origin = {}));
exports.Origin = Origin;
var OriginGroup;
(function (OriginGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.OriginGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OriginGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-origingroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.OriginGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_10.OriginGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OriginGroup shouldn't use an unstable API (cdn.v20240601preview.OriginGroup). A compatible replacement can be found at 'cdn.OriginGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OriginGroup || (OriginGroup = {}));
exports.OriginGroup = OriginGroup;
var Policy;
(function (Policy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Policy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Policy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-policy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Policy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_11.Policy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Policy shouldn't use an unstable API (cdn.v20240601preview.Policy). A compatible replacement can be found at 'cdn.Policy'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Policy || (Policy = {}));
exports.Policy = Policy;
var Profile;
(function (Profile) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Profile).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Profile.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-profile-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Profile).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_12.Profile, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Profile shouldn't use an unstable API (cdn.v20240601preview.Profile). A compatible replacement can be found at 'cdn.Profile'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Profile || (Profile = {}));
exports.Profile = Profile;
var Route;
(function (Route) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Route).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Route.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-route-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Route).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_13.Route, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Route shouldn't use an unstable API (cdn.v20240601preview.Route). A compatible replacement can be found at 'cdn.Route'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Route || (Route = {}));
exports.Route = Route;
var Rule;
(function (Rule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Rule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Rule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-rule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Rule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_14.Rule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Rule shouldn't use an unstable API (cdn.v20240601preview.Rule). A compatible replacement can be found at 'cdn.Rule'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Rule || (Rule = {}));
exports.Rule = Rule;
var RuleSet;
(function (RuleSet) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.RuleSet).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RuleSet.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-ruleset-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.RuleSet).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_15.RuleSet, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RuleSet shouldn't use an unstable API (cdn.v20240601preview.RuleSet). A compatible replacement can be found at 'cdn.RuleSet'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RuleSet || (RuleSet = {}));
exports.RuleSet = RuleSet;
var Secret;
(function (Secret) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Secret).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Secret.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-secret-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.Secret).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_16.Secret, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Secret shouldn't use an unstable API (cdn.v20240601preview.Secret). A compatible replacement can be found at 'cdn.Secret'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Secret || (Secret = {}));
exports.Secret = Secret;
var SecurityPolicy;
(function (SecurityPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.SecurityPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    SecurityPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-securitypolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.SecurityPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_17.SecurityPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure SecurityPolicy shouldn't use an unstable API (cdn.v20240601preview.SecurityPolicy). A compatible replacement can be found at 'cdn.SecurityPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(SecurityPolicy || (SecurityPolicy = {}));
exports.SecurityPolicy = SecurityPolicy;
var TunnelPolicy;
(function (TunnelPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.TunnelPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    TunnelPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-cdn-v20240601preview-tunnelpolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (cdn.v20240601preview.TunnelPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20240601preview_18.TunnelPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure TunnelPolicy shouldn't use an unstable API (cdn.v20240601preview.TunnelPolicy). A compatible replacement can be found at 'cdn.TunnelPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["cdn"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(TunnelPolicy || (TunnelPolicy = {}));
exports.TunnelPolicy = TunnelPolicy;
