// @pulumi/azure-compliance-policies/azurenative/machinelearningservices/v20241001preview/index.js
"use strict";
// Copyright 2016-2024, Pulumi Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ------------------------------- WARNING -------------------------------------
// This file was programmatically generated. Do not edit unless you know what
// you're doing.
// ------------------------------- WARNING -------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceConnection = exports.Workspace = exports.ServerlessEndpoint = exports.Schedule = exports.RegistryModelVersion = exports.RegistryModelContainer = exports.RegistryEnvironmentVersion = exports.RegistryEnvironmentContainer = exports.RegistryDataVersion = exports.RegistryDataContainer = exports.RegistryComponentVersion = exports.RegistryComponentContainer = exports.RegistryCodeVersion = exports.RegistryCodeContainer = exports.Registry = exports.RaiPolicy = exports.PrivateEndpointConnection = exports.OnlineEndpoint = exports.OnlineDeployment = exports.ModelVersion = exports.ModelContainer = exports.MarketplaceSubscription = exports.ManagedNetworkSettingsRule = exports.Job = exports.InferencePool = exports.InferenceGroup = exports.InferenceEndpoint = exports.FeaturestoreEntityVersion = exports.FeaturestoreEntityContainerEntity = exports.FeaturesetVersion = exports.FeaturesetContainerEntity = exports.EnvironmentVersion = exports.EnvironmentContainer = exports.EndpointDeployment = exports.Datastore = exports.DataVersion = exports.DataContainer = exports.ConnectionRaiPolicy = exports.ConnectionRaiBlocklistItem = exports.ConnectionRaiBlocklist = exports.ConnectionDeployment = exports.Compute = exports.ComponentVersion = exports.ComponentContainer = exports.CodeVersion = exports.CodeContainer = exports.CapabilityHost = exports.BatchEndpoint = exports.BatchDeployment = void 0;
const policy_1 = require("@pulumi/policy");
const compliance_policy_manager_1 = require("@pulumi/compliance-policy-manager");
const v20241001preview_1 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_2 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_3 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_4 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_5 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_6 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_7 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_8 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_9 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_10 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_11 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_12 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_13 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_14 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_15 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_16 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_17 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_18 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_19 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_20 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_21 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_22 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_23 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_24 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_25 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_26 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_27 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_28 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_29 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_30 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_31 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_32 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_33 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_34 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_35 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_36 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_37 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_38 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_39 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_40 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_41 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_42 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_43 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_44 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_45 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_46 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_47 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_48 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
const v20241001preview_49 = require("@pulumi/azure-native/machinelearningservices/v20241001preview");
var BatchDeployment;
(function (BatchDeployment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.BatchDeployment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BatchDeployment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-batchdeployment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.BatchDeployment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_1.BatchDeployment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BatchDeployment shouldn't use an unstable API (machinelearningservices.v20241001preview.BatchDeployment). A compatible replacement can be found at 'machinelearningservices.BatchDeployment'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BatchDeployment || (BatchDeployment = {}));
exports.BatchDeployment = BatchDeployment;
var BatchEndpoint;
(function (BatchEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.BatchEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    BatchEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-batchendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.BatchEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_2.BatchEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure BatchEndpoint shouldn't use an unstable API (machinelearningservices.v20241001preview.BatchEndpoint). A compatible replacement can be found at 'machinelearningservices.BatchEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(BatchEndpoint || (BatchEndpoint = {}));
exports.BatchEndpoint = BatchEndpoint;
var CapabilityHost;
(function (CapabilityHost) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.CapabilityHost).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CapabilityHost.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-capabilityhost-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.CapabilityHost).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_3.CapabilityHost, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CapabilityHost shouldn't use an unstable API (machinelearningservices.v20241001preview.CapabilityHost). A compatible replacement can be found at 'machinelearningservices.CapabilityHost'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CapabilityHost || (CapabilityHost = {}));
exports.CapabilityHost = CapabilityHost;
var CodeContainer;
(function (CodeContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.CodeContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CodeContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-codecontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.CodeContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_4.CodeContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CodeContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.CodeContainer). A compatible replacement can be found at 'machinelearningservices.CodeContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CodeContainer || (CodeContainer = {}));
exports.CodeContainer = CodeContainer;
var CodeVersion;
(function (CodeVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.CodeVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    CodeVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-codeversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.CodeVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_5.CodeVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure CodeVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.CodeVersion). A compatible replacement can be found at 'machinelearningservices.CodeVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(CodeVersion || (CodeVersion = {}));
exports.CodeVersion = CodeVersion;
var ComponentContainer;
(function (ComponentContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ComponentContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ComponentContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-componentcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ComponentContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_6.ComponentContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ComponentContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.ComponentContainer). A compatible replacement can be found at 'machinelearningservices.ComponentContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ComponentContainer || (ComponentContainer = {}));
exports.ComponentContainer = ComponentContainer;
var ComponentVersion;
(function (ComponentVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ComponentVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ComponentVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-componentversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ComponentVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_7.ComponentVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ComponentVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.ComponentVersion). A compatible replacement can be found at 'machinelearningservices.ComponentVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ComponentVersion || (ComponentVersion = {}));
exports.ComponentVersion = ComponentVersion;
var Compute;
(function (Compute) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Compute).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Compute.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-compute-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Compute).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_8.Compute, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Compute shouldn't use an unstable API (machinelearningservices.v20241001preview.Compute). A compatible replacement can be found at 'machinelearningservices.Compute'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Compute || (Compute = {}));
exports.Compute = Compute;
var ConnectionDeployment;
(function (ConnectionDeployment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ConnectionDeployment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConnectionDeployment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-connectiondeployment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ConnectionDeployment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_9.ConnectionDeployment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConnectionDeployment shouldn't use an unstable API (machinelearningservices.v20241001preview.ConnectionDeployment). A compatible replacement can be found at 'machinelearningservices.ConnectionDeployment'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConnectionDeployment || (ConnectionDeployment = {}));
exports.ConnectionDeployment = ConnectionDeployment;
var ConnectionRaiBlocklist;
(function (ConnectionRaiBlocklist) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ConnectionRaiBlocklist).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConnectionRaiBlocklist.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-connectionraiblocklist-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ConnectionRaiBlocklist).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_10.ConnectionRaiBlocklist, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConnectionRaiBlocklist shouldn't use an unstable API (machinelearningservices.v20241001preview.ConnectionRaiBlocklist). A compatible replacement can be found at 'machinelearningservices.ConnectionRaiBlocklist'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConnectionRaiBlocklist || (ConnectionRaiBlocklist = {}));
exports.ConnectionRaiBlocklist = ConnectionRaiBlocklist;
var ConnectionRaiBlocklistItem;
(function (ConnectionRaiBlocklistItem) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ConnectionRaiBlocklistItem).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConnectionRaiBlocklistItem.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-connectionraiblocklistitem-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ConnectionRaiBlocklistItem).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_11.ConnectionRaiBlocklistItem, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConnectionRaiBlocklistItem shouldn't use an unstable API (machinelearningservices.v20241001preview.ConnectionRaiBlocklistItem). A compatible replacement can be found at 'machinelearningservices.ConnectionRaiBlocklistItem'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConnectionRaiBlocklistItem || (ConnectionRaiBlocklistItem = {}));
exports.ConnectionRaiBlocklistItem = ConnectionRaiBlocklistItem;
var ConnectionRaiPolicy;
(function (ConnectionRaiPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ConnectionRaiPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ConnectionRaiPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-connectionraipolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ConnectionRaiPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_12.ConnectionRaiPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ConnectionRaiPolicy shouldn't use an unstable API (machinelearningservices.v20241001preview.ConnectionRaiPolicy). A compatible replacement can be found at 'machinelearningservices.ConnectionRaiPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ConnectionRaiPolicy || (ConnectionRaiPolicy = {}));
exports.ConnectionRaiPolicy = ConnectionRaiPolicy;
var DataContainer;
(function (DataContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.DataContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DataContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-datacontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.DataContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_13.DataContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DataContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.DataContainer). A compatible replacement can be found at 'machinelearningservices.DataContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DataContainer || (DataContainer = {}));
exports.DataContainer = DataContainer;
var DataVersion;
(function (DataVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.DataVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    DataVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-dataversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.DataVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_14.DataVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure DataVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.DataVersion). A compatible replacement can be found at 'machinelearningservices.DataVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(DataVersion || (DataVersion = {}));
exports.DataVersion = DataVersion;
var Datastore;
(function (Datastore) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Datastore).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Datastore.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-datastore-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Datastore).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_15.Datastore, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Datastore shouldn't use an unstable API (machinelearningservices.v20241001preview.Datastore). A compatible replacement can be found at 'machinelearningservices.Datastore'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Datastore || (Datastore = {}));
exports.Datastore = Datastore;
var EndpointDeployment;
(function (EndpointDeployment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.EndpointDeployment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EndpointDeployment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-endpointdeployment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.EndpointDeployment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_16.EndpointDeployment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EndpointDeployment shouldn't use an unstable API (machinelearningservices.v20241001preview.EndpointDeployment). A compatible replacement can be found at 'machinelearningservices.EndpointDeployment'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EndpointDeployment || (EndpointDeployment = {}));
exports.EndpointDeployment = EndpointDeployment;
var EnvironmentContainer;
(function (EnvironmentContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.EnvironmentContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EnvironmentContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-environmentcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.EnvironmentContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_17.EnvironmentContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EnvironmentContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.EnvironmentContainer). A compatible replacement can be found at 'machinelearningservices.EnvironmentContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EnvironmentContainer || (EnvironmentContainer = {}));
exports.EnvironmentContainer = EnvironmentContainer;
var EnvironmentVersion;
(function (EnvironmentVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.EnvironmentVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    EnvironmentVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-environmentversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.EnvironmentVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_18.EnvironmentVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure EnvironmentVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.EnvironmentVersion). A compatible replacement can be found at 'machinelearningservices.EnvironmentVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(EnvironmentVersion || (EnvironmentVersion = {}));
exports.EnvironmentVersion = EnvironmentVersion;
var FeaturesetContainerEntity;
(function (FeaturesetContainerEntity) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.FeaturesetContainerEntity).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FeaturesetContainerEntity.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-featuresetcontainerentity-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.FeaturesetContainerEntity).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_19.FeaturesetContainerEntity, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FeaturesetContainerEntity shouldn't use an unstable API (machinelearningservices.v20241001preview.FeaturesetContainerEntity). A compatible replacement can be found at 'machinelearningservices.FeaturesetContainerEntity'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FeaturesetContainerEntity || (FeaturesetContainerEntity = {}));
exports.FeaturesetContainerEntity = FeaturesetContainerEntity;
var FeaturesetVersion;
(function (FeaturesetVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.FeaturesetVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FeaturesetVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-featuresetversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.FeaturesetVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_20.FeaturesetVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FeaturesetVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.FeaturesetVersion). A compatible replacement can be found at 'machinelearningservices.FeaturesetVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FeaturesetVersion || (FeaturesetVersion = {}));
exports.FeaturesetVersion = FeaturesetVersion;
var FeaturestoreEntityContainerEntity;
(function (FeaturestoreEntityContainerEntity) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.FeaturestoreEntityContainerEntity).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FeaturestoreEntityContainerEntity.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-featurestoreentitycontainerentity-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.FeaturestoreEntityContainerEntity).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_21.FeaturestoreEntityContainerEntity, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FeaturestoreEntityContainerEntity shouldn't use an unstable API (machinelearningservices.v20241001preview.FeaturestoreEntityContainerEntity). A compatible replacement can be found at 'machinelearningservices.FeaturestoreEntityContainerEntity'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FeaturestoreEntityContainerEntity || (FeaturestoreEntityContainerEntity = {}));
exports.FeaturestoreEntityContainerEntity = FeaturestoreEntityContainerEntity;
var FeaturestoreEntityVersion;
(function (FeaturestoreEntityVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.FeaturestoreEntityVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    FeaturestoreEntityVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-featurestoreentityversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.FeaturestoreEntityVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_22.FeaturestoreEntityVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure FeaturestoreEntityVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.FeaturestoreEntityVersion). A compatible replacement can be found at 'machinelearningservices.FeaturestoreEntityVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(FeaturestoreEntityVersion || (FeaturestoreEntityVersion = {}));
exports.FeaturestoreEntityVersion = FeaturestoreEntityVersion;
var InferenceEndpoint;
(function (InferenceEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.InferenceEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    InferenceEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-inferenceendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.InferenceEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_23.InferenceEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure InferenceEndpoint shouldn't use an unstable API (machinelearningservices.v20241001preview.InferenceEndpoint). A compatible replacement can be found at 'machinelearningservices.InferenceEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(InferenceEndpoint || (InferenceEndpoint = {}));
exports.InferenceEndpoint = InferenceEndpoint;
var InferenceGroup;
(function (InferenceGroup) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.InferenceGroup).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    InferenceGroup.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-inferencegroup-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.InferenceGroup).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_24.InferenceGroup, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure InferenceGroup shouldn't use an unstable API (machinelearningservices.v20241001preview.InferenceGroup). A compatible replacement can be found at 'machinelearningservices.InferenceGroup'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(InferenceGroup || (InferenceGroup = {}));
exports.InferenceGroup = InferenceGroup;
var InferencePool;
(function (InferencePool) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.InferencePool).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    InferencePool.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-inferencepool-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.InferencePool).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_25.InferencePool, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure InferencePool shouldn't use an unstable API (machinelearningservices.v20241001preview.InferencePool). A compatible replacement can be found at 'machinelearningservices.InferencePool'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(InferencePool || (InferencePool = {}));
exports.InferencePool = InferencePool;
var Job;
(function (Job) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Job).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Job.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-job-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Job).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_26.Job, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Job shouldn't use an unstable API (machinelearningservices.v20241001preview.Job). A compatible replacement can be found at 'machinelearningservices.Job'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Job || (Job = {}));
exports.Job = Job;
var ManagedNetworkSettingsRule;
(function (ManagedNetworkSettingsRule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ManagedNetworkSettingsRule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ManagedNetworkSettingsRule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-managednetworksettingsrule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ManagedNetworkSettingsRule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_27.ManagedNetworkSettingsRule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ManagedNetworkSettingsRule shouldn't use an unstable API (machinelearningservices.v20241001preview.ManagedNetworkSettingsRule). A compatible replacement can be found at 'machinelearningservices.ManagedNetworkSettingsRule'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ManagedNetworkSettingsRule || (ManagedNetworkSettingsRule = {}));
exports.ManagedNetworkSettingsRule = ManagedNetworkSettingsRule;
var MarketplaceSubscription;
(function (MarketplaceSubscription) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.MarketplaceSubscription).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    MarketplaceSubscription.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-marketplacesubscription-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.MarketplaceSubscription).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_28.MarketplaceSubscription, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure MarketplaceSubscription shouldn't use an unstable API (machinelearningservices.v20241001preview.MarketplaceSubscription). A compatible replacement can be found at 'machinelearningservices.MarketplaceSubscription'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(MarketplaceSubscription || (MarketplaceSubscription = {}));
exports.MarketplaceSubscription = MarketplaceSubscription;
var ModelContainer;
(function (ModelContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ModelContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ModelContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-modelcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ModelContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_29.ModelContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ModelContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.ModelContainer). A compatible replacement can be found at 'machinelearningservices.ModelContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ModelContainer || (ModelContainer = {}));
exports.ModelContainer = ModelContainer;
var ModelVersion;
(function (ModelVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ModelVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ModelVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-modelversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ModelVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_30.ModelVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ModelVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.ModelVersion). A compatible replacement can be found at 'machinelearningservices.ModelVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ModelVersion || (ModelVersion = {}));
exports.ModelVersion = ModelVersion;
var OnlineDeployment;
(function (OnlineDeployment) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.OnlineDeployment).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OnlineDeployment.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-onlinedeployment-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.OnlineDeployment).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_31.OnlineDeployment, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OnlineDeployment shouldn't use an unstable API (machinelearningservices.v20241001preview.OnlineDeployment). A compatible replacement can be found at 'machinelearningservices.OnlineDeployment'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OnlineDeployment || (OnlineDeployment = {}));
exports.OnlineDeployment = OnlineDeployment;
var OnlineEndpoint;
(function (OnlineEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.OnlineEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    OnlineEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-onlineendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.OnlineEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_32.OnlineEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure OnlineEndpoint shouldn't use an unstable API (machinelearningservices.v20241001preview.OnlineEndpoint). A compatible replacement can be found at 'machinelearningservices.OnlineEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(OnlineEndpoint || (OnlineEndpoint = {}));
exports.OnlineEndpoint = OnlineEndpoint;
var PrivateEndpointConnection;
(function (PrivateEndpointConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.PrivateEndpointConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    PrivateEndpointConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-privateendpointconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.PrivateEndpointConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_33.PrivateEndpointConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure PrivateEndpointConnection shouldn't use an unstable API (machinelearningservices.v20241001preview.PrivateEndpointConnection). A compatible replacement can be found at 'machinelearningservices.PrivateEndpointConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(PrivateEndpointConnection || (PrivateEndpointConnection = {}));
exports.PrivateEndpointConnection = PrivateEndpointConnection;
var RaiPolicy;
(function (RaiPolicy) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RaiPolicy).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RaiPolicy.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-raipolicy-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RaiPolicy).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_34.RaiPolicy, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RaiPolicy shouldn't use an unstable API (machinelearningservices.v20241001preview.RaiPolicy). A compatible replacement can be found at 'machinelearningservices.RaiPolicy'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RaiPolicy || (RaiPolicy = {}));
exports.RaiPolicy = RaiPolicy;
var Registry;
(function (Registry) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Registry).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Registry.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registry-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Registry).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_35.Registry, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Registry shouldn't use an unstable API (machinelearningservices.v20241001preview.Registry). A compatible replacement can be found at 'machinelearningservices.Registry'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Registry || (Registry = {}));
exports.Registry = Registry;
var RegistryCodeContainer;
(function (RegistryCodeContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryCodeContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryCodeContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registrycodecontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryCodeContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_36.RegistryCodeContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryCodeContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryCodeContainer). A compatible replacement can be found at 'machinelearningservices.RegistryCodeContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryCodeContainer || (RegistryCodeContainer = {}));
exports.RegistryCodeContainer = RegistryCodeContainer;
var RegistryCodeVersion;
(function (RegistryCodeVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryCodeVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryCodeVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registrycodeversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryCodeVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_37.RegistryCodeVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryCodeVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryCodeVersion). A compatible replacement can be found at 'machinelearningservices.RegistryCodeVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryCodeVersion || (RegistryCodeVersion = {}));
exports.RegistryCodeVersion = RegistryCodeVersion;
var RegistryComponentContainer;
(function (RegistryComponentContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryComponentContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryComponentContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registrycomponentcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryComponentContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_38.RegistryComponentContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryComponentContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryComponentContainer). A compatible replacement can be found at 'machinelearningservices.RegistryComponentContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryComponentContainer || (RegistryComponentContainer = {}));
exports.RegistryComponentContainer = RegistryComponentContainer;
var RegistryComponentVersion;
(function (RegistryComponentVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryComponentVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryComponentVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registrycomponentversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryComponentVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_39.RegistryComponentVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryComponentVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryComponentVersion). A compatible replacement can be found at 'machinelearningservices.RegistryComponentVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryComponentVersion || (RegistryComponentVersion = {}));
exports.RegistryComponentVersion = RegistryComponentVersion;
var RegistryDataContainer;
(function (RegistryDataContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryDataContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryDataContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registrydatacontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryDataContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_40.RegistryDataContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryDataContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryDataContainer). A compatible replacement can be found at 'machinelearningservices.RegistryDataContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryDataContainer || (RegistryDataContainer = {}));
exports.RegistryDataContainer = RegistryDataContainer;
var RegistryDataVersion;
(function (RegistryDataVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryDataVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryDataVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registrydataversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryDataVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_41.RegistryDataVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryDataVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryDataVersion). A compatible replacement can be found at 'machinelearningservices.RegistryDataVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryDataVersion || (RegistryDataVersion = {}));
exports.RegistryDataVersion = RegistryDataVersion;
var RegistryEnvironmentContainer;
(function (RegistryEnvironmentContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryEnvironmentContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryEnvironmentContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registryenvironmentcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryEnvironmentContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_42.RegistryEnvironmentContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryEnvironmentContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryEnvironmentContainer). A compatible replacement can be found at 'machinelearningservices.RegistryEnvironmentContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryEnvironmentContainer || (RegistryEnvironmentContainer = {}));
exports.RegistryEnvironmentContainer = RegistryEnvironmentContainer;
var RegistryEnvironmentVersion;
(function (RegistryEnvironmentVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryEnvironmentVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryEnvironmentVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registryenvironmentversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryEnvironmentVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_43.RegistryEnvironmentVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryEnvironmentVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryEnvironmentVersion). A compatible replacement can be found at 'machinelearningservices.RegistryEnvironmentVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryEnvironmentVersion || (RegistryEnvironmentVersion = {}));
exports.RegistryEnvironmentVersion = RegistryEnvironmentVersion;
var RegistryModelContainer;
(function (RegistryModelContainer) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryModelContainer).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryModelContainer.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registrymodelcontainer-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryModelContainer).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_44.RegistryModelContainer, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryModelContainer shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryModelContainer). A compatible replacement can be found at 'machinelearningservices.RegistryModelContainer'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryModelContainer || (RegistryModelContainer = {}));
exports.RegistryModelContainer = RegistryModelContainer;
var RegistryModelVersion;
(function (RegistryModelVersion) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryModelVersion).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    RegistryModelVersion.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-registrymodelversion-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.RegistryModelVersion).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_45.RegistryModelVersion, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure RegistryModelVersion shouldn't use an unstable API (machinelearningservices.v20241001preview.RegistryModelVersion). A compatible replacement can be found at 'machinelearningservices.RegistryModelVersion'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(RegistryModelVersion || (RegistryModelVersion = {}));
exports.RegistryModelVersion = RegistryModelVersion;
var Schedule;
(function (Schedule) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Schedule).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Schedule.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-schedule-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Schedule).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_46.Schedule, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Schedule shouldn't use an unstable API (machinelearningservices.v20241001preview.Schedule). A compatible replacement can be found at 'machinelearningservices.Schedule'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Schedule || (Schedule = {}));
exports.Schedule = Schedule;
var ServerlessEndpoint;
(function (ServerlessEndpoint) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ServerlessEndpoint).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    ServerlessEndpoint.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-serverlessendpoint-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.ServerlessEndpoint).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_47.ServerlessEndpoint, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure ServerlessEndpoint shouldn't use an unstable API (machinelearningservices.v20241001preview.ServerlessEndpoint). A compatible replacement can be found at 'machinelearningservices.ServerlessEndpoint'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(ServerlessEndpoint || (ServerlessEndpoint = {}));
exports.ServerlessEndpoint = ServerlessEndpoint;
var Workspace;
(function (Workspace) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Workspace).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    Workspace.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-workspace-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.Workspace).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_48.Workspace, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure Workspace shouldn't use an unstable API (machinelearningservices.v20241001preview.Workspace). A compatible replacement can be found at 'machinelearningservices.Workspace'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(Workspace || (Workspace = {}));
exports.Workspace = Workspace;
var WorkspaceConnection;
(function (WorkspaceConnection) {
    /**
     * Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.WorkspaceConnection).
     *
     * @severity medium
     * @frameworks none
     * @topics api, preview, unstable
     * @link https://learn.microsoft.com/en-us/rest/api/azure/
     */
    WorkspaceConnection.disallowPreviewResource = compliance_policy_manager_1.policyManager.registerPolicy({
        resourceValidationPolicy: {
            name: "azurenative-machinelearningservices-v20241001preview-workspaceconnection-disallow-preview-resource",
            description: "Disallow the use of non-stable (Preview) Azure resouces (machinelearningservices.v20241001preview.WorkspaceConnection).",
            configSchema: compliance_policy_manager_1.policyManager.policyConfigSchema,
            enforcementLevel: "advisory",
            validateResource: (0, policy_1.validateResourceOfType)(v20241001preview_49.WorkspaceConnection, (_, args, reportViolation) => {
                if (!compliance_policy_manager_1.policyManager.shouldEvalPolicy(args)) {
                    return;
                }
                reportViolation("Azure WorkspaceConnection shouldn't use an unstable API (machinelearningservices.v20241001preview.WorkspaceConnection). A compatible replacement can be found at 'machinelearningservices.WorkspaceConnection'.");
            }),
        },
        vendors: ["azure"],
        services: ["machinelearningservices"],
        severity: "medium",
        topics: ["api", "unstable", "preview"],
    });
})(WorkspaceConnection || (WorkspaceConnection = {}));
exports.WorkspaceConnection = WorkspaceConnection;
