// @pulumi/azure/appconfiguration/licationLoadBalancerFrontend.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicationLoadBalancerFrontend = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Application Gateway for Containers Frontend.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.appconfiguration.LicationLoadBalancer("example", {
 *     name: "example",
 *     resourceGroupName: "example",
 *     location: "West Europe",
 * });
 * const exampleLicationLoadBalancerFrontend = new azure.appconfiguration.LicationLoadBalancerFrontend("example", {
 *     name: "example",
 *     applicationLoadBalancerId: example.id,
 * });
 * ```
 *
 * ## Import
 *
 * Application Gateway for Containers Frontend can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:appconfiguration/licationLoadBalancerFrontend:LicationLoadBalancerFrontend example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.ServiceNetworking/trafficControllers/alb1/frontends/frontend1
 * ```
 */
class LicationLoadBalancerFrontend extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["applicationLoadBalancerId"] = state ? state.applicationLoadBalancerId : undefined;
            resourceInputs["fullyQualifiedDomainName"] = state ? state.fullyQualifiedDomainName : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.applicationLoadBalancerId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'applicationLoadBalancerId'");
            }
            resourceInputs["applicationLoadBalancerId"] = args ? args.applicationLoadBalancerId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["fullyQualifiedDomainName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LicationLoadBalancerFrontend.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing LicationLoadBalancerFrontend resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LicationLoadBalancerFrontend(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LicationLoadBalancerFrontend.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LicationLoadBalancerFrontend.__pulumiType;
    }
}
exports.LicationLoadBalancerFrontend = LicationLoadBalancerFrontend;
/** @internal */
LicationLoadBalancerFrontend.__pulumiType = 'azure:appconfiguration/licationLoadBalancerFrontend:LicationLoadBalancerFrontend';
